<div id="tabs">
    <ul>
        <li><a href="#tabs-1">Overview</a></li>
        <li><a href="#tabs-2">Input & Parameters</a></li>
        <li><a href="#tabs-3">References</a></li>
    </ul>
    <div id="tabs-1" class="greenlink">
        Ali2D which takes a sequence alignment as input, performs PSIPRED and MEMSAT2 on each of the sequences and
        finally plots information about secondary structure, transmembrane regions and amino acids on the alignment.
    </div>
    <div id="tabs-2" class="greenlink">
        <ul class="accordion" data-accordion data-allow-all-closed='true'>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Input</a>
                <div class="accordion-content" data-tab-content>
                    You have the option to enter a multiple sequence alignment (MSA) in FASTA or CLUSTAL format. The
                    number of sequences is currently limited to 100.<br/><br/>
                    @views.html.help.common.msa()
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">% identity cutoff to invoke a new PSIPRED run</a>
                <div class="accordion-content" data-tab-content>
                    This option specifies the percentage sequence identity below which a new Psipred run is invoked.
                </div>
            </li>
        </ul>
    </div>
    <div id="tabs-3" class="greenlink">
        @views.html.help.common.toolkit_citation()
    </div>
</div>