@()
<div id="tabs">
    <ul>
        <li><a href="#tabs-1">Overview</a></li>
        <li><a href="#tabs-2">Parameters</a></li>
        <li><a href="#tabs-3">References</a></li>
    </ul>
    <div id="tabs-1" class="greenlink">
        The Backtranslator translates an amino acid sequence back into a possible DNA sequence.
    </div>
    <div id="tabs-2" class="greenlink">
        <ul class="accordion" data-accordion data-allow-all-closed='true'>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Input</a>
                <div class="accordion-content" data-tab-content>
                    <p>Enter a single protein sequence in FASTA format, for example:</p><br/>
                    <pre>>gi|576099|pdb|1ENA|  Staphylococcal Nuclease (E.C.3.1.31.1) Mutation<br/>LHKEPATLIKAIDGETVKLMYKGQPMTFRLLLVDTPETKHPKKGVEKYGPEASAFTKKMVENAKKIEVEF<br/>DKGQRTDKYGRGLAYIYADGKMVNEALVRQGLAKVAYVYKPNNTHEQHLRKSEAQAKKEKLNIWS</pre><br/>
                    <p>The sequence name (and an optional description) must be contained in a single name line beginning with the ">" symbol and followed directly by the sequence name. The residue data is contained in one or more lines of arbitrary length following the name line. No empty lines should be used. Upper and lower case letters are allowed. Spaces and gaps (-.) will be removed.</p>
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Include amino acid sequence in output</a>
                <div class="accordion-content" data-tab-content>
                    <p>Activate this option to include the amino acid sequence in the output. If activated amino acids will be printed above their encoding DNA codon. For example, results may look like this:</p><br/>
                    <pre>> Possible DNA-sequence of Protein: gi|576099|pdb|1ENA|  Staphylococcal Nuclease (E.C.3.1.31.1) Mutation<br/> L   H   K   E   P   A   T   L   I   K   A   I   D   G   E   T   V   K   L   M  <br/>TTG CAC AAG GAG CCT GCA ACC CTT ATC AAG GCA ATA GAC GGA GAA ACG GTA AAA CTC ATG <br/><br/> Y   K   G   Q   P   M   T   F   R   L   L   L   V   D   T   P   E   T   K   H  <br/>TAC AAA GGG CAG CCT ATG ACG TTT CGG CTA CTT CTG GTC GAC ACG CCG GAA ACG AAA CAT <br/><br/> P   K   K   G   V   E   K   Y   G   P   E   A   S   A   F   T   K   K   M   V  <br/>CCG AAG AAA GGG GTC GAA AAA TAT GGT CCG GAG GCC TCG GCA TTC ACA AAA AAG ATG GTA <br/><br/> E   N   A   K   K   I   E   V   E   F   D   K   G   Q   R   T   D   K   Y   G  <br/>GAG AAT GCT AAA AAA ATT GAG GTG GAG TTT GAT AAA GGT CAA CGA ACT GAC AAG TAT GGT <br/><br/> R   G   L   A   Y   I   Y   A   D   G   K   M   V   N   E   A   L   V   R   Q  <br/>AGA GGT TTA GCT TAC ATC TAC GCT GAT GGC AAG ATG GTC AAT GAA GCC CTC GTT CGG CAA <br/><br/> G   L   A   K   V   A   Y   V   Y   K   P   N   N   T   H   E   Q   H   L   R  <br/>GGC CTT GCA AAA GTG GCT TAT GTG TAT AAA CCG AAC AAT ACT CAC GAA CAA CAC CTT CGG <br/><br/> K   S   E   A   Q   A   K   K   E   K   L   N   I   W   S  <br/>AAA TCA GAG GCT CAG GCA AAA AAG GAA AAA TTA AAT ATC TGG TCG </pre><br/><br/>With the option turned of the above output would look like this:<br/><br/><br/><pre>> Possible DNA-sequence of Protein: gi|576099|pdb|1ENA|  Staphylococcal Nuclease (E.C.3.1.31.1) Mutation <br/>CTTCATAAAGAACCAGCTACCCTTATCAAGGCGATCGACGGTGAAACTGTGAAGCTCATGTACAAGGGGCAAC<br/>CAATGACGTTCCGCCTGCTGCTTGTGGACACACCCGAAACTAAGCACCCCAAAAAAGGTGTAGAGAAATACGG<br/>GCCTGAAGCTTCTGCCTTCACAAAAAAGATGGTGGAAAACGCAAAGAAGATTGAGGTCGAATTCGATAAGGGC<br/>CAAAGAACGGATAAGTACGGTCGAGGGCTGGCCTATATTTACGCTGACGGCAAGATGGTGAATGAAGCGCTCG<br/>TGCGTCAAGGACTGGCCAAAGTAGCATACGTTTACAAACCAAATAATACTCATGAGCAACACCTCCGAAAGAG<br/>TGAGGCGCAAGCTAAAAAGGAAAAGCTCAACATCTGGTCC</pre>
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Genetic code</a>
                <div class="accordion-content" data-tab-content>
                    Select the correct genetic code for your organism. The genetic codes are identical to those available on the <a href = https://www.ncbi.nlm.nih.gov/taxonomy target="_blank" rel="noopener">NCBI homepage</a>.
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Codon usage table</a>
                <div class="accordion-content" data-tab-content>
                    Select this option if you want to use a codon usage table. If this option is not selected all possible codons will have the same probability.
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Codon usage table of organism</a>
                <div class="accordion-content" data-tab-content>
                    Here you can search for a codon usage table of an organism. The codon usage database (identical to the one of <a href = http://www.kazusa.or.jp/codon target="_blank" rel="noopener">http://www.kazusa.or.jp/codon/</a>) is based of the <a href = https://www.ncbi.nlm.nih.gov/genbank target="_blank" rel="noopener">NCBI GenBank</a> (GenBank Release 160.0 [June 15 2007]). Please enter the name of the organism in the box.
                </div>
            </li>
        </ul>
    </div>
    <div id="tabs-3" class="greenlink">
        @help.citations.toolkit_citation()
        <p>Nakamura Y., Gojobori T., Ikemura T. (2000) <b>Codon usage tabulated from international DNA sequence databases: status for the year 2000. </b>Nucleic Acids Res 28(1):292. <a href = https://www.ncbi.nlm.nih.gov/pubmed/10592250 target="_blank" rel="noopener">PMID: 10592250</a></p>
    </div>
</div>

<script>
        $(".accordion").foundation();
</script>