@()
<div id="tabs">
    <ul>
        <li><a href="#tabs-1">Overview</a></li>
        <li><a href="#tabs-2">Parameters</a></li>
        <li><a href="#tabs-3">References</a></li>
    </ul>
    <div id="tabs-1" class="greenlink">
        <p>For an accurate sequence-based prediction of protein structure and function, a highly sensitive sequence-search method is crucial. Today, profile-profile and HMM-HMM alignment methods are custom when it comes to identification and alignment of templates for 3D-homology modeling. As these methods are generally too slow for iteratively searching through large databases, we offer HHblits as a 'lightning-fast' HMM-HMM-based iterative sequence search tool. The profile-profile alignment prefilter of HHblits reduces the number of full HMM-HMM alignments from many millions to a few thousand, making it faster than PSI-BLAST but still as sensitive as HHsearch. Prefiltering is essential for speed and sensitivity. Compared to PSI-BLAST, HHblits is faster, up to twice as sensitive and produces more accurate alignments.</p>
        <p>The HHblits software is part of the open source package HHsuite.</p>
    </div>
    <div id="tabs-2" class="greenlink">
        <ul class="accordion" data-accordion data-allow-all-closed='true'>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Paste sequence or MSA</a>
                <div class="accordion-content" data-tab-content>
                    You have the option to paste either a single protein sequence or a Multiple Sequence Alignment (MSA). HHblits builds high-quality MSAs starting from single sequences or from MSAs. It transforms these into a query HMM and iteratively searches through the selected databases.
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Select input format</a>
                <div class="accordion-content" data-tab-content>
                    <p><b>Entering a single query sequence</b></p>
                    <p>You can enter a single query sequence in FASTA format (with or without name line). The query sequence in FASTA can look like this:</p>
                    <pre>>gi|532319|pir|TVFV2E|TVFV2E envelope protein<br/>ELRLRYCAPAGFALLKCNDADYDGFKTNCSNVSVVHCTNLMNTTVTTGLLLNGSYSENRT<br/>QIWQKHRTSNDSALILLNKHYNLTVTCKRPGNKTVLPVTIMAGLVFHSQKYNLRLRQAWC<br/>HFPSNWKGAWKEVKEEIVNLPKERYRGTNDPKRIFFQRQWGDPETANLWFNCHGEFFYCK<br/>MDWFLNYLNNLTVDADHNECKNTSGTKSGNKRAPGPCVQR</pre><br/>
                    <p>or this</p>
                    <pre>>gi|532319|pir|TVFV2E|TVFV2E envelope protein<br/>ELRLRYCAPA GFALLKCNDA DYDGFKTNCS NVSVVHCTNL MNTTVTTGLL LNGSYSENRT QIWQKHRTSN <br/>DSALILLNKH YNLTVTCKRP GNKTVLPVTI MAGLVFHSQK YNLRLRQAWC HFPSNWKGAW KEVKEEIVNL <br/>PKERYRGTND PKRIFFQRQW GDPETANLWF NCHGEFFYCK MDWFLNYLNN LTVDADHNEC KNTSGTKSGN <br/>KRAPGPCVQR</pre><br/>
                    <p>or simply this</p>
                    <pre>ELRLRYCAPAGFALLKCNDADYDGFKTNCSNVSVVHCTNLMNTTVTTGLLLNGSYSENRTQIWQKHRTSNDSALILLNKHYNLTVTCKRPGNK<br/>TVLPVTIMAGLVFHSQKYNLRLRQAWCHFPSNWKGAWKEVKEEIVNLPKERYRGTNDPKRIFFQRQWGDPETANLWFNCHGEFFYCKMDWFLNYLNNLTVDADHNE<br/>CKNTSGTKSGNKRAPGPCVQR</pre><br/>
                    <p>The sequence may contain a nameline that must start with a > symbol, followed directly by the sequence name and an optional description. The residue data is contained in one or more lines of arbitrary length following the name line. No empty lines should be used. Upper and lower case letters are allowed. Spaces and gaps (-.) will be removed. </p><br/>
                    <p><b>Entering a multiple alignment</b></p>
                    <p>If you already have a query alignment then you can enter it here. You can enter alignments in different formats, which are listed in the dropdown menu. Format descriptions can be found in the help pages of the reformat tool.</p><br/>
                    <p><b>Example for aligned FASTA format:</b></p><br/>
                    <pre>>d1a1x__ b.63.1.1 (-) p13-MTCP1 {Human (Homo sapiens)}<br/>PPDHLWVHQEGIYRDEYQRTWVAVVEE--E--T--SF---------LR----------ARVQQIQVPLG-------DAARPSHLLTS-----QLPLMWQLYP<br/>>gi|6678257|ref|NP_033363.1|:(7-103) T-cell lymphoma breakpoint 1 [Mus musculus]<br/>HPNRLWIWEKHVYLDEFRRSWLPVVIK--S--N--EK---------FQ----------VILRQEDVTLG-------EAMSPSQLVPY-----ELPLMWQLYP<br/>>gi|7305557|ref|NP_038800.1|:(8-103) T-cell leukemia/lymphoma 1B, 3 [Mus musculus]<br/>PPRFLVCTRDDIYEDENGRQWVVAKVE--T--S--RSpygsrietcIT----------VHLQHMTTIPQ-------EPTPQQPINNN-----SLPTMWRLES<br/>>gi|11415028|ref|NP_068801.1|:(2-106) T-cell lymphoma-1; T-cell lymphoma-1A [Homo sapiens]<br/>HPDRLWAWEKFVYLDEKQHAWLPLTIEikD--R--LQ---------LR----------VLLRREDVVLG-------RPMTPTQIGPS-----LLPIMWQLYP<br/>>gi|7305561|ref|NP_038804.1|:(7-103) T-cell leukemia/lymphoma 1B, 5 [Mus musculus]<br/>----------GIYEDEHHRVWIAVNVE--T--S--HS---------SHgnrietcvt-VHLQHMTTLPQ-------EPTPQQPINNN-----SLPTMWRLES<br/>>gi|7305553|ref|NP_038801.1|:(5-103) T-cell leukemia/lymphoma 1B, 1 [Mus musculus]<br/>LPVYLVSVRLGIYEDEHHRVWIVANVE--TshS--SH---------GN----------RRRTHVTVHLW-------KLIPQQVIPFNplnydFLPTTWKLES<br/>>gi|27668591|ref|XP_234504.1|:(7-103) similar to Chain A, Crystal Structure Of Murine Tcl1<br/>-PDRLWLWEKHVYLDEFRRSWLPIVIK--S--N--GK---------FQ----------VIMRQKDVILG-------DSMTPSQLVPY-----ELPLMWQLYP<br/>>gi|27668589|ref|XP_234503.1|:(9-91) similar to T-cell leukemia/lymphoma 1B, 5;<br/>-PHILTLRTHGIYEDEHHRLWVVLDLQ--A--ShlSF---------SN----------RLLIYLTVYLQqgvafplESTPPSPMNLN-----GLPRRWTLRT<br/>>gi|7305559|ref|NP_038802.1|:(8-102) T-cell leukemia/lymphoma 1B, 4 [Mus musculus] <br/>PPCFLVCTRDDIYEDEHGRQWVAAKVE--T--S--SH---------SPycskietcvtVHLWQMTTLFQ-------EPSPDSLKTFN-----FLPRTWRLES<br/>>gi|7305555|ref|NP_038803.1|:(9-102) T-cell leukemia/lymphoma 1B, 2 [Mus musculus]<br/>---------PGFYEDEHHRLWMVAKLE--T--C--SH---------SPycnkietcvtVHLWQMTRYPQ-------EPAPYNPMNYN-----FLPMTWRLAS</pre><br/>
                    <p>In aligned FASTA the gaps are written with '-' and the n'th letter of each sequence (except newlines) is understood to build the n'th column of the multiple alignment. The sequence residues may be broken up into several lines containing arbitrary numbers of residues.</p><br/>
                    <p><b>The same alignment in CLUSTAL format looks like this:</b></p><br/>
                    <pre>CLUSTAL X (1.83) multiple sequence alignment<br/><br/><br/>d1a1x__                             PPDHLWVHQEGIYRDEYQRTWVAVVEE--E--T--SF---------LR----------AR<br/>gi|6678257|ref|NP_033363.1|__7      HPNRLWIWEKHVYLDEFRRSWLPVVIK--S--N--EK---------FQ----------VI<br/>gi|7305557|ref|NP_038800.1|__8      PPRFLVCTRDDIYEDENGRQWVVAKVE--T--S--RSPYGSRIETCIT----------VH<br/>gi|11415028|ref|NP_068801.1|__      HPDRLWAWEKFVYLDEKQHAWLPLTIEIKD--R--LQ---------LR----------VL<br/>gi|7305561|ref|NP_038804.1|__7      ----------GIYEDEHHRVWIAVNVE--T--S--HS---------SHGNRIETCVT-VH<br/>gi|7305553|ref|NP_038801.1|__5      LPVYLVSVRLGIYEDEHHRVWIVANVE--TSHS--SH---------GN----------RR<br/>gi|27668591|ref|XP_234504.1|__      -PDRLWLWEKHVYLDEFRRSWLPIVIK--S--N--GK---------FQ----------VI<br/>gi|27668589|ref|XP_234503.1|__      -PHILTLRTHGIYEDEHHRLWVVLDLQ--A--SHLSF---------SN----------RL<br/>gi|7305559|ref|NP_038802.1|__8      PPCFLVCTRDDIYEDEHGRQWVAAKVE--T--S--SH---------SPYCSKIETCVTVH<br/>gi|7305555|ref|NP_038803.1|__9      ---------PGFYEDEHHRLWMVAKLE--T--C--SH---------SPYCNKIETCVTVH<br/>                                               .* **  : *:    :                       <br/><br/>d1a1x__                             VQQIQVPLG-------DAARPSHLLTS-----QLPLMWQLYP<br/>gi|6678257|ref|NP_033363.1|__7      LRQEDVTLG-------EAMSPSQLVPY-----ELPLMWQLYP<br/>gi|7305557|ref|NP_038800.1|__8      LQHMTTIPQ-------EPTPQQPINNN-----SLPTMWRLES<br/>gi|11415028|ref|NP_068801.1|__      LRREDVVLG-------RPMTPTQIGPS-----LLPIMWQLYP<br/>gi|7305561|ref|NP_038804.1|__7      LQHMTTLPQ-------EPTPQQPINNN-----SLPTMWRLES<br/>gi|7305553|ref|NP_038801.1|__5      RTHVTVHLW-------KLIPQQVIPFNPLNYDFLPTTWKLES<br/>gi|27668591|ref|XP_234504.1|__      MRQKDVILG-------DSMTPSQLVPY-----ELPLMWQLYP<br/>gi|27668589|ref|XP_234503.1|__      LIYLTVYLQQGVAFPLESTPPSPMNLN-----GLPRRWTLRT<br/>gi|7305559|ref|NP_038802.1|__8      LWQMTTLFQ-------EPSPDSLKTFN-----FLPRTWRLES<br/>gi|7305555|ref|NP_038803.1|__9      LWQMTRYPQ-------EPAPYNPMNYN-----FLPMTWRLAS<br/>                                                                     **  * * .</pre><br/>
                    <p>The first line must begin with CLUSTAL, the rest of the line is ignored. The sequences come in blocks with the sequence name, cut to a constant number of characters, followed by some whitespace and the sequence residues and gaps (-). The number of alignment columns per sequence line is arbitrary but constant within a block. Thus the whole alignment may be written in one long block. At the end of each block is an optional line indicating the conservation of each alignment column, followed by an obligatory empty line that separates the block from the next. Capital and lower case characters may be used for the amino acids.</p><br/>
                    <p><b>The same alignment in Stockholm format is here:</b></p><br/>
                    <pre># STOCKHOLM 1.0<br/><br/>#=GC RF                          XXXXXXXXXXXXXXXXXXXXXXXXXXX--X--X--XX---------XX----------XXXXXXXXXXX<br/>d1a1x__                          PPDHLWVHQEGIYRDEYQRTWVAVVEE--E--T--SF---------LR----------ARVQQIQVPLG<br/>gi|6678257|ref|NP_033363.1|:(7-1 HPNRLWIWEKHVYLDEFRRSWLPVVIK--S--N--EK---------FQ----------VILRQEDVTLG<br/>gi|7305557|ref|NP_038800.1|:(8-1 PPRFLVCTRDDIYEDENGRQWVVAKVE--T--S--RSpygsrietcIT----------VHLQHMTTIPQ<br/>gi|11415028|ref|NP_068801.1|:(2- HPDRLWAWEKFVYLDEKQHAWLPLTIEikD--R--LQ---------LR----------VLLRREDVVLG<br/>gi|7305561|ref|NP_038804.1|:(7-1 ----------GIYEDEHHRVWIAVNVE--T--S--HS---------SHgnrietcvt-VHLQHMTTLPQ<br/>gi|7305553|ref|NP_038801.1|:(5-1 LPVYLVSVRLGIYEDEHHRVWIVANVE--TshS--SH---------GN----------RRRTHVTVHLW<br/>gi|27668591|ref|XP_234504.1|:(7- -PDRLWLWEKHVYLDEFRRSWLPIVIK--S--N--GK---------FQ----------VIMRQKDVILG<br/>gi|27668589|ref|XP_234503.1|:(9- -PHILTLRTHGIYEDEHHRLWVVLDLQ--A--ShlSF---------SN----------RLLIYLTVYLQ<br/>gi|7305559|ref|NP_038802.1|:(8-1 PPCFLVCTRDDIYEDEHGRQWVAAKVE--T--S--SH---------SPycskietcvtVHLWQMTTLFQ<br/>gi|7305555|ref|NP_038803.1|:(9-1 ---------PGFYEDEHHRLWMVAKLE--T--C--SH---------SPycnkietcvtVHLWQMTRYPQ<br/><br/>#=GC RF                          -------XXXXXXXXXXX-----XXXXXXXXXX<br/>d1a1x__                          -------DAARPSHLLTS-----QLPLMWQLYP<br/>gi|6678257|ref|NP_033363.1|:(7-1 -------EAMSPSQLVPY-----ELPLMWQLYP<br/>gi|7305557|ref|NP_038800.1|:(8-1 -------EPTPQQPINNN-----SLPTMWRLES<br/>gi|11415028|ref|NP_068801.1|:(2- -------RPMTPTQIGPS-----LLPIMWQLYP<br/>gi|7305561|ref|NP_038804.1|:(7-1 -------EPTPQQPINNN-----SLPTMWRLES<br/>gi|7305553|ref|NP_038801.1|:(5-1 -------KLIPQQVIPFNplnydFLPTTWKLES<br/>gi|27668591|ref|XP_234504.1|:(7- -------DSMTPSQLVPY-----ELPLMWQLYP<br/>gi|27668589|ref|XP_234503.1|:(9- qgvafplESTPPSPMNLN-----GLPRRWTLRT<br/>gi|7305559|ref|NP_038802.1|:(8-1 -------EPSPDSLKTFN-----FLPRTWRLES<br/>gi|7305555|ref|NP_038803.1|:(9-1 -------EPAPYNPMNYN-----FLPMTWRLAS<br/>//</pre><br/>
                    <p>Stockholm format is similar to Clustal format, but offers a number of markups to include annotation and additional information referring to the alignment, its columns, or sequences. It was developed by Sean Eddy for Pfam and the HMMer software package. Each alignment may be separated into an arbitrary number of blocks with an arbitrary number of alignment columns contained per line. Gaps may be written '-' or '.'. The line starting with #=GC RF is an example of an optional markup line that contains one character per alignment columns. In this case it indicates which are the match/delete columns (as opposed to insert columns).<br/><br/><br/><b>The same alignment in A2M format looks like this:</b></p><br/>
                    <pre>>d1a1x__ b.63.1.1 (-) p13-MTCP1 {Human (Homo sapiens)}<br/>PPDHLWVHQEGIYRDEYQRTWVAVVEE..E..T..SF.........LR..........ARVQQIQVPLG.......DAARPSHLLTS.....QLPLMWQLYP<br/>>gi|6678257|ref|NP_033363.1|:(7-103) T-cell lymphoma breakpoint 1 [Mus musculus]<br/>HPNRLWIWEKHVYLDEFRRSWLPVVIK..S..N..EK.........FQ..........VILRQEDVTLG.......EAMSPSQLVPY.....ELPLMWQLYP<br/>>gi|7305557|ref|NP_038800.1|:(8-103) T-cell leukemia/lymphoma 1B, 3 [Mus musculus]<br/>PPRFLVCTRDDIYEDENGRQWVVAKVE..T..S..RSpygsrietcIT..........VHLQHMTTIPQ.......EPTPQQPINNN.....SLPTMWRLES<br/>>gi|11415028|ref|NP_068801.1|:(2-106) T-cell lymphoma-1; T-cell lymphoma-1A [Homo sapiens]<br/>HPDRLWAWEKFVYLDEKQHAWLPLTIEikD..R..LQ.........LR..........VLLRREDVVLG.......RPMTPTQIGPS.....LLPIMWQLYP<br/>>gi|7305561|ref|NP_038804.1|:(7-103) T-cell leukemia/lymphoma 1B, 5 [Mus musculus]<br/>----------GIYEDEHHRVWIAVNVE..T..S..HS.........SHgnrietcvt.VHLQHMTTLPQ.......EPTPQQPINNN.....SLPTMWRLES<br/>>gi|7305553|ref|NP_038801.1|:(5-103) T-cell leukemia/lymphoma 1B, 1 [Mus musculus]<br/>LPVYLVSVRLGIYEDEHHRVWIVANVE..TshS..SH.........GN..........RRRTHVTVHLW.......KLIPQQVIPFNplnydFLPTTWKLES<br/>>gi|27668591|ref|XP_234504.1|:(7-103) similar to Chain A, Crystal Structure Of Murine Tcl1<br/>-PDRLWLWEKHVYLDEFRRSWLPIVIK..S..N..GK.........FQ..........VIMRQKDVILG.......DSMTPSQLVPY.....ELPLMWQLYP<br/>>gi|27668589|ref|XP_234503.1|:(9-91) similar to T-cell leukemia/lymphoma 1B, 5;<br/>-PHILTLRTHGIYEDEHHRLWVVLDLQ..A..ShlSF.........SN..........RLLIYLTVYLQqgvafplESTPPSPMNLN.....GLPRRWTLRT<br/>>gi|7305559|ref|NP_038802.1|:(8-102) T-cell leukemia/lymphoma 1B, 4 [Mus musculus] <br/>PPCFLVCTRDDIYEDEHGRQWVAAKVE..T..S..SH.........SPycskietcvtVHLWQMTTLFQ.......EPSPDSLKTFN.....FLPRTWRLES<br/>>gi|7305555|ref|NP_038803.1|:(9-102) T-cell leukemia/lymphoma 1B, 2 [Mus musculus]<br/>---------PGFYEDEHHRLWMVAKLE..T..C..SH.........SPycnkietcvtVHLWQMTRYPQ.......EPAPYNPMNYN.....FLPMTWRLAS</pre><br/>
                    <p>A2M format is derived from aligned FASTA format. It looks very similar, but it distinguishes between match/delete columns and insert columns. This information is used to specify how an alignment is transformed into an HMM. The match/delete columns use upper case letters for residues and the '-' symbol for deletions (gaps). The insert columns use lower case letters for the inserted residues. Gaps aligned to inserted residues are written as '.'</p><br/>
                    <p><b>The same alignment in A3M is here:</b></p><br/>
                    <pre>>d1a1x__ b.63.1.1 (-) p13-MTCP1 {Human (Homo sapiens)}<br/>PPDHLWVHQEGIYRDEYQRTWVAVVEEETSFLRARVQQIQVPLGDAARPSHLLTSQLPLMWQLYP<br/>>gi|6678257|ref|NP_033363.1|:(7-103) T-cell lymphoma breakpoint 1 [Mus musculus]<br/>HPNRLWIWEKHVYLDEFRRSWLPVVIKSNEKFQVILRQEDVTLGEAMSPSQLVPYELPLMWQLYP<br/>>gi|7305557|ref|NP_038800.1|:(8-103) T-cell leukemia/lymphoma 1B, 3 [Mus musculus]<br/>PPRFLVCTRDDIYEDENGRQWVVAKVETSRSpygsrietcITVHLQHMTTIPQEPTPQQPINNNSLPTMWRLES<br/>>gi|11415028|ref|NP_068801.1|:(2-106) T-cell lymphoma-1; T-cell lymphoma-1A [Homo sapiens]<br/>HPDRLWAWEKFVYLDEKQHAWLPLTIEikDRLQLRVLLRREDVVLGRPMTPTQIGPSLLPIMWQLYP<br/>>gi|7305561|ref|NP_038804.1|:(7-103) T-cell leukemia/lymphoma 1B, 5 [Mus musculus]<br/>----------GIYEDEHHRVWIAVNVETSHSSHgnrietcvtVHLQHMTTLPQEPTPQQPINNNSLPTMWRLES<br/>>gi|7305553|ref|NP_038801.1|:(5-103) T-cell leukemia/lymphoma 1B, 1 [Mus musculus]<br/>LPVYLVSVRLGIYEDEHHRVWIVANVETshSSHGNRRRTHVTVHLWKLIPQQVIPFNplnydFLPTTWKLES<br/>>gi|27668591|ref|XP_234504.1|:(7-103) similar to Chain A, Crystal Structure Of Murine Tcl1<br/>-PDRLWLWEKHVYLDEFRRSWLPIVIKSNGKFQVIMRQKDVILGDSMTPSQLVPYELPLMWQLYP<br/>>gi|27668589|ref|XP_234503.1|:(9-91) similar to T-cell leukemia/lymphoma 1B, 5;<br/>-PHILTLRTHGIYEDEHHRLWVVLDLQAShlSFSNRLLIYLTVYLQqgvafplESTPPSPMNLNGLPRRWTLRT<br/>>gi|7305559|ref|NP_038802.1|:(8-102) T-cell leukemia/lymphoma 1B, 4 [Mus musculus] <br/>PPCFLVCTRDDIYEDEHGRQWVAAKVETSSHSPycskietcvtVHLWQMTTLFQEPSPDSLKTFNFLPRTWRLES<br/>>gi|7305555|ref|NP_038803.1|:(9-102) T-cell leukemia/lymphoma 1B, 2 [Mus musculus]<br/>---------PGFYEDEHHRLWMVAKLETCSHSPycnkietcvtVHLWQMTRYPQEPAPYNPMNYNFLPMTWRLAS</pre><br/>
                    <p>The A3M format is a condensed version of A2M format. It is obtained by omitting all '.' symbols from A2M format. Hence residues emitted by Match states of the HMM are in upper case, residues emitted by Insert states are in lower case and deletions are written '-'. A3M-formatted alignments are not really human readable but they can be reformatted to other more readable formats like FASTA, A2M CLUSTAL and many others with the reformatting utility in our toolbox. A3M is not an official format. Likewise, the name A3M is our personal invention for this very practical and space-efficient format which we use to store the alignments internally.</p>
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Match state assignment</a>
                <div class="accordion-content" data-tab-content>
                    <p>With this option you can choose the match state assignment for alignment with a format different to A2M/A3M. This value is the percentage threshold up to which percentage of gaps a column should be assigned as a match state, e.g. a value of 50 means that all columns with less than 50% gaps are assigned as match states. With the value 'first' all columns with a residue in 1st sequence are assigned as match states.</p>
                    <p>This parameter is not available with input formats a2m and a3m, because with these input formats the match state assignment is taken from the input.</p>
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Select HMM databases</a>
                <div class="accordion-content" data-tab-content>
                    Select the database(s) of template HMMs against which you want to compare the query sequence or alignment.<br/><br/><a href = https://uniclust.mmseqs.com>Uniclust30</a><br/>The Uniclust30 database is a clustered version of the Uniclust database down to a maximum pairwise sequence identity of 30%.
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Max. number of iterations</a>
                <div class="accordion-content" data-tab-content>
                    This specifies the number of iterations that are performed by HHblits. The iterations stop in any case when no further sequences are found.
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Alignment mode</a>
                <div class="accordion-content" data-tab-content>
                    <p>In local mode, HHblits will use <u>local</u> alignments to score your query sequence with the database sequences. In global mode it uses global alignments with no end gap penalties (sometimes called 'global-local' alignments), i.e. the query-template alignments extend up to the beginning and end of the query or template sequence. The use of local mode is strongly recommended if you want to know which are your most likely remote homologs. If you have already identified your template with a previous run in local mode, you can use the global mode to get global alignments with an already identified template.</p><p>HHblits was shown to be significantly more sensitive to detect remote homologs in local than in global mode. The reason is that most remote homologs are really only homologous in a common core that define their superfamily. This core has been adorned by other secondary structure elements in different ways in different families. Also, the statistical evaluation is more error-prone for the global alignment mode, since the negatives score distribution for global alignments does not follow an extreme-value distribution.</p><p>You may generate global alignments and at the same time score alignments with the more sensitive local Viterbi alignment by selecting the 'local' option in combination with the option 'Realign with MAC algorithm' and setting the 'MAC realignment threshold' to 0.0.</p>
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Realign with MAC algorithm</a>
                <div class="accordion-content" data-tab-content>
                    <p>After the search with the standard Viterbi algorithm, this option will cause HHblits to realign all query-template alignments with the more accurate Maximum ACcuracy alignment algorithm.</p><p>Since for technical reasons the scores, E-values, and probabilities from the Viterbi alignment and NOT the MAC alignment are displayed and used for ranking, the MAC realignment option may lead to high-scoring alignments of length 1. This happens when the MAC algorithm cannot find any acceptable alignment path, which is an indication for a non-homologous relationship.</p><p>When the MAC realignment option is activated, the 'global' and 'local' options now refer to both the Viterbi search stage as well as the MAC realignment stage. With -global (-local), the posterior probability matrix will be calculated for global (local) alignment.</p>
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">MAC realignment threshold</a>
                <div class="accordion-content" data-tab-content>
                    Only active when the 'Realign with MAC algorithm' option is active. This parameter controls the MAC alignment algorithm's greediness. More precisely, the MAC algorithm finds the alignment that maximizes the sum of posterior probabilities minus this threshold for each aligned pair. Global alignments are generated with 0, whereas 0.7 will produce quite conservative local alignments with average probabilities for each aligned pair of residues above 70%. Default value is 0.5.
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Disable most filters</a>
                <div class="accordion-content" data-tab-content>
                    This option disable the most additional filter steps of HHblits with exception of the profile-profile prefilter. This results in a slighty more sensitive performance but it increases the runtime of HHblits.
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">E-value inclusion threshold</a>
                <div class="accordion-content" data-tab-content>
                    All HMM matches ("hits") with an E-value better than this threshold are included in the alignment for the next iteration or, in the last iteration, for building the final alignment shown in the 'Show alignment' result tab.
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Min. coverage of hits</a>
                <div class="accordion-content" data-tab-content>
                    Minimal coverage of HMM matches to be included in the evolving query profile.
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Show representative sequences per HMM</a>
                <div class="accordion-content" data-tab-content>
                    Setting this to 10, for instance, will generate alignments containing 10 representative sequences from both query and template alignments. This can help to get a better feeling for the conservation along the alignment. The representative sequences shown are selected for maximum pairwise dissimilarity. The maximum value for this parameter is 10.
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Width of alignments (in columns)</a>
                <div class="accordion-content" data-tab-content>
                    Number of alignment columns per line in the HMM-HMM alignments. Leave at default value except if your browser cuts off parts of the alignment.
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Minimum probability in hit list</a>
                <div class="accordion-content" data-tab-content>
                    Only hits above this cut-off probability will be displayed.
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Max. number of hits in hit list</a>
                <div class="accordion-content" data-tab-content>
                    This parameter controls how many matches will be displayed in the results.
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Coloring query-template multiple alignments in results</a>
                <div class="accordion-content" data-tab-content>
                    You can choose to color the sequences in the query-template alignments on the results page by physico-chemical properties of the amino acids, either by using colored fonts or by background coloring.
                </div>
            </li>
        </ul>
    </div>
    <div id="tabs-3" class="greenlink">
        <p>Alva V., Nam SZ., Söding J., Lupas AN. (2016) <b>The MPI bioinformatics Toolkit as an integrative platform for advanced protein sequence and structure analysis.</b> Nucleic Acids Res. pii: gkw348. <a href=https://www.ncbi.nlm.nih.gov/pubmed/27131380>PMID: 27131380</a><br/><br/>Remmert M., Biegert A., Hauser A., Söding J. (2011) <b>HHblits: Lightning-fast iterative protein sequence searching by HMM-HMM alignment.</b> Nat Methods. 9(2):173-5. doi: 10.1038/nmeth.1818. <a href=https://www.ncbi.nlm.nih.gov/pubmed/22198341>PMID: 22198341</a></p>
    </div>
</div>

<script>
        $(".accordion").foundation();
</script>