@()
<div id="tabs">
    <ul>
        <li><a href="#tabs-1">Overview</a></li>
        <li><a href="#tabs-2">Input & Parameters</a></li>
        <li><a href="#tabs-3">References</a></li>
    </ul>
    <div id="tabs-1" class="greenlink">
        <b>Background</b>
        <p>For an accurate sequence-based prediction of protein structure and function, a highly sensitive sequence-search method
            is crucial. Today, profile-profile and HMM-HMM alignment methods are custom when it comes to identification
            and alignment of templates for 3D-homology modeling. As these methods are generally too slow for iteratively
            searching through large databases, we offer HHblits as a 'lightning-fast' HMM-HMM-based iterative sequence
            search tool. The profile-profile alignment prefilter of HHblits reduces the number of full HMM-HMM alignments
            from many millions to a few thousand, making it faster than PSI-BLAST, but still as sensitive as HHsearch.
            Prefiltering is essential for speed and sensitivity. Compared to PSI-BLAST, HHblits is faster, up to twice
            as sensitive and produces more accurate alignments. The HHblits software is part of the open source<br/>
            package <a href = https://github.com/soedinglab/hh-suite target="_blank" rel="noopener">HHsuite</a>.</p><br/>

        <b>Output</b><br/>
        The results of HHblits are organized into five different tabs:<br/><br/>

        The 'Results' tab provides an interactive and detailed summary of the obtained hits. Here, a selection of the
        found hits (either the aligned regions or the full-length sequences) or the query A3M can be forwarded to other
        tools (e.g. HHpred, HHomp). A3Ms for matched templates can be retrieved in the 'Alignments' section by clicking
        on the 'Template alignment' link.<br/><br/>

        The 'Raw Output' tab displays the search results in text format and also offers it for download.<br/><br/>

        The E-value distribution of the obtained hits is presented in the 'E-value Plot' tab.<br/><br/>

        The query-template and query alignments can be visualized, downloaded, and forwarded to other tools in the
        'Query Template MSA' and 'Query Alignment' tabs.<br/>


    </div>
    <div id="tabs-2" class="greenlink">
        <ul class="accordion" data-accordion data-allow-all-closed='true'>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Input</a>
                <div class="accordion-content" data-tab-content>
                    You have the option to enter or upload either a single protein sequence or a multiple sequence alignment (MSA) in FASTA or CLUSTAL format.<br/><br/>
                        <p><b>Single query sequence:</b></p>
                    <p>You can enter a single query sequence in FASTA format (with or without name line):</p>
                    <pre>>gi|532319|pir|TVFV2E|TVFV2E envelope protein<br/>ELRLRYCAPAGFALLKCNDADYDGFKTNCSNVSVVHCTNLMNTTVTTGLLLNGSYSENRT<br/>QIWQKHRTSNDSALILLNKHYNLTVTCKRPGNKTVLPVTIMAGLVFHSQKYNLRLRQAWC<br/>HFPSNWKGAWKEVKEEIVNLPKERYRGTNDPKRIFFQRQWGDPETANLWFNCHGEFFYCK<br/>MDWFLNYLNNLTVDADHNECKNTSGTKSGNKRAPGPCVQR</pre>
                    <p>or this</p>
                    <pre>ELRLRYCAPAGFALLKCNDADYDGFKTNCSNVSVVHCTNLMNTTVTTGLLLNGSYSENRT<br/>QIWQKHRTSNDSALILLNKHYNLTVTCKRPGNKTVLPVTIMAGLVFHSQKYNLRLRQAWC<br/>HFPSNWKGAWKEVKEEIVNLPKERYRGTNDPKRIFFQRQWGDPETANLWFNCHGEFFYCK<br/>MDWFLNYLNNLTVDADHNECKNTSGTKSGNKRAPGPCVQR</pre><br/>
                    <p><b>Multiple sequence alignment</b></p>
                    <p>You can enter alignments in FASTA or CLUSTAL formats.</p>
                    <p>Example for aligned FASTA format:</p>
                    <pre>>d1a1x__ b.63.1.1 (-) p13-MTCP1 {Human (Homo sapiens)}<br/>PPDHLWVHQEGIYRDEYQRTWVAVVEE--E--T--SF---------LR----------ARVQQIQVPLG-------DAARPSHLLTS-----QLPLMWQLYP<br/>>gi|6678257|ref|NP_033363.1|:(7-103) T-cell lymphoma breakpoint 1 [Mus musculus]<br/>HPNRLWIWEKHVYLDEFRRSWLPVVIK--S--N--EK---------FQ----------VILRQEDVTLG-------EAMSPSQLVPY-----ELPLMWQLYP<br/>>gi|7305557|ref|NP_038800.1|:(8-103) T-cell leukemia/lymphoma 1B, 3 [Mus musculus]<br/>PPRFLVCTRDDIYEDENGRQWVVAKVE--T--S--RSpygsrietcIT----------VHLQHMTTIPQ-------EPTPQQPINNN-----SLPTMWRLES<br/>>gi|11415028|ref|NP_068801.1|:(2-106) T-cell lymphoma-1; T-cell lymphoma-1A [Homo sapiens]<br/>HPDRLWAWEKFVYLDEKQHAWLPLTIEikD--R--LQ---------LR----------VLLRREDVVLG-------RPMTPTQIGPS-----LLPIMWQLYP<br/>>gi|7305561|ref|NP_038804.1|:(7-103) T-cell leukemia/lymphoma 1B, 5 [Mus musculus]<br/>----------GIYEDEHHRVWIAVNVE--T--S--HS---------SHgnrietcvt-VHLQHMTTLPQ-------EPTPQQPINNN-----SLPTMWRLES<br/>>gi|7305553|ref|NP_038801.1|:(5-103) T-cell leukemia/lymphoma 1B, 1 [Mus musculus]<br/>LPVYLVSVRLGIYEDEHHRVWIVANVE--TshS--SH---------GN----------RRRTHVTVHLW-------KLIPQQVIPFNplnydFLPTTWKLES<br/>>gi|27668591|ref|XP_234504.1|:(7-103) similar to Chain A, Crystal Structure Of Murine Tcl1<br/>-PDRLWLWEKHVYLDEFRRSWLPIVIK--S--N--GK---------FQ----------VIMRQKDVILG-------DSMTPSQLVPY-----ELPLMWQLYP<br/>>gi|27668589|ref|XP_234503.1|:(9-91) similar to T-cell leukemia/lymphoma 1B, 5;<br/>-PHILTLRTHGIYEDEHHRLWVVLDLQ--A--ShlSF---------SN----------RLLIYLTVYLQqgvafplESTPPSPMNLN-----GLPRRWTLRT<br/>>gi|7305559|ref|NP_038802.1|:(8-102) T-cell leukemia/lymphoma 1B, 4 [Mus musculus] <br/>PPCFLVCTRDDIYEDEHGRQWVAAKVE--T--S--SH---------SPycskietcvtVHLWQMTTLFQ-------EPSPDSLKTFN-----FLPRTWRLES<br/>>gi|7305555|ref|NP_038803.1|:(9-102) T-cell leukemia/lymphoma 1B, 2 [Mus musculus]<br/>---------PGFYEDEHHRLWMVAKLE--T--C--SH---------SPycnkietcvtVHLWQMTRYPQ-------EPAPYNPMNYN-----FLPMTWRLAS</pre><br/>
                    <p>The same alignment in CLUSTAL format:</p>
                    <pre>CLUSTAL X (1.83) multiple sequence alignment<br/><br/>d1a1x__                             PPDHLWVHQEGIYRDEYQRTWVAVVEE--E--T--SF---------LR----------AR<br/>gi|6678257|ref|NP_033363.1|__7      HPNRLWIWEKHVYLDEFRRSWLPVVIK--S--N--EK---------FQ----------VI<br/>gi|7305557|ref|NP_038800.1|__8      PPRFLVCTRDDIYEDENGRQWVVAKVE--T--S--RSPYGSRIETCIT----------VH<br/>gi|11415028|ref|NP_068801.1|__      HPDRLWAWEKFVYLDEKQHAWLPLTIEIKD--R--LQ---------LR----------VL<br/>gi|7305561|ref|NP_038804.1|__7      ----------GIYEDEHHRVWIAVNVE--T--S--HS---------SHGNRIETCVT-VH<br/>gi|7305553|ref|NP_038801.1|__5      LPVYLVSVRLGIYEDEHHRVWIVANVE--TSHS--SH---------GN----------RR<br/>gi|27668591|ref|XP_234504.1|__      -PDRLWLWEKHVYLDEFRRSWLPIVIK--S--N--GK---------FQ----------VI<br/>gi|27668589|ref|XP_234503.1|__      -PHILTLRTHGIYEDEHHRLWVVLDLQ--A--SHLSF---------SN----------RL<br/>gi|7305559|ref|NP_038802.1|__8      PPCFLVCTRDDIYEDEHGRQWVAAKVE--T--S--SH---------SPYCSKIETCVTVH<br/>gi|7305555|ref|NP_038803.1|__9      ---------PGFYEDEHHRLWMVAKLE--T--C--SH---------SPYCNKIETCVTVH<br/>                                               .* **  : *:    :                       <br/><br/>d1a1x__                             VQQIQVPLG-------DAARPSHLLTS-----QLPLMWQLYP<br/>gi|6678257|ref|NP_033363.1|__7      LRQEDVTLG-------EAMSPSQLVPY-----ELPLMWQLYP<br/>gi|7305557|ref|NP_038800.1|__8      LQHMTTIPQ-------EPTPQQPINNN-----SLPTMWRLES<br/>gi|11415028|ref|NP_068801.1|__      LRREDVVLG-------RPMTPTQIGPS-----LLPIMWQLYP<br/>gi|7305561|ref|NP_038804.1|__7      LQHMTTLPQ-------EPTPQQPINNN-----SLPTMWRLES<br/>gi|7305553|ref|NP_038801.1|__5      RTHVTVHLW-------KLIPQQVIPFNPLNYDFLPTTWKLES<br/>gi|27668591|ref|XP_234504.1|__      MRQKDVILG-------DSMTPSQLVPY-----ELPLMWQLYP<br/>gi|27668589|ref|XP_234503.1|__      LIYLTVYLQQGVAFPLESTPPSPMNLN-----GLPRRWTLRT<br/>gi|7305559|ref|NP_038802.1|__8      LWQMTTLFQ-------EPSPDSLKTFN-----FLPRTWRLES<br/>gi|7305555|ref|NP_038803.1|__9      LWQMTRYPQ-------EPAPYNPMNYN-----FLPMTWRLAS<br/>                                                                     **  * * .</pre>
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Select database</a>
                <div class="accordion-content" data-tab-content>
                    Database of template HMMs against which the query sequence or alignment is compared. The <a href = https://uniclust.mmseqs.com target="_blank" rel="noopener">Uniclust30 database</a> is a version of the Uniclust database clustered down to a maximum pairwise sequence identity of 30%.
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">E-value inclusion threshold</a>
                <div class="accordion-content" data-tab-content>
                    E-value cutoff for inclusion in result alignment.
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Max. number of iterations</a>
                <div class="accordion-content" data-tab-content>
                    This specifies the number of iterations that are performed by HHblits. The iterations stop in any case when no further sequences are found.
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Minimum probability in hit list (> 10%)</a>
                <div class="accordion-content" data-tab-content>
                    Only hits above this cut-off probability will be displayed.
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">No. of target sequences</a>
                <div class="accordion-content" data-tab-content>
                    This parameter controls how many matches will be displayed in the results.
                </div>
            </li>
        </ul>
    </div>
    <div id="tabs-3" class="greenlink">
        @help.citations.toolkit_citation()
        <p>HHblits: lightning-fast iterative protein sequence searching by HMM-HMM alignment.
            Remmert M, Biegert A, Hauser A, Söding J. <a href = http://academic.oup.com/nar/article-lookup/doi/10.1093/nar/gkw348 target="_blank" rel="noopener">Nat Methods. 2011 Dec 25;9(2):173-5</a>.</p>
    </div>
</div>

<script>
        $(".accordion").foundation();
</script>