@()
<div id="tabs">
    <ul>
        <li><a href="#tabs-1">Overview</a></li>
        <li><a href="#tabs-2">Parameters</a></li>
        <li><a href="#tabs-3">References</a></li>
    </ul>
    <div id="tabs-1" class="greenlink">
        <h5><b>HHpred server for protein remote homolog detection and 3D structure prediction</b></h5>
        <p>The primary aim in developing HHpred was to provide biologists with a method for sequence database searching and structure prediction that is as easy to use as BLAST or PSI-BLAST and that is at the same time much more sensitive in finding remote homologs. In fact, HHpred's sensitivity is competitive with the most powerful servers for structure prediction currently available.</p>
        <p>HHpred is the first server that is based on the pairwise comparison of profile hidden Markov models (HMMs). Whereas most conventional sequence search methods search sequence databases such as UniProt or the NR, HHpred searches alignment databases, like Pfam or SMART. This greatly simplifies the list of hits to a number of sequence families instead of a clutter of single sequences. All major publicly available profile and alignment databases are available through HHpred.</p>
        <p>HHpred accepts a single query sequence or a multiple alignment as input. Within only a few minutes it returns the search results in an easy-to-read format similar to that of PSI-BLAST. Search options include local or global alignment and scoring secondary structure similarity. HHpred can produce pairwise query-template sequence alignments, merged query-template multiple alignments (e.g. for transitive searches), as well as 3D structural models calculated by the MODELLER software from HHpred alignments.</p>
        <h5><b>When can HHpred be useful for you?</b></h5>
        <p>It is well known that sequence search methods such as BLAST, FASTA, or PSI-BLAST are of prime importance for biological research because functional information of a protein or gene can be inferred from homologous proteins or genes identified in a sequence search. But quite often no significant relationship to a protein of known function can be established. This is certainly the case for the most interesting group of proteins, those for which no ortholog has yet been studied.</p>
        <p>It is less well known that in cases where conventional sequence search methods fail, the recently developed, highly sensitive methods for homology detection or structure prediction (see descriptions and links at <a href = http://meta.bioinfo.pl/submit_wizard.pl>http://meta.bioinfo.pl/submit_wizard.pl</a>) quite often allow to make inferences from more remotely homologous relationships [<a href = https://www.ncbi.nlm.nih.gov/pubmed/10871397?dopt=Abstract>Venclovas (2000)</a>, <a href = https://www.ncbi.nlm.nih.gov/pubmed/15688435?dopt=Abstract>Zheng (2005)</a>, <a href = https://www.ncbi.nlm.nih.gov/pubmed/14983005?dopt=Abstract>Ginalski (2004)</a>, <a href = https://www.ncbi.nlm.nih.gov/pubmed/15452342?dopt=Abstract>Rand (2004)</a>, <a href = https://www.ncbi.nlm.nih.gov/pubmed/15684412?dopt=Abstract>Pawlak (2005)</a>, <a href = https://www.ncbi.nlm.nih.gov/pubmed/15048836?dopt=Abstract>Kihara (2004)</a>]. If the relationship is so remote that no common function can be assumed, one can generally still derive hypotheses about possible mechanisms, active site positions and residues, or the class of substrate bound [<a href = https://www.ncbi.nlm.nih.gov/pubmed/11286560?dopt=Abstract>Todd (2001)</a>, <a href = https://www.ncbi.nlm.nih.gov/pubmed/10902155?dopt=Abstract>Pawlowski (2000)</a>]. When a homologous protein with known structure can be identified, its structure can be used as a template to model the 3D structure for the protein of interest [<a href = https://www.ncbi.nlm.nih.gov/pubmed/9710568?dopt=Abstract>Rychlewski (1998)</a>], since even remotely homologous proteins generally have quite similar 3D structure [<a href = https://www.ncbi.nlm.nih.gov/pubmed/12127461?dopt=Abstract>Kinch (2002)</a>]. The 3D model may then help to generate hypotheses to guide experiments.</p>
        <h5><b>What is HMM-HMM comparison and why is it so powerful?</b></h5>
        <p>When searching for remote homologs, it is wise to make use of as much information about the query and database proteins as possible in order to better distinguish true from false positives and to produce optimal alignments. This is the reason why sequence-sequence comparison is inferior to profile-sequence comparison. Sequence profiles contain for each column of a multiple alignment the frequencies of the 20 amino acids. They therefore contain detailed information about the conservation of each residue position, i.e. how important each position is for defining other members of the protein family, and about the preferred amino acids. Profile Hidden Markov Models (<b>HMMs</b>) are similar to simple sequence profiles, but in addition to the amino acid frequencies in the columns of a multiple sequence alignment they contain information about the frequency of inserts and deletions at each column. Using profile HMMs in place of simple sequence profiles should therefore further improve sensitivity. HHpred is the first server to employ HMM-HMM comparison, based on a novel statistical method that we have developed recently. Using HMMs both on the query and the database side greatly enhances the sensitivity and selectivity over sequence-profile based methods such as PSI-BLAST.</p>
        <h5><b>Structure prediction with HHpred</b></h5>
        <p>The most successful techniques for protein structure prediction rely on identifying homologous sequences with known structure to be used as template. This works so well because structures diverge much more slowly than sequences and homologous proteins may have very similar structures even when their sequences have diverged beyond recognition. But sensitivity in homology detection is crucial for success since many proteins have only remote relatives in the structure database.</p>
    </div>
    <div id="tabs-2" class="greenlink">
        <ul class="accordion" data-accordion data-allow-all-closed='true'>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Paste single sequence or multiple alignment</a>
                <div class="accordion-content" data-tab-content>
                    <p>Paste the query sequence with which you want to do the search either as raw sequence (e.g. 'ELRLRYCAPAGFALLKCN...') or as FASTA sequence (i.e. with a single line '>sequence_name optional description' preceding the raw sequence, see next subsection). HHpred will either use HHBlits (Uniprot database filtered to 20% ) or PSI-BLAST (non-redundant sequence database from NCBI) to build a multiple alignment (MSA) from the respective databases. This alignment will then be used to construct a concise statistical representation (a hidden Markov model or HMM) and to search the selected HMM databases for homologs.</p>
                    <p>You may also paste a query multiple alignment (for allowed formats see below). The alignment will be used to jumpstart the MSA (multiple sequence alignment) generation method. An expert-built or expert-edited alignment or one that contains homologs not yet contained in the non-redundant sequence database may significantly increase the sensitivity and reliability of the HHpred search. I recommend NOT to use automatic multiple alignment programs such as CLUSTAL, as the MSA generation tools tend to give clearly better alignments for the purpose of detecting homologs. Note: if you paste a multiple alignment please don't forget to select the appropriate input format below.</p>
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Select input format</a>
                <div class="accordion-content" data-tab-content>
                    <b>Entering a single query sequence</b><br/><br/>You can enter a single query sequence in FASTA format (with or without name line). HHsearch/HHpred will build a multiple alignment of homologs from it with PsiBlast. The query sequence in FASTA can look like this:<br/><br/><pre>>gi|532319|pir|TVFV2E|TVFV2E envelope protein<br/>ELRLRYCAPAGFALLKCNDADYDGFKTNCSNVSVVHCTNLMNTTVTTGLLLNGSYSENRT<br/>QIWQKHRTSNDSALILLNKHYNLTVTCKRPGNKTVLPVTIMAGLVFHSQKYNLRLRQAWC<br/>HFPSNWKGAWKEVKEEIVNLPKERYRGTNDPKRIFFQRQWGDPETANLWFNCHGEFFYCK<br/>MDWFLNYLNNLTVDADHNECKNTSGTKSGNKRAPGPCVQR</pre><br/><br/>or this<pre><br/><br/>>gi|532319|pir|TVFV2E|TVFV2E envelope protein<br/>ELRLRYCAPA GFALLKCNDA DYDGFKTNCS NVSVVHCTNL MNTTVTTGLL LNGSYSENRT QIWQKHRTSN <br/>DSALILLNKH YNLTVTCKRP GNKTVLPVTI MAGLVFHSQK YNLRLRQAWC HFPSNWKGAW KEVKEEIVNL <br/>PKERYRGTND PKRIFFQRQW GDPETANLWF NCHGEFFYCK MDWFLNYLNN LTVDADHNEC KNTSGTKSGN <br/>KRAPGPCVQR<br/><br/></pre>or simply this<br/><br/><pre>ELRLRYCAPAGFALLKCNDADYDGFKTNCSNVSVVHCTNLMNTTVTTGLLLNGSYSENRTQIWQKHRTSNDSALILLNKHYNLTVTCKRPGNK<br/>TVLPVTIMAGLVFHSQKYNLRLRQAWCHFPSNWKGAWKEVKEEIVNLPKERYRGTNDPKRIFFQRQWGDPETANLWFNCHGEFFYCKMDWFLNYLNNLTVDADHNE<br/>CKNTSGTKSGNKRAPGPCVQR</pre><br/><br/>The sequence may contain a nameline that must start with a > symbol, followed directly by the sequence name and an optional description. The residue data is contained in one or more lines of arbitrary length following the name line. No empty lines should be used. Upper and lower case letters are allowed. Spaces and gaps (-.) will be removed. <br/><br/><br/><b>Entering a multiple alignment</b><br/><br/>If you already have a query alignment then you can enter it here. You can enter alignments in different formats, which are listed in the dropdown menu. Format descriptions can be found in the help pages of the reformat tool.<br/><br/><br/>Example for aligned FASTA format:<br/><br/><br/></pre>>d1a1x__ b.63.1.1 (-) p13-MTCP1 {Human (Homo sapiens)}<br/>PPDHLWVHQEGIYRDEYQRTWVAVVEE--E--T--SF---------LR----------ARVQQIQVPLG-------DAARPSHLLTS-----QLPLMWQLYP<br/>>gi|6678257|ref|NP_033363.1|:(7-103) T-cell lymphoma breakpoint 1 [Mus musculus]<br/>HPNRLWIWEKHVYLDEFRRSWLPVVIK--S--N--EK---------FQ----------VILRQEDVTLG-------EAMSPSQLVPY-----ELPLMWQLYP<br/>>gi|7305557|ref|NP_038800.1|:(8-103) T-cell leukemia/lymphoma 1B, 3 [Mus musculus]<br/>PPRFLVCTRDDIYEDENGRQWVVAKVE--T--S--RSpygsrietcIT----------VHLQHMTTIPQ-------EPTPQQPINNN-----SLPTMWRLES<br/>>gi|11415028|ref|NP_068801.1|:(2-106) T-cell lymphoma-1; T-cell lymphoma-1A [Homo sapiens]<br/>HPDRLWAWEKFVYLDEKQHAWLPLTIEikD--R--LQ---------LR----------VLLRREDVVLG-------RPMTPTQIGPS-----LLPIMWQLYP<br/>>gi|7305561|ref|NP_038804.1|:(7-103) T-cell leukemia/lymphoma 1B, 5 [Mus musculus]<br/>----------GIYEDEHHRVWIAVNVE--T--S--HS---------SHgnrietcvt-VHLQHMTTLPQ-------EPTPQQPINNN-----SLPTMWRLES<br/>>gi|7305553|ref|NP_038801.1|:(5-103) T-cell leukemia/lymphoma 1B, 1 [Mus musculus]<br/>LPVYLVSVRLGIYEDEHHRVWIVANVE--TshS--SH---------GN----------RRRTHVTVHLW-------KLIPQQVIPFNplnydFLPTTWKLES<br/>>gi|27668591|ref|XP_234504.1|:(7-103) similar to Chain A, Crystal Structure Of Murine Tcl1<br/>-PDRLWLWEKHVYLDEFRRSWLPIVIK--S--N--GK---------FQ----------VIMRQKDVILG-------DSMTPSQLVPY-----ELPLMWQLYP<br/>>gi|27668589|ref|XP_234503.1|:(9-91) similar to T-cell leukemia/lymphoma 1B, 5;<br/>-PHILTLRTHGIYEDEHHRLWVVLDLQ--A--ShlSF---------SN----------RLLIYLTVYLQqgvafplESTPPSPMNLN-----GLPRRWTLRT<br/>>gi|7305559|ref|NP_038802.1|:(8-102) T-cell leukemia/lymphoma 1B, 4 [Mus musculus] <br/>PPCFLVCTRDDIYEDEHGRQWVAAKVE--T--S--SH---------SPycskietcvtVHLWQMTTLFQ-------EPSPDSLKTFN-----FLPRTWRLES<br/>>gi|7305555|ref|NP_038803.1|:(9-102) T-cell leukemia/lymphoma 1B, 2 [Mus musculus]<br/>---------PGFYEDEHHRLWMVAKLE--T--C--SH---------SPycnkietcvtVHLWQMTRYPQ-------EPAPYNPMNYN-----FLPMTWRLAS</pre><br/><br/>In aligned FASTA the gaps are written with '-' and the n'th letter of each sequence (except newlines) is understood to build the n'th column of the multiple alignment. The sequence residues may be broken up into several lines containing arbitrary numbers of residues.<br/><br/><br/><b>The same alignment in CLUSTAL format looks like this:</b><br/><br/><br/><pre>CLUSTAL X (1.83) multiple sequence alignment<br/><br/><br/>d1a1x__                             PPDHLWVHQEGIYRDEYQRTWVAVVEE--E--T--SF---------LR----------AR<br/>gi|6678257|ref|NP_033363.1|__7      HPNRLWIWEKHVYLDEFRRSWLPVVIK--S--N--EK---------FQ----------VI<br/>gi|7305557|ref|NP_038800.1|__8      PPRFLVCTRDDIYEDENGRQWVVAKVE--T--S--RSPYGSRIETCIT----------VH<br/>gi|11415028|ref|NP_068801.1|__      HPDRLWAWEKFVYLDEKQHAWLPLTIEIKD--R--LQ---------LR----------VL<br/>gi|7305561|ref|NP_038804.1|__7      ----------GIYEDEHHRVWIAVNVE--T--S--HS---------SHGNRIETCVT-VH<br/>gi|7305553|ref|NP_038801.1|__5      LPVYLVSVRLGIYEDEHHRVWIVANVE--TSHS--SH---------GN----------RR<br/>gi|27668591|ref|XP_234504.1|__      -PDRLWLWEKHVYLDEFRRSWLPIVIK--S--N--GK---------FQ----------VI<br/>gi|27668589|ref|XP_234503.1|__      -PHILTLRTHGIYEDEHHRLWVVLDLQ--A--SHLSF---------SN----------RL<br/>gi|7305559|ref|NP_038802.1|__8      PPCFLVCTRDDIYEDEHGRQWVAAKVE--T--S--SH---------SPYCSKIETCVTVH<br/>gi|7305555|ref|NP_038803.1|__9      ---------PGFYEDEHHRLWMVAKLE--T--C--SH---------SPYCNKIETCVTVH<br/>                                               .* **  : *:    :                       <br/><br/>d1a1x__                             VQQIQVPLG-------DAARPSHLLTS-----QLPLMWQLYP<br/>gi|6678257|ref|NP_033363.1|__7      LRQEDVTLG-------EAMSPSQLVPY-----ELPLMWQLYP<br/>gi|7305557|ref|NP_038800.1|__8      LQHMTTIPQ-------EPTPQQPINNN-----SLPTMWRLES<br/>gi|11415028|ref|NP_068801.1|__      LRREDVVLG-------RPMTPTQIGPS-----LLPIMWQLYP<br/>gi|7305561|ref|NP_038804.1|__7      LQHMTTLPQ-------EPTPQQPINNN-----SLPTMWRLES<br/>gi|7305553|ref|NP_038801.1|__5      RTHVTVHLW-------KLIPQQVIPFNPLNYDFLPTTWKLES<br/>gi|27668591|ref|XP_234504.1|__      MRQKDVILG-------DSMTPSQLVPY-----ELPLMWQLYP<br/>gi|27668589|ref|XP_234503.1|__      LIYLTVYLQQGVAFPLESTPPSPMNLN-----GLPRRWTLRT<br/>gi|7305559|ref|NP_038802.1|__8      LWQMTTLFQ-------EPSPDSLKTFN-----FLPRTWRLES<br/>gi|7305555|ref|NP_038803.1|__9      LWQMTRYPQ-------EPAPYNPMNYN-----FLPMTWRLAS<br/>                                                                     **  * * .</pre><br/><br/><br/>The first line must begin with CLUSTAL, the rest of the line is ignored. The sequences come in blocks with the sequence name, cut to a constant number of characters, followed by some whitespace and the sequence residues and gaps (-). The number of alignment columns per sequence line is arbitrary but constant within a block. Thus the whole alignment may be written in one long block. At the end of each block is an optional line indicating the conservation of each alignment column, followed by an obligatory empty line that separates the block from the next. Capital and lower case characters may be used for the amino acids.<br/><br/><br/><b>The same alignment in Stockholm format is here:</b><br/><br/><br/><pre># STOCKHOLM 1.0<br/><br/>#=GC RF                          XXXXXXXXXXXXXXXXXXXXXXXXXXX--X--X--XX---------XX----------XXXXXXXXXXX<br/>d1a1x__                          PPDHLWVHQEGIYRDEYQRTWVAVVEE--E--T--SF---------LR----------ARVQQIQVPLG<br/>gi|6678257|ref|NP_033363.1|:(7-1 HPNRLWIWEKHVYLDEFRRSWLPVVIK--S--N--EK---------FQ----------VILRQEDVTLG<br/>gi|7305557|ref|NP_038800.1|:(8-1 PPRFLVCTRDDIYEDENGRQWVVAKVE--T--S--RSpygsrietcIT----------VHLQHMTTIPQ<br/>gi|11415028|ref|NP_068801.1|:(2- HPDRLWAWEKFVYLDEKQHAWLPLTIEikD--R--LQ---------LR----------VLLRREDVVLG<br/>gi|7305561|ref|NP_038804.1|:(7-1 ----------GIYEDEHHRVWIAVNVE--T--S--HS---------SHgnrietcvt-VHLQHMTTLPQ<br/>gi|7305553|ref|NP_038801.1|:(5-1 LPVYLVSVRLGIYEDEHHRVWIVANVE--TshS--SH---------GN----------RRRTHVTVHLW<br/>gi|27668591|ref|XP_234504.1|:(7- -PDRLWLWEKHVYLDEFRRSWLPIVIK--S--N--GK---------FQ----------VIMRQKDVILG<br/>gi|27668589|ref|XP_234503.1|:(9- -PHILTLRTHGIYEDEHHRLWVVLDLQ--A--ShlSF---------SN----------RLLIYLTVYLQ<br/>gi|7305559|ref|NP_038802.1|:(8-1 PPCFLVCTRDDIYEDEHGRQWVAAKVE--T--S--SH---------SPycskietcvtVHLWQMTTLFQ<br/>gi|7305555|ref|NP_038803.1|:(9-1 ---------PGFYEDEHHRLWMVAKLE--T--C--SH---------SPycnkietcvtVHLWQMTRYPQ<br/><br/>#=GC RF                          -------XXXXXXXXXXX-----XXXXXXXXXX<br/>d1a1x__                          -------DAARPSHLLTS-----QLPLMWQLYP<br/>gi|6678257|ref|NP_033363.1|:(7-1 -------EAMSPSQLVPY-----ELPLMWQLYP<br/>gi|7305557|ref|NP_038800.1|:(8-1 -------EPTPQQPINNN-----SLPTMWRLES<br/>gi|11415028|ref|NP_068801.1|:(2- -------RPMTPTQIGPS-----LLPIMWQLYP<br/>gi|7305561|ref|NP_038804.1|:(7-1 -------EPTPQQPINNN-----SLPTMWRLES<br/>gi|7305553|ref|NP_038801.1|:(5-1 -------KLIPQQVIPFNplnydFLPTTWKLES<br/>gi|27668591|ref|XP_234504.1|:(7- -------DSMTPSQLVPY-----ELPLMWQLYP<br/>gi|27668589|ref|XP_234503.1|:(9- qgvafplESTPPSPMNLN-----GLPRRWTLRT<br/>gi|7305559|ref|NP_038802.1|:(8-1 -------EPSPDSLKTFN-----FLPRTWRLES<br/>gi|7305555|ref|NP_038803.1|:(9-1 -------EPAPYNPMNYN-----FLPMTWRLAS<br/>//</pre><br/><br/><br/>Stockholm format is similar to Clustal format, but offers a number of markups to include annotation and additional information referring to the alignment, its columns, or sequences. It was developed by Sean Eddy for Pfam and the HMMer software package. Each alignment may be separated into an arbitrary number of blocks with an arbitrary number of alignment columns contained per line. Gaps may be written '-' or '.'. The line starting with #=GC RF is an example of an optional markup line that contains one character per alignment columns. In this case it indicates which are the match/delete columns (as opposed to insert columns).<br/><br/><br/><b>The same alignment in A2M format looks like this:</b><br/><br/><br/><pre>>d1a1x__ b.63.1.1 (-) p13-MTCP1 {Human (Homo sapiens)}<br/>PPDHLWVHQEGIYRDEYQRTWVAVVEE..E..T..SF.........LR..........ARVQQIQVPLG.......DAARPSHLLTS.....QLPLMWQLYP<br/>>gi|6678257|ref|NP_033363.1|:(7-103) T-cell lymphoma breakpoint 1 [Mus musculus]<br/>HPNRLWIWEKHVYLDEFRRSWLPVVIK..S..N..EK.........FQ..........VILRQEDVTLG.......EAMSPSQLVPY.....ELPLMWQLYP<br/>>gi|7305557|ref|NP_038800.1|:(8-103) T-cell leukemia/lymphoma 1B, 3 [Mus musculus]<br/>PPRFLVCTRDDIYEDENGRQWVVAKVE..T..S..RSpygsrietcIT..........VHLQHMTTIPQ.......EPTPQQPINNN.....SLPTMWRLES<br/>>gi|11415028|ref|NP_068801.1|:(2-106) T-cell lymphoma-1; T-cell lymphoma-1A [Homo sapiens]<br/>HPDRLWAWEKFVYLDEKQHAWLPLTIEikD..R..LQ.........LR..........VLLRREDVVLG.......RPMTPTQIGPS.....LLPIMWQLYP<br/>>gi|7305561|ref|NP_038804.1|:(7-103) T-cell leukemia/lymphoma 1B, 5 [Mus musculus]<br/>----------GIYEDEHHRVWIAVNVE..T..S..HS.........SHgnrietcvt.VHLQHMTTLPQ.......EPTPQQPINNN.....SLPTMWRLES<br/>>gi|7305553|ref|NP_038801.1|:(5-103) T-cell leukemia/lymphoma 1B, 1 [Mus musculus]<br/>LPVYLVSVRLGIYEDEHHRVWIVANVE..TshS..SH.........GN..........RRRTHVTVHLW.......KLIPQQVIPFNplnydFLPTTWKLES<br/>>gi|27668591|ref|XP_234504.1|:(7-103) similar to Chain A, Crystal Structure Of Murine Tcl1<br/>-PDRLWLWEKHVYLDEFRRSWLPIVIK..S..N..GK.........FQ..........VIMRQKDVILG.......DSMTPSQLVPY.....ELPLMWQLYP<br/>>gi|27668589|ref|XP_234503.1|:(9-91) similar to T-cell leukemia/lymphoma 1B, 5;<br/>-PHILTLRTHGIYEDEHHRLWVVLDLQ..A..ShlSF.........SN..........RLLIYLTVYLQqgvafplESTPPSPMNLN.....GLPRRWTLRT<br/>>gi|7305559|ref|NP_038802.1|:(8-102) T-cell leukemia/lymphoma 1B, 4 [Mus musculus] <br/>PPCFLVCTRDDIYEDEHGRQWVAAKVE..T..S..SH.........SPycskietcvtVHLWQMTTLFQ.......EPSPDSLKTFN.....FLPRTWRLES<br/>>gi|7305555|ref|NP_038803.1|:(9-102) T-cell leukemia/lymphoma 1B, 2 [Mus musculus]<br/>---------PGFYEDEHHRLWMVAKLE..T..C..SH.........SPycnkietcvtVHLWQMTRYPQ.......EPAPYNPMNYN.....FLPMTWRLAS</pre><br/><br/><br/>A2M format is derived from aligned FASTA format. It looks very similar, but it distinguishes between match/delete columns and insert columns. This information is used to specify how an alignment is transformed into an HMM. The match/delete columns use upper case letters for residues and the '-' symbol for deletions (gaps). The insert columns use lower case letters for the inserted residues. Gaps aligned to inserted residues are written as '.'<br/><br/><br/><b>The same alignment in A3M is here:</b><br/><br/><br/><pre>>d1a1x__ b.63.1.1 (-) p13-MTCP1 {Human (Homo sapiens)}<br/>PPDHLWVHQEGIYRDEYQRTWVAVVEEETSFLRARVQQIQVPLGDAARPSHLLTSQLPLMWQLYP<br/>>gi|6678257|ref|NP_033363.1|:(7-103) T-cell lymphoma breakpoint 1 [Mus musculus]<br/>HPNRLWIWEKHVYLDEFRRSWLPVVIKSNEKFQVILRQEDVTLGEAMSPSQLVPYELPLMWQLYP<br/>>gi|7305557|ref|NP_038800.1|:(8-103) T-cell leukemia/lymphoma 1B, 3 [Mus musculus]<br/>PPRFLVCTRDDIYEDENGRQWVVAKVETSRSpygsrietcITVHLQHMTTIPQEPTPQQPINNNSLPTMWRLES<br/>>gi|11415028|ref|NP_068801.1|:(2-106) T-cell lymphoma-1; T-cell lymphoma-1A [Homo sapiens]<br/>HPDRLWAWEKFVYLDEKQHAWLPLTIEikDRLQLRVLLRREDVVLGRPMTPTQIGPSLLPIMWQLYP<br/>>gi|7305561|ref|NP_038804.1|:(7-103) T-cell leukemia/lymphoma 1B, 5 [Mus musculus]<br/>----------GIYEDEHHRVWIAVNVETSHSSHgnrietcvtVHLQHMTTLPQEPTPQQPINNNSLPTMWRLES<br/>>gi|7305553|ref|NP_038801.1|:(5-103) T-cell leukemia/lymphoma 1B, 1 [Mus musculus]<br/>LPVYLVSVRLGIYEDEHHRVWIVANVETshSSHGNRRRTHVTVHLWKLIPQQVIPFNplnydFLPTTWKLES<br/>>gi|27668591|ref|XP_234504.1|:(7-103) similar to Chain A, Crystal Structure Of Murine Tcl1<br/>-PDRLWLWEKHVYLDEFRRSWLPIVIKSNGKFQVIMRQKDVILGDSMTPSQLVPYELPLMWQLYP<br/>>gi|27668589|ref|XP_234503.1|:(9-91) similar to T-cell leukemia/lymphoma 1B, 5;<br/>-PHILTLRTHGIYEDEHHRLWVVLDLQAShlSFSNRLLIYLTVYLQqgvafplESTPPSPMNLNGLPRRWTLRT<br/>>gi|7305559|ref|NP_038802.1|:(8-102) T-cell leukemia/lymphoma 1B, 4 [Mus musculus] <br/>PPCFLVCTRDDIYEDEHGRQWVAAKVETSSHSPycskietcvtVHLWQMTTLFQEPSPDSLKTFNFLPRTWRLES<br/>>gi|7305555|ref|NP_038803.1|:(9-102) T-cell leukemia/lymphoma 1B, 2 [Mus musculus]<br/>---------PGFYEDEHHRLWMVAKLETCSHSPycnkietcvtVHLWQMTRYPQEPAPYNPMNYNFLPMTWRLAS</pre><br/><br/><br/>The A3M format is a condensed version of A2M format. It is obtained by omitting all '.' symbols from A2M format. Hence residues emitted by Match states of the HMM are in upper case, residues emitted by Insert states are in lower case and deletions are written '-'. A3M-formatted alignments are not really human readable but they can be reformatted to other more readable formats like FASTA, A2M CLUSTAL and many others with the reformatting utility in our toolbox. A3M is not an official format. Likewise, the name A3M is our personal invention for this very practical and space-efficient format which we use to store the HHsearch alignments internally.",
                    "Select HMM databases ": "Select the database(s) of template HMMs against which you want to compare the query HMM. Hold down the Control key to select more than one database.<br/><br/><b>Proteomes</b><br/>Here you can select databases of HMMs representing each of the proteins of an organism. Searching through the proteome of an organism using HHsearch (or HHblits) can be useful when you suspect a diverged ortholog or homolog of your query protein to hide in an organism's genome (and you cannot find it using BLAST or PSI-BLAST).<br/><br/><a href = http://www.rcsb.org/pdb/home/home.do><b>pdb70</b></a><br/><br/>A search against the our pdb70 database is the first step when trying to predict a structure with HHpred. It is based on the protein data bank (PDB) which contains all publicly available 3D structures of proteins. For each protein chain in the PDB we build a multiple alignment with iterated PSI-BLAST searches and transform these alignments into HMMs. HHpred can compare the query HMM to the database of HMMs based on PDB chains and generate query-template alignments or it can build structural models for the query sequence. We use a filtered version with a maximum sequence identity of 70% [J. Söding, unpublished]. We check the PDB for new structures every night and update our database whenever new files are found.<br/><font size = 1><a href = https://www.ncbi.nlm.nih.gov/pubmed/14681399?dopt=Abstract>Bourne PE, Addess KJ, Bluhm WF, Chen L, Deshpande N, Feng Z, Fleri W, Green R, Merino-Ott JC, Townsend-Merino W, Weissig H, Westbrook J, Berman HM. The distribution and query systems of the RCSB Protein Data Bank. NAR 2004, 32:D223.</a></font><br/><br/><a href = http://scop.berkeley.edu/><b>scop70</b></a><br/><br/>The scop70 alignment database is built from sequences in SCOP in a similar way as pdb70. SCOP is a hand-curated database with protein domains of known structure. It is hierarchically organized into families, superfamilies, folds, and classes to reflect evolutionary relationships. Only members of the same superfamily can be savely assumed to be homologous. Members of the same fold but different superfamily are assumed not to be homologous, although many exceptions to this rule are now known. HHpred reports the SCOP code of the family the domain belongs to. For example, a.118.8.1 stands for domain class a (all alpha helical), fold 118 (alpha-alpha superhelices), superfamily 8 (TPR-like), family 1 (Tetratricopeptide repeats (TPR)). We use a filtered version of the SCOP sequence set from the <a href = http://scop.berkeley.edu/astral/><b>ASTRAL server</b></a>, e.g. scop70_1.69 is version 1.69 of SCOP filtered to 70% maximum sequence identity. Our scop70 alignment database is similar to the <a href = http://supfam.mrc-lmb.cam.ac.uk/SUPERFAMILY/><b>SUPERFAMILY</b></a> database. SCOP is updated irregularly, approximately every twelve months.<br/><font size = 1><a href = https://www.ncbi.nlm.nih.gov/pubmed/10592240?dopt=Abstract>Lo Conte L, Ailey B, Hubbard TJ, Brenner SE, Murzin AG, Chothia C. SCOP: a structural classification of proteins database. NAR 2000, 28:257</a></font><br/><br/><a href = https://www.ncbi.nlm.nih.gov/Structure/cdd/cdd.shtml><b>CDD</b></a><br/><br/>The Conserved Domain Database (CDD) at the <a href = https://www.ncbi.nlm.nih.gov/><b>NCBI</b></a> is a composite database combining the alignments from Pfam, SMART, and COG/KOG, as well as containing alignments of conserved domains curated at the NCBI. The source of the alignments as shown in the HHpred results is part of the identifiers. Identifiers beginning with cd or LOAD indicate that an alignment was curated at the NCBI. The CDD database contained 18640 alignments on November 17, 2004. It is updated irregularly every 6 to 12 months. HHpred checks once a week for updates.<br/><font size = 1><a href = https://www.ncbi.nlm.nih.gov/pubmed/11752315?dopt=Abstract>Marchler-Bauer A, Panchenko AR, Shoemaker BA, Thiessen PA, Geer LY, Bryant SH. CDD: a database of conserved domain alignments with links to domain three-dimensional structure. NAR 2002, 30:281.</a></font><br/><br/><a href = http://www.ebi.ac.uk/interpro/><b>InterPro</b></a><br/><br/>Our version of InterPro contains HMMs from Pfam, SMART, PANTHER, TIGRFAMs, PIRSF, SUPERFAMILY, and Gene3D/CATH (see below). It encodes full-length protein families, domain families, and motifs in which identifiable features from known proteins can be applied to unknown protein sequences. In the version offered through HHpred, the PRINTS and PROSITE sequene motif databases are excluded, as are explicit transmembrane helix prediction by TMHMM and signal peptide prediction by SignalP. InterPro_12.0 contains already more than 31000 HMMs, therefore a search will take a few minutes. (InterPro offers only HMMs and not the underlying alignments. Therefore, HHpred does not have access to alignments for PANTHER, TIGRFAMs, PIRSF, SUPERFAMILY, and CATH/Gene3D. Hence, when using the option \"Merge query and template alignments\", only the single consensus sequences instead of full template alignments can be merged with your query alignment.)<br/><font size = 1><a href = https://www.ncbi.nlm.nih.gov/pubmed/15608177?dopt=Abstract>Mulder NJ, Apweiler R, Attwood TK, Bairoch A, Bateman A, Binns D, Bradley P, Bork P, Bucher P, Cerutti L, Copley R, Courcelle E, Das U, Durbin R, Fleischmann W, Gough J, Haft D, Harte N, Hulo N, Kahn D, Kanapin A, Krestyaninova M, Lonsdale D, Lopez R, Letunic I, Madera M, Maslen J, McDowall J, Mitchell A, Nikolskaya AN, Orchard S, Pagni M, Ponting CP, Quevillon E, Selengut J, Sigrist CJ, Silventoinen V, Studholme DJ, Vaughan R, Wu CH. InterPro, progress and status in 2005. Nucleic Acids Res. 2005 Jan 1;33(Database issue):D201-5.</a></font><br/><br/><a href = http://pfam.xfam.org/><b>PfamA</b></a><br/><br/>PfamA is a large collection of curated multiple sequence alignments covering many common protein domains and families. Many PfamA families are well documented with links to literature, and about a third of them have at least one member with known structure. 74% of protein sequences have at least one match to PfamA_16.0 with its 7503 alignments (August 2004). PfamA gets updated approximately every two to six months. We automatically download two versions of the PfamA database: one directly from <a href = ftp://ftp.ebi.ac.uk/pub/databases/Pfam/>ftp://ftp.ebi.ac.uk/pub/databases/Pfam/</a> and one as a part of the CDD database from NCBI. The version directly from Pfam has a version number in its name whereas the version from NCBI carries the date when it was downloaded. The original Pfam is generally more up to date and contains more sequences in each alignment than the version from NCBI (for some unkown reason).<br/><font size = 1><a href = https://www.ncbi.nlm.nih.gov/pubmed/9399864?dopt=Abstract>Sonnhammer EL, Eddy SR, Birney E, Bateman A, Durbin R. Pfam: multiple sequence alignments and HMM-profiles of protein domains. NAR 1998 26:320.</a></font><br/><br/><a href = http://smart.embl-heidelberg.de/><b>SMART</b></a><br/><br/>SMART is a database of curated alignments of genetically mobile domains found in signalling, extracellular and chromatin-associated proteins, predominantly in metazoans. These domains are extensively annotated with respect to phyletic distributions, functional class, tertiary structures and functionally important residues. In January 2004 SMART contained 685 alignments. We download the SMART database as a part of the CDD database from NCBI and check for updates every week. Note that the version at NCBI may get updated only be a few weeks after a new SMART version has been released.<br/><font size = 1><a href = https://www.ncbi.nlm.nih.gov/pubmed/9847187?dopt=Abstract>Ponting CP, Schultz J, Milpetz F, Bork P. SMART: identification and annotation of domains from signalling and extracellular protein sequences. NAR 1999, 27:229.</a></font><br/><br/><a href = http://www.pantherdb.org/about.jsp><b>PANTHER</b></a><br/><br/>PANTHER is a large collection of protein families that have been subdivided into functionally related subfamilies, using human expertise. These subfamilies model the divergence of specific functions within protein families, allowing more accurate association with function (human-curated molecular function and biological process classifications and pathway diagrams), as well as inference of amino acids important for functional specificity. Hidden Markov models (HMMs) are built for each family and subfamily for classifying additional protein sequences. Only family alignments are included in HHpred.<br/><font size = 1><a href = https://www.ncbi.nlm.nih.gov/pubmed/15608197?dopt=Abstract>Mi H, Lazareva-Ulitsky B, Loo R, Kejariwal A, Vandergriff J, Rabkin S, Guo N, Muruganujan A, Doremieux O, Campbell MJ, Kitano H, Thomas PD. (2005) The PANTHER database of protein families, subfamilies, functions and pathways. NAR 33:D284-288.</a></font><br/><br/><a href = http://www.jcvi.org/cgi-bin/tigrfams/index.cgi><b>TIGRFAMs</b></a><br/><br/>TIGRFAMs is a collection of protein families, featuring curated multiple sequence alignments, hidden Markov models (HMMs) and annotation, which provides a tool for identifying functionally related proteins based on sequence homology. Those entries which are \"equivalogs\" group homologous proteins which are conserved with respect to function.<br/><font size = 1><a href = https://www.ncbi.nlm.nih.gov/pubmed/12520025?dopt=Abstract>Haft DH, Selengut JD, White O. (2003) The TIGRFAMs database of protein families. NAR 31:371-373.</a></font><br/><br/><a href = http://pir.georgetown.edu/pirwww/dbinfo/iproclass.shtml><b>PIRSF</b></a><br/><br/>The PIRSF protein classification system is a network with multiple levels of sequence diversity from superfamilies to subfamilies that reflects the evolutionary relationship of full-length proteins and domains. The primary PIRSF classification unit is the homeomorphic family, whose members are both homologous (evolved from a common ancestor) and homeomorphic (sharing full-length sequence similarity and a common domain architecture).<br/><font size = 1><a href = https://www.ncbi.nlm.nih.gov/pubmed/14681371?dopt=Abstract>Wu CH, Nikolskaya A, Huang H, Yeh LS, Natale DA, Vinayaka CR, Hu ZZ, Mazumder R, Kumar S, Kourtesis P, Ledley RS, Suzek BE, Arminski L, Chen Y, Zhang J, Cardenas JL, Chung S, Castro-Alvear J, Dinkov G, Barker WC. (2004) PIRSF: family classification system at the Protein Information Resource. NAR 32:D112-114.</a></font><br/><br/><a href = http://supfam.mrc-lmb.cam.ac.uk/SUPERFAMILY/><b>SUPERFAMILY</b></a><br/><br/>SUPERFAMILY is a library of profile hidden Markov models that represent all proteins of known structure. The library is based on the SCOP classification of proteins: each model corresponds to a SCOP domain and aims to represent the entire SCOP superfamily that the domain belongs to. SUPERFAMILY has been used to carry out structural assignments to all completely sequenced genomes.<br/><font size = 1><a href = https://www.ncbi.nlm.nih.gov/pubmed/11697912?dopt=Abstract>Gough J, Karplus K, Hughey R, Chothia C. (2001) Assignment of Homology to Genome Sequences using a Library of Hidden Markov Models that Represent all Proteins of Known Structure. JMB 313:903-919.</a></font><br/><br/><a href = http://www.cathdb.info/Gene3D/><b>CATH/Gene3D</b></a><br/><br/>Gene3D is an alignment database built from sequences of the CATH database. CATH is a database for structural classification of protein domains, similar to SCOP. Gene3D describes protein families and domain architectures in complete genomes. Protein families are formed using a Markov clustering algorithm, followed by multi-linkage clustering according to sequence identity. Mapping of predicted structure and sequence domains is undertaken using hidden Markov models libraries representing CATH and Pfam domains. Functional annotation is provided to proteins from multiple resources. Functional prediction and analysis of domain architectures is available from the Gene3D website.<br/><font size = 1><a href = https://www.ncbi.nlm.nih.gov/pubmed/15608188?dopt=Abstract>Pearl F, Todd A, Sillitoe I, Dibley M, Redfern O, Lewis T, Bennett C, Marsden R, Grant A, Lee D, Akpor A, Maibaum M, Harrison A, Dallman T, Reeves G, Diboun I, Addou S, Lise S, Johnston C, Sillero A, Thornton J, Orengo C. (2005) The CATH Domain Structure Database and related resources Gene3D and DHS provide comprehensive domain family information for genome analysis. NAR 33:D247-D251</a></font><br/><br/><a href = https://www.ncbi.nlm.nih.gov/COG/><b>COG/KOG</b></a><br/><br/>COG/KOG (Clusters of Orthologous Groups of proteins) is a database of alignments of homologous, conserved domains. For the 2003 update, sequences from over 60 completeley sequences prokaryotes (COGs) and 7 completely sequenced euKaryotes (KOGs) were automatically divided into domains. The domain sequences are clustered in such a way that the alignments consist mainly of orthologous proteins from different lineages and paralogous proteins from the same lineage. The COG database is therefore particularly useful for functional annotation. But beware: HHpred is very sensitive in detecting distant homologous relationships. Therefore, a significant match with a COG only implies homology, but not necessarily a similar function.<br/><font size = 1><a href = https://www.ncbi.nlm.nih.gov/pubmed/12969510?dopt=Abstract>Tatusov RL, Fedorova ND, Jackson JD, Jacobs AR, Kiryutin B, Koonin EV, Krylov DM, Mazumder R, Mekhedov SL, Nikolskaya AN, Rao BS, Smirnov S, Sverdlov AV, Vasudevan S, Wolf YI, Yin JJ, Natale DA. The COG database: an updated version includes eukaryotes. BMC Bioinformatics. 2003, 4:41.</a></font><br/><br/><a href = http://pfam.xfam.org/><b>PfamB</b></a><br/><br/>Please note: PfamB contains over 14000 unannotated, mainly small and insignificant alignments. To avoid wasting time, we recommend to search PfamB only for doing intermediate HMM searching, when you can't find any significant match to an alignment from the other databases. PfamB is derived from the ProDom database by leaving out all parts of alignments which are already covered by PfamA. ProDom in turn is built by clustering homologous sequence segments into alignments through an iterative scheme employing PSI-BLAST. PfamB alignments are not annotated, many contain only a few sequences or are quite short, and most represent only parts of domains. To keep the number of HMMs in PfamB reasonable, we include only alignments with at least three sequences of a certain minimum diversity.
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">MSA generation method</a>
                <div class="accordion-content" data-tab-content>
                    This option sets the Multiple Sequence Alignment Generation method used by HHpred. Detailed information on the methods can be found here:<br/>HHBlits : <a href = http://wwwuser.gwdg.de/~compbiol/pdf/Preprint-2011-HHblits-inkl-Suppl2.pdf>Download pdf file</a><br/>PSI-Blast : <a href = https://www.ncbi.nlm.nih.gov/pubmed/9254694>View Article</a>
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Max. MSA generation iterations</a>
                <div class="accordion-content" data-tab-content>
                    This specifies the number of iterations performed by the MSA generation algorithm, to build up an alignment from the input sequence or input alignment. The iterations stop in any case when no further sequences are found. If you want to use exactly your input alignment choose 0 here. This will ensure that only predicted secondary structure is added to your input alignment.
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Store secondary structure</a>
                <div class="accordion-content" data-tab-content>
                    If you choose "yes", the secondary structure of the query will be predicted with <a href = https://www.ncbi.nlm.nih.gov/pubmed/10493868?dopt=Abstract>PSIPRED</a> and compared to the actual secondary structure of the database sequences. If the actual structure of the database sequence is unknown, the PSIPRED-predicted secondary structure will be used. To force HHpred to use predicted instead of actual secondary structure, choose "predicted vs predicted only". Scoring the secondary structure similarity improves sensitivity and also alignment quality significantly. Since the weight of the secondary structure score is small (15% of the amino acid similarity score), the risk for high-scoring non-homologous matches is still small.
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Alignment mode</a>
                <div class="accordion-content" data-tab-content>
                    <p>In local mode, HHpred will use <u>local</u> alignments to score your query sequence with the database sequences. In global mode it uses global alignments with no end gap penalties (sometimes called 'global-local' alignments), i.e. the query-template alignments extend up to the beginning and end of the query or template sequence. The use of local mode is strongly recommended if you want to know which are your most likely remote homologs. If you have already identified your template with a previous run in local mode, you can use the global mode to get global alignments with an already identified template.</p>
                    <p>HHpred was shown to be significantly more sensitive to detect remote homologs in local than in global mode. The reason is that most remote homologs are really only homologous in a common core that define their superfamily. This core has been adorned by other secondary structure elements in different ways in different families. Also, the statistical evaluation is more error-prone for the global alignment mode, since the negatives score distribution for global alignments does not follow an extreme-value distribution.</p>
                    <p>You may generate global alignments and at the same time score alignments with the more sensitive local Viterbi alignment by selecting the 'local' option in combination with the option 'Realign with MAC algorithm' and setting the 'MAC realignment threshold' to 0.0 (hidden under 'More options').</p>
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Realign with MAC algorithm</a>
                <div class="accordion-content" data-tab-content>
                    After the search with the standard Viterbi algorithm, this option will cause HHpred to realign all query-template alignments with the more accurate Maximum ACcuracy alignment algorithm.<br/>Since for technical reasons the scores, E-values, and probabilities from the Viterbi alignment and NOT the MAC alignment are displayed and used for ranking, the MAC realignment option may lead to high-scoring alignments of length 1. This happens when the MAC algorithm cannot find any acceptable alignment path, which is an indication for a non-homologous relationship.<br/>When the MAC realignment option is activated, the 'global' and 'local' options now refer to both the Viterbi search stage as well as the MAC realignment stage. With -global (-local), the posterior probability matrix will be calculated for global (local) alignment.
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">E-value threshold for MSA generation</a>
                <div class="accordion-content" data-tab-content>
                    All sequence matches ("hits") with an E-value better than this threshold are included in the alignment for the next MSA Generation iteration or, in the last iteration, for building the HMM for HHsearch.
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Min. coverage of MSA hits</a>
                <div class="accordion-content" data-tab-content>
                    This parameter controls what percentage of the query residues needs to be covered by the hits of the MSA algorithm in order for the hit to be included into the alignment for the next iteration. More precisely, the coverage of a sequence found by the MSA algorithm is defined as the number of residues aligned with the query divided by the length of the query.
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Min. sequence identity of MSA hits with query</a>
                <div class="accordion-content" data-tab-content>
                    This parameter controls what percentage of residues from a hit of the MSA algorithm needs to be identical to their aligned query residues in order for the hit to be included in the MSA alignment. More precisely, the sequence identity of a hit by the MSA algorithm is defined as the number of residues identical with aligned query residues divided by the length of the MSA hit.
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">MSA realignment threshold</a>
                <div class="accordion-content" data-tab-content>
                    Only active when the 'Realign with MAC algorithm' and the 'local' option is active. This parameter controls the MAC alignment algorithm's greediness. More precisely, the MAC algorithm finds the alignment that maximizes the sum of posterior probabilities minus this threshold for each aligned pair. Global alignments are generated with 0, whereas 0.5 will produce quite conservative local alignments with average probabilities for each aligned pair of residues above 50%. Default value is 0.3, which produces alignments of roughly the same length as the Viterbi algorithm.
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Compositional bias correction</a>
                <div class="accordion-content" data-tab-content>
                    When two proteins have a similar, biased amino acid distribution, such as alpha-helical membrane proteins, they may look similar because the average score between any random pair of columns may be slightly positive. Activating this option will prevent protein pairs from looking similar just because their average amino acid distributions are similarly biased.
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Min. sequence identity of MSA hits with query</a>
                <div class="accordion-content" data-tab-content>
                    This parameter controls what percentage of residues from a hit of the MSA algorithm needs to be identical to their aligned query residues in order for the hit to be included in the MSA alignment. More precisely, the sequence identity of a hit by the MSA algorithm is defined as the number of residues identical with aligned query residues divided by the length of the MSA hit.
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Show representative sequences per HMM</a>
                <div class="accordion-content" data-tab-content>
                    Setting this to 10, for instance, will generate alignments containing 10 representative sequences from both query and template alignments. This can help to get a better feeling for the conservation along the alignment. The representative sequences shown are selected for maximum pairwise dissimilarity. The maximum value for this parameter is 10.
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Width of alignments (in columns)</a>
                <div class="accordion-content" data-tab-content>
                    Number of alignment columns per line in the HMM-HMM alignments. Leave at default value except if your browser cuts off parts of the alignment.
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Minimum probability in hit list</a>
                <div class="accordion-content" data-tab-content>
                    Only hits above this cut-off probability will be displayed.
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Max. number of hits in hit list</a>
                <div class="accordion-content" data-tab-content>
                    This parameter controls how many matches will be displayed in the results.
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Show hits as bar graph</a>
                <div class="accordion-content" data-tab-content>
                    With this option you can display a bar graph indicating the region in your query protein that is covered by the matches. The significance of the hits is color-coded, with most significant hits in red (>95% probability) and orange (>85%), then yellow and green (>70%), and finally cyan (>50%) and blue (>40%) for potential true positives hits.
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Match state assignment</a>
                <div class="accordion-content" data-tab-content>
                    <p>With this option you can choose the match state assignment for alignment with a format different to A2M/A3M. This value is the percentage threshold up to which percentage of gaps a column should be assigned as a match state, e.g. a value of 50 means that all columns with less than 50% gaps are assigned as match states. With the value 'first' all columns with a residue in 1st sequence are assigned as match states.</p>
                    <p>This parameter is not available with input formats a2m and a3m, because with these input formats the match state assignment is taken from the input.</p>
                </div>
            </li>
        </ul>
    </div>
    <div id="tabs-3" class="greenlink">
        <p>Alva V., Nam SZ., Söding J., Lupas AN. (2016) <b>The MPI Bioinformatics Toolkit as an integrative platform for advanced protein sequence and structure analysis.</b> Nucleic Acids Res. pii: gkw348. <a href=https://www.ncbi.nlm.nih.gov/pubmed/27131380>PMID: 27131380</a></p>
        <p>Söding J. (2005) <b>Protein homology detection by HMM-HMM comparison.</b> Bioinformatics 21: 951-960. <a href = https://www.ncbi.nlm.nih.gov/pubmed/15531603>PMID: 15531603</a></p>
        <p>Hildebrand A., Remmert A., Biegert A., Söding J. (2009) <b>Fast and accurate automatic structure prediction with HHpred.</b> Proteins 77(Suppl 9):128-32. doi: 10.1002/prot.22499. <a href = https://www.ncbi.nlm.nih.gov/pubmed/19626712>PMID: 19626712</a></p>
        <p>Meier A., Söding J. (2015) <b>Automatic Prediction of Protein 3D Structures by Probabilistic Multi-template Homology Modeling.</b> PLoS Comput Biol. 11(10):e1004343. doi: 10.1371/journal.pcbi.1004343. <a href = https://www.ncbi.nlm.nih.gov/pubmed/26496371>PMID: 26496371</a></p>
    </div>
</div>

<script>
        $(".accordion").foundation();
</script>