@()
<div id="tabs">
    <ul>
        <li><a href="#tabs-1">Overview</a></li>
        <li><a href="#tabs-2">Input & Parameters</a></li>
        <li><a href="#tabs-3">References</a></li>
    </ul>
    <div id="tabs-1" class="greenlink">
        <p>HHrepID is a novel automated procedure for the de novo identification of repeats in protein sequences. It is able to detect the sequence signature of structural repeats in many proteins that have not yet been known to possess internal sequence symmetry, such as TIM barrels and outer membrane beta-barrels. HHrepID uses HMM-HMM comparison to exploit evolutionary information in the form of the multiple sequence alignment of homologs, but in contrast to HHrep, the new method has several novel characteristics: (1) automated generation of a multiple alignment of repeats; (2) utilization of the transitive nature of homology through a novel merging procedure based on the fully probabilistic treatment of alignments; (3) superior alignment quality through an algorithm that maximizes the expected accuracy of an alignment; (4) the ability to identify different repeats within complicated architectures or multiple domains through automatic domain boundary detection, (5) new statistical treatment yielding improved sensitivity.</p>
        <h5><b>Length of submitted sequences</b></h5>
        <p>HHrepID works best on single-domain sequences or sequences containing only one kind of tandem repeat. If you don't know where your domains are, perform an HHpred search through SCOP and PfamA. Also, you could perform a preliminary HHrepID analysis to home in on the repeat regions and then cut the sequence further down to include only regions with one kind of repeat structure. Sequences should best have between 100 and 300 residues length. Sequences that are too short will gather too few homologs in the PSI-BLAST searches, thus loosing sensitivity. Also, the more repeat units are contained in a sequence, the more information from transitivity of pairwise alignments (A-B, B-C => A-C) can be used by HHrepID through its "Merge alignments" functionality. Sequences longer than 800 residues are truncated by HHrepID for practical reasons.</p>
        <h5><b>Submitting an alignment</b></h5>
        <p>If you already have a trustworthy multiple sequence alignment of your query with some homologs it is recommended that you start with this instead of from a single sequence. This will help PSI-BLAST to avoid alignment errors.</p>
        <h5><b>Why is HHrepID so sensitive?</b></h5>
        <p>The reason for HHrepID's sensitivity lies in the use of evolutionary information when comparing the query sequence with itself. This evolutionary information comes from the multiple sequence alignment of the sequence with homologous stretches of proteins. The information in the multiple sequence alignment is used in a condensed form, a profile HMM. A profile HMM is almost the same as a sequence profile. This in turn is just a table with 20 rows and L columns (where L is the length of the query protein). Each of the L columns contains the frequencies of the 20 amino acids in the corresponding column of the multiple sequence alignment. This table is then compared with itself in order to find stretches of columns with similar amino acid distributions. By looking at many related sequences with very similar structures that contribute to the sequence profile, profile-profile (or HMM-HMM) comparison effectively averages away those features of the sequence that are there just by chance and not by necessity and concentrates on those that determine the protein family's properties.</p>
        <p>A method that also employs HMM-HMM comparison for very sensitive protein-protein comparison is implemented in HHpred, our interactive function and structure prediction server. (For a short description of HHpred, please go to the HHpred help.)</p>
    </div>
    <div id="tabs-2" class="greenlink">
        <ul class="accordion" data-accordion data-allow-all-closed='true'>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Input</a>
                <div class="accordion-content" data-tab-content>
                    You have the option to enter or upload either a single protein sequence or a multiple sequence alignment (MSA) in FASTA, CLUSTAL, or A3M format.<br/><br/>
                    <p><b>Single query sequence:</b></p>
                    <p>You can enter a single query sequence in FASTA format (with or without name line):</p>
                    <pre>>gi|532319|pir|TVFV2E|TVFV2E envelope protein<br/>ELRLRYCAPAGFALLKCNDADYDGFKTNCSNVSVVHCTNLMNTTVTTGLLLNGSYSENRT<br/>QIWQKHRTSNDSALILLNKHYNLTVTCKRPGNKTVLPVTIMAGLVFHSQKYNLRLRQAWC<br/>HFPSNWKGAWKEVKEEIVNLPKERYRGTNDPKRIFFQRQWGDPETANLWFNCHGEFFYCK<br/>MDWFLNYLNNLTVDADHNECKNTSGTKSGNKRAPGPCVQR</pre>
                    <p>or this</p>
                    <pre>ELRLRYCAPAGFALLKCNDADYDGFKTNCSNVSVVHCTNLMNTTVTTGLLLNGSYSENRT<br/>QIWQKHRTSNDSALILLNKHYNLTVTCKRPGNKTVLPVTIMAGLVFHSQKYNLRLRQAWC<br/>HFPSNWKGAWKEVKEEIVNLPKERYRGTNDPKRIFFQRQWGDPETANLWFNCHGEFFYCK<br/>MDWFLNYLNNLTVDADHNECKNTSGTKSGNKRAPGPCVQR</pre><br/>
                    <p><b>Multiple sequence alignment</b></p>
                    <p>You can enter alignments in FASTA or CLUSTAL formats. The number of aligned sequences is currently limited to 2,000.</p>
                    <p>Example for aligned FASTA format:</p>
                    <pre>>d1a1x__ b.63.1.1 (-) p13-MTCP1 {Human (Homo sapiens)}<br/>PPDHLWVHQEGIYRDEYQRTWVAVVEE--E--T--SF---------LR----------ARVQQIQVPLG-------DAARPSHLLTS-----QLPLMWQLYP<br/>>gi|6678257|ref|NP_033363.1|:(7-103) T-cell lymphoma breakpoint 1 [Mus musculus]<br/>HPNRLWIWEKHVYLDEFRRSWLPVVIK--S--N--EK---------FQ----------VILRQEDVTLG-------EAMSPSQLVPY-----ELPLMWQLYP<br/>>gi|7305557|ref|NP_038800.1|:(8-103) T-cell leukemia/lymphoma 1B, 3 [Mus musculus]<br/>PPRFLVCTRDDIYEDENGRQWVVAKVE--T--S--RSpygsrietcIT----------VHLQHMTTIPQ-------EPTPQQPINNN-----SLPTMWRLES<br/>>gi|11415028|ref|NP_068801.1|:(2-106) T-cell lymphoma-1; T-cell lymphoma-1A [Homo sapiens]<br/>HPDRLWAWEKFVYLDEKQHAWLPLTIEikD--R--LQ---------LR----------VLLRREDVVLG-------RPMTPTQIGPS-----LLPIMWQLYP<br/>>gi|7305561|ref|NP_038804.1|:(7-103) T-cell leukemia/lymphoma 1B, 5 [Mus musculus]<br/>----------GIYEDEHHRVWIAVNVE--T--S--HS---------SHgnrietcvt-VHLQHMTTLPQ-------EPTPQQPINNN-----SLPTMWRLES<br/>>gi|7305553|ref|NP_038801.1|:(5-103) T-cell leukemia/lymphoma 1B, 1 [Mus musculus]<br/>LPVYLVSVRLGIYEDEHHRVWIVANVE--TshS--SH---------GN----------RRRTHVTVHLW-------KLIPQQVIPFNplnydFLPTTWKLES<br/>>gi|27668591|ref|XP_234504.1|:(7-103) similar to Chain A, Crystal Structure Of Murine Tcl1<br/>-PDRLWLWEKHVYLDEFRRSWLPIVIK--S--N--GK---------FQ----------VIMRQKDVILG-------DSMTPSQLVPY-----ELPLMWQLYP<br/>>gi|27668589|ref|XP_234503.1|:(9-91) similar to T-cell leukemia/lymphoma 1B, 5;<br/>-PHILTLRTHGIYEDEHHRLWVVLDLQ--A--ShlSF---------SN----------RLLIYLTVYLQqgvafplESTPPSPMNLN-----GLPRRWTLRT<br/>>gi|7305559|ref|NP_038802.1|:(8-102) T-cell leukemia/lymphoma 1B, 4 [Mus musculus] <br/>PPCFLVCTRDDIYEDEHGRQWVAAKVE--T--S--SH---------SPycskietcvtVHLWQMTTLFQ-------EPSPDSLKTFN-----FLPRTWRLES<br/>>gi|7305555|ref|NP_038803.1|:(9-102) T-cell leukemia/lymphoma 1B, 2 [Mus musculus]<br/>---------PGFYEDEHHRLWMVAKLE--T--C--SH---------SPycnkietcvtVHLWQMTRYPQ-------EPAPYNPMNYN-----FLPMTWRLAS</pre><br/>
                    <p>The same alignment in CLUSTAL format:</p>
                    <pre>CLUSTAL X (1.83) multiple sequence alignment<br/><br/>d1a1x__                             PPDHLWVHQEGIYRDEYQRTWVAVVEE--E--T--SF---------LR----------AR<br/>gi|6678257|ref|NP_033363.1|__7      HPNRLWIWEKHVYLDEFRRSWLPVVIK--S--N--EK---------FQ----------VI<br/>gi|7305557|ref|NP_038800.1|__8      PPRFLVCTRDDIYEDENGRQWVVAKVE--T--S--RSPYGSRIETCIT----------VH<br/>gi|11415028|ref|NP_068801.1|__      HPDRLWAWEKFVYLDEKQHAWLPLTIEIKD--R--LQ---------LR----------VL<br/>gi|7305561|ref|NP_038804.1|__7      ----------GIYEDEHHRVWIAVNVE--T--S--HS---------SHGNRIETCVT-VH<br/>gi|7305553|ref|NP_038801.1|__5      LPVYLVSVRLGIYEDEHHRVWIVANVE--TSHS--SH---------GN----------RR<br/>gi|27668591|ref|XP_234504.1|__      -PDRLWLWEKHVYLDEFRRSWLPIVIK--S--N--GK---------FQ----------VI<br/>gi|27668589|ref|XP_234503.1|__      -PHILTLRTHGIYEDEHHRLWVVLDLQ--A--SHLSF---------SN----------RL<br/>gi|7305559|ref|NP_038802.1|__8      PPCFLVCTRDDIYEDEHGRQWVAAKVE--T--S--SH---------SPYCSKIETCVTVH<br/>gi|7305555|ref|NP_038803.1|__9      ---------PGFYEDEHHRLWMVAKLE--T--C--SH---------SPYCNKIETCVTVH<br/>                                               .* **  : *:    :                       <br/><br/>d1a1x__                             VQQIQVPLG-------DAARPSHLLTS-----QLPLMWQLYP<br/>gi|6678257|ref|NP_033363.1|__7      LRQEDVTLG-------EAMSPSQLVPY-----ELPLMWQLYP<br/>gi|7305557|ref|NP_038800.1|__8      LQHMTTIPQ-------EPTPQQPINNN-----SLPTMWRLES<br/>gi|11415028|ref|NP_068801.1|__      LRREDVVLG-------RPMTPTQIGPS-----LLPIMWQLYP<br/>gi|7305561|ref|NP_038804.1|__7      LQHMTTLPQ-------EPTPQQPINNN-----SLPTMWRLES<br/>gi|7305553|ref|NP_038801.1|__5      RTHVTVHLW-------KLIPQQVIPFNPLNYDFLPTTWKLES<br/>gi|27668591|ref|XP_234504.1|__      MRQKDVILG-------DSMTPSQLVPY-----ELPLMWQLYP<br/>gi|27668589|ref|XP_234503.1|__      LIYLTVYLQQGVAFPLESTPPSPMNLN-----GLPRRWTLRT<br/>gi|7305559|ref|NP_038802.1|__8      LWQMTTLFQ-------EPSPDSLKTFN-----FLPRTWRLES<br/>gi|7305555|ref|NP_038803.1|__9      LWQMTRYPQ-------EPAPYNPMNYN-----FLPMTWRLAS<br/>                                                                     **  * * .</pre>

                <em>Note:</em><p>The alignment will be used to jumpstart PSI-BLAST. An expert-built or expert-edited alignment or one that contains homologs not yet contained in the non-redundant sequence database may significantly increase the sensitivity and reliability of HHrepID. I do NOT recommend to use automatic multiple alignment programs such as CLUSTAL, as PSI-BLAST tends to give clearly better alignments for the purpose of homologous relationships. Note: if you paste a multiple alignment please don't forget to select the appropriate input format below.</p>
                </div>
            </li>



            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Maximal no. of MSA generation steps</a>
                <div class="accordion-content" data-tab-content>
                    This specifies the number of PSI-BLAST iterations that are performed to build up an alignment from the input sequence or input alignment.
                    The iterations stop in any case when no further sequences are found.
                    If you want to use exactly your input alignment, choose 0 here.
                    This will ensure that only predicted secondary structure is added to your input alignment.
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Score secondary structure</a>
                <div class="accordion-content" data-tab-content>
                    If you choose "yes", the secondary structure of the query will be predicted with
                    <a href = https://www.ncbi.nlm.nih.gov/pubmed/10493868?dopt=Abstract target="_blank" rel="noopener">PSIPRED</a>
                    and a secondary structure similarity score is added to the total score.
                    Scoring the secondary structure similarity improves sensitivity and also alignment quality significantly.
                    Since the weight of the secondary structure score is small (12% of the amino acid similarity score),
                    the risk for high-scoring non-homologous matches is still small.
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Repeat family P-value threshold</a>
                <div class="accordion-content" data-tab-content>
                    The statistical significance threshold for including a repeat family in the repeat detection results.
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Self-alignment P-value threshold</a>
                <div class="accordion-content" data-tab-content>
                    The statistical significance threshold for including a suboptimal self-alignment in the total poster probability matrix.
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Merge rounds</a>
                <div class="accordion-content" data-tab-content>
                    Number of merge iterations to be performed before repeat are extracted from the posterior probability matrix.
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">MAC threshold</a>
                <div class="accordion-content" data-tab-content>
                    Probability threshold for MAC alignment algorithm. The lower the threshold the longer the MAC alignments and vice versa.
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Domain boundary detection</a>
                <div class="accordion-content" data-tab-content>
                    Turn this option off if you are certain that your query contains only a single domain.
                </div>
            </li>
        </ul>
    </div>
    <div id="tabs-3" class="greenlink">
        @help.citations.toolkit_citation()
        <p>Biegert A., Söding J. (2008) <b>HHrepID: de novo protein repeat identification by probabilistic consistency. </b>Bioinformatics 24(6):807-814.<a href = https://academic.oup.com/bioinformatics/article/24/6/807/194276/De-novo-identification-of-highly-diverged-protein target="_blank" rel="noopener">(download pdf)</a><br/><a href = https://www.ncbi.nlm.nih.gov/pubmed/18245125 target="_blank" rel="noopener">PMID: 18245125</a></p>
    </div>
</div>

<script>
        $(".accordion").foundation();
</script>