@()
<div id="tabs">
    <ul>
        <li><a href="#tabs-1">Overview</a></li>
        <li><a href="#tabs-2">Input & Parameters</a></li>
        <li><a href="#tabs-3">References</a></li>
    </ul>
    <div id="tabs-1" class="greenlink">
        <a href = https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3125773 target="_blank" rel="noopener">HMMER</a> is a
        tool for fast and sensitive homology searching based on profile Hidden Markov models. It accepts a single
        sequence or a multiple alignment in aligned FASTA/CLUSTAL format as input. In case the input is a single sequence,
        HHblits is run over Uniprot20 to build an alignment required for the calculation of query HMM.
    </div>
    <div id="tabs-2" class="greenlink">
        <ul class="accordion" data-accordion data-allow-all-closed='true'>
            <li class="accordion-item" data-accordion-item>
            <a class="accordion-title">Input</a>
            <div class="accordion-content" data-tab-content>
                You have the option to enter or upload either a single protein sequence or a multiple sequence alignment (MSA) in FASTA or CLUSTAL format.<br/><br/>
                <p><b>Single query sequence:</b></p>
                <p>You can enter a single query sequence in FASTA format (with or without name line):</p>
                <pre>>gi|532319|pir|TVFV2E|TVFV2E envelope protein<br/>ELRLRYCAPAGFALLKCNDADYDGFKTNCSNVSVVHCTNLMNTTVTTGLLLNGSYSENRT<br/>QIWQKHRTSNDSALILLNKHYNLTVTCKRPGNKTVLPVTIMAGLVFHSQKYNLRLRQAWC<br/>HFPSNWKGAWKEVKEEIVNLPKERYRGTNDPKRIFFQRQWGDPETANLWFNCHGEFFYCK<br/>MDWFLNYLNNLTVDADHNECKNTSGTKSGNKRAPGPCVQR</pre>
                <p>or this</p>
                <pre>ELRLRYCAPAGFALLKCNDADYDGFKTNCSNVSVVHCTNLMNTTVTTGLLLNGSYSENRT<br/>QIWQKHRTSNDSALILLNKHYNLTVTCKRPGNKTVLPVTIMAGLVFHSQKYNLRLRQAWC<br/>HFPSNWKGAWKEVKEEIVNLPKERYRGTNDPKRIFFQRQWGDPETANLWFNCHGEFFYCK<br/>MDWFLNYLNNLTVDADHNECKNTSGTKSGNKRAPGPCVQR</pre><br/>
                <p><b>Multiple sequence alignment</b></p>
                <p>You can enter alignments in FASTA or CLUSTAL formats.</p>
                <p>Example for aligned FASTA format:</p>
                <pre>>d1a1x__ b.63.1.1 (-) p13-MTCP1 {Human (Homo sapiens)}<br/>PPDHLWVHQEGIYRDEYQRTWVAVVEE--E--T--SF---------LR----------ARVQQIQVPLG-------DAARPSHLLTS-----QLPLMWQLYP<br/>>gi|6678257|ref|NP_033363.1|:(7-103) T-cell lymphoma breakpoint 1 [Mus musculus]<br/>HPNRLWIWEKHVYLDEFRRSWLPVVIK--S--N--EK---------FQ----------VILRQEDVTLG-------EAMSPSQLVPY-----ELPLMWQLYP<br/>>gi|7305557|ref|NP_038800.1|:(8-103) T-cell leukemia/lymphoma 1B, 3 [Mus musculus]<br/>PPRFLVCTRDDIYEDENGRQWVVAKVE--T--S--RSpygsrietcIT----------VHLQHMTTIPQ-------EPTPQQPINNN-----SLPTMWRLES<br/>>gi|11415028|ref|NP_068801.1|:(2-106) T-cell lymphoma-1; T-cell lymphoma-1A [Homo sapiens]<br/>HPDRLWAWEKFVYLDEKQHAWLPLTIEikD--R--LQ---------LR----------VLLRREDVVLG-------RPMTPTQIGPS-----LLPIMWQLYP<br/>>gi|7305561|ref|NP_038804.1|:(7-103) T-cell leukemia/lymphoma 1B, 5 [Mus musculus]<br/>----------GIYEDEHHRVWIAVNVE--T--S--HS---------SHgnrietcvt-VHLQHMTTLPQ-------EPTPQQPINNN-----SLPTMWRLES<br/>>gi|7305553|ref|NP_038801.1|:(5-103) T-cell leukemia/lymphoma 1B, 1 [Mus musculus]<br/>LPVYLVSVRLGIYEDEHHRVWIVANVE--TshS--SH---------GN----------RRRTHVTVHLW-------KLIPQQVIPFNplnydFLPTTWKLES<br/>>gi|27668591|ref|XP_234504.1|:(7-103) similar to Chain A, Crystal Structure Of Murine Tcl1<br/>-PDRLWLWEKHVYLDEFRRSWLPIVIK--S--N--GK---------FQ----------VIMRQKDVILG-------DSMTPSQLVPY-----ELPLMWQLYP<br/>>gi|27668589|ref|XP_234503.1|:(9-91) similar to T-cell leukemia/lymphoma 1B, 5;<br/>-PHILTLRTHGIYEDEHHRLWVVLDLQ--A--ShlSF---------SN----------RLLIYLTVYLQqgvafplESTPPSPMNLN-----GLPRRWTLRT<br/>>gi|7305559|ref|NP_038802.1|:(8-102) T-cell leukemia/lymphoma 1B, 4 [Mus musculus] <br/>PPCFLVCTRDDIYEDEHGRQWVAAKVE--T--S--SH---------SPycskietcvtVHLWQMTTLFQ-------EPSPDSLKTFN-----FLPRTWRLES<br/>>gi|7305555|ref|NP_038803.1|:(9-102) T-cell leukemia/lymphoma 1B, 2 [Mus musculus]<br/>---------PGFYEDEHHRLWMVAKLE--T--C--SH---------SPycnkietcvtVHLWQMTRYPQ-------EPAPYNPMNYN-----FLPMTWRLAS</pre><br/>
                <p>The same alignment in CLUSTAL format:</p>
                <pre>CLUSTAL X (1.83) multiple sequence alignment<br/><br/>d1a1x__                             PPDHLWVHQEGIYRDEYQRTWVAVVEE--E--T--SF---------LR----------AR<br/>gi|6678257|ref|NP_033363.1|__7      HPNRLWIWEKHVYLDEFRRSWLPVVIK--S--N--EK---------FQ----------VI<br/>gi|7305557|ref|NP_038800.1|__8      PPRFLVCTRDDIYEDENGRQWVVAKVE--T--S--RSPYGSRIETCIT----------VH<br/>gi|11415028|ref|NP_068801.1|__      HPDRLWAWEKFVYLDEKQHAWLPLTIEIKD--R--LQ---------LR----------VL<br/>gi|7305561|ref|NP_038804.1|__7      ----------GIYEDEHHRVWIAVNVE--T--S--HS---------SHGNRIETCVT-VH<br/>gi|7305553|ref|NP_038801.1|__5      LPVYLVSVRLGIYEDEHHRVWIVANVE--TSHS--SH---------GN----------RR<br/>gi|27668591|ref|XP_234504.1|__      -PDRLWLWEKHVYLDEFRRSWLPIVIK--S--N--GK---------FQ----------VI<br/>gi|27668589|ref|XP_234503.1|__      -PHILTLRTHGIYEDEHHRLWVVLDLQ--A--SHLSF---------SN----------RL<br/>gi|7305559|ref|NP_038802.1|__8      PPCFLVCTRDDIYEDEHGRQWVAAKVE--T--S--SH---------SPYCSKIETCVTVH<br/>gi|7305555|ref|NP_038803.1|__9      ---------PGFYEDEHHRLWMVAKLE--T--C--SH---------SPYCNKIETCVTVH<br/>                                               .* **  : *:    :                       <br/><br/>d1a1x__                             VQQIQVPLG-------DAARPSHLLTS-----QLPLMWQLYP<br/>gi|6678257|ref|NP_033363.1|__7      LRQEDVTLG-------EAMSPSQLVPY-----ELPLMWQLYP<br/>gi|7305557|ref|NP_038800.1|__8      LQHMTTIPQ-------EPTPQQPINNN-----SLPTMWRLES<br/>gi|11415028|ref|NP_068801.1|__      LRREDVVLG-------RPMTPTQIGPS-----LLPIMWQLYP<br/>gi|7305561|ref|NP_038804.1|__7      LQHMTTLPQ-------EPTPQQPINNN-----SLPTMWRLES<br/>gi|7305553|ref|NP_038801.1|__5      RTHVTVHLW-------KLIPQQVIPFNPLNYDFLPTTWKLES<br/>gi|27668591|ref|XP_234504.1|__      MRQKDVILG-------DSMTPSQLVPY-----ELPLMWQLYP<br/>gi|27668589|ref|XP_234503.1|__      LIYLTVYLQQGVAFPLESTPPSPMNLN-----GLPRRWTLRT<br/>gi|7305559|ref|NP_038802.1|__8      LWQMTTLFQ-------EPSPDSLKTFN-----FLPRTWRLES<br/>gi|7305555|ref|NP_038803.1|__9      LWQMTRYPQ-------EPAPYNPMNYN-----FLPMTWRLAS<br/>                                                                     **  * * .</pre>
            </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Select database</a>
                <div class="accordion-content" data-tab-content>
                    The selected database is used by HMMER to perform the search.
                        <ul>
                            <li><b>nr50, nr70, nr90:</b> NCBI's nonredundant protein sequence database filtered for a
                                maximum pairwise sequence identity of 50%, 70% and 90%, respectively.</li>
                            <li><b>env70, env90:</b> NCBI's environmental sequence database, usually from unknown organisms,
                                filtered for a maximum pairwise sequence identity of 70% and 90%, respectively.</li>
                            <li><b>nr_euk*, nr_bac*, nr_arc*, nr_pro*, nr_vir*:</b> eukaryotic, bacterial, archaeal,
                                prokaryotic, and viral sequences from the nonredundant database nr.</li>
                            <li><b>pdb_nr: </b>Sequences from proteins whose structures have been deposited in the Protein
                                Data Bank PDB.</li>
                            <li><b>Uniprot_sprot:</b> manually annotated and reviewed sequences from the
                                <a href="http://www.expasy.org/sprot/" target="_blank" rel="noopener">UniProtKB.</a></li>
                            <li><b>Uniprot_trembl:</b> automatically annotated and not reviewed sequences from the
                                <a href="http://www.ebi.ac.uk/swissprot/" target="_blank" rel="noopener">UniProtKB.</a></li>
                        </ul>
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">MSA enrichment iterations using HHblits</a>
                <div class="accordion-content" data-tab-content>
                    Specifies the number of HHblits iterations over Uniprot20 to build an MSA when the input is a single sequence.
                </div>
            </li>

            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">E-value</a>
                <div class="accordion-content" data-tab-content>
                    Report target sequences with an E-value of <= the specified value.
                </div>
        <li class="accordion-item" data-accordion-item>
            <a class="accordion-title">No. of target sequences</a>
            <div class="accordion-content" data-tab-content>
                This parameter controls how many matches will be displayed in the results.
            </div>
        </li>
            </li>
        </ul>
    </div>
    <div id="tabs-3" class="greenlink">
        @help.citations.toolkit_citation()
        <p>Finn RD., Clements J., Eddy SR. (2011) HMMER web server: interactive sequence similarity searching. Nucleic Acids Res. 39(Web Server issue): W29â€“W37. <a href = https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3125773 target="_blank" rel="noopener">PMCID: PMC3125773</a></p>
    </div>
</div>

<script>
        $(".accordion").foundation();
</script>