@()
<div id="tabs">
    <ul>
        <li><a href="#tabs-1">Overview</a></li>
        <li><a href="#tabs-2">Input & Parameters</a></li>
        <li><a href="#tabs-3">References</a></li>
    </ul>
    <div id="tabs-1" class="greenlink">
        <p>Kalign - an accurate and fast multiple sequence alignment algorithm.</p>
        <p>Timo Lassmann and Erik LL Sonnhammer (2005) Kalign - an accurate and fast multiple sequence alignment algorithm<br/>BMC Bioinformatics 2005, 6:298 doi:10.1186/1471-2105-6-298</p>
    </div>
    <div id="tabs-2" class="greenlink">
        <ul class="accordion" data-accordion data-allow-all-closed='true'>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Input</a>
                <div class="accordion-content" data-tab-content>
                    <p>Enter multiple sequences in FASTA format. A sequence in FASTA format begins with a single-line description, followed by lines of sequence data. The description line is distinguished from the sequence data by a greater-than (">") symbol in the first column. It is recommended that all lines of text be shorter than 80 characters in length. Currently, up to 2,000 sequences are supported. An example sequence in FASTA format is:</p><br/>
                    <pre>>gi|532319|pir|TVFV2E|TVFV2E envelope protein<br/>ELRLRYCAPAGFALLKCNDADYDGFKTNCSNVSVVHCTNLMNTTVTTGLLLNGSYSENRT<br/>QIWQKHRTSNDSALILLNKHYNLTVTCKRPGNKTVLPVTIMAGLVFHSQKYNLRLRQAWC<br/>HFPSNWKGAWKEVKEEIVNLPKERYRGTNDPKRIFFQRQWGDPETANLWFNCHGEFFYCK<br/>MDWFLNYLNNLTVDADHNECKNTSGTKSGNKRAPGPCVQRTYVACHIRSVIIWLETISKK<br/>TYAPPREGHLECTSTVTGMTVELNYIPKNRTNVTLSPQIESIWAAELDRYKLVEITPIGF<br/>APTEVRRYTGGHERQKRVPFVXXXXXXXXXXXXXXXXXXXXXXVQSQHLLAGILQQQKNL<br/>LAAVEAQQQMLKLTIWGVK</pre><br/>
                    <p>Sequences are expected to be represented in the standard IUB/IUPAC amino acid codes, with these exceptions: lower-case letters are accepted and are mapped into upper-case; a single hyphen or dash can be used to represent a gap of indeterminate length; and in amino acid sequences, U and * are acceptable letters (see below). Before submitting a request, any numerical digits in the query sequence should either be removed or replaced by appropriate letter codes (e.g., X for unknown amino acid residue).</p>
                    <p>The accepted amino acid codes are:</p><br/>
                    <pre>    A  alanine                         P  proline<br/>    B  aspartate or asparagine         Q  glutamine<br/>    C  cystine                         R  arginine<br/>    D  aspartate                       S  serine<br/>    E  glutamate                       T  threonine<br/>    F  phenylalanine                   U  selenocysteine<br/>    G  glycine                         V  valine<br/>    H  histidine                       W  tryptophan<br/>    I  isoleucine                      Y  tyrosine<br/>    K  lysine                          Z  glutamate or glutamine<br/>    L  leucine                         X  any<br/>    M  methionine                      *  translation stop<br/>    N  asparagine                      -  gap of indeterminate length</pre>
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Output the alignment in</a>
                <div class="accordion-content" data-tab-content>
                    <p>Specify the order in which the aligned sequences appear within the resulting multiple sequence alignment</p>
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Gap open penalty</a>
                <div class="accordion-content" data-tab-content>
                    Specify the gap open penalty (numbers greater than 0 are valid).
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Gap extension penalty</a>
                <div class="accordion-content" data-tab-content>
                    Specify the gap extension penalty.
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Terminal gap penalties</a>
                <div class="accordion-content" data-tab-content>
                    Specify the terminal gap penalties.
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Bonus score</a>
                <div class="accordion-content" data-tab-content>
                    Here you can specify a constant added to the substitution matrix.
                </div>
            </li>
        </ul>
    </div>
    <div id="tabs-3" class="greenlink">
        @help.citations.toolkit_citation()
        <p>Lassmann T., Sonnhammer EL. (2005) <b>Kalign--an accurate and fast multiple sequence alignment algorithm.</b> BMC Bioinformatics. 6:298.<a href = https://www.ncbi.nlm.nih.gov/pubmed/16343337 target="_blank" rel="noopener">PMID: 16343337</a></p>
    </div>
</div>

<script>
        $(".accordion").foundation();
</script>