@()
<div id="tabs">
    <ul>
        <li><a href="#tabs-1">Overview</a></li>
        <li><a href="#tabs-2">Parameters</a></li>
        <li><a href="#tabs-3">References</a></li>
    </ul>
    <div id="tabs-1" class="greenlink">
        Multiple alignment program for amino acid or nucleotide sequences<br/>[MAFFT v7; <a href = http://mafft.cbrc.jp/alignment/software target="_blank">http://mafft.cbrc.jp/alignment/software/</a>]
    </div>
    <div id="tabs-2" class="greenlink">
        <ul class="accordion" data-accordion data-allow-all-closed='true'>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Input format</a>
                <div class="accordion-content" data-tab-content>
                    Enter multiple sequences in fasta format. A sequence in FASTA format begins with a single-line description, followed by lines of sequence data. The description line is distinguished from the sequence data by a greater-than (\">\") symbol in the first column. It is recommended that all lines of text be shorter than 80 characters in length. An example sequence in FASTA format is:<br/><br/><br/><pre>>gi|532319|pir|TVFV2E|TVFV2E envelope protein<br/>ELRLRYCAPAGFALLKCNDADYDGFKTNCSNVSVVHCTNLMNTTVTTGLLLNGSYSENRT<br/>QIWQKHRTSNDSALILLNKHYNLTVTCKRPGNKTVLPVTIMAGLVFHSQKYNLRLRQAWC<br/>HFPSNWKGAWKEVKEEIVNLPKERYRGTNDPKRIFFQRQWGDPETANLWFNCHGEFFYCK<br/>MDWFLNYLNNLTVDADHNECKNTSGTKSGNKRAPGPCVQRTYVACHIRSVIIWLETISKK<br/>TYAPPREGHLECTSTVTGMTVELNYIPKNRTNVTLSPQIESIWAAELDRYKLVEITPIGF<br/>APTEVRRYTGGHERQKRVPFVXXXXXXXXXXXXXXXXXXXXXXVQSQHLLAGILQQQKNL<br/>LAAVEAQQQMLKLTIWGVK</pre><br/><br/>Sequences are expected to be represented in the standard IUB/IUPAC amino acid codes, with these exceptions: lower-case letters are accepted and are mapped into upper-case; a single hyphen or dash can be used to represent a gap of indeterminate length; and in amino acid sequences, U and * are acceptable letters (see below). Before submitting a request, any numerical digits in the query sequence should either be removed or replaced by appropriate letter codes (e.g., X for unknown amino acid residue).<br/><br/>The accepted amino acid codes are:<br/><br/><pre>    A  alanine                         P  proline<br/>    B  aspartate or asparagine         Q  glutamine<br/>    C  cystine                         R  arginine<br/>    D  aspartate                       S  serine<br/>    E  glutamate                       T  threonine<br/>    F  phenylalanine                   U  selenocysteine<br/>    G  glycine                         V  valine<br/>    H  histidine                       W  tryptophan<br/>    I  isoleucine                      Y  tyrosine<br/>    K  lysine                          Z  glutamate or glutamine<br/>    L  leucine                         X  any<br/>    M  methionine                      *  translation stop<br/>    N  asparagine                      -  gap of indeterminate length</pre>
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">CLUSTAL output</a>
                <div class="accordion-content" data-tab-content>
                    Activate this option to print the computed alignment in CLUSTAL format.
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Alignment order</a>
                <div class="accordion-content" data-tab-content>
                    Activate this option to print sequences in alignment order rather than input order.
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Gap open penalty</a>
                <div class="accordion-content" data-tab-content>
                    Specify the gap open penalty (numbers greater than 0 are valid).
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Offset</a>
                <div class="accordion-content" data-tab-content>
                    This option specifies the offset parameter which works like a gap extension penalty (numbers greater than 0 are valid).
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Advanced options</a>
                <div class="accordion-content" data-tab-content>
                    Use this textfield to specify advanced options via commandline. The options supported by MAFFT are:<br/><br/> <pre>Options:<br/>      --localpair :       All pairwise local alignment information is included<br/>                          to the objective function. default: off<br/>      --globalpair :      All pairwise global alignment information is included<br/>                          to the objective function. default: off<br/>      --bl #, --jtt # :   Scoring matrix. default: BLOSUM62<br/>                          Alternatives are BLOSUM (--bl) 30, 45, 62, 80,<br/>                          or JTT (--jtt) # PAM.<br/>      --nuc or --amino :  Sequence type. default: auto<br/>      --retree # :        The number of tree building in progressive method<br/>                          (see the paper for detail). default: 2<br/>      --maxiterate # :    Maximum number of iterative refinement. default: 0<br/>      --fft or --nofft:   FFT is enabled or disabled. default: enabled<br/>      --memsave:          Memory saving mode (beta). default: off<br/>      --quiet :           Do not report progress.</pre>
                </div>
            </li>
        </ul>
    </div>
    <div id="tabs-3" class="greenlink">
        @help.citations.toolkit_citation()
        <p>Katoh K., Misawa K., Kuma K., Miyata T. (2002) <b>MAFFT: a novel method for rapid multiple sequence alignment based on fast Fourier transform.</b> Nucleic Acid Res 30(14):3059-3066. <a href = https://www.ncbi.nlm.nih.gov/pubmed/12136088 target="_blank">PMID: 12136088</a></p>
    </div>
</div>

<script>
        $(".accordion").foundation();
</script>