@()
<div id="tabs">
    <ul>
        <li><a href="#tabs-1">Overview</a></li>
        <li><a href="#tabs-2">Input & Parameters</a></li>
        <li><a href="#tabs-3">References</a></li>
    </ul>
    <div id="tabs-1" class="greenlink">
        MARCOIL is a hidden Markov model-based program that predicts existence and location of potential coiled-coil domains in protein sequences. The external MARCOIL Homepage can be found <a href = http://bioinf.wehi.edu.au/folders/mauro/Marcoil/index.html target="_blank" rel="noopener">here</a>.
    </div>
    <div id="tabs-2" class="greenlink">
        <ul class="accordion" data-accordion data-allow-all-closed='true'>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Input</a>
                <div class="accordion-content" data-tab-content>
                    <p>Enter a single protein sequence in FASTA format, for example:</p><br/>
                    <pre>>gi|576099|pdb|1ENA|  Staphylococcal Nuclease (E.C.3.1.31.1) Mutation<br/>LHKEPATLIKAIDGETVKLMYKGQPMTFRLLLVDTPETKHPKKGVEKYGPEASAFTKKMVENAKKIEVEF<br/>DKGQRTDKYGRGLAYIYADGKMVNEALVRQGLAKVAYVYKPNNTHEQHLRKSEAQAKKEKLNIWS</pre><br/>
                    <p>The sequence name (and an optional description) must be contained in a single name line beginning with the ">" symbol and followed directly by the sequence name. The residue data is contained in one or more lines of arbitrary length following the name line. No empty lines should be used. Upper and lower case letters are allowed.
                    Letters not from the canonical one-letter-code are treated as unknown residues</p>
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Matrix</a>
                <div class="accordion-content" data-tab-content>
                    <p>The three matrices accessible through the web interface are the one used in the paper and trained on 9 "families" of proteins (9FAM). This is a matrix of amino-acid probabilities derived from a large dataset of coiled-coil domains. It is unspecific, as the dataset contains all kind of domains and these differ in the number of helices, the orientation, the length and the hydrophobicity. The matrix is meant for first-pass genomic screenings.</p>
                    <p>It generalises the two matrices proposed by A. Lupas and collaborators and used by the program COILS. These matrices are MTIDK, derived from 5 and MTK derived from three "families" of proteins. Those are matrices of frequency ratios and we computed from them, by using an estimate of absolute amino-acid frequencies, the other two matrices that can be used.</p>
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Transition Probability</a>
                <div class="accordion-content" data-tab-content>
                    Specify whether to use high or low transition probability.
                </div>
            </li>
        </ul>
    </div>
    <div id="tabs-3" class="greenlink">
        @help.citations.toolkit_citation()
        <p>Delorenzi M., Speed T. (2002) <b>An HMM model for coiled-coil domains and a comparison with PSSM-based predictions. </b>Bioinformatics 18(4):617-625. <a href = https://www.ncbi.nlm.nih.gov/pubmed/12016059 target="_blank" rel="noopener">PMID: 12016059</a></p>
    </div>
</div>

<script>
        $(".accordion").foundation();
</script>