@()
<div id="tabs">
    <ul>
        <li><a href="#tabs-1">Overview</a></li>
        <li><a href="#tabs-2">Parameters</a></li>
        <li><a href="#tabs-3">References</a></li>
    </ul>
    <div id="tabs-1" class="greenlink">
        MARCOIL is a hidden Markov model-based program that predicts existence and location of potential coiled-coil domains in protein sequences. The external MARCOIL Homepage can be found <a href = http://bioinf.wehi.edu.au/folders/mauro/Marcoil/index.html target="_blank">here</a>.
    </div>
    <div id="tabs-2" class="greenlink">
        <ul class="accordion" data-accordion data-allow-all-closed='true'>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Input format</a>
                <div class="accordion-content" data-tab-content>
                    <p>Please use only fasta format, as only this was tested.</p>
                    <p>The format is not checked by the program and you could get false results.</p>
                    <p>The number of sequences is unlimited.</p>
                    <p>Only the 20 standard amino acids are represented in the model,other letters are interpreted as unknown amino acids.</p>
                    <p>Lowercase is converted to uppercase, numbers, spaces,newlines, tabs and other special signs in the sequence are ignored.</p>
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Matrices</a>
                <div class="accordion-content" data-tab-content>
                    <p>The three matrices accessible through the web interface are the one used in the paper and trained on 9 "families" of proteins (9FAM). This is a matrix of amino-acid probabilities derived from a large dataset of coiled-coil domains. It is unspecific, as the dataset contains all kind of domains and these differ in the number of helices, the orientation, the length and the hydrophobicity. The matrix is meant for first-pass genomic screenings.</p>
                    <p>It generalises the two matrices proposed by A. Lupas and collaborators and used by the program COILS. These matrices are MTIDK, derived from 5 and MTK derived from three "families" of proteins. Those are matrices of frequency ratios and we computed from them, by using an estimate of absolute amino-acid frequencies, the other two matrices that can be used.</p>
                </div>
            </li>
        </ul>
    </div>
    <div id="tabs-3" class="greenlink">
        <p>Alva V., Nam SZ., SÃ¶ding J., Lupas AN. (2016) <b>The MPI bioinformatics Toolkit as an integrative platform for advanced protein sequence and structure analysis.</b> Nucleic Acids Res. pii: gkw348. <a href=https://www.ncbi.nlm.nih.gov/pubmed/27131380 target="_blank">PMID: 27131380</a></p>
        <p>Delorenzi M., Speed T. (2002) <b>An HMM model for coiled-coil domains and a comparison with PSSM-based predictions. </b>Bioinformatics 18(4):617-625. <a href = https://www.ncbi.nlm.nih.gov/pubmed/12016059 target="_blank">PMID: 12016059</a></p>
    </div>
</div>

<script>
        $(".accordion").foundation();
</script>