@()
<div id="tabs">
    <ul>
        <li><a href="#tabs-1">Overview</a></li>
        <li><a href="#tabs-2">Parameters</a></li>
        <li><a href="#tabs-3">References</a></li>
    </ul>
    <div id="tabs-1" class="greenlink">
        TPRpred uses the profile representation of the known repeats to detect Tetratrico Peptide Repeats (TPRs), Pentatrico Peptide Repeats(PPRs) and SEL1-like repeats from the query sequence and computes the statistical significance for their occurrence.
    </div>
    <div id="tabs-2" class="greenlink">
        <ul class="accordion" data-accordion data-allow-all-closed='true'>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Input format</a>
                <div class="accordion-content" data-tab-content>
                    A sequence in FASTA format begins with a single-line description, followed by lines of sequence data. The description line is distinguished from the sequence data by a greater-than (">") symbol in the first column. It is recommended that all lines of text be shorter than 80 characters in length. An example sequence in FASTA format is:<br/><br/><br/><pre>>HcpC_a HcpC Chain A<br/>DPKELVGLGAKSYKEKDFTQAKKYFEKACDLKENSGCFNLGVLYYQGQGV<br/>EKNLKKAASFYAKACDLNYSNGCHLLGNLYYSGQGVSQNTNKALQYYSKA<br/>CDLKYAEGCASLGGIYHDGKVVTRDFKKAVEYFTKACDLNDGDGCTILGS<br/>LYDAGRGTPKDLKKALASYDKACDLKDSPGCFNAGNMYHHGEGATKNFKE<br/>ALARYSKACELENGGGCFNLGAMQYNGEGVTRNEKQAIENFKKGCKLGAK<br/>GACDILKQLKIKVHHH</pre><br/><br/>Sequences are expected to be represented in the standard IUB/IUPAC amino acid acid codes, with these exceptions: lower-case letters are accepted and are mapped into upper-case; a single hyphen or dash can be used to represent a gap of indeterminate length; and in amino acid sequences, U and * are acceptable letters (see below). Before submitting a request, any numerical digits in the query sequence should either be removed or replaced by appropriate letter codes (e.g., X for unknown amino acid residue). The amino acid codes supported are:<br/><br/><pre>    A  alanine                         P  proline<br/>    B  aspartate or asparagine         Q  glutamine<br/>    C  cystine                         R  arginine<br/>    D  aspartate                       S  serine<br/>    E  glutamate                       T  threonine<br/>    F  phenylalanine                   U  selenocysteine<br/>    G  glycine                         V  valine<br/>    H  histidine                       W  tryptophan<br/>    I  isoleucine                      Y  tyrosine<br/>    K  lysine                          Z  glutamate or glutamine<br/>    L  leucine                         X  any<br/>    M  methionine                      *  translation stop<br/>    N  asparagine                      -  gap of indeterminate length</pre>
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Maximum whole protein E-value</a>
                <div class="accordion-content" data-tab-content>
                    Maximum whole-protein e-value for output of repeat residues to alifile.
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Minimum number of single-repeat hits</a>
                <div class="accordion-content" data-tab-content>
                    Minimum number of single repeat hits for output to alifile.
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Maximum number of rows in hitlist</a>
                <div class="accordion-content" data-tab-content>
                    As the name suggests.
                </div>
            </li>
        </ul>
    </div>
    <div id="tabs-3" class="greenlink">
        <p>Alva V., Nam SZ., SÃ¶ding J., Lupas AN. (2016) <b>The MPI bioinformatics Toolkit as an integrative platform for advanced protein sequence and structure analysis.</b> Nucleic Acids Res. pii: gkw348. <a href=https://www.ncbi.nlm.nih.gov/pubmed/27131380 target="_blank">PMID: 27131380</a></p>
        <p>Magis C., Taly JF., Bussotti G., Chang JM., Di Tommaso P., Erb I., Espinosa-Carrasco J., Notredame C. (2014) <b>T-Coffee: Tree-based consistency objective function for alignment evaluation. </b>Methods Mol Biol 1079:117-129. <a href = https://www.ncbi.nlm.nih.gov/pubmed/24170398 target="_blank">PMID: 24170398</a></p>
    </div>
</div>

<script>
        $(".accordion").foundation();
</script>