import Component = Mithril.Component;

let accordion : ElementConfig,
    accordionContent : any,
    accordionItem : ElementConfig,
    exampleContent : any,
    helpContent : any,
    helpModalReveal : ElementConfig,
    helpModalTabs : ElementConfig,
    parameterContent : any;

helpModalReveal = function(elem : any, isInit : boolean) : any {
    if (!isInit) {
        elem.setAttribute("data-reveal", "data-reveal");
        return $(elem).foundation();
    }
};

helpModalTabs = function(elem : any, isInit : boolean) : any {
    if (!isInit) {
        return $(elem).tabs();
    }
};

parameterContent = {
    //PSI-BLAST
    "Alignment": "Enter single protein sequence or protein sequence alignment and choose its format from the dropdown menu.",
    "Database": "The selected databases are used by BLAST to perform the search. <br /> - <b>nr:</b> the non-redundant sequence database at the NCBI, updated weekly. (See NCBI's BLAST tutorial). <br /> - <b>env:</b> NCBI's env database with environnmental sequences, usually from unknown organisms. <br /> - <b>nre:</b> nr+env database. Updated weekly. <br /> - <b>nr90, nr70, env90 etc.:</b> representative sequences, filtered for a maximum pairwise sequence identity of ~90% or 70% with kClust. Updated bimonthly. <br /> - <b>nr_euk, nr_bac, nr_arc, nr_pro, nr_vir:</b> eukaryotic, bacterial, archaeal, prokaryotic, and viral sequences from the non-redundant database nr. Updated weekly. <br /> - <b>uniprot_sprot:</b> http://www.expasy.org/sprot/ Updated weekly. <br /> - <b>uniprot_trembl:</b> http://www.ebi.ac.uk/swissprot/ Updated weekly. <br /> - <b>pdb_nr: Sequences</b> from proteins whose structures have been deposited in the Protein Data Bank PDB. This database uses the more complete SEQRES records of the PDB files (not the ATOM records that contain the atomic coordinates). This is the database used to construct the HHpred PDB alignment database. Updated weekly from the PDB. <br /> ",
    "Matrix": "Specify an alternate scoring matrix for PSIBLAST.",
    "Number of iterations": "Specify the maximum number of rounds of search. After each successive round the search alignment is updated.",
    "E-Value": "The statistical significance threshold for reporting matches against database sequences; the default value is 10, meaning that 10 matches are expected to be found merely by chance, according to the stochastic model of Karlin and Altschul (1990). If the statistical significance ascribed to a match is greater than the e-value threshold, the match will not be reported. Lower e-value thresholds are more stringent, leading to fewer chance matches being reported. Increasing the threshold shows less stringent matches. Fractional values are acceptable.",
    "E-value inclusion threshold": "The statistical significance threshold for including a sequence in the model used by PSI-BLAST to create the PSSM on the next iteration.",
    "Filter (low complexity)": "Mask off segments of the query sequence that have low compositional complexity, as determined by the SEG program of Wootton & Federhen (Computers and Chemistry, 1993) or, for BLASTN, by the DUST program of Tatusov and Lipman (in preparation). Filtering can eliminate statistically significant but biologically uninteresting reports from the blast output (e.g., hits against common acidic-, basic- or proline-rich regions), leaving the more biologically interesting regions of the query sequence available for specific matching against database sequences. Filtering is only applied to the query sequence (or its translation products), not to database sequences. Default filtering is DUST for BLASTN, SEG for other programs. It is not unusual for nothing at all to be masked by SEG, when applied to sequences in SWISS-PROT, so filtering should not be expected to always yield an effect. Furthermore, in some cases, sequences are masked in their entirety, indicating that the statistical significance of any matches reported against the unfiltered query sequence should be suspect.",
    "Compute Smith-Waterman alignment": "Compute locally optimal Smith-Waterman alignments.",
    "Use nr70 for all but last iteration": " If this option is selected the program will use the nr70-database for all but last round. In the last round, the database you selected in the database section will be used.",
    "Alignments and descriptions": "Restricts the number of short descriptions of matching sequences reported to the number specified; default limit is 100 descriptions. Restricts also database sequences to the number specified for which high-scoring segment pairs (HSPs) are reported. If more database sequences than this happen to satisfy the statistical significance threshold for reporting, only the matches ascribed the greatest statistical significance are reported.",
    //HHblits
    "Paste sequence or MSA": "You have the option to paste either a single protein sequence or a Multiple Sequence Alignment (MSA). HHblits builds high-quality MSAs starting from single sequences or from MSAs. It transforms these into a query HMM and iteratively searches through the selected databases.",
    "Select input format": "<b>Entering a single query sequence</b><br/><br/>You can enter a single query sequence in FASTA format (with or without name line). The query sequence in FASTA can look like this:<br/><br/><pre>>gi|532319|pir|TVFV2E|TVFV2E envelope protein<br/>ELRLRYCAPAGFALLKCNDADYDGFKTNCSNVSVVHCTNLMNTTVTTGLLLNGSYSENRT<br/>QIWQKHRTSNDSALILLNKHYNLTVTCKRPGNKTVLPVTIMAGLVFHSQKYNLRLRQAWC<br/>HFPSNWKGAWKEVKEEIVNLPKERYRGTNDPKRIFFQRQWGDPETANLWFNCHGEFFYCK<br/>MDWFLNYLNNLTVDADHNECKNTSGTKSGNKRAPGPCVQR</pre><br/><br/>or this<br/><br/><pre>>gi|532319|pir|TVFV2E|TVFV2E envelope protein<br/>ELRLRYCAPA GFALLKCNDA DYDGFKTNCS NVSVVHCTNL MNTTVTTGLL LNGSYSENRT QIWQKHRTSN <br/>DSALILLNKH YNLTVTCKRP GNKTVLPVTI MAGLVFHSQK YNLRLRQAWC HFPSNWKGAW KEVKEEIVNL <br/>PKERYRGTND PKRIFFQRQW GDPETANLWF NCHGEFFYCK MDWFLNYLNN LTVDADHNEC KNTSGTKSGN <br/>KRAPGPCVQR</pre><br/><br/>or simply this<br/><br/><pre>ELRLRYCAPAGFALLKCNDADYDGFKTNCSNVSVVHCTNLMNTTVTTGLLLNGSYSENRTQIWQKHRTSNDSALILLNKHYNLTVTCKRPGNK<br/>TVLPVTIMAGLVFHSQKYNLRLRQAWCHFPSNWKGAWKEVKEEIVNLPKERYRGTNDPKRIFFQRQWGDPETANLWFNCHGEFFYCKMDWFLNYLNNLTVDADHNE<br/>CKNTSGTKSGNKRAPGPCVQR</pre><br/><br/>The sequence may contain a nameline that must start with a > symbol, followed directly by the sequence name and an optional description. The residue data is contained in one or more lines of arbitrary length following the name line. No empty lines should be used. Upper and lower case letters are allowed. Spaces and gaps (-.) will be removed. <br/><br/><br/><b>Entering a multiple alignment</b><br/><br/>If you already have a query alignment then you can enter it here. You can enter alignments in different formats, which are listed in the dropdown menu. Format descriptions can be found in the help pages of the reformat tool.<br/><br/><br/>Example for aligned FASTA format:<br/><br/><br/><pre>>d1a1x__ b.63.1.1 (-) p13-MTCP1 {Human (Homo sapiens)}<br/>PPDHLWVHQEGIYRDEYQRTWVAVVEE--E--T--SF---------LR----------ARVQQIQVPLG-------DAARPSHLLTS-----QLPLMWQLYP<br/>>gi|6678257|ref|NP_033363.1|:(7-103) T-cell lymphoma breakpoint 1 [Mus musculus]<br/>HPNRLWIWEKHVYLDEFRRSWLPVVIK--S--N--EK---------FQ----------VILRQEDVTLG-------EAMSPSQLVPY-----ELPLMWQLYP<br/>>gi|7305557|ref|NP_038800.1|:(8-103) T-cell leukemia/lymphoma 1B, 3 [Mus musculus]<br/>PPRFLVCTRDDIYEDENGRQWVVAKVE--T--S--RSpygsrietcIT----------VHLQHMTTIPQ-------EPTPQQPINNN-----SLPTMWRLES<br/>>gi|11415028|ref|NP_068801.1|:(2-106) T-cell lymphoma-1; T-cell lymphoma-1A [Homo sapiens]<br/>HPDRLWAWEKFVYLDEKQHAWLPLTIEikD--R--LQ---------LR----------VLLRREDVVLG-------RPMTPTQIGPS-----LLPIMWQLYP<br/>>gi|7305561|ref|NP_038804.1|:(7-103) T-cell leukemia/lymphoma 1B, 5 [Mus musculus]<br/>----------GIYEDEHHRVWIAVNVE--T--S--HS---------SHgnrietcvt-VHLQHMTTLPQ-------EPTPQQPINNN-----SLPTMWRLES<br/>>gi|7305553|ref|NP_038801.1|:(5-103) T-cell leukemia/lymphoma 1B, 1 [Mus musculus]<br/>LPVYLVSVRLGIYEDEHHRVWIVANVE--TshS--SH---------GN----------RRRTHVTVHLW-------KLIPQQVIPFNplnydFLPTTWKLES<br/>>gi|27668591|ref|XP_234504.1|:(7-103) similar to Chain A, Crystal Structure Of Murine Tcl1<br/>-PDRLWLWEKHVYLDEFRRSWLPIVIK--S--N--GK---------FQ----------VIMRQKDVILG-------DSMTPSQLVPY-----ELPLMWQLYP<br/>>gi|27668589|ref|XP_234503.1|:(9-91) similar to T-cell leukemia/lymphoma 1B, 5;<br/>-PHILTLRTHGIYEDEHHRLWVVLDLQ--A--ShlSF---------SN----------RLLIYLTVYLQqgvafplESTPPSPMNLN-----GLPRRWTLRT<br/>>gi|7305559|ref|NP_038802.1|:(8-102) T-cell leukemia/lymphoma 1B, 4 [Mus musculus] <br/>PPCFLVCTRDDIYEDEHGRQWVAAKVE--T--S--SH---------SPycskietcvtVHLWQMTTLFQ-------EPSPDSLKTFN-----FLPRTWRLES<br/>>gi|7305555|ref|NP_038803.1|:(9-102) T-cell leukemia/lymphoma 1B, 2 [Mus musculus]<br/>---------PGFYEDEHHRLWMVAKLE--T--C--SH---------SPycnkietcvtVHLWQMTRYPQ-------EPAPYNPMNYN-----FLPMTWRLAS</pre><br/><br/>In aligned FASTA the gaps are written with '-' and the n'th letter of each sequence (except newlines) is understood to build the n'th column of the multiple alignment. The sequence residues may be broken up into several lines containing arbitrary numbers of residues.<br/><br/><br/><b>The same alignment in CLUSTAL format looks like this:</b><br/><br/><br/><pre>CLUSTAL X (1.83) multiple sequence alignment<br/><br/><br/>d1a1x__                             PPDHLWVHQEGIYRDEYQRTWVAVVEE--E--T--SF---------LR----------AR<br/>gi|6678257|ref|NP_033363.1|__7      HPNRLWIWEKHVYLDEFRRSWLPVVIK--S--N--EK---------FQ----------VI<br/>gi|7305557|ref|NP_038800.1|__8      PPRFLVCTRDDIYEDENGRQWVVAKVE--T--S--RSPYGSRIETCIT----------VH<br/>gi|11415028|ref|NP_068801.1|__      HPDRLWAWEKFVYLDEKQHAWLPLTIEIKD--R--LQ---------LR----------VL<br/>gi|7305561|ref|NP_038804.1|__7      ----------GIYEDEHHRVWIAVNVE--T--S--HS---------SHGNRIETCVT-VH<br/>gi|7305553|ref|NP_038801.1|__5      LPVYLVSVRLGIYEDEHHRVWIVANVE--TSHS--SH---------GN----------RR<br/>gi|27668591|ref|XP_234504.1|__      -PDRLWLWEKHVYLDEFRRSWLPIVIK--S--N--GK---------FQ----------VI<br/>gi|27668589|ref|XP_234503.1|__      -PHILTLRTHGIYEDEHHRLWVVLDLQ--A--SHLSF---------SN----------RL<br/>gi|7305559|ref|NP_038802.1|__8      PPCFLVCTRDDIYEDEHGRQWVAAKVE--T--S--SH---------SPYCSKIETCVTVH<br/>gi|7305555|ref|NP_038803.1|__9      ---------PGFYEDEHHRLWMVAKLE--T--C--SH---------SPYCNKIETCVTVH<br/>                                               .* **  : *:    :                       <br/><br/>d1a1x__                             VQQIQVPLG-------DAARPSHLLTS-----QLPLMWQLYP<br/>gi|6678257|ref|NP_033363.1|__7      LRQEDVTLG-------EAMSPSQLVPY-----ELPLMWQLYP<br/>gi|7305557|ref|NP_038800.1|__8      LQHMTTIPQ-------EPTPQQPINNN-----SLPTMWRLES<br/>gi|11415028|ref|NP_068801.1|__      LRREDVVLG-------RPMTPTQIGPS-----LLPIMWQLYP<br/>gi|7305561|ref|NP_038804.1|__7      LQHMTTLPQ-------EPTPQQPINNN-----SLPTMWRLES<br/>gi|7305553|ref|NP_038801.1|__5      RTHVTVHLW-------KLIPQQVIPFNPLNYDFLPTTWKLES<br/>gi|27668591|ref|XP_234504.1|__      MRQKDVILG-------DSMTPSQLVPY-----ELPLMWQLYP<br/>gi|27668589|ref|XP_234503.1|__      LIYLTVYLQQGVAFPLESTPPSPMNLN-----GLPRRWTLRT<br/>gi|7305559|ref|NP_038802.1|__8      LWQMTTLFQ-------EPSPDSLKTFN-----FLPRTWRLES<br/>gi|7305555|ref|NP_038803.1|__9      LWQMTRYPQ-------EPAPYNPMNYN-----FLPMTWRLAS<br/>                                                                     **  * * .</pre><br/><br/><br/>The first line must begin with CLUSTAL, the rest of the line is ignored. The sequences come in blocks with the sequence name, cut to a constant number of characters, followed by some whitespace and the sequence residues and gaps (-). The number of alignment columns per sequence line is arbitrary but constant within a block. Thus the whole alignment may be written in one long block. At the end of each block is an optional line indicating the conservation of each alignment column, followed by an obligatory empty line that separates the block from the next. Capital and lower case characters may be used for the amino acids.<br/><br/><br/><b>The same alignment in Stockholm format is here:</b><br/><br/><br/><pre># STOCKHOLM 1.0<br/><br/>#=GC RF                          XXXXXXXXXXXXXXXXXXXXXXXXXXX--X--X--XX---------XX----------XXXXXXXXXXX<br/>d1a1x__                          PPDHLWVHQEGIYRDEYQRTWVAVVEE--E--T--SF---------LR----------ARVQQIQVPLG<br/>gi|6678257|ref|NP_033363.1|:(7-1 HPNRLWIWEKHVYLDEFRRSWLPVVIK--S--N--EK---------FQ----------VILRQEDVTLG<br/>gi|7305557|ref|NP_038800.1|:(8-1 PPRFLVCTRDDIYEDENGRQWVVAKVE--T--S--RSpygsrietcIT----------VHLQHMTTIPQ<br/>gi|11415028|ref|NP_068801.1|:(2- HPDRLWAWEKFVYLDEKQHAWLPLTIEikD--R--LQ---------LR----------VLLRREDVVLG<br/>gi|7305561|ref|NP_038804.1|:(7-1 ----------GIYEDEHHRVWIAVNVE--T--S--HS---------SHgnrietcvt-VHLQHMTTLPQ<br/>gi|7305553|ref|NP_038801.1|:(5-1 LPVYLVSVRLGIYEDEHHRVWIVANVE--TshS--SH---------GN----------RRRTHVTVHLW<br/>gi|27668591|ref|XP_234504.1|:(7- -PDRLWLWEKHVYLDEFRRSWLPIVIK--S--N--GK---------FQ----------VIMRQKDVILG<br/>gi|27668589|ref|XP_234503.1|:(9- -PHILTLRTHGIYEDEHHRLWVVLDLQ--A--ShlSF---------SN----------RLLIYLTVYLQ<br/>gi|7305559|ref|NP_038802.1|:(8-1 PPCFLVCTRDDIYEDEHGRQWVAAKVE--T--S--SH---------SPycskietcvtVHLWQMTTLFQ<br/>gi|7305555|ref|NP_038803.1|:(9-1 ---------PGFYEDEHHRLWMVAKLE--T--C--SH---------SPycnkietcvtVHLWQMTRYPQ<br/><br/>#=GC RF                          -------XXXXXXXXXXX-----XXXXXXXXXX<br/>d1a1x__                          -------DAARPSHLLTS-----QLPLMWQLYP<br/>gi|6678257|ref|NP_033363.1|:(7-1 -------EAMSPSQLVPY-----ELPLMWQLYP<br/>gi|7305557|ref|NP_038800.1|:(8-1 -------EPTPQQPINNN-----SLPTMWRLES<br/>gi|11415028|ref|NP_068801.1|:(2- -------RPMTPTQIGPS-----LLPIMWQLYP<br/>gi|7305561|ref|NP_038804.1|:(7-1 -------EPTPQQPINNN-----SLPTMWRLES<br/>gi|7305553|ref|NP_038801.1|:(5-1 -------KLIPQQVIPFNplnydFLPTTWKLES<br/>gi|27668591|ref|XP_234504.1|:(7- -------DSMTPSQLVPY-----ELPLMWQLYP<br/>gi|27668589|ref|XP_234503.1|:(9- qgvafplESTPPSPMNLN-----GLPRRWTLRT<br/>gi|7305559|ref|NP_038802.1|:(8-1 -------EPSPDSLKTFN-----FLPRTWRLES<br/>gi|7305555|ref|NP_038803.1|:(9-1 -------EPAPYNPMNYN-----FLPMTWRLAS<br/>//</pre><br/><br/><br/>Stockholm format is similar to Clustal format, but offers a number of markups to include annotation and additional information referring to the alignment, its columns, or sequences. It was developed by Sean Eddy for Pfam and the HMMer software package. Each alignment may be separated into an arbitrary number of blocks with an arbitrary number of alignment columns contained per line. Gaps may be written '-' or '.'. The line starting with #=GC RF is an example of an optional markup line that contains one character per alignment columns. In this case it indicates which are the match/delete columns (as opposed to insert columns).<br/><br/><br/><b>The same alignment in A2M format looks like this:</b><br/><br/><br/><pre>>d1a1x__ b.63.1.1 (-) p13-MTCP1 {Human (Homo sapiens)}<br/>PPDHLWVHQEGIYRDEYQRTWVAVVEE..E..T..SF.........LR..........ARVQQIQVPLG.......DAARPSHLLTS.....QLPLMWQLYP<br/>>gi|6678257|ref|NP_033363.1|:(7-103) T-cell lymphoma breakpoint 1 [Mus musculus]<br/>HPNRLWIWEKHVYLDEFRRSWLPVVIK..S..N..EK.........FQ..........VILRQEDVTLG.......EAMSPSQLVPY.....ELPLMWQLYP<br/>>gi|7305557|ref|NP_038800.1|:(8-103) T-cell leukemia/lymphoma 1B, 3 [Mus musculus]<br/>PPRFLVCTRDDIYEDENGRQWVVAKVE..T..S..RSpygsrietcIT..........VHLQHMTTIPQ.......EPTPQQPINNN.....SLPTMWRLES<br/>>gi|11415028|ref|NP_068801.1|:(2-106) T-cell lymphoma-1; T-cell lymphoma-1A [Homo sapiens]<br/>HPDRLWAWEKFVYLDEKQHAWLPLTIEikD..R..LQ.........LR..........VLLRREDVVLG.......RPMTPTQIGPS.....LLPIMWQLYP<br/>>gi|7305561|ref|NP_038804.1|:(7-103) T-cell leukemia/lymphoma 1B, 5 [Mus musculus]<br/>----------GIYEDEHHRVWIAVNVE..T..S..HS.........SHgnrietcvt.VHLQHMTTLPQ.......EPTPQQPINNN.....SLPTMWRLES<br/>>gi|7305553|ref|NP_038801.1|:(5-103) T-cell leukemia/lymphoma 1B, 1 [Mus musculus]<br/>LPVYLVSVRLGIYEDEHHRVWIVANVE..TshS..SH.........GN..........RRRTHVTVHLW.......KLIPQQVIPFNplnydFLPTTWKLES<br/>>gi|27668591|ref|XP_234504.1|:(7-103) similar to Chain A, Crystal Structure Of Murine Tcl1<br/>-PDRLWLWEKHVYLDEFRRSWLPIVIK..S..N..GK.........FQ..........VIMRQKDVILG.......DSMTPSQLVPY.....ELPLMWQLYP<br/>>gi|27668589|ref|XP_234503.1|:(9-91) similar to T-cell leukemia/lymphoma 1B, 5;<br/>-PHILTLRTHGIYEDEHHRLWVVLDLQ..A..ShlSF.........SN..........RLLIYLTVYLQqgvafplESTPPSPMNLN.....GLPRRWTLRT<br/>>gi|7305559|ref|NP_038802.1|:(8-102) T-cell leukemia/lymphoma 1B, 4 [Mus musculus] <br/>PPCFLVCTRDDIYEDEHGRQWVAAKVE..T..S..SH.........SPycskietcvtVHLWQMTTLFQ.......EPSPDSLKTFN.....FLPRTWRLES<br/>>gi|7305555|ref|NP_038803.1|:(9-102) T-cell leukemia/lymphoma 1B, 2 [Mus musculus]<br/>---------PGFYEDEHHRLWMVAKLE..T..C..SH.........SPycnkietcvtVHLWQMTRYPQ.......EPAPYNPMNYN.....FLPMTWRLAS</pre><br/><br/><br/>A2M format is derived from aligned FASTA format. It looks very similar, but it distinguishes between match/delete columns and insert columns. This information is used to specify how an alignment is transformed into an HMM. The match/delete columns use upper case letters for residues and the '-' symbol for deletions (gaps). The insert columns use lower case letters for the inserted residues. Gaps aligned to inserted residues are written as '.'<br/><br/><br/><b>The same alignment in A3M is here:</b><br/><br/><br/><pre>>d1a1x__ b.63.1.1 (-) p13-MTCP1 {Human (Homo sapiens)}<br/>PPDHLWVHQEGIYRDEYQRTWVAVVEEETSFLRARVQQIQVPLGDAARPSHLLTSQLPLMWQLYP<br/>>gi|6678257|ref|NP_033363.1|:(7-103) T-cell lymphoma breakpoint 1 [Mus musculus]<br/>HPNRLWIWEKHVYLDEFRRSWLPVVIKSNEKFQVILRQEDVTLGEAMSPSQLVPYELPLMWQLYP<br/>>gi|7305557|ref|NP_038800.1|:(8-103) T-cell leukemia/lymphoma 1B, 3 [Mus musculus]<br/>PPRFLVCTRDDIYEDENGRQWVVAKVETSRSpygsrietcITVHLQHMTTIPQEPTPQQPINNNSLPTMWRLES<br/>>gi|11415028|ref|NP_068801.1|:(2-106) T-cell lymphoma-1; T-cell lymphoma-1A [Homo sapiens]<br/>HPDRLWAWEKFVYLDEKQHAWLPLTIEikDRLQLRVLLRREDVVLGRPMTPTQIGPSLLPIMWQLYP<br/>>gi|7305561|ref|NP_038804.1|:(7-103) T-cell leukemia/lymphoma 1B, 5 [Mus musculus]<br/>----------GIYEDEHHRVWIAVNVETSHSSHgnrietcvtVHLQHMTTLPQEPTPQQPINNNSLPTMWRLES<br/>>gi|7305553|ref|NP_038801.1|:(5-103) T-cell leukemia/lymphoma 1B, 1 [Mus musculus]<br/>LPVYLVSVRLGIYEDEHHRVWIVANVETshSSHGNRRRTHVTVHLWKLIPQQVIPFNplnydFLPTTWKLES<br/>>gi|27668591|ref|XP_234504.1|:(7-103) similar to Chain A, Crystal Structure Of Murine Tcl1<br/>-PDRLWLWEKHVYLDEFRRSWLPIVIKSNGKFQVIMRQKDVILGDSMTPSQLVPYELPLMWQLYP<br/>>gi|27668589|ref|XP_234503.1|:(9-91) similar to T-cell leukemia/lymphoma 1B, 5;<br/>-PHILTLRTHGIYEDEHHRLWVVLDLQAShlSFSNRLLIYLTVYLQqgvafplESTPPSPMNLNGLPRRWTLRT<br/>>gi|7305559|ref|NP_038802.1|:(8-102) T-cell leukemia/lymphoma 1B, 4 [Mus musculus] <br/>PPCFLVCTRDDIYEDEHGRQWVAAKVETSSHSPycskietcvtVHLWQMTTLFQEPSPDSLKTFNFLPRTWRLES<br/>>gi|7305555|ref|NP_038803.1|:(9-102) T-cell leukemia/lymphoma 1B, 2 [Mus musculus]<br/>---------PGFYEDEHHRLWMVAKLETCSHSPycnkietcvtVHLWQMTRYPQEPAPYNPMNYNFLPMTWRLAS</pre><br/><br/><br/>The A3M format is a condensed version of A2M format. It is obtained by omitting all '.' symbols from A2M format. Hence residues emitted by Match states of the HMM are in upper case, residues emitted by Insert states are in lower case and deletions are written '-'. A3M-formatted alignments are not really human readable but they can be reformatted to other more readable formats like FASTA, A2M CLUSTAL and many others with the reformatting utility in our toolbox. A3M is not an official format. Likewise, the name A3M is our personal invention for this very practical and space-efficient format which we use to store the alignments internally.",
    "Match state assignment": "With this option you can choose the match state assignment for alignment with a format different to A2M/A3M. This value is the percentage threshold up to which percentage of gaps a column should be assigned as a match state, e.g. a value of 50 means that all columns with less than 50% gaps are assigned as match states. With the value 'first' all columns with a residue in 1st sequence are assigned as match states.<br/><br/>This parameter is not available with input formats a2m and a3m, because with these input formats the match state assignment is taken from the input.",
    "Select HMM databases": "Select the database(s) of template HMMs against which you want to compare the query sequence or alignment.<br/><br/><a href = https://uniclust.mmseqs.com//>Uniclust30</a><br/>The Uniclust30 database is a clustered version of the Uniclust database down to a maximum pairwise sequence identity of 30%.",
    "Max. number of iterations": "This specifies the number of iterations that are performed by HHblits. The iterations stop in any case when no further sequences are found.",
    "Alignment mode": "In local mode, HHblits will use <u>local</u> alignments to score your query sequence with the database sequences. In global mode it uses global alignments with no end gap penalties (sometimes called 'global-local' alignments), i.e. the query-template alignments extend up to the beginning and end of the query or template sequence. The use of local mode is strongly recommended if you want to know which are your most likely remote homologs. If you have already identified your template with a previous run in local mode, you can use the global mode to get global alignments with an already identified template.<br/><br/>HHblits was shown to be significantly more sensitive to detect remote homologs in local than in global mode. The reason is that most remote homologs are really only homologous in a common core that define their superfamily. This core has been adorned by other secondary structure elements in different ways in different families. Also, the statistical evaluation is more error-prone for the global alignment mode, since the negatives score distribution for global alignments does not follow an extreme-value distribution.<br/><br/>You may generate global alignments and at the same time score alignments with the more sensitive local Viterbi alignment by selecting the 'local' option in combination with the option 'Realign with MAC algorithm' and setting the 'MAC realignment threshold' to 0.0.",
    "Realign with MAC algorithm": "After the search with the standard Viterbi algorithm, this option will cause HHblits to realign all query-template alignments with the more accurate Maximum ACcuracy alignment algorithm.<br/><br/>Since for technical reasons the scores, E-values, and probabilities from the Viterbi alignment and NOT the MAC alignment are displayed and used for ranking, the MAC realignment option may lead to high-scoring alignments of length 1. This happens when the MAC algorithm cannot find any acceptable alignment path, which is an indication for a non-homologous relationship.<br/><br/>When the MAC realignment option is activated, the 'global' and 'local' options now refer to both the Viterbi search stage as well as the MAC realignment stage. With -global (-local), the posterior probability matrix will be calculated for global (local) alignment.",
    "MAC realignment threshold": "Only active when the 'Realign with MAC algorithm' option is active. This parameter controls the MAC alignment algorithm's greediness. More precisely, the MAC algorithm finds the alignment that maximizes the sum of posterior probabilities minus this threshold for each aligned pair. Global alignments are generated with 0, whereas 0.7 will produce quite conservative local alignments with average probabilities for each aligned pair of residues above 70%. Default value is 0.5.",
    "Disable most filters": "This option disable the most additional filter steps of HHblits with exception of the profile-profile prefilter. This results in a slighty more sensitive performance but it increases the runtime of HHblits.",
    "E-value inclusion threshold ": "All HMM matches (\"hits\") with an E-value better than this threshold are included in the alignment for the next iteration or, in the last iteration, for building the final alignment shown in the 'Show alignment' result tab.",
    "Min. coverage of hits": " Minimal coverage of HMM matches to be included in the evolving query profile.",
    "Show representative sequences per HMM": "Setting this to 10, for instance, will generate alignments containing 10 representative sequences from both query and template alignments. This can help to get a better feeling for the conservation along the alignment. The representative sequences shown are selected for maximum pairwise dissimilarity. The maximum value for this parameter is 10.",
    "Width of alignments (in columns)": "Number of alignment columns per line in the HMM-HMM alignments. Leave at default value except if your browser cuts off parts of the alignment.",
    "Minimum probability in hit list": "Only hits above this cut-off probability will be displayed.",
    "Max. number of hits in hit list": "This parameter controls how many matches will be displayed in the results.",
    "Coloring query-template multiple alignments in results": "You can choose to color the sequences in the query-template alignments on the results page by physico-chemical properties of the amino acids, either by using colored fonts or by background coloring.",
    //HHpred
    "Paste single sequence or multiple alignment": "Paste the query sequence with which you want to do the search either as raw sequence (e.g. 'ELRLRYCAPAGFALLKCN...') or as FASTA sequence (i.e. with a single line '>sequence_name optional description' preceding the raw sequence, see next subsection). HHpred will either use HHBlits (Uniprot database filtered to 20% ) or PSI-BLAST (non-redundant sequence database from NCBI) to build a multiple alignment (MSA) from the respective databases. This alignment will then be used to construct a concise statistical representation (a hidden Markov model or HMM) and to search the selected HMM databases for homologs.<br/>You may also paste a query multiple alignment (for allowed formats see below). The alignment will be used to jumpstart the MSA (multiple sequence alignment) generation method. An expert-built or expert-edited alignment or one that contains homologs not yet contained in the non-redundant sequence database may significantly increase the sensitivity and reliability of the HHpred search. I recommend NOT to use automatic multiple alignment programs such as CLUSTAL, as the MSA generation tools tend to give clearly better alignments for the purpose of detecting homologs. Note: if you paste a multiple alignment please don't forget to select the appropriate input format below.",
    "Select input format ": "<b>Entering a single query sequence</b><br/><br/>You can enter a single query sequence in FASTA format (with or without name line). HHsearch/HHpred will build a multiple alignment of homologs from it with PsiBlast. The query sequence in FASTA can look like this:<br/><br/><pre>>gi|532319|pir|TVFV2E|TVFV2E envelope protein<br/>ELRLRYCAPAGFALLKCNDADYDGFKTNCSNVSVVHCTNLMNTTVTTGLLLNGSYSENRT<br/>QIWQKHRTSNDSALILLNKHYNLTVTCKRPGNKTVLPVTIMAGLVFHSQKYNLRLRQAWC<br/>HFPSNWKGAWKEVKEEIVNLPKERYRGTNDPKRIFFQRQWGDPETANLWFNCHGEFFYCK<br/>MDWFLNYLNNLTVDADHNECKNTSGTKSGNKRAPGPCVQR</pre><br/><br/>or this<pre><br/><br/>>gi|532319|pir|TVFV2E|TVFV2E envelope protein<br/>ELRLRYCAPA GFALLKCNDA DYDGFKTNCS NVSVVHCTNL MNTTVTTGLL LNGSYSENRT QIWQKHRTSN <br/>DSALILLNKH YNLTVTCKRP GNKTVLPVTI MAGLVFHSQK YNLRLRQAWC HFPSNWKGAW KEVKEEIVNL <br/>PKERYRGTND PKRIFFQRQW GDPETANLWF NCHGEFFYCK MDWFLNYLNN LTVDADHNEC KNTSGTKSGN <br/>KRAPGPCVQR<br/><br/></pre>or simply this<br/><br/><pre>ELRLRYCAPAGFALLKCNDADYDGFKTNCSNVSVVHCTNLMNTTVTTGLLLNGSYSENRTQIWQKHRTSNDSALILLNKHYNLTVTCKRPGNK<br/>TVLPVTIMAGLVFHSQKYNLRLRQAWCHFPSNWKGAWKEVKEEIVNLPKERYRGTNDPKRIFFQRQWGDPETANLWFNCHGEFFYCKMDWFLNYLNNLTVDADHNE<br/>CKNTSGTKSGNKRAPGPCVQR</pre><br/><br/>The sequence may contain a nameline that must start with a > symbol, followed directly by the sequence name and an optional description. The residue data is contained in one or more lines of arbitrary length following the name line. No empty lines should be used. Upper and lower case letters are allowed. Spaces and gaps (-.) will be removed. <br/><br/><br/><b>Entering a multiple alignment</b><br/><br/>If you already have a query alignment then you can enter it here. You can enter alignments in different formats, which are listed in the dropdown menu. Format descriptions can be found in the help pages of the reformat tool.<br/><br/><br/>Example for aligned FASTA format:<br/><br/><br/></pre>>d1a1x__ b.63.1.1 (-) p13-MTCP1 {Human (Homo sapiens)}<br/>PPDHLWVHQEGIYRDEYQRTWVAVVEE--E--T--SF---------LR----------ARVQQIQVPLG-------DAARPSHLLTS-----QLPLMWQLYP<br/>>gi|6678257|ref|NP_033363.1|:(7-103) T-cell lymphoma breakpoint 1 [Mus musculus]<br/>HPNRLWIWEKHVYLDEFRRSWLPVVIK--S--N--EK---------FQ----------VILRQEDVTLG-------EAMSPSQLVPY-----ELPLMWQLYP<br/>>gi|7305557|ref|NP_038800.1|:(8-103) T-cell leukemia/lymphoma 1B, 3 [Mus musculus]<br/>PPRFLVCTRDDIYEDENGRQWVVAKVE--T--S--RSpygsrietcIT----------VHLQHMTTIPQ-------EPTPQQPINNN-----SLPTMWRLES<br/>>gi|11415028|ref|NP_068801.1|:(2-106) T-cell lymphoma-1; T-cell lymphoma-1A [Homo sapiens]<br/>HPDRLWAWEKFVYLDEKQHAWLPLTIEikD--R--LQ---------LR----------VLLRREDVVLG-------RPMTPTQIGPS-----LLPIMWQLYP<br/>>gi|7305561|ref|NP_038804.1|:(7-103) T-cell leukemia/lymphoma 1B, 5 [Mus musculus]<br/>----------GIYEDEHHRVWIAVNVE--T--S--HS---------SHgnrietcvt-VHLQHMTTLPQ-------EPTPQQPINNN-----SLPTMWRLES<br/>>gi|7305553|ref|NP_038801.1|:(5-103) T-cell leukemia/lymphoma 1B, 1 [Mus musculus]<br/>LPVYLVSVRLGIYEDEHHRVWIVANVE--TshS--SH---------GN----------RRRTHVTVHLW-------KLIPQQVIPFNplnydFLPTTWKLES<br/>>gi|27668591|ref|XP_234504.1|:(7-103) similar to Chain A, Crystal Structure Of Murine Tcl1<br/>-PDRLWLWEKHVYLDEFRRSWLPIVIK--S--N--GK---------FQ----------VIMRQKDVILG-------DSMTPSQLVPY-----ELPLMWQLYP<br/>>gi|27668589|ref|XP_234503.1|:(9-91) similar to T-cell leukemia/lymphoma 1B, 5;<br/>-PHILTLRTHGIYEDEHHRLWVVLDLQ--A--ShlSF---------SN----------RLLIYLTVYLQqgvafplESTPPSPMNLN-----GLPRRWTLRT<br/>>gi|7305559|ref|NP_038802.1|:(8-102) T-cell leukemia/lymphoma 1B, 4 [Mus musculus] <br/>PPCFLVCTRDDIYEDEHGRQWVAAKVE--T--S--SH---------SPycskietcvtVHLWQMTTLFQ-------EPSPDSLKTFN-----FLPRTWRLES<br/>>gi|7305555|ref|NP_038803.1|:(9-102) T-cell leukemia/lymphoma 1B, 2 [Mus musculus]<br/>---------PGFYEDEHHRLWMVAKLE--T--C--SH---------SPycnkietcvtVHLWQMTRYPQ-------EPAPYNPMNYN-----FLPMTWRLAS</pre><br/><br/>In aligned FASTA the gaps are written with '-' and the n'th letter of each sequence (except newlines) is understood to build the n'th column of the multiple alignment. The sequence residues may be broken up into several lines containing arbitrary numbers of residues.<br/><br/><br/><b>The same alignment in CLUSTAL format looks like this:</b><br/><br/><br/><pre>CLUSTAL X (1.83) multiple sequence alignment<br/><br/><br/>d1a1x__                             PPDHLWVHQEGIYRDEYQRTWVAVVEE--E--T--SF---------LR----------AR<br/>gi|6678257|ref|NP_033363.1|__7      HPNRLWIWEKHVYLDEFRRSWLPVVIK--S--N--EK---------FQ----------VI<br/>gi|7305557|ref|NP_038800.1|__8      PPRFLVCTRDDIYEDENGRQWVVAKVE--T--S--RSPYGSRIETCIT----------VH<br/>gi|11415028|ref|NP_068801.1|__      HPDRLWAWEKFVYLDEKQHAWLPLTIEIKD--R--LQ---------LR----------VL<br/>gi|7305561|ref|NP_038804.1|__7      ----------GIYEDEHHRVWIAVNVE--T--S--HS---------SHGNRIETCVT-VH<br/>gi|7305553|ref|NP_038801.1|__5      LPVYLVSVRLGIYEDEHHRVWIVANVE--TSHS--SH---------GN----------RR<br/>gi|27668591|ref|XP_234504.1|__      -PDRLWLWEKHVYLDEFRRSWLPIVIK--S--N--GK---------FQ----------VI<br/>gi|27668589|ref|XP_234503.1|__      -PHILTLRTHGIYEDEHHRLWVVLDLQ--A--SHLSF---------SN----------RL<br/>gi|7305559|ref|NP_038802.1|__8      PPCFLVCTRDDIYEDEHGRQWVAAKVE--T--S--SH---------SPYCSKIETCVTVH<br/>gi|7305555|ref|NP_038803.1|__9      ---------PGFYEDEHHRLWMVAKLE--T--C--SH---------SPYCNKIETCVTVH<br/>                                               .* **  : *:    :                       <br/><br/>d1a1x__                             VQQIQVPLG-------DAARPSHLLTS-----QLPLMWQLYP<br/>gi|6678257|ref|NP_033363.1|__7      LRQEDVTLG-------EAMSPSQLVPY-----ELPLMWQLYP<br/>gi|7305557|ref|NP_038800.1|__8      LQHMTTIPQ-------EPTPQQPINNN-----SLPTMWRLES<br/>gi|11415028|ref|NP_068801.1|__      LRREDVVLG-------RPMTPTQIGPS-----LLPIMWQLYP<br/>gi|7305561|ref|NP_038804.1|__7      LQHMTTLPQ-------EPTPQQPINNN-----SLPTMWRLES<br/>gi|7305553|ref|NP_038801.1|__5      RTHVTVHLW-------KLIPQQVIPFNPLNYDFLPTTWKLES<br/>gi|27668591|ref|XP_234504.1|__      MRQKDVILG-------DSMTPSQLVPY-----ELPLMWQLYP<br/>gi|27668589|ref|XP_234503.1|__      LIYLTVYLQQGVAFPLESTPPSPMNLN-----GLPRRWTLRT<br/>gi|7305559|ref|NP_038802.1|__8      LWQMTTLFQ-------EPSPDSLKTFN-----FLPRTWRLES<br/>gi|7305555|ref|NP_038803.1|__9      LWQMTRYPQ-------EPAPYNPMNYN-----FLPMTWRLAS<br/>                                                                     **  * * .</pre><br/><br/><br/>The first line must begin with CLUSTAL, the rest of the line is ignored. The sequences come in blocks with the sequence name, cut to a constant number of characters, followed by some whitespace and the sequence residues and gaps (-). The number of alignment columns per sequence line is arbitrary but constant within a block. Thus the whole alignment may be written in one long block. At the end of each block is an optional line indicating the conservation of each alignment column, followed by an obligatory empty line that separates the block from the next. Capital and lower case characters may be used for the amino acids.<br/><br/><br/><b>The same alignment in Stockholm format is here:</b><br/><br/><br/><pre># STOCKHOLM 1.0<br/><br/>#=GC RF                          XXXXXXXXXXXXXXXXXXXXXXXXXXX--X--X--XX---------XX----------XXXXXXXXXXX<br/>d1a1x__                          PPDHLWVHQEGIYRDEYQRTWVAVVEE--E--T--SF---------LR----------ARVQQIQVPLG<br/>gi|6678257|ref|NP_033363.1|:(7-1 HPNRLWIWEKHVYLDEFRRSWLPVVIK--S--N--EK---------FQ----------VILRQEDVTLG<br/>gi|7305557|ref|NP_038800.1|:(8-1 PPRFLVCTRDDIYEDENGRQWVVAKVE--T--S--RSpygsrietcIT----------VHLQHMTTIPQ<br/>gi|11415028|ref|NP_068801.1|:(2- HPDRLWAWEKFVYLDEKQHAWLPLTIEikD--R--LQ---------LR----------VLLRREDVVLG<br/>gi|7305561|ref|NP_038804.1|:(7-1 ----------GIYEDEHHRVWIAVNVE--T--S--HS---------SHgnrietcvt-VHLQHMTTLPQ<br/>gi|7305553|ref|NP_038801.1|:(5-1 LPVYLVSVRLGIYEDEHHRVWIVANVE--TshS--SH---------GN----------RRRTHVTVHLW<br/>gi|27668591|ref|XP_234504.1|:(7- -PDRLWLWEKHVYLDEFRRSWLPIVIK--S--N--GK---------FQ----------VIMRQKDVILG<br/>gi|27668589|ref|XP_234503.1|:(9- -PHILTLRTHGIYEDEHHRLWVVLDLQ--A--ShlSF---------SN----------RLLIYLTVYLQ<br/>gi|7305559|ref|NP_038802.1|:(8-1 PPCFLVCTRDDIYEDEHGRQWVAAKVE--T--S--SH---------SPycskietcvtVHLWQMTTLFQ<br/>gi|7305555|ref|NP_038803.1|:(9-1 ---------PGFYEDEHHRLWMVAKLE--T--C--SH---------SPycnkietcvtVHLWQMTRYPQ<br/><br/>#=GC RF                          -------XXXXXXXXXXX-----XXXXXXXXXX<br/>d1a1x__                          -------DAARPSHLLTS-----QLPLMWQLYP<br/>gi|6678257|ref|NP_033363.1|:(7-1 -------EAMSPSQLVPY-----ELPLMWQLYP<br/>gi|7305557|ref|NP_038800.1|:(8-1 -------EPTPQQPINNN-----SLPTMWRLES<br/>gi|11415028|ref|NP_068801.1|:(2- -------RPMTPTQIGPS-----LLPIMWQLYP<br/>gi|7305561|ref|NP_038804.1|:(7-1 -------EPTPQQPINNN-----SLPTMWRLES<br/>gi|7305553|ref|NP_038801.1|:(5-1 -------KLIPQQVIPFNplnydFLPTTWKLES<br/>gi|27668591|ref|XP_234504.1|:(7- -------DSMTPSQLVPY-----ELPLMWQLYP<br/>gi|27668589|ref|XP_234503.1|:(9- qgvafplESTPPSPMNLN-----GLPRRWTLRT<br/>gi|7305559|ref|NP_038802.1|:(8-1 -------EPSPDSLKTFN-----FLPRTWRLES<br/>gi|7305555|ref|NP_038803.1|:(9-1 -------EPAPYNPMNYN-----FLPMTWRLAS<br/>//</pre><br/><br/><br/>Stockholm format is similar to Clustal format, but offers a number of markups to include annotation and additional information referring to the alignment, its columns, or sequences. It was developed by Sean Eddy for Pfam and the HMMer software package. Each alignment may be separated into an arbitrary number of blocks with an arbitrary number of alignment columns contained per line. Gaps may be written '-' or '.'. The line starting with #=GC RF is an example of an optional markup line that contains one character per alignment columns. In this case it indicates which are the match/delete columns (as opposed to insert columns).<br/><br/><br/><b>The same alignment in A2M format looks like this:</b><br/><br/><br/><pre>>d1a1x__ b.63.1.1 (-) p13-MTCP1 {Human (Homo sapiens)}<br/>PPDHLWVHQEGIYRDEYQRTWVAVVEE..E..T..SF.........LR..........ARVQQIQVPLG.......DAARPSHLLTS.....QLPLMWQLYP<br/>>gi|6678257|ref|NP_033363.1|:(7-103) T-cell lymphoma breakpoint 1 [Mus musculus]<br/>HPNRLWIWEKHVYLDEFRRSWLPVVIK..S..N..EK.........FQ..........VILRQEDVTLG.......EAMSPSQLVPY.....ELPLMWQLYP<br/>>gi|7305557|ref|NP_038800.1|:(8-103) T-cell leukemia/lymphoma 1B, 3 [Mus musculus]<br/>PPRFLVCTRDDIYEDENGRQWVVAKVE..T..S..RSpygsrietcIT..........VHLQHMTTIPQ.......EPTPQQPINNN.....SLPTMWRLES<br/>>gi|11415028|ref|NP_068801.1|:(2-106) T-cell lymphoma-1; T-cell lymphoma-1A [Homo sapiens]<br/>HPDRLWAWEKFVYLDEKQHAWLPLTIEikD..R..LQ.........LR..........VLLRREDVVLG.......RPMTPTQIGPS.....LLPIMWQLYP<br/>>gi|7305561|ref|NP_038804.1|:(7-103) T-cell leukemia/lymphoma 1B, 5 [Mus musculus]<br/>----------GIYEDEHHRVWIAVNVE..T..S..HS.........SHgnrietcvt.VHLQHMTTLPQ.......EPTPQQPINNN.....SLPTMWRLES<br/>>gi|7305553|ref|NP_038801.1|:(5-103) T-cell leukemia/lymphoma 1B, 1 [Mus musculus]<br/>LPVYLVSVRLGIYEDEHHRVWIVANVE..TshS..SH.........GN..........RRRTHVTVHLW.......KLIPQQVIPFNplnydFLPTTWKLES<br/>>gi|27668591|ref|XP_234504.1|:(7-103) similar to Chain A, Crystal Structure Of Murine Tcl1<br/>-PDRLWLWEKHVYLDEFRRSWLPIVIK..S..N..GK.........FQ..........VIMRQKDVILG.......DSMTPSQLVPY.....ELPLMWQLYP<br/>>gi|27668589|ref|XP_234503.1|:(9-91) similar to T-cell leukemia/lymphoma 1B, 5;<br/>-PHILTLRTHGIYEDEHHRLWVVLDLQ..A..ShlSF.........SN..........RLLIYLTVYLQqgvafplESTPPSPMNLN.....GLPRRWTLRT<br/>>gi|7305559|ref|NP_038802.1|:(8-102) T-cell leukemia/lymphoma 1B, 4 [Mus musculus] <br/>PPCFLVCTRDDIYEDEHGRQWVAAKVE..T..S..SH.........SPycskietcvtVHLWQMTTLFQ.......EPSPDSLKTFN.....FLPRTWRLES<br/>>gi|7305555|ref|NP_038803.1|:(9-102) T-cell leukemia/lymphoma 1B, 2 [Mus musculus]<br/>---------PGFYEDEHHRLWMVAKLE..T..C..SH.........SPycnkietcvtVHLWQMTRYPQ.......EPAPYNPMNYN.....FLPMTWRLAS</pre><br/><br/><br/>A2M format is derived from aligned FASTA format. It looks very similar, but it distinguishes between match/delete columns and insert columns. This information is used to specify how an alignment is transformed into an HMM. The match/delete columns use upper case letters for residues and the '-' symbol for deletions (gaps). The insert columns use lower case letters for the inserted residues. Gaps aligned to inserted residues are written as '.'<br/><br/><br/><b>The same alignment in A3M is here:</b><br/><br/><br/><pre>>d1a1x__ b.63.1.1 (-) p13-MTCP1 {Human (Homo sapiens)}<br/>PPDHLWVHQEGIYRDEYQRTWVAVVEEETSFLRARVQQIQVPLGDAARPSHLLTSQLPLMWQLYP<br/>>gi|6678257|ref|NP_033363.1|:(7-103) T-cell lymphoma breakpoint 1 [Mus musculus]<br/>HPNRLWIWEKHVYLDEFRRSWLPVVIKSNEKFQVILRQEDVTLGEAMSPSQLVPYELPLMWQLYP<br/>>gi|7305557|ref|NP_038800.1|:(8-103) T-cell leukemia/lymphoma 1B, 3 [Mus musculus]<br/>PPRFLVCTRDDIYEDENGRQWVVAKVETSRSpygsrietcITVHLQHMTTIPQEPTPQQPINNNSLPTMWRLES<br/>>gi|11415028|ref|NP_068801.1|:(2-106) T-cell lymphoma-1; T-cell lymphoma-1A [Homo sapiens]<br/>HPDRLWAWEKFVYLDEKQHAWLPLTIEikDRLQLRVLLRREDVVLGRPMTPTQIGPSLLPIMWQLYP<br/>>gi|7305561|ref|NP_038804.1|:(7-103) T-cell leukemia/lymphoma 1B, 5 [Mus musculus]<br/>----------GIYEDEHHRVWIAVNVETSHSSHgnrietcvtVHLQHMTTLPQEPTPQQPINNNSLPTMWRLES<br/>>gi|7305553|ref|NP_038801.1|:(5-103) T-cell leukemia/lymphoma 1B, 1 [Mus musculus]<br/>LPVYLVSVRLGIYEDEHHRVWIVANVETshSSHGNRRRTHVTVHLWKLIPQQVIPFNplnydFLPTTWKLES<br/>>gi|27668591|ref|XP_234504.1|:(7-103) similar to Chain A, Crystal Structure Of Murine Tcl1<br/>-PDRLWLWEKHVYLDEFRRSWLPIVIKSNGKFQVIMRQKDVILGDSMTPSQLVPYELPLMWQLYP<br/>>gi|27668589|ref|XP_234503.1|:(9-91) similar to T-cell leukemia/lymphoma 1B, 5;<br/>-PHILTLRTHGIYEDEHHRLWVVLDLQAShlSFSNRLLIYLTVYLQqgvafplESTPPSPMNLNGLPRRWTLRT<br/>>gi|7305559|ref|NP_038802.1|:(8-102) T-cell leukemia/lymphoma 1B, 4 [Mus musculus] <br/>PPCFLVCTRDDIYEDEHGRQWVAAKVETSSHSPycskietcvtVHLWQMTTLFQEPSPDSLKTFNFLPRTWRLES<br/>>gi|7305555|ref|NP_038803.1|:(9-102) T-cell leukemia/lymphoma 1B, 2 [Mus musculus]<br/>---------PGFYEDEHHRLWMVAKLETCSHSPycnkietcvtVHLWQMTRYPQEPAPYNPMNYNFLPMTWRLAS</pre><br/><br/><br/>The A3M format is a condensed version of A2M format. It is obtained by omitting all '.' symbols from A2M format. Hence residues emitted by Match states of the HMM are in upper case, residues emitted by Insert states are in lower case and deletions are written '-'. A3M-formatted alignments are not really human readable but they can be reformatted to other more readable formats like FASTA, A2M CLUSTAL and many others with the reformatting utility in our toolbox. A3M is not an official format. Likewise, the name A3M is our personal invention for this very practical and space-efficient format which we use to store the HHsearch alignments internally.",
    "Select HMM databases ": "Select the database(s) of template HMMs against which you want to compare the query HMM. Hold down the Control key to select more than one database.<br/><br/><b>Proteomes</b><br/>Here you can select databases of HMMs representing each of the proteins of an organism. Searching through the proteome of an organism using HHsearch (or HHblits) can be useful when you suspect a diverged ortholog or homolog of your query protein to hide in an organism's genome (and you cannot find it using BLAST or PSI-BLAST).<br/><br/><a href = http://www.rcsb.org/pdb/home/home.do><b>pdb70</b></a><br/><br/>A search against the our pdb70 database is the first step when trying to predict a structure with HHpred. It is based on the protein data bank (PDB) which contains all publicly available 3D structures of proteins. For each protein chain in the PDB we build a multiple alignment with iterated PSI-BLAST searches and transform these alignments into HMMs. HHpred can compare the query HMM to the database of HMMs based on PDB chains and generate query-template alignments or it can build structural models for the query sequence. We use a filtered version with a maximum sequence identity of 70% [J. Söding, unpublished]. We check the PDB for new structures every night and update our database whenever new files are found.<br/><font size = 1><a href = https://www.ncbi.nlm.nih.gov/pubmed/14681399?dopt=Abstract>Bourne PE, Addess KJ, Bluhm WF, Chen L, Deshpande N, Feng Z, Fleri W, Green R, Merino-Ott JC, Townsend-Merino W, Weissig H, Westbrook J, Berman HM. The distribution and query systems of the RCSB Protein Data Bank. NAR 2004, 32:D223.</a></font><br/><br/><a href = http://scop.berkeley.edu/><b>scop70</b></a><br/><br/>The scop70 alignment database is built from sequences in SCOP in a similar way as pdb70. SCOP is a hand-curated database with protein domains of known structure. It is hierarchically organized into families, superfamilies, folds, and classes to reflect evolutionary relationships. Only members of the same superfamily can be savely assumed to be homologous. Members of the same fold but different superfamily are assumed not to be homologous, although many exceptions to this rule are now known. HHpred reports the SCOP code of the family the domain belongs to. For example, a.118.8.1 stands for domain class a (all alpha helical), fold 118 (alpha-alpha superhelices), superfamily 8 (TPR-like), family 1 (Tetratricopeptide repeats (TPR)). We use a filtered version of the SCOP sequence set from the <a href = http://scop.berkeley.edu/astral/><b>ASTRAL server</b></a>, e.g. scop70_1.69 is version 1.69 of SCOP filtered to 70% maximum sequence identity. Our scop70 alignment database is similar to the <a href = http://supfam.mrc-lmb.cam.ac.uk/SUPERFAMILY/><b>SUPERFAMILY</b></a> database. SCOP is updated irregularly, approximately every twelve months.<br/><font size = 1><a href = https://www.ncbi.nlm.nih.gov/pubmed/10592240?dopt=Abstract>Lo Conte L, Ailey B, Hubbard TJ, Brenner SE, Murzin AG, Chothia C. SCOP: a structural classification of proteins database. NAR 2000, 28:257</a></font><br/><br/><a href = https://www.ncbi.nlm.nih.gov/Structure/cdd/cdd.shtml><b>CDD</b></a><br/><br/>The Conserved Domain Database (CDD) at the <a href = https://www.ncbi.nlm.nih.gov/><b>NCBI</b></a> is a composite database combining the alignments from Pfam, SMART, and COG/KOG, as well as containing alignments of conserved domains curated at the NCBI. The source of the alignments as shown in the HHpred results is part of the identifiers. Identifiers beginning with cd or LOAD indicate that an alignment was curated at the NCBI. The CDD database contained 18640 alignments on November 17, 2004. It is updated irregularly every 6 to 12 months. HHpred checks once a week for updates.<br/><font size = 1><a href = https://www.ncbi.nlm.nih.gov/pubmed/11752315?dopt=Abstract>Marchler-Bauer A, Panchenko AR, Shoemaker BA, Thiessen PA, Geer LY, Bryant SH. CDD: a database of conserved domain alignments with links to domain three-dimensional structure. NAR 2002, 30:281.</a></font><br/><br/><a href = http://www.ebi.ac.uk/interpro/><b>InterPro</b></a><br/><br/>Our version of InterPro contains HMMs from Pfam, SMART, PANTHER, TIGRFAMs, PIRSF, SUPERFAMILY, and Gene3D/CATH (see below). It encodes full-length protein families, domain families, and motifs in which identifiable features from known proteins can be applied to unknown protein sequences. In the version offered through HHpred, the PRINTS and PROSITE sequene motif databases are excluded, as are explicit transmembrane helix prediction by TMHMM and signal peptide prediction by SignalP. InterPro_12.0 contains already more than 31000 HMMs, therefore a search will take a few minutes. (InterPro offers only HMMs and not the underlying alignments. Therefore, HHpred does not have access to alignments for PANTHER, TIGRFAMs, PIRSF, SUPERFAMILY, and CATH/Gene3D. Hence, when using the option \"Merge query and template alignments\", only the single consensus sequences instead of full template alignments can be merged with your query alignment.)<br/><font size = 1><a href = https://www.ncbi.nlm.nih.gov/pubmed/15608177?dopt=Abstract>Mulder NJ, Apweiler R, Attwood TK, Bairoch A, Bateman A, Binns D, Bradley P, Bork P, Bucher P, Cerutti L, Copley R, Courcelle E, Das U, Durbin R, Fleischmann W, Gough J, Haft D, Harte N, Hulo N, Kahn D, Kanapin A, Krestyaninova M, Lonsdale D, Lopez R, Letunic I, Madera M, Maslen J, McDowall J, Mitchell A, Nikolskaya AN, Orchard S, Pagni M, Ponting CP, Quevillon E, Selengut J, Sigrist CJ, Silventoinen V, Studholme DJ, Vaughan R, Wu CH. InterPro, progress and status in 2005. Nucleic Acids Res. 2005 Jan 1;33(Database issue):D201-5.</a></font><br/><br/><a href = http://pfam.xfam.org/><b>PfamA</b></a><br/><br/>PfamA is a large collection of curated multiple sequence alignments covering many common protein domains and families. Many PfamA families are well documented with links to literature, and about a third of them have at least one member with known structure. 74% of protein sequences have at least one match to PfamA_16.0 with its 7503 alignments (August 2004). PfamA gets updated approximately every two to six months. We automatically download two versions of the PfamA database: one directly from <a href = ftp://ftp.ebi.ac.uk/pub/databases/Pfam/>ftp://ftp.ebi.ac.uk/pub/databases/Pfam/</a> and one as a part of the CDD database from NCBI. The version directly from Pfam has a version number in its name whereas the version from NCBI carries the date when it was downloaded. The original Pfam is generally more up to date and contains more sequences in each alignment than the version from NCBI (for some unkown reason).<br/><font size = 1><a href = https://www.ncbi.nlm.nih.gov/pubmed/9399864?dopt=Abstract>Sonnhammer EL, Eddy SR, Birney E, Bateman A, Durbin R. Pfam: multiple sequence alignments and HMM-profiles of protein domains. NAR 1998 26:320.</a></font><br/><br/><a href = http://smart.embl-heidelberg.de/><b>SMART</b></a><br/><br/>SMART is a database of curated alignments of genetically mobile domains found in signalling, extracellular and chromatin-associated proteins, predominantly in metazoans. These domains are extensively annotated with respect to phyletic distributions, functional class, tertiary structures and functionally important residues. In January 2004 SMART contained 685 alignments. We download the SMART database as a part of the CDD database from NCBI and check for updates every week. Note that the version at NCBI may get updated only be a few weeks after a new SMART version has been released.<br/><font size = 1><a href = https://www.ncbi.nlm.nih.gov/pubmed/9847187?dopt=Abstract>Ponting CP, Schultz J, Milpetz F, Bork P. SMART: identification and annotation of domains from signalling and extracellular protein sequences. NAR 1999, 27:229.</a></font><br/><br/><a href = http://www.pantherdb.org/about.jsp><b>PANTHER</b></a><br/><br/>PANTHER is a large collection of protein families that have been subdivided into functionally related subfamilies, using human expertise. These subfamilies model the divergence of specific functions within protein families, allowing more accurate association with function (human-curated molecular function and biological process classifications and pathway diagrams), as well as inference of amino acids important for functional specificity. Hidden Markov models (HMMs) are built for each family and subfamily for classifying additional protein sequences. Only family alignments are included in HHpred.<br/><font size = 1><a href = https://www.ncbi.nlm.nih.gov/pubmed/15608197?dopt=Abstract>Mi H, Lazareva-Ulitsky B, Loo R, Kejariwal A, Vandergriff J, Rabkin S, Guo N, Muruganujan A, Doremieux O, Campbell MJ, Kitano H, Thomas PD. (2005) The PANTHER database of protein families, subfamilies, functions and pathways. NAR 33:D284-288.</a></font><br/><br/><a href = http://www.jcvi.org/cgi-bin/tigrfams/index.cgi><b>TIGRFAMs</b></a><br/><br/>TIGRFAMs is a collection of protein families, featuring curated multiple sequence alignments, hidden Markov models (HMMs) and annotation, which provides a tool for identifying functionally related proteins based on sequence homology. Those entries which are \"equivalogs\" group homologous proteins which are conserved with respect to function.<br/><font size = 1><a href = https://www.ncbi.nlm.nih.gov/pubmed/12520025?dopt=Abstract>Haft DH, Selengut JD, White O. (2003) The TIGRFAMs database of protein families. NAR 31:371-373.</a></font><br/><br/><a href = http://pir.georgetown.edu/pirwww/dbinfo/iproclass.shtml><b>PIRSF</b></a><br/><br/>The PIRSF protein classification system is a network with multiple levels of sequence diversity from superfamilies to subfamilies that reflects the evolutionary relationship of full-length proteins and domains. The primary PIRSF classification unit is the homeomorphic family, whose members are both homologous (evolved from a common ancestor) and homeomorphic (sharing full-length sequence similarity and a common domain architecture).<br/><font size = 1><a href = https://www.ncbi.nlm.nih.gov/pubmed/14681371?dopt=Abstract>Wu CH, Nikolskaya A, Huang H, Yeh LS, Natale DA, Vinayaka CR, Hu ZZ, Mazumder R, Kumar S, Kourtesis P, Ledley RS, Suzek BE, Arminski L, Chen Y, Zhang J, Cardenas JL, Chung S, Castro-Alvear J, Dinkov G, Barker WC. (2004) PIRSF: family classification system at the Protein Information Resource. NAR 32:D112-114.</a></font><br/><br/><a href = http://supfam.mrc-lmb.cam.ac.uk/SUPERFAMILY/><b>SUPERFAMILY</b></a><br/><br/>SUPERFAMILY is a library of profile hidden Markov models that represent all proteins of known structure. The library is based on the SCOP classification of proteins: each model corresponds to a SCOP domain and aims to represent the entire SCOP superfamily that the domain belongs to. SUPERFAMILY has been used to carry out structural assignments to all completely sequenced genomes.<br/><font size = 1><a href = https://www.ncbi.nlm.nih.gov/pubmed/11697912?dopt=Abstract>Gough J, Karplus K, Hughey R, Chothia C. (2001) Assignment of Homology to Genome Sequences using a Library of Hidden Markov Models that Represent all Proteins of Known Structure. JMB 313:903-919.</a></font><br/><br/><a href = http://www.cathdb.info/Gene3D/><b>CATH/Gene3D</b></a><br/><br/>Gene3D is an alignment database built from sequences of the CATH database. CATH is a database for structural classification of protein domains, similar to SCOP. Gene3D describes protein families and domain architectures in complete genomes. Protein families are formed using a Markov clustering algorithm, followed by multi-linkage clustering according to sequence identity. Mapping of predicted structure and sequence domains is undertaken using hidden Markov models libraries representing CATH and Pfam domains. Functional annotation is provided to proteins from multiple resources. Functional prediction and analysis of domain architectures is available from the Gene3D website.<br/><font size = 1><a href = https://www.ncbi.nlm.nih.gov/pubmed/15608188?dopt=Abstract>Pearl F, Todd A, Sillitoe I, Dibley M, Redfern O, Lewis T, Bennett C, Marsden R, Grant A, Lee D, Akpor A, Maibaum M, Harrison A, Dallman T, Reeves G, Diboun I, Addou S, Lise S, Johnston C, Sillero A, Thornton J, Orengo C. (2005) The CATH Domain Structure Database and related resources Gene3D and DHS provide comprehensive domain family information for genome analysis. NAR 33:D247-D251</a></font><br/><br/><a href = https://www.ncbi.nlm.nih.gov/COG/><b>COG/KOG</b></a><br/><br/>COG/KOG (Clusters of Orthologous Groups of proteins) is a database of alignments of homologous, conserved domains. For the 2003 update, sequences from over 60 completeley sequences prokaryotes (COGs) and 7 completely sequenced euKaryotes (KOGs) were automatically divided into domains. The domain sequences are clustered in such a way that the alignments consist mainly of orthologous proteins from different lineages and paralogous proteins from the same lineage. The COG database is therefore particularly useful for functional annotation. But beware: HHpred is very sensitive in detecting distant homologous relationships. Therefore, a significant match with a COG only implies homology, but not necessarily a similar function.<br/><font size = 1><a href = https://www.ncbi.nlm.nih.gov/pubmed/12969510?dopt=Abstract>Tatusov RL, Fedorova ND, Jackson JD, Jacobs AR, Kiryutin B, Koonin EV, Krylov DM, Mazumder R, Mekhedov SL, Nikolskaya AN, Rao BS, Smirnov S, Sverdlov AV, Vasudevan S, Wolf YI, Yin JJ, Natale DA. The COG database: an updated version includes eukaryotes. BMC Bioinformatics. 2003, 4:41.</a></font><br/><br/><a href = http://pfam.xfam.org/><b>PfamB</b></a><br/><br/>Please note: PfamB contains over 14000 unannotated, mainly small and insignificant alignments. To avoid wasting time, we recommend to search PfamB only for doing intermediate HMM searching, when you can't find any significant match to an alignment from the other databases. PfamB is derived from the ProDom database by leaving out all parts of alignments which are already covered by PfamA. ProDom in turn is built by clustering homologous sequence segments into alignments through an iterative scheme employing PSI-BLAST. PfamB alignments are not annotated, many contain only a few sequences or are quite short, and most represent only parts of domains. To keep the number of HMMs in PfamB reasonable, we include only alignments with at least three sequences of a certain minimum diversity.",
    "MSA generation method": "This option sets the Multiple Sequence Alignment Generation method used by HHpred. Detailed information on the methods can be found here:<br/>HHBlits : <a href = http://wwwuser.gwdg.de/~compbiol/pdf/Preprint-2011-HHblits-inkl-Suppl2.pdf>Download pdf file</a><br/>PSI-Blast : <a href = https://www.ncbi.nlm.nih.gov/pubmed/9254694>View Article</a>",
    "Max. MSA generation iterations": " This specifies the number of iterations performed by the MSA generation algorithm, to build up an alignment from the input sequence or input alignment. The iterations stop in any case when no further sequences are found. If you want to use exactly your input alignment choose 0 here. This will ensure that only predicted secondary structure is added to your input alignment.",
    "Score secondary structure": "If you choose \"yes\", the secondary structure of the query will be predicted with <a href = https://www.ncbi.nlm.nih.gov/pubmed/10493868?dopt=Abstract>PSIPRED</a> and compared to the actual secondary structure of the database sequences. If the actual structure of the database sequence is unknown, the PSIPRED-predicted secondary structure will be used. To force HHpred to use predicted instead of actual secondary structure, choose \"predicted vs predicted only\". Scoring the secondary structure similarity improves sensitivity and also alignment quality significantly. Since the weight of the secondary structure score is small (15% of the amino acid similarity score), the risk for high-scoring non-homologous matches is still small.",
    "Alignment mode ": "In local mode, HHpred will use <u>local</u> alignments to score your query sequence with the database sequences. In global mode it uses global alignments with no end gap penalties (sometimes called 'global-local' alignments), i.e. the query-template alignments extend up to the beginning and end of the query or template sequence. The use of local mode is strongly recommended if you want to know which are your most likely remote homologs. If you have already identified your template with a previous run in local mode, you can use the global mode to get global alignments with an already identified template.<br/><br/>HHpred was shown to be significantly more sensitive to detect remote homologs in local than in global mode. The reason is that most remote homologs are really only homologous in a common core that define their superfamily. This core has been adorned by other secondary structure elements in different ways in different families. Also, the statistical evaluation is more error-prone for the global alignment mode, since the negatives score distribution for global alignments does not follow an extreme-value distribution.<br/><br/>You may generate global alignments and at the same time score alignments with the more sensitive local Viterbi alignment by selecting the 'local' option in combination with the option 'Realign with MAC algorithm' and setting the 'MAC realignment threshold' to 0.0 (hidden under 'More options').",
    "Realign with MAC algorithm ": "After the search with the standard Viterbi algorithm, this option will cause HHpred to realign all query-template alignments with the more accurate Maximum ACcuracy alignment algorithm.<br/>Since for technical reasons the scores, E-values, and probabilities from the Viterbi alignment and NOT the MAC alignment are displayed and used for ranking, the MAC realignment option may lead to high-scoring alignments of length 1. This happens when the MAC algorithm cannot find any acceptable alignment path, which is an indication for a non-homologous relationship.<br/>When the MAC realignment option is activated, the 'global' and 'local' options now refer to both the Viterbi search stage as well as the MAC realignment stage. With -global (-local), the posterior probability matrix will be calculated for global (local) alignment.",
    "E-value threshold for MSA generation": "All sequence matches (\"hits\") with an E-value better than this threshold are included in the alignment for the next MSA Generation iteration or, in the last iteration, for building the HMM for HHsearch.",
    "Min. coverage of MSA hits": "This parameter controls what percentage of the query residues needs to be covered by the hits of the MSA algorithm in order for the hit to be included into the alignment for the next iteration. More precisely, the coverage of a sequence found by the MSA algorithm is defined as the number of residues aligned with the query divided by the length of the query.",
    "Min. sequence identity of MSA hits with query": "This parameter controls what percentage of residues from a hit of the MSA algorithm needs to be identical to their aligned query residues in order for the hit to be included in the MSA alignment. More precisely, the sequence identity of a hit by the MSA algorithm is defined as the number of residues identical with aligned query residues divided by the length of the MSA hit.",
    "MAC realignment threshold ": "Only active when the 'Realign with MAC algorithm' and the 'local' option is active. This parameter controls the MAC alignment algorithm's greediness. More precisely, the MAC algorithm finds the alignment that maximizes the sum of posterior probabilities minus this threshold for each aligned pair. Global alignments are generated with 0, whereas 0.5 will produce quite conservative local alignments with average probabilities for each aligned pair of residues above 50%. Default value is 0.3, which produces alignments of roughly the same length as the Viterbi algorithm.",
    "Compositional bias correction": "When two proteins have a similar, biased amino acid distribution, such as alpha-helical membrane proteins, they may look similar because the average score between any random pair of columns may be slightly positive. Activating this option will prevent protein pairs from looking similar just because their average amino acid distributions are similarly biased.",
    //"Show representative sequences per HMM"
    //"Width of alignments (in columns)"
    //"Minimum probability in hit list"
    //"Max. number of hits in hit list"
    "Show hits as bar graph": "With this option you can display a bar graph indicating the region in your query protein that is covered by the matches. The significance of the hits is color-coded, with most significant hits in red (>95% probability) and orange (>85%), then yellow and green (>70%), and finally cyan (>50%) and blue (>40%) for potential true positives hits.",
    //"Match state assignment"
    //HMMER
    "Input": "Enter protein sequence alignment in FASTA/CLUSTAL format.",
    "Input format": "Accepted formats are listed in the dropdown menu.",
    "Database ": "The selected databases are used by HMMAccel to perform the search. You may press CTRL to select multiple databases.",
    "Convert input to uppercase": "Convert all residues in the input alignment to upper case.",
    "Compute consensus sequence": "Create a consensus sequence of the alignment and use that as seed for psi-blast (alternative: use the first sequence in the alignment).",
    "Extract sequences up to E-value": "Up to which E-value will you want sequences to be extracted for? (your filtered database will consist of these sequences).",
    "HMM build options": "<pre>   -h     : help; print brief help on version and usage<br/>   -n &lts&gt : name; name this (first) HMM &lts&gt<br/>   -o &ltf&gt : re-save annotated alignment to &ltf&gt<br/>   -A     : append; append this HMM to <br/>   -F     : force; allow overwriting of <br/><br/>  Alternative search algorithm styles: (default: hmmls domain alignment)<br/>   -f     : multi-hit local (hmmfs style)<br/>   -g     : global alignment (hmms style, Needleman/Wunsch)<br/>   -s     : local alignment (hmmsw style, Smith/Waterman)<br/><br/>  Alternative model construction strategies: (default: MAP)<br/>   --fast        : Krogh/Haussler fast heuristic construction (see --gapmax)<br/>   --hand        : manual construction (requires annotated alignment)<br/><br/>  Expert customization of parameters and priors:<br/>   --null  &ltf&gt   : read null (random sequence) model from &ltf&gt<br/>   --pam   &ltf&gt   : heuristic PAM-based prior, using BLAST PAM matrix in &ltf&gt<br/>   --prior &ltf&gt   : read Dirichlet prior parameters from &ltf&gt<br/><br/>  Alternative sequence weighting strategies: (default: GSC weights)<br/>   --wblosum     : Henikoff simple filter weights (see --idlevel)<br/>   --wgsc        : Gerstein/Sonnhammer/Chothia tree weights (default)<br/>   --wme         : maximum entropy (ME)<br/>   --wpb         : Henikoff position-based weights<br/>   --wvoronoi    : Sibbald/Argos Voronoi weights<br/>   --wnone       : don't do any weighting<br/>   --noeff       : don't use effective sequence number; just use nseq<br/>   --pbswitch &ltn&gt: set switch from GSC to position-based wgts at > n seqs<br/><br/>  Forcing an alphabet: (normally autodetected)<br/>   --amino   : override autodetection, assert that seqs are protein<br/>   --nucleic : override autodetection, assert that seqs are DNA/RNA<br/><br/>  Other expert options:<br/>   --archpri &ltx&gt : set architecture size prior to &ltx&gt {0.85} [0..1]<br/>   --binary      : save the model in binary format, not ASCII text<br/>   --cfile &ltf&gt   : save count vectors to &ltf&gt<br/>   --gapmax &ltx&gt  : max fraction of gaps in mat column {0.50} [0..1]<br/>   --idlevel &ltx&gt : set frac. id level used by eff. nseq and --wblosum {0.62}<br/>   --informat &lts&gt: input alignment is in format &lts&gt, not Stockholm<br/>   --pamwgt &ltx&gt  : set weight on PAM-based prior to &ltx&gt {20.}[>=0]<br/>   --swentry &ltx&gt : set S/W aggregate entry prob. to &ltx&gt {0.5}<br/>   --swexit &ltx&gt  : set S/W aggregate exit prob. to &ltx&gt  {0.5}<br/>   --verbose     : print boring information</pre>",
    "HMM calibrate options": "<pre>  -h             : print short usage and version info, then exit<br/>  --cpu &ltn&gt      : run &ltn&gt threads in parallel (if threaded)<br/>  --fixed &ltn&gt    : fix random sequence length at &ltn&gt<br/>  --histfile &ltf&gt : save histogram(s) to file &ltf&gt<br/>  --mean &ltx&gt     : set random seq length mean at &ltx&gt [350]<br/>  --num &ltn&gt      : set number of sampled seqs to &ltn&gt [5000]<br/>  --pvm          : run on a Parallel Virtual Machine (PVM)<br/>  --sd &ltx&gt       : set random seq length std. dev to &ltx&gt [350]<br/>  --seed &ltn&gt     : set random seed to &ltn&gt [time()]</pre>",
    "HMM search options": "<pre>   -h        : help; print brief help on version and usage<br/>   -A &ltn&gt    : sets alignment output limit to &ltn&gt best domain alignments<br/>   -E &ltx&gt    : sets E value cutoff (globE) to <= x<br/>   -T &ltx&gt    : sets T bit threshold (globT) to >= x<br/>   -Z &ltn&gt    : sets Z (# seqs) for E-value calculation<br/><br/>   --compat  : make best effort to use last version's output style<br/>   --cpu &ltn&gt : run &ltn&gt threads in parallel (if threaded)<br/>   --cut_ga  : use Pfam GA gathering threshold cutoffs<br/>   --cut_nc  : use Pfam NC noise threshold cutoffs<br/>   --cut_tc  : use Pfam TC trusted threshold cutoffs<br/>   --domE &ltx&gt: sets domain Eval cutoff (2nd threshold) to <= x<br/>   --domT &ltx&gt: sets domain T bit thresh (2nd threshold) to >= x<br/>   --forward : use the full Forward() algorithm instead of Viterbi<br/>   --informat &lts&gt: sequence file is in format &lts&gt, not FASTA<br/>   --null2   : turn OFF the post hoc second null model<br/>   --pvm     : run on a Parallel Virtual Machine (PVM)<br/>   --xnu     : turn ON XNU filtering of target protein sequences</pre>",
    "HMM align options": "<pre>   -h     : help; print brief help on version and usage<br/>   -m     : only print symbols aligned to match states<br/>   -o &ltf&gt : save alignment in file &ltf&gt in SELEX format<br/>   -q     : quiet - suppress verbose banner<br/><br/>   --informat &lts&gt: sequence file is in format &lts&gt, not FASTA<br/>   --mapali &ltf&gt  : include alignment in file &ltf&gt using map in HMM<br/>   --withali &ltf&gt : include alignment to (fixed) alignment in file &ltf&gt</pre>",
    //PatternSearch
    "Input format ": "<pre>Prosite grammar:<br/><br/>Example:<br/>G[VLIM](2)x(3,10)W{A}Dxx[ST]<br/><br/>A prosite pattern is described using the following conventions:<br/><br/>-> The standard IUPAC one-letter codes for the amino acids are used.<br/>-> The symbol 'x' is used for a position where any amino acid is accepted. <br/>-> Ambiguities are indicated by listing the acceptable amino acids for a given position, <br/>   between square parentheses '[ ]'. For example: [ALT] stands for Ala or Leu or Thr. <br/>-> Ambiguities are also indicated by listing between a pair of curly brackets '{ }' the <br/>   amino acids that are not accepted at a given position. For example: {AM} stands for any <br/>   amino acid except Ala and Met. <br/>-> Repetition of an element of the pattern can be indicated by following that element with <br/>   a numerical value or a numerical range between parenthesis. <br/>   Examples: x(3) corresponds to xxx, x(2,4) corresponds to xx or xxx or xxxx. <br/>-> When a pattern is restricted to either the N- or C-terminal of a sequence, <br/>   that pattern either starts with a '<' symbol or respectively ends with a '>' symbol. <br/>   In some rare cases (e.g. PS00267 or PS00539), '>' can also occur inside square brackets for the <br/>   C-terminal element. 'F[GSTV]PRL[G>]' means that either 'F[GSTV]PRLG' or 'F[GSTV]PRL>' are considered.<br/><br/>(<a href = http://prosite.expasy.org/#convent35>Prosite user manual</a>)<br/><br/>Regular expression:<br/><br/>Example:<br/>G[VLIM]{2}.{3,10}W[^A]D..[ST]<br/><br/>-> A character class, defined by square parentheses '[ ]', matches a single amino acid out <br/>   of all the possibilities offered by the character class.<br/>   For example: [ALT] stands for Ala or Leu or Thr<br/>-> A caret '^' immediately after the opening '[' negates the character class.<br/>   For example: [^ALT] stands for all amino acid except Ala and Leu and Thr<br/>-> A dot '.' matches any amino acid.<br/>-> A pair of curly brackets '{}' repeats the previous item exactly n times '{n}' <br/>   or between n and m times '{n,m}'.<br/>   For example: A{3} correspond to AAA, .{2,4} corresponds to .. or ... or ....<br/>-> A caret '^' match the start of the sequence.<br/>-> A dollar '$' match the end of the sequence.</pre>",
    "Select database": "The selected databases are used in the pattern search. You may press CTRL to select multiple databases.",
    "Sequence count cutoff": "You can select the max. count of sequences that will be fetched by your Pattern Search. Default value is 1000 sequences. You can increase the cutoff for degenerate Prosite Patterns, however the Bioinformatics Toolkit may not be able to display all results due to to size constraints of the output file.",
    "FASTA sequences": " Here you can enter your own sequences in which the pattern will be searched.",
    //Clustal Omega
    "Input format  ": "Enter multiple sequences in fasta format. A sequence in FASTA format begins with a single-line description, followed by lines of sequence data. The description line is distinguished from the sequence data by a greater-than (\">\") symbol in the first column. It is recommended that all lines of text be shorter than 80 characters in length. An example sequence in FASTA format is:<br/><br/><br/><pre>>gi|532319|pir|TVFV2E|TVFV2E envelope protein<br/>ELRLRYCAPAGFALLKCNDADYDGFKTNCSNVSVVHCTNLMNTTVTTGLLLNGSYSENRT<br/>QIWQKHRTSNDSALILLNKHYNLTVTCKRPGNKTVLPVTIMAGLVFHSQKYNLRLRQAWC<br/>HFPSNWKGAWKEVKEEIVNLPKERYRGTNDPKRIFFQRQWGDPETANLWFNCHGEFFYCK<br/>MDWFLNYLNNLTVDADHNECKNTSGTKSGNKRAPGPCVQRTYVACHIRSVIIWLETISKK<br/>TYAPPREGHLECTSTVTGMTVELNYIPKNRTNVTLSPQIESIWAAELDRYKLVEITPIGF<br/>APTEVRRYTGGHERQKRVPFVXXXXXXXXXXXXXXXXXXXXXXVQSQHLLAGILQQQKNL<br/>LAAVEAQQQMLKLTIWGVK</pre><br/><br/>Sequences are expected to be represented in the standard IUB/IUPAC amino acid and nucleic acid codes, with these exceptions: lower-case letters are accepted and are mapped into upper-case; a single hyphen or dash can be used to represent a gap of indeterminate length; and in amino acid sequences, U and * are acceptable letters (see below). Before submitting a request, any numerical digits in the query sequence should either be removed or replaced by appropriate letter codes (e.g., N for unknown nucleic acid residue or X for unknown amino acid residue).<br/>The nucleic acid codes supported are:<br/><br/><br/><pre>   A --> adenosine           M --> A C (amino)<br/>   C --> cytidine            S --> G C (strong)<br/>   G --> guanine             W --> A T (weak)<br/>   T --> thymidine           B --> G T C<br/>   U --> uridine             D --> G A T<br/>   R --> G A (purine)        H --> A C T<br/>   Y --> T C (pyrimidine)    V --> G C A<br/>   K --> G T (keto)          N --> A G C T (any)<br/>                             -  gap of indeterminate length</pre><br/><br/>The accepted amino acid codes are:<br/><br/><br/>    <pre>    A  alanine                         P  proline<br/>    B  aspartate or asparagine         Q  glutamine<br/>    C  cystine                         R  arginine<br/>    D  aspartate                       S  serine<br/>    E  glutamate                       T  threonine<br/>    F  phenylalanine                   U  selenocysteine<br/>    G  glycine                         V  valine<br/>    H  histidine                       W  tryptophan<br/>    I  isoleucine                      Y  tyrosine<br/>    K  lysine                          Z  glutamate or glutamine<br/>    L  leucine                         X  any<br/>    M  methionine                      *  translation stop<br/>    N  asparagine                      -  gap of indeterminate length</pre>",
    "Clustal Omega: Other advanced options": "Use this textfield to specify advanced options via commandline. After removing some options not useful in this toolkit web environment, remaining options supported by Clustal Omega are (for ClustalW options see next section):<br/><br/><pre>Sequence Input:<br/>  --dealign                 Dealign input sequences<br/>  --is-profile              disable check if profile, force profile (default no)<br/><br/>Clustering:<br/>  --pileup                  Sequentially align sequences<br/>  --full                    Use full distance matrix for guide-tree calculation (might be slow; mBed is default)<br/>  --full-iter               Use full distance matrix for guide-tree calculation during iteration (might be slowish; mBed is default)<br/>  --cluster-size=&ltn&gt        soft maximum of sequences in sub-clusters<br/>  --trans=&ltn&gt               use transitivity (default: 0)<br/>  --use-kimura              use Kimura distance correction for aligned sequences (default no)<br/>  --percent-id              convert distances into percent identities (default no)<br/>                            <br/>Alignment Output:<br/>  --residuenumber, --resno  print residue numbers (default no)<br/>  --wrap=&ltn&gt                number of residues before line-wrap in output<br/>  --output-order={input-order,tree-order} MSA output order like in input/guide-tree<br/>                            <br/>Iteration:<br/>  --iterations, --iter=&ltn&gt  Number of (combined guide-tree/HMM) iterations<br/>  --max-guidetree-iterations=&ltn&gt Maximum number of guidetree iterations<br/>  --max-hmm-iterations=&ltn&gt  Maximum number of HMM iterations<br/>                            <br/>Limits (will exit early, if exceeded):<br/>  --maxnumseq=&ltn&gt           Maximum allowed number of sequences<br/>  --maxseqlen=&ltl&gt           Maximum allowed sequence length<br/>                            <br/>Miscellaneous:<br/>  -h, --help                Print this help and exit<br/>  -v, --verbose             Verbose output (increases if given multiple times)<br/>  --version                 Print version information and exit<br/>  --long-version            Print long version information and exit</pre>",
    "ClustalW: Other advanced options": "Use this text field to specify advanced options via commandline. Options supported by ClustalW are (for Clustal Omega options see previous section):<br/><br/><pre>***General settings:****<br/>-INTERACTIVE :read command line, then enter normal interactive menus<br/>-QUICKTREE   :use FAST algorithm for the alignment guide tree<br/>-TYPE=       :PROTEIN or DNA sequences<br/>-NEGATIVE    :protein alignment with negative values in matrix<br/>-OUTFILE=    :sequence alignment file name<br/>-OUTPUT=     :GCG, GDE, PHYLIP, PIR or NEXUS<br/>-OUTORDER=   :INPUT or ALIGNED<br/>-CASE        :LOWER or UPPER (for GDE output only)<br/>-SEQNOS=     :OFF or ON (for Clustal output only)<br/> <br/> <br/>***Fast Pairwise Alignments:***<br/>-KTUPLE=n    :word size<br/>-TOPDIAGS=n  :number of best diags.<br/>-WINDOW=n    :window around best diags.<br/>-PAIRGAP=n   :gap penalty<br/>-SCORE       :PERCENT or ABSOLUTE<br/> <br/> <br/>***Slow Pairwise Alignments:***<br/>-PWMATRIX=    :Protein weight matrix=BLOSUM, PAM, GONNET, ID or filename<br/>-PWDNAMATRIX= :DNA weight matrix=IUB, CLUSTALW or filename<br/>-PWGAPOPEN=f  :gap opening penalty<br/>-PWGAPEXT=f   :gap opening penalty<br/> <br/> <br/>***Multiple Alignments:***<br/>-NEWTREE=      :file for new guide tree<br/>-USETREE=      :file for old guide tree<br/>-MATRIX=       :Protein weight matrix=BLOSUM, PAM, GONNET, ID or filename<br/>-DNAMATRIX=    :DNA weight matrix=IUB, CLUSTALW or filename<br/>-GAPOPEN=f     :gap opening penalty<br/>-GAPEXT=f      :gap extension penalty<br/>-ENDGAPS       :no end gap separation pen.<br/>-GAPDIST=n     :gap separation pen. range<br/>-NOPGAP        :residue-specific gaps off<br/>-NOHGAP        :hydrophilic gaps off<br/>-HGAPRESIDUES= :list hydrophilic res.<br/>-MAXDIV=n      :% ident. for delay<br/>-TYPE=         :PROTEIN or DNA<br/>-TRANSWEIGHT=f :transitions weighting<br/> <br/> <br/>***Profile Alignments:***<br/>-PROFILE      :Merge two alignments by profile alignment<br/>-NEWTREE1=    :file for new guide tree for profile1<br/>-NEWTREE2=    :file for new guide tree for profile2<br/>-USETREE1=    :file for old guide tree for profile1<br/>-USETREE2=    :file for old guide tree for profile2<br/> <br/> <br/>***Sequence to Profile Alignments:***<br/>-SEQUENCES   :Sequentially add profile2 sequences to profile1 alignment<br/>-NEWTREE=    :file for new guide tree<br/>-USETREE=    :file for old guide tree<br/> <br/> <br/>***Structure Alignments:***<br/>-NOSECSTR1     :do not use secondary structure-gap penalty mask for profile 1<br/>-NOSECSTR2     :do not use secondary structure-gap penalty mask for profile 2<br/>-SECSTROUT=STRUCTURE or MASK or BOTH or NONE   :output in alignment file<br/>-HELIXGAP=n    :gap penalty for helix core residues<br/>-STRANDGAP=n   :gap penalty for strand core residues<br/>-LOOPGAP=n     :gap penalty for loop regions<br/>-TERMINALGAP=n :gap penalty for structure termini<br/>-HELIXENDIN=n  :number of residues inside helix to be treated as terminal<br/>-HELIXENDOUT=n :number of residues outside helix to be treated as terminal<br/>-STRANDENDIN=n :number of residues inside strand to be treated as terminal<br/>-STRANDENDOUT=n:number of residues outside strand to be treated as terminal<br/> <br/> <br/>***Trees:***<br/>-OUTPUTTREE=nj OR phylip OR dist OR nexus<br/>-SEED=n        :seed number for bootstraps.<br/>-KIMURA        :use Kimura's correction.<br/>-TOSSGAPS      :ignore positions with gaps.<br/>-BOOTLABELS=node OR branch :position of bootstrap values in tree display</pre>",
    //Kalign
    "Input format   ": "Enter multiple sequences in fasta format. A sequence in FASTA format begins with a single-line description, followed by lines of sequence data. The description line is distinguished from the sequence data by a greater-than (\">\") symbol in the first column. It is recommended that all lines of text be shorter than 80 characters in length. An example sequence in FASTA format is:<br/><br/><br/><pre>>gi|532319|pir|TVFV2E|TVFV2E envelope protein<br/>ELRLRYCAPAGFALLKCNDADYDGFKTNCSNVSVVHCTNLMNTTVTTGLLLNGSYSENRT<br/>QIWQKHRTSNDSALILLNKHYNLTVTCKRPGNKTVLPVTIMAGLVFHSQKYNLRLRQAWC<br/>HFPSNWKGAWKEVKEEIVNLPKERYRGTNDPKRIFFQRQWGDPETANLWFNCHGEFFYCK<br/>MDWFLNYLNNLTVDADHNECKNTSGTKSGNKRAPGPCVQRTYVACHIRSVIIWLETISKK<br/>TYAPPREGHLECTSTVTGMTVELNYIPKNRTNVTLSPQIESIWAAELDRYKLVEITPIGF<br/>APTEVRRYTGGHERQKRVPFVXXXXXXXXXXXXXXXXXXXXXXVQSQHLLAGILQQQKNL<br/>LAAVEAQQQMLKLTIWGVK</pre><br/><br/>Sequences are expected to be represented in the standard IUB/IUPAC amino acid codes, with these exceptions: lower-case letters are accepted and are mapped into upper-case; a single hyphen or dash can be used to represent a gap of indeterminate length; and in amino acid sequences, U and * are acceptable letters (see below). Before submitting a request, any numerical digits in the query sequence should either be removed or replaced by appropriate letter codes (e.g., X for unknown amino acid residue).<br/><br/>The accepted amino acid codes are:<br/><br/><pre>    A  alanine                         P  proline<br/>    B  aspartate or asparagine         Q  glutamine<br/>    C  cystine                         R  arginine<br/>    D  aspartate                       S  serine<br/>    E  glutamate                       T  threonine<br/>    F  phenylalanine                   U  selenocysteine<br/>    G  glycine                         V  valine<br/>    H  histidine                       W  tryptophan<br/>    I  isoleucine                      Y  tyrosine<br/>    K  lysine                          Z  glutamate or glutamine<br/>    L  leucine                         X  any<br/>    M  methionine                      *  translation stop<br/>    N  asparagine                      -  gap of indeterminate length</pre>",
    "Output sequence order": "Use this radio button to specify the output order of the sequences.",
    "Gap open penalty": "Specify the gap open penalty (numbers greater than 0 are valid).",
    "Gap extension penalty": "Specify the gap extension penalty.",
    "Terminal gap penalties": "Specify the terminal gap penalties.",
    "Bonus score": " Here you can specify a constant added to the substitution matrix.",
    //MAFFT
    //"Input format  "
    "CLUSTAL output": " Activate this option to print the computed alignment in CLUSTAL format.",
    "Alignment order": "Activate this option to print sequences in alignment order rather than input order.",
    //"Gap open penalty"
    "Offset": "This option specifies the offset parameter which works like a gap extension penalty (numbers greater than 0 are valid).",
    "Advanced options": "Use this textfield to specify advanced options via commandline. The options supported by MAFFT are:<br/><br/> <pre>Options:<br/>      --localpair :       All pairwise local alignment information is included<br/>                          to the objective function. default: off<br/>      --globalpair :      All pairwise global alignment information is included<br/>                          to the objective function. default: off<br/>      --bl #, --jtt # :   Scoring matrix. default: BLOSUM62<br/>                          Alternatives are BLOSUM (--bl) 30, 45, 62, 80,<br/>                          or JTT (--jtt) # PAM.<br/>      --nuc or --amino :  Sequence type. default: auto<br/>      --retree # :        The number of tree building in progressive method<br/>                          (see the paper for detail). default: 2<br/>      --maxiterate # :    Maximum number of iterative refinement. default: 0<br/>      --fft or --nofft:   FFT is enabled or disabled. default: enabled<br/>      --memsave:          Memory saving mode (beta). default: off<br/>      --quiet :           Do not report progress.</pre>",
    //MSAprobs
    //"Input format  "
    "Other advanced options": "The advanced options are not yet supported",
    //MUSCLE
    //"Input format   "
    "Select output format": "Use this radio button to specify if the output should be in Fasta or in ClustalW format.",
    "Maximum number of iterations": "Specify the maximum number of iterations.",
    "Other advanced options ": "Use this text field to specify advanced options via commandline.",
    //T-Coffee
    //"Input format  "
    "Pairwise method": "<table><tr><td>mlalign_id_pair</td>	<td>Produces local alignments. The default for lalign_id_pair is GOP=-10, GEP=-4, MATRIX=blosum50mt.</td></tr><tr><td>mfast_pair</td>	<td>Makes a global fasta style pairwise alignment. For proteins, matrix=blosum62mt, GEP=-1, GOP=-10, ktup=2. For DNA, matrix=idmat (id=10), gep=-1, gop=-20, ktup=5. Each pair of residue is given a score function of the weighting mode defined by -weight.</td></tr><tr><td>mslow_pair</td>	<td>Identical to fast pair, but does a full dynamic programming, using the Myers and Miller algorithm. This method is recommended if your sequences are distantly related.</td></tr></table>",
    //"Other advanced options "
    "Character transformation": "Changes sequence letters to upper or lower case",
    "Residue number": "Causes the output alignment to contain residue numbers at the end of each line.",
    "Output in alignment order": "Sets the order of the sequences in the output alignment. Default (unchecked) means to keep the sequences in the original order. Checking the box means the sequences come in the order indicated by the tree.",
    //Aln2plot
    //"Input"
    //"Input format"
    //FRpred
    "Input ": "Enter a protein sequence alignment and choose its format from the dropdown menu or enter a single query sequence in FASTA format. FRpred will build a multiple alignment of homologs from it with PSI-BLAST.<br/>Its length should be at most 477 amino acids.",
    "Input format    ": "Accepted formats are listed in the dropdown menu. FASTA input format accepts a single raw sequence without headers as well. The grouped FASTA format has been defined for this program specifically. Use a line starting with '#' in order to separate sequences belonging to different subgroups.<br/> <br/><pre>>sequence_name_1_belonging_to_subgroup_1<br/>SE-QVENCED--ATAXDDEFIVLIPSPTRCMH<br/>>sequence_name_2_belonging_to_subgroup_1<br/>SE-QVENCED--AT-ADEEFIVLIPS-TRCKH<br/>>sequence_name_3_belonging_to_subgroup_1<br/>SE-QVENCED--ATADDDEYILLIPSPTRCWK<br/>>sequence_name_4_belonging_to_subgroup_1<br/>SE-QVENCED--AT-ADEEFIVLIPS-TRCNR<br/>#<br/>>sequence_name_1_belonging_to_subgroup_2<br/>TE-QVQDCED---SAXDDEFIVLIPSPTRCMH<br/>>sequence_name_2_belonging_to_subgroup_2<br/>TE-QVQDCED---S-ADEEFIVLIPS-TRCKH<br/>>sequence_name_3_belonging_to_subgroup_2<br/>TE-QVQDCED---SADDDEYILLIPSPTRCWK<br/>>sequence_name_4_belonging_to_subgroup_2<br/>TE-QVQDCED---S-ADEEFIVLIPS-TRCNR<br/>#<br/>>sequence_name_1_belonging_to_subgroup_3<br/>TE-QVQDCED---SAXDDEFIVLIKKGARCMH<br/>>sequence_name_2_belonging_to_subgroup_3<br/>TE-QVQDCED---S-ADEEFIVLIKKG-RCKH<br/>>sequence_name_3_belonging_to_subgroup_3<br/>TE-QVQDCED---SADDDEYILLIPKK--RCWK</pre><br/><br/><b>PLEASE make sure that the query sequence is the first sequence in your input alignment!</b>",
    "PDB file": "Optionally it is possible to upload a file containing structural information that represents your alignment/sequence in PDB format. For each method, a RasMol-script is generated, which visualizes high-scoring residues. The sequence of the PDB file is aligned to the query (first sequence in FastA alignment). Therefore, please make sure that the PDB file is unambiguously alignable to the query sequence. The safest way is to use the part of the PDB sequence that corresponds to your alignment as the query sequence. Please do not forget to specify the appropriate chain in the PDB file (if applicable).",
    "Chain": "Please do not forget to specify the appropriate chain if you use a PDB file. Even if there is only one chain in your file it might have a name. Only if it does not have a name you can leave this field free.",
    "Extend sequence/alignment": "This checkbox triggers an extension of your alignment kickstarting PSI-BLAST. For single sequence input this is done automatically.",
    "Maximum number of PSI-BLAST iterations": "Maximum number of PSI-BLAST iterations used to build an alignment around a seed sequence.",
    "PSI-BLAST E-value threshold": "PSI-BLAST E-value threshold for building alignments around seed sequences.",
    "Minimum coverage": "Minimum coverage of PSI-BLAST hits with the seed sequence in the PSI-BLAST iterations.",
    "Minimum sequence identity": "Minimum sequence identity of PSI-BLAST hits with the seed sequence in the PSI-BLAST iterations.",
    "Catalytic residues":  "If checked, this option scores residues that are possibly catalytically active higher than hydrophobic ones. Use this option if you are looking for ligand binding sites. Do not use it if you are analyzing differences in structure.",
    "Sequence-centered analysis": "If you are interested in learning about the protein family of your alignment, it is best to leave this box unchecked. However, if you are mainly interested in the sequence, the checked box performs a sequence-centered analysis of the UPGMA tree. That is, it takes only those subgroups into account that contain information about the query sequence. For single sequence input this is done automatically.",
    //HHrepID
    "Paste single sequence or multiple alignment ": "Paste the query sequence with which you want to do the search either as raw sequence (e.g. 'ELRLRYCAPAGFALLKCN...'), as FASTA sequence (i.e. with a single line '>sequence_name optional description' preceding the raw sequence, see next subsection). HHrep will use PSI-BLAST searches against the non-redundant sequence database from NCBI to build a multiple alignment from it. This alignment will then be used to construct a concise statistical representation (a hidden Markov model or HMM) and to search the selected HMM databases for homologs.<br/>You may also paste a query multiple alignment (for allowed formats see below). The alignment will be used to jumpstart PSI-BLAST. An expert-built or expert-edited alignment or one that contains homologs not yet contained in the non-redundant sequence database may significantly increase the sensitivity and reliability of HHrep. I recommend NOT to use automatic multiple alignment programs such as CLUSTAL, as PSI-BLAST tends to give clearly better alignments for the purpose of homologous relationships. Note: if you paste a multiple alignment please don't forget to select the appropriate input format below.",
    //"Select input format"
    "Max. number of PSI-BLAST iterations": "This specifies the number of PSI-BLAST iterations that are performed to build up an alignment from the input sequence or input alignment. The iterations stop in any case when no further sequences are found. If you want to use exactly your input alignment choose 0 here. This will ensure that only predicted secondary structure is added to your input alignment.",
    "Score secondary structure ": "If you choose \"yes\", the secondary structure of the query will be predicted with <a href = https://www.ncbi.nlm.nih.gov/pubmed/10493868?dopt=Abstract>PSIPRED</a> and a secondary structure similarity score is added to the total score. Scoring the secondary structure similarity improves sensitivity and also alignment quality significantly. Since the weight of the secondary structure score is small (12% of the amino acid similarity score), the risk for high-scoring non-homologous matches is still small.",
    "Repeat family P-value threshold": "The statistical significance threshold for including a repeat family in the repeat detection results.",
    "Self-alignment P-value threshold": "The statistical significance threshold for including a suboptimal self-alignment in the total poster probability matrix.",
    "Merge rounds": "Number of merge iterations to be performed before repeat are extracted from the posterior probability matrix.",
    "MAC threshold": "Probability threshold for MAC alignment algorithm. The lower the threshold the longer the MAC alignments and vice versa.",
    "Alignment stringency": "To avoid non-homologous or wrongly aligned sequences in the multiple alignment from which the HMM for self-comparison is constructed, all sequences with a BLOSUM62 similarity lower than 0.0 - 0.03 bits per column to the first sequence are filtered out. Reducing this number increases the sensitivity to detect diverged repeats but increases the risk of detecting spurious similarities.",
    "Domain boundary detection": "Turn this option off if you are certain that your query contains only a single domain.",
    //"Coloring query-template multiple alignments in results"
    //MARCOIL
    "Input  ": "Please use only fasta format, as only this was tested.<br/>The format is not checked by the program and you could get false results.<br/>The number of sequences is unlimited.<br/>Only the 20 standard amino acids are represented in the model,other letters are interpreted as unknown amino acids.<br/>Lowercase is converted to uppercase, numbers, spaces,newlines, tabs and other special signs in the sequence are ignored.",
    "Matrices": "The three matrices accessible through the web interface are the one used in the paper and trained on 9 \"families\" of proteins (9FAM). This is a matrix of amino-acid probabilities derived from a large dataset of coiled-coil domains. It is unspecific, as the dataset contains all kind of domains and these differ in the number of helices, the orientation, the length and the hydrophobicity. The matrix is meant for first-pass genomic screenings.<br/>It generalises the two matrices proposed by A. Lupas and collaborators and used by the program COILS. These matrices are MTIDK, derived from 5 and MTK derived from three \"families\" of proteins. Those are matrices of frequency ratios and we computed from them, by using an estimate of absolute amino-acid frequencies, the other two matrices that can be used.",
    //PCOILS
    //"Input"
    //"Input format"
    "Input mode": "Use input as given: Performs all calculations on the basis of a given sequence or a given sequence alignment in FASTA format. Either put in a sequence or sequence alignment of your choice.<br/><br/>Run PSI-BLAST: Performs a PSI-BLAST run on the given sequence (two iterations, E-value for inclusion 0.0001). Hits are aligned and filtered to a minimum sequence identity of 40% with the query sequence and a minimum coverage of 20%. This alignment is then the basis for the input profile. In case you are not happy with the used PSI-BLAST parameters, run PSI-BLAST separately with the parameters of your choice, create a multiple alignment in FASTA format, and use this as COILS input.",
    "Matrix ": "MTIDK - is a matrix derived from myosins, paramyosins, tropomyosins, intermediate filaments type I - V, desmosomal proteins and kinesins. The matrix was compiled by weighting the residue frequencies of the different protein families according to the following scheme:<br/>0.2 MYOSINS - 0.5 myosins - 0.5 paramyosins<br/>0.2 TROPOMYOSINS<br/>0.2 INTERMEDIATE FILAMENTS - 0.2 type I (keratin) - 0.2 type II (keratin) - 0.2 type III (desmin, vimentin, GFAP, peripherin) - 0.2 type IV (NF light, medium and heavy chains) - 0.2 type V (lamins A and B)<br/>0.2 DESMOSOMAL PROTEINS - 0.33 desmoplakin - 0.33 plectin - 0.33 hemidesmosomal plaque prot. (bullous pemphigoid)<br/>0.2 KINESINS<br/><br/>PDB - is a matrix derived from coiled coils in the PDB. <br/>",
    "Weighting": " Weighting: Because coiled coils are generally fibrous, solvent-exposed structures, all but the internal a and d positions have a high likelihood of being occupied by hydrophilic residues. A program that gives equal weight to all positions is therefore going to be biased towards hydrophilic, charge-rich sequences. While this does not pose a problem for the vast majority of natural sequences, some highly charged sequences obtain high coiled-coil probabilities in the obvious absence of heptad periodicity and coiled-coil-forming potential. An extreme case is that of polyglutamate which obtains a coiled-coil-forming probability > 99%. To counter this problem, COILS contains a weighting option, which allows the user to assign the the same weight to the two hydrophobic positions a and d as to the five hydrophilic positions b, c, e, f and g. This leads to an only slightly worse performance of the program and permits the identification of false positives. It is recommended to run a weighted and and unweighted scan and to compare the outputs. A drop of more than 20-30% in the probability is a clear indication of a highly-charged false positive.",
    //REPPER
    //"Input"
    //"Input format"
    "Input mode ": "Use input as given: Performs all calculations on the basis of a given sequence or a given sequence alignment in FASTA format. Either put in a sequence or sequence alignment of your choice.<br/><br/>Run PSI-BLAST: Performs a PSI-BLAST run on the given sequence (two iterations, E-value for inclusion 0.0001). Hits are aligned and filtered to a minimum sequence identity of 40% with the query sequence and a minimum coverage of 20%. This alignment is then the basis for the input profile. In case you are not happy with the used PSI-BLAST parameters, run PSI-BLAST separately with the parameters of your choice, create a multiple alignment in FASTA format, and use this as REPPER input.",
    "Window size (FTwin and REPwin": "Size of the sliding window. Must be smaller than sequence length.",
    "Periodicity range (FTwin)": "Periodicity range for Fourier Transform. Minimal periodicity must be at least 2, maximal periodicity should not exceed the window size (otherwise it is set to window size automatically).",
    "Scale (FTwin)": "A Fourier Transform cannot deal with amino acids. Amino acids are therefore represented by numbers. Depending on what one is looking for (e.g. repeat patterns of hydrophobic residues), those numbers represent a certain property (in this case hydrophobicity). The default values are set to the Kyte-Doolittle hydrophobicity scale, but you can also switch to an empirical set of binary hydrophobic values (standard binary) by clicking the button or you can manually put in values for your own purposes , either by changing each value manually or by pasting all amino acids into the prepared form (own scale).",
    "FTwin threshold": "Refinement option. For each window, the FTwin output graph shows all periodicities above<br/>average+threshold*standard_deviation,<br/>while \"average\" is the average intensity of the W Fourier transform periodicities of the given window (W = window size), \"threshold\" is the parameter given by the user, and \"standard_deviation\" is the standard deviation of the intensities of the W Fourier transform periodicities of the given window.<br/>Whether the threshold parameter is reasonable depends on the values that are attributed to the amino acids, but also on the window size. Decrease this value if you see too much in the FTwin graph, increase it if you see too little or nothing.",
    "REPwin threshold": "Refinement option. The default value of 2 is reasonable for most applications. Decrease this value if you see too much in the REPwin graph, increase it if you see too little or nothing.",
    "COILS parameters": "Matrix: COILS compares an input sequence (or an input alignment) with a coiled-coil profile. The profile information is stored in a matrix. The MTIDK matrix is derived from myosins, paramyosins, tropomyosins, intermediate filaments type I - V, desmosomal proteins and kinesins, whereas the PDB matrix is derived from known coiled coils in the Protein Data Bank.<br/><br/>Weighting: Because coiled coils are generally fibrous, solvent-exposed structures, all but the internal a and d positions have a high likelihood of being occupied by hydrophilic residues. A program that gives equal weight to all positions is therefore going to be biased towards hydrophilic, charge-rich sequences. While this does not pose a problem for the vast majority of natural sequences, some highly charged sequences obtain high coiled-coil probabilities in the obvious absence of heptad periodicity and coiled-coil-forming potential. An extreme case is that of polyglutamate which obtains a coiled-coil-forming probability > 99%. To counter this problem, COILS contains a weighting option, which allows the user to assign the the same weight to the two hydrophobic positions a and d as to the five hydrophilic positions b, c, e, f and g. This leads to an only slightly worse performance of the program and permits the identification of false positives. It is recommended to run a weighted and and unweighted scan and to compare the outputs. A drop of more than 20-30% in the probability is a clear indication of a highly-charged false positive.",
    //TPRpred
    "Input format     ": "A sequence in FASTA format begins with a single-line description, followed by lines of sequence data. The description line is distinguished from the sequence data by a greater-than (\">\") symbol in the first column. It is recommended that all lines of text be shorter than 80 characters in length. An example sequence in FASTA format is:<br/><br/><br/><pre>>HcpC_a HcpC Chain A<br/>DPKELVGLGAKSYKEKDFTQAKKYFEKACDLKENSGCFNLGVLYYQGQGV<br/>EKNLKKAASFYAKACDLNYSNGCHLLGNLYYSGQGVSQNTNKALQYYSKA<br/>CDLKYAEGCASLGGIYHDGKVVTRDFKKAVEYFTKACDLNDGDGCTILGS<br/>LYDAGRGTPKDLKKALASYDKACDLKDSPGCFNAGNMYHHGEGATKNFKE<br/>ALARYSKACELENGGGCFNLGAMQYNGEGVTRNEKQAIENFKKGCKLGAK<br/>GACDILKQLKIKVHHH</pre><br/>     <br/><br/>Sequences are expected to be represented in the standard IUB/IUPAC amino acid acid codes, with these exceptions: lower-case letters are accepted and are mapped into upper-case; a single hyphen or dash can be used to represent a gap of indeterminate length; and in amino acid sequences, U and * are acceptable letters (see below). Before submitting a request, any numerical digits in the query sequence should either be removed or replaced by appropriate letter codes (e.g., X for unknown amino acid residue). The amino acid codes supported are:<br/><br/><pre>    A  alanine                         P  proline<br/>    B  aspartate or asparagine         Q  glutamine<br/>    C  cystine                         R  arginine<br/>    D  aspartate                       S  serine<br/>    E  glutamate                       T  threonine<br/>    F  phenylalanine                   U  selenocysteine<br/>    G  glycine                         V  valine<br/>    H  histidine                       W  tryptophan<br/>    I  isoleucine                      Y  tyrosine<br/>    K  lysine                          Z  glutamate or glutamine<br/>    L  leucine                         X  any<br/>    M  methionine                      *  translation stop<br/>    N  asparagine                      -  gap of indeterminate length</pre>",
    "Profiles": "TPR - Tetratrico Peptide Repeat<br/>PPR - Pentatrico Peptide Repeat<br/>SEL1-Like repeat",
    "Maximum whole protein E-value": "Maximum whole-protein e-value for output of repeat residues to alifile.",
    "Minimum number of single-repeat hits": "Minimum number of single repeat hits for output to alifile.",
    "Maximum number of rows in hitlist": "As the name suggests.",
    //Ali2d
    //"Input"
    //"Input format"
    "Percentage sequence identity": "This option specifies the percentage sequence identity below which a new Psipred run is invoked.",
    //Quick2d
    //"Input"
    //"input format"
    "Do PSI-BLAST search": "Search for homologous protein sequences and generate a profile for the predictions. Alignments are extended using the jump-start option of PSI-BLAST.",
    //Modeller
    "Input format      ": "Paste a multiple alignment in one of the formats, which are listed in the dropdown menu.<br/><br/>The first sequence in the multiple alignment must be the query sequence. The other sequences are the template sequences. In each template the first word in the header has to be the PDB- or SCOP-identifier. A FASTA formatted example imput is shown below:<br/><br/><br/><pre>>gi|50121231|ref|YP_050398.1|<br/>AHQEGDFIVRAGSATV-RPSESSD-NVLGSLGSFQVNNNTQLGLTFSYMV----T-DNIGVELLAATPF-<br/>----NHNVGLRSTGTIAEVKHLPPSLVAQYYFGDKEDTLRPYLGVGVNYTLFFDEKFNDTGTNAGLSDLS<br/>-LKNSWGIAAQAGLDYNLDKNWLLNMSVWWMDIDTDVKFKAGNDQQSVHTRLDPWAFMFGVGYRF<br/>>1qj8<br/>--ATSTVTGGYAQSD-A------Q-G----------QMNKMGGFNLKYRYEEDNSPLGVIGSFTYTEKSR<br/>TASS------GD-----YNKNQYYGITAGPAY-RINDWASIYGVVGVGYGKFQTT-------EY-PTYKN<br/>-DTSDYGFSYGAGLQFNPMENVALDFSYEQSRI----------------RSVDVGTWI-AGVGYR<br/>>1qjp<br/>APKDNTWYTGAKLGWSQ-------H-----------ENKLGAGAFGGYQV----N-PYVGFEMGYDWLG-<br/>----RMPY---------AYKAQGVQLTAKLGY-PITDDLDIYTRLGGMVWRADTY-----SNVY---GKN<br/>H-DTGVSPVFAGGVEYAITPEIATRLEYQWT---------------------N-GMLSLGVSYRF<br/>>1p4t<br/>-EGASGFYVQADAAHA-KASSSLG-S----------AKGFSPRISAGYRI----N--DLRFAVDYTRYK-<br/>----N-YKAPST-----DFKLYSIGASAIYDF-DTQSPVKPYLGARLSLNRASVD--------LGG-SDS<br/>FSQTSIGLGVLTGVSYAVTPNVDLDAGYRYNYI----------GKVNTVKNVRSGELSAGVRVKF</pre>",
    "Modeller key": "The MODELLER license requires you to enter a MODELLER-key to use this tool. This key is freely available for academic users and easily obtainable at: <a href = http://salilab.org/modeller/registration.shtml>http://salilab.org/modeller/registration.shtml</a>.",
    //SamCC
    "Input format       ": "A PDB file containing a four-helical bundle motif. Missing or unusual residues, like selenomethionine, are not allowed. The structure should not contain alternative conformations.",
    "Definition line for helix 1-4": "In these four lines, every helix of a four helical bundle is defined. In each line you need to enter the chain identifier of each helix (one character or digit) and its start and end positions (use PDB file numbering). Residues localized at positions defined by \"from\" values form the first layer of the bundle, while those in \"to\" the last one. You also need to provide the register of residues located in the first defined layer (one letter, a-g for 7 periodicity, a-k for 11 periodicity etc.). The registers of subsequent layers are automatically defined based on periodicity and helix direction (parallel vs antiparallel).<br/>IMPORTANT NOTE #1: If the \"from\" value is larger than \"to\" then program assumes that given helix is antiparallel (see also Example 2).<br/>IMPORTANT NOTE #2: Helices pairs 1&2 and 3&4 have to be located on the diagonals (see Figures 1 and 2).",
    "Periodicity": "The program needs to know the periodicity in order to calculate Crick angle deviation (observed Crick angle - expected Crick angle).",
    //ANCESCON
    //"Input"
    //"Input format"
    "Use long sequence names": "If the option is not checked the sequence header truncated at the first white space character is used as the sequence name. Otherwise up to the first 150 characters are used as the sequence name. Nonetheless no more than the first 30 characters will be displayed by our tree labeling tool.",
    "Upload local tree file": "Here you can upload a file with a tree structure. Otherwise the tool will generate a tree structure. You don't need to submit a special parameter for your tree file.",
    "Other advanced options  ": "Use this textfield to specify advanced options via commandline. The options supported by ANCESCON are:<br/><br/><pre>-O:  optimize alfa with maximum likelihood method<br/>-C:  Alignment-Based alfa<br/>-D:  Use Joint method to reconstruct ancestral sequences<br/>-R:  Reconstruct sequence for biological root and all internal nodes<br/>-RO: Only reconstruct sequence for the biological root<br/>-PP: Optimize PI vector with Powell method<br/>-PD: Optimize PI vector with Downhill Simplex Method in Multidimensions<br/>-G value:  Non-gap number or percentage (>=3 or =0~1), no estimated alpha used at that site <br/>           if non-gap number is less<br/><br/>Recently added options (July, 2014):<br/><br/>-RI: Reconstruct sequences for all internal nodes<br/>-PA: Optimize PI vector with Continuous Minimization by Simulated Annealing<br/>-PS: Optimize PI vector for each site with Continuous Minimization by Simulated Annealing<br/>-T: Upload own tree file (is set automatically if tree file is uploaded)<br/><br/>Not available:<br/><br/>-Z: Predict functional sites, should be used with -R/-RO</pre>",
    //CLANS
    "Select search tool": "Choose whether to use BLASTP or PSIBLAST to calculate the pairwise sequence similarities",
    //"Input format  "
    "Database  ": "The selected databases are used by Clustering to perform the search. You may press CTRL to select multiple databases.",
    //"Matrix"
    //"Number of iterations"
    //"E-value"
    "E-value for inclusion in first iteration": " The E-value for which sequences should be added to the evolving profile (relevant only for PSI-BLAST and if multiple rounds of searching are performed).",
    "Filter for low complexity regions": "Mask off segments of the query sequence that have low compositional complexity, as determined by the SEG program of Wootton & Federhen (Computers and Chemistry, 1993) or, for BLASTN, by the DUST program of Tatusov and Lipman (in preparation). Filtering can eliminate statistically significant but biologically uninteresting reports from the blast output (e.g., hits against common acidic-, basic- or proline-rich regions), leaving the more biologically interesting regions of the query sequence available for specific matching against database sequences.<br/>Filtering is only applied to the query sequence (or its translation products), not to database sequences. Default filtering is DUST for BLASTN, SEG for other programs.<br/>It is not unusual for nothing at all to be masked by SEG, when applied to sequences in SWISS-PROT, so filtering should not be expected to always yield an effect. Furthermore, in some cases, sequences are masked in their entirety, indicating that the statistical significance of any matches reported against the unfiltered query sequence should be suspect.",
    "Perform ungapped alignment": "Set BLAST/PSIBLAST parameter changing the alignment behavior of these programs.",
    "Cutoff": " The P-value cutoff. Only hits with P-values better than specified will be used in further calculations.",
    "Min. sequences": "Determines the minimal number of sequences necessary for a \"cluster\" to be regarded as such.",
    "Cluster by...": "Choose the clustering methodology: linkage; convex or network.<br/><br/><ul><li>Linkage: performs a linkage clustering (you specify the linkage number)</li><li>Convex: adds sequences to clusters depending on their average \"attraction\" (proportional to -log P-value). Will fail if a cluster is not \"convex\", i.e. the majority of sequences do not share an above average pairwise similarity (one parameter \"X\": the average attraction of a sequence be to a cluster has to be greater than the average global attraction+(\"X\"*STDEV) for it to be added to a cluster).</li><li>Network: A neural net approach. Each sequence is represented by a neuron and emits a number to all other neurons. The input weights are proportional to -log P-value. in the first round each sequence emits it's own sequence numbers. in consecutive rounds, each sequence emits the number which had the largest input weight. This process iterates until no further neurons change their emission number (i.e. the cluster assignment has converged)</li></ul><br/>Alternative:<br/>View in CLANS. This creates a CLANS file and allows viewing of the results in a more interactive fashion in an java applet (and provides all of the above analyses. You can download a working version of CLANS <a href = ftp://ftp.tuebingen.mpg.de/pub/protevo/CLANS/>here</a>)",
    //PhyML
    //"Input"
    //"Input format"
    "Enable bootstrapping": "Checking this box enables bootstrapping. The data is resampled using the program <a href = http://evolution.genetics.washington.edu/phylip/doc/seqboot.html>SEQBOOT</a> (regular bootstrap mode) of the PHYLIP package. A consensus tree can be used to determine the reliability of a phylogenetic tree.",
    "Number of replicates": "Number of resampled datasets asked by <a href = http://evolution.genetics.washington.edu/phylip/doc/seqboot.html>SEQBOOT</a>.",
    "DNA or Protein": "Please enter the biochemical type of your input sequences.<br/><br/>\"DNA\" refers to nucleic acid sequences using the following one letter code.<br/><br/>   <pre>   A --> adenosine           M --> A C (amino)<br/>   C --> cytidine            S --> G C (strong)<br/>   G --> guanine             W --> A T (weak)<br/>   T --> thymidine           B --> G T C<br/>   U --> uridine             D --> G A T<br/>   R --> G A (purine)        H --> A C T<br/>   Y --> T C (pyrimidine)    V --> G C A<br/>   K --> G T (keto)          N --> A G C T (any)<br/>                             -  gap of indeterminate length</pre><br/><br/><br/>\"Protein\" refers to amino acid sequences using the following one letter code.<br/><br/>    <pre>    A  alanine                         P  proline<br/>    B  aspartate or asparagine         Q  glutamine<br/>    C  cystine                         R  arginine<br/>    D  aspartate                       S  serine<br/>    E  glutamate                       T  threonine<br/>    F  phenylalanine                   U  selenocysteine<br/>    G  glycine                         V  valine<br/>    H  histidine                       W  tryptophan<br/>    I  isoleucine                      Y  tyrosine<br/>    K  lysine                          Z  glutamate or glutamine<br/>    L  leucine                         X  any<br/>    M  methionine                      N  asparagine  <br/>                                       -  gap of indeterminate length</pre>",
    "Model of amino acid replacement": "Amino acid substitution model that is used for the computation of the distance matrix by <a href = http://evolution.genetics.washington.edu/phylip/doc/protdist.html>PROTDIST</a> (Please have a look at the link for a more detailed description).",
    "Model of nucleotide substitution": "Nucleotide substitution model that is used for the computation of the distance matrix by <a href = http://evolution.genetics.washington.edu/phylip/doc/dnadist.html>DNADIST</a> (Please take the link for a more detailed description).",
    "Gamma distributed rates across sites": "Enable distance correction for unequal rates of change at different amino acid positions. This correction assumes that the distribution of rates of change among amino acid positions follows a Gamma distribution.<br/>Further information can be found here: <a href = http://evolution.genetics.washington.edu/phylip/doc/dnadist.html>DNADIST</a> or <a href = http://evolution.genetics.washington.edu/phylip/doc/protdist.html>PROTDIST</a>.",
    "Amount of variation": "Enter the value that determines the amount of variation of rates among amino acid positions. Instead of the more widely-known coefficient alpha, PROTDIST uses the coefficient of variation (ratio of the standard deviation to the mean) of rates among amino acid positions. So if there is 20% variation in rates, the CV is is 0.20. The square of the C.V. is also the reciprocal of the better-known \"shape parameter\" alpha, of the Gamma distribution, so in this case the shape parameter alpha = 1/(0.20*0.20) = 25. If you want to achieve a particular value of alpha, such as 10, you will want to use a CV of 1/sqrt(100) = 1/10 = 0.1.<br/>Further information can be found here <a href = http://evolution.genetics.washington.edu/phylip/doc/dnadist.html>DNADIST</a> or <a href = http://evolution.genetics.washington.edu/phylip/doc/protdist.html>PROTDIST</a>.",
    //6FrameTranslation
    "Input format        ": "A sequence in FASTA format begins with a single-line description, followed by lines of sequence data. The description line is distinguished from the sequence data by a greater-than (\">\") symbol in the first column. It is recommended that all lines of text be shorter than 80 characters in length. An example sequence in FASTA format is:<br/><br/><br/><pre>>gi|555|emb|X65215.1|BTMISATN B.taurus microsatellite DNA (624bp)<br/>ACCTCCACTAGCTTTGTTTGTAGTGATGCTCTGTAGCACCACTGGGAAGCCCTTTAATGAATGTGCCTTTCCGCAAATCA<br/>CACACACACAAATACACTTATAGAAACAAGGTGATTTTCTTGAAATAATAAAACAAAATTTGGAAGAAGATTTTTACTGT<br/>CTTAGGAAAAGTAAGGCATTGGAAGGTGGCTAGGTATGACATATGAAGTTGCATTTTAAAACTGGAATTGGACAACTGAT<br/>ATTCAGTGATATTTATGCTACTACCTTCTAGAATCGAGAGCATGCACCCCACTCTGTACTCTTGCCTGGAGAATCCATGA<br/>TGAGAGCCTGGTAGGCTGCAGTCCATGGGGTCACACAGAGTCGGACATGACTGAGCGACTTCACTTTCACTTTTCAATTT<br/>CATGCATTGGAGCCGGAAATGGCAACCCACTCCAGTGTTCTTGCCTGGAGAATCCCAGGGATGGGGAAGCCTGGTGGGCT<br/>GCTGTCTATGGGGTCGCAGAGAGTCAGACACGACTGAAGTGACTTAGCAGCAACCTTCTGGAATAAACGCCTCAGGCTTT<br/>AAACTCTGGCTTGACCATTCACTAGCCATGGGATCCACTAGAGTCGACCTGCAGGCATGCAAGC</pre><br/><br/>Sequences are expected to be represented in the standard IUB/IUPAC amino acid and nucleic acid codes, with these exceptions: lower-case letters are accepted and are mapped into upper-case; a single hyphen or dash can be used to represent a gap of indeterminate length; and in amino acid sequences, U and * are acceptable letters (see below). Before submitting a request, any numerical digits in the query sequence should either be removed or replaced by appropriate letter codes (e.g., N for unknown nucleic acid residue or X for unknown amino acid residue).<br/>The nucleic acid codes supported are:<br/><br/><br/>   <pre>   A --> adenosine           M --> A C (amino)<br/>   C --> cytidine            S --> G C (strong)<br/>   G --> guanine             W --> A T (weak)<br/>   T --> thymidine           B --> G T C<br/>   U --> uridine             D --> G A T<br/>   R --> G A (purine)        H --> A C T<br/>   Y --> T C (pyrimidine)    V --> G C A<br/>   K --> G T (keto)          N --> A G C T (any)<br/>                             -  gap of indeterminate length</pre>",
    "Include nucleic acid sequence in output": "Without this option, the output will be in fasta format like this:<br/><br/><br/><pre>>gi|555|emb|X65215.1|BTMISATN B.taurus microsatellite DNA (624bp)_+1<br/>TSTSFVCSDAL*HHWEAL**MCLSANHTHTNTLIETR*FS*NNKTKFGRRFLLS*EK*GIGRWLGMTYEVAF*...<br/>>gi|555|emb|X65215.1|BTMISATN B.taurus microsatellite DNA (624bp)_+2<br/>PPLALFVVMLCSTTGKPFNECAFPQITHTQIHL*KQGDFLEIIKQNLEEDFYCLRKSKALEGG*V*HMKLHFK...<br/>>gi|555|emb|X65215.1|BTMISATN B.taurus microsatellite DNA (624bp)_+3<br/>LH*LCL**CSVAPLGSPLMNVPFRKSHTHKYTYRNKVIFLK**NKIWKKIFTVLGKVRHWKVARYDI*SCILK...<br/>>gi|555|emb|X65215.1|BTMISATN B.taurus microsatellite DNA (624bp)_-1<br/>ACMPAGRL*WIPWLVNGQARV*SLRRLFQKVAAKSLQSCLTLCDPIDSSPPGFPIPGILQARTLEWVAISGSN...<br/>>gi|555|emb|X65215.1|BTMISATN B.taurus microsatellite DNA (624bp)_-2<br/>LACLQVDSSGSHG**MVKPEFKA*GVYSRRLLLSHFSRV*LSATP*TAAHQASPSLGFSRQEHWSGLPFPAPM...<br/>>gi|555|emb|X65215.1|BTMISATN B.taurus microsatellite DNA (624bp)_-3<br/>LHACRSTLVDPMASEWSSQSLKPEAFIPEGCC*VTSVVSDSLRPHRQQPTRLPHPWDSPGKNTGVGCHFRLQC...</pre><br/><br/>If you activate this option the nucleic acid sequence will be in the output too. The results look like this:<br/><br/><br/><pre>>gi|555|emb|X65215.1|BTMISATN B.taurus microsatellite DNA (624bp)<br/>ACCTCCACTAGCTTTGTTTGTAGTGATGCTCTGTAGCACCACTGGGAAGCCCTTTAATGAATGTGCCTTTCCGCA...<br/>>gi|555|emb|X65215.1|BTMISATN B.taurus microsatellite DNA (624bp)_+1<br/> T  S  T  S  F  V  C  S  D  A  L  *  H  H  W  E  A  L  *  *  M  C  L  S  A ...<br/>>gi|555|emb|X65215.1|BTMISATN B.taurus microsatellite DNA (624bp)_+2<br/>  P  P  L  A  L  F  V  V  M  L  C  S  T  T  G  K  P  F  N  E  C  A  F  P  Q...<br/>>gi|555|emb|X65215.1|BTMISATN B.taurus microsatellite DNA (624bp)_+3<br/>   L  H  *  L  C  L  *  *  C  S  V  A  P  L  G  S  P  L  M  N  V  P  F  R  ...<br/>>gi|555|emb|X65215.1|BTMISATN B.taurus microsatellite DNA (624bp)_reverse<br/>GCTTGCATGCCTGCAGGTCGACTCTAGTGGATCCCATGGCTAGTGAATGGTCAAGCCAGAGTTTAAAGCCTGAGG...<br/>>gi|555|emb|X65215.1|BTMISATN B.taurus microsatellite DNA (624bp)_-1<br/> A  C  M  P  A  G  R  L  *  W  I  P  W  L  V  N  G  Q  A  R  V  *  S  L  R ...<br/>>gi|555|emb|X65215.1|BTMISATN B.taurus microsatellite DNA (624bp)_-2<br/>  L  A  C  L  Q  V  D  S  S  G  S  H  G  *  *  M  V  K  P  E  F  K  A  *  G...<br/>>gi|555|emb|X65215.1|BTMISATN B.taurus microsatellite DNA (624bp)_-3<br/>   L  H  A  C  R  S  T  L  V  D  P  M  A  S  E  W  S  S  Q  S  L  K  P  E  ...</pre>",
    "Print amino acids in relation to nucleotide sequence": "Without this option, the output will be in fasta format like thione of the two output formats mentioned above.<br/><br/>If you activate this option the corresponding outputs will change. The results look like this:<br/><br/><br/><pre>include nucleic acid sequence in output=off:<br/><br/>>gi|555|emb|X65215.1|BTMISATN B.taurus microsatellite DNA (624bp)<br/>    L  H  *  L  C  L  *  *  C  S  V  A  P  L  G  S  P  L  M  N  V  P  F  R  K  S  H  T  H  K  Y  T  Y  R  N  K  V  I  F  L  K  *  *  N  K  I  W  K  K  I  F  T  V  L  G  K  V  R  H  W  K  V  A  R  Y  D  I  *  S  C  I  L  K  L  E  L  D  N  *  Y  S  V  I  F  M  L  L  P  S  R  I  E  S  M  H  P  T  L  Y  S  C  L  E  N  P  *  *  E  P  G  R  L  Q  S  M  G  S  H  R  V  G  H  D  *  A  T  S  L  S  L  F  N  F  M  H  W  S  R  K  W  Q  P  T  P  V  F  L  P  G  E  S  Q  G  W  G  S  L  V  G  C  C  L  W  G  R  R  E  S  D  T  T  E  V  T  *  Q  Q  P  S  G  I  N  A  S  G  F  K  L  W  L  D  H  S  L  A  M  G  S  T  R  V  D  L  Q  A  C  K  - <br/>   P  P  L  A  L  F  V  V  M  L  C  S  T  T  G  K  P  F  N  E  C  A  F  P  Q  I  T  H  T  Q  I  H  L  *  K  Q  G  D  F  L  E  I  I  K  Q  N  L  E  E  D  F  Y  C  L  R  K  S  K  A  L  E  G  G  *  V  *  H  M  K  L  H  F  K  T  G  I  G  Q  L  I  F  S  D  I  Y  A  T  T  F  *  N  R  E  H  A  P  H  S  V  L  L  P  G  E  S  M  M  R  A  W  *  A  A  V  H  G  V  T  Q  S  R  T  *  L  S  D  F  T  F  T  F  Q  F  H  A  L  E  P  E  M  A  T  H  S  S  V  L  A  W  R  I  P  G  M  G  K  P  G  G  L  L  S  M  G  S  Q  R  V  R  H  D  *  S  D  L  A  A  T  F  W  N  K  R  L  R  L  *  T  L  A  *  P  F  T  S  H  G  I  H  *  S  R  P  A  G  M  Q  A <br/>  T  S  T  S  F  V  C  S  D  A  L  *  H  H  W  E  A  L  *  *  M  C  L  S  A  N  H  T  H  T  N  T  L  I  E  T  R  *  F  S  *  N  N  K  T  K  F  G  R  R  F  L  L  S  *  E  K  *  G  I  G  R  W  L  G  M  T  Y  E  V  A  F  *  N  W  N  W  T  T  D  I  Q  *  Y  L  C  Y  Y  L  L  E  S  R  A  C  T  P  L  C  T  L  A  W  R  I  H  D  E  S  L  V  G  C  S  P  W  G  H  T  E  S  D  M  T  E  R  L  H  F  H  F  S  I  S  C  I  G  A  G  N  G  N  P  L  Q  C  S  C  L  E  N  P  R  D  G  E  A  W  W  A  A  V  Y  G  V  A  E  S  Q  T  R  L  K  *  L  S  S  N  L  L  E  *  T  P  Q  A  L  N  S  G  L  T  I  H  *  P  W  D  P  L  E  S  T  C  R  H  A  S <br/> ACCTCCACTAGCTTTGTTTGTAGTGATGCTCTGTAGCACCACTGGGAAGCCCTTTAATGAATGTGCCTTTCCGCAAATCACACACACACAAATACACTTATAGAAACAAGGTGATTTTCTTGAAATAATAAAACAAAATTTGGAAGAAGATTTTTACTGTCTTAGGAAAAGTAAGGCATTGGAAGGTGGCTAGGTATGACATATGAAGTTGCATTTTAAAACTGGAATTGGACAACTGATATTCAGTGATATTTATGCTACTACCTTCTAGAATCGAGAGCATGCACCCCACTCTGTACTCTTGCCTGGAGAATCCATGATGAGAGCCTGGTAGGCTGCAGTCCATGGGGTCACACAGAGTCGGACATGACTGAGCGACTTCACTTTCACTTTTCAATTTCATGCATTGGAGCCGGAAATGGCAACCCACTCCAGTGTTCTTGCCTGGAGAATCCCAGGGATGGGGAAGCCTGGTGGGCTGCTGTCTATGGGGTCGCAGAGAGTCAGACACGACTGAAGTGACTTAGCAGCAACCTTCTGGAATAAACGCCTCAGGCTTTAAACTCTGGCTTGACCATTCACTAGCCATGGGATCCACTAGAGTCGACCTGCAGGCATGCAAGC<br/>--><br/><--<br/> TGGAGGTGATCGAAACAAACATCACTACGAGACATCGTGGTGACCCTTCGGGAAATTACTTACACGGAAAGGCGTTTAGTGTGTGTGTGTTTATGTGAATATCTTTGTTCCACTAAAAGAACTTTATTATTTTGTTTTAAACCTTCTTCTAAAAATGACAGAATCCTTTTCATTCCGTAACCTTCCACCGATCCATACTGTATACTTCAACGTAAAATTTTGACCTTAACCTGTTGACTATAAGTCACTATAAATACGATGATGGAAGATCTTAGCTCTCGTACGTGGGGTGAGACATGAGAACGGACCTCTTAGGTACTACTCTCGGACCATCCGACGTCAGGTACCCCAGTGTGTCTCAGCCTGTACTGACTCGCTGAAGTGAAAGTGAAAAGTTAAAGTACGTAACCTCGGCCTTTACCGTTGGGTGAGGTCACAAGAACGGACCTCTTAGGGTCCCTACCCCTTCGGACCACCCGACGACAGATACCCCAGCGTCTCTCAGTCTGTGCTGACTTCACTGAATCGTCGTTGGAAGACCTTATTTGCGGAGTCCGAAATTTGAGACCGAACTGGTAAGTGATCGGTACCCTAGGTGATCTCAGCTGGACGTCCGTACGTTCG<br/>  G  G  S  A  K  N  T  T  I  S  Q  L  V  V  P  F  G  K  L  S  H  A  K  G  C  I  V  C  V  C  I  C  K  Y  F  C  P  S  K  R  S  I  I  F  C  F  K  S  S  S  K  *  Q  R  L  F  L  L  A  N  S  P  P  *  T  H  C  I  F  N  C  K  L  V  P  I  P  C  S  I  N  L  S  I  *  A  V  V  K  *  F  R  S  C  A  G  W  E  T  S  K  G  P  S  D  M  I  L  A  Q  Y  A  A  T  W  P  T  V  C  L  R  V  H  S  L  S  K  V  K  V  K  *  N  *  A  N  S  G  S  I  A  V  W  E  L  T  R  A  Q  L  I  G  P  I  P  F  G  P  P  S  S  D  I  P  D  C  L  T  L  C  S  Q  L  S  K  A  A  V  K  Q  F  L  R  R  L  S  *  V  R  A  Q  G  N  V  L  W  P  I  W  *  L  R  G  A  P  M  C  A <br/> V  E  V  L  K  T  Q  L  S  A  R  Y  C  W  Q  S  A  R  *  H  I  H  R  E  A  F  *  V  C  V  F  V  S  I  S  V  L  H  N  E  Q  F  L  L  V  F  N  P  L  L  N  K  S  D  *  S  F  Y  P  M  P  L  H  S  P  I  V  Y  S  T  A  N  *  F  Q  F  Q  V  V  S  I  *  H  Y  K  H  *  *  R  R  S  D  L  A  H  V  G  S  Q  V  R  A  Q  L  I  W  S  S  L  R  T  P  Q  L  G  H  P  *  V  S  D  S  M  V  S  R  S  *  K  *  K  E  I  E  H  M  P  A  P  F  P  L  G  S  W  H  E  Q  R  S  F  G  L  S  P  S  A  Q  H  A  A  T  *  P  T  A  S  L  *  V  R  S  F  H  S  L  L  L  R  R  S  Y  V  G  *  A  K  F  E  P  K  V  M  *  *  G  H  S  G  S  S  D  V  Q  L  C  A  L  <br/>-  R  W  *  S  Q  K  Y  H  H  E  T  A  G  S  P  L  G  K  I  F  T  G  K  R  L  D  C  V  C  L  Y  V  *  L  F  L  T  I  K  K  F  Y  Y  F  L  I  Q  F  F  I  K  V  T  K  P  F  T  L  C  Q  F  T  A  L  Y  S  M  H  L  Q  M  K  F  S  S  N  S  L  Q  Y  E  T  I  N  I  S  S  G  E  L  I  S  L  M  C  G  V  R  Y  E  Q  R  S  F  G  H  H  S  G  P  L  S  C  D  M  P  D  C  L  T  P  C  S  Q  A  V  E  S  E  S  K  L  K  M  C  Q  L  R  F  H  C  G  V  G  T  N  K  G  P  S  D  W  P  H  P  L  R  T  P  Q  Q  R  H  P  R  L  S  D  S  V  V  S  T  V  *  C  C  G  E  P  I  F  A  E  P  K  L  S  Q  S  S  W  E  S  A  M  P  D  V  L  T  S  R  C  A  H  L   <br/><br/><br/><br/>include nucleic acid sequence in output=on:<br/><br/>>gi|555|emb|X65215.1|BTMISATN B.taurus microsatellite DNA (624bp)<br/>ACCTCCACTAGCTTTGTTTGTAGTGATGCTCTGTAGCACCACTGGGAAGCCCTTTAATGAATGTGCCTTTCCGCAAATCACACACACACAAATACACTTATAGAAACAAGGTGATTTTCTTGAAATAATAAAACAAAATTTGGAAGAAGATTTTTACTGTCTTAGGAAAAGTAAGGCATTGGAAGGTGGCTAGGTATGACATATGAAGTTGCATTTTAAAACTGGAATTGGACAACTGATATTCAGTGATATTTATGCTACTACCTTCTAGAATCGAGAGCATGCACCCCACTCTGTACTCTTGCCTGGAGAATCCATGATGAGAGCCTGGTAGGCTGCAGTCCATGGGGTCACACAGAGTCGGACATGACTGAGCGACTTCACTTTCACTTTTCAATTTCATGCATTGGAGCCGGAAATGGCAACCCACTCCAGTGTTCTTGCCTGGAGAATCCCAGGGATGGGGAAGCCTGGTGGGCTGCTGTCTATGGGGTCGCAGAGAGTCAGACACGACTGAAGTGACTTAGCAGCAACCTTCTGGAATAAACGCCTCAGGCTTTAAACTCTGGCTTGACCATTCACTAGCCATGGGATCCACTAGAGTCGACCTGCAGGCATGCAAGC<br/>>gi|555|emb|X65215.1|BTMISATN B.taurus microsatellite DNA (624bp)_+1<br/> T  S  T  S  F  V  C  S  D  A  L  *  H  H  W  E  A  L  *  *  M  C  L  S  A  N  H  T  H  T  N  T  L  I  E  T  R  *  F  S  *  N  N  K  T  K  F  G  R  R  F  L  L  S  *  E  K  *  G  I  G  R  W  L  G  M  T  Y  E  V  A  F  *  N  W  N  W  T  T  D  I  Q  *  Y  L  C  Y  Y  L  L  E  S  R  A  C  T  P  L  C  T  L  A  W  R  I  H  D  E  S  L  V  G  C  S  P  W  G  H  T  E  S  D  M  T  E  R  L  H  F  H  F  S  I  S  C  I  G  A  G  N  G  N  P  L  Q  C  S  C  L  E  N  P  R  D  G  E  A  W  W  A  A  V  Y  G  V  A  E  S  Q  T  R  L  K  *  L  S  S  N  L  L  E  *  T  P  Q  A  L  N  S  G  L  T  I  H  *  P  W  D  P  L  E  S  T  C  R  H  A  S <br/>>gi|555|emb|X65215.1|BTMISATN B.taurus microsatellite DNA (624bp)_+2<br/>  P  P  L  A  L  F  V  V  M  L  C  S  T  T  G  K  P  F  N  E  C  A  F  P  Q  I  T  H  T  Q  I  H  L  *  K  Q  G  D  F  L  E  I  I  K  Q  N  L  E  E  D  F  Y  C  L  R  K  S  K  A  L  E  G  G  *  V  *  H  M  K  L  H  F  K  T  G  I  G  Q  L  I  F  S  D  I  Y  A  T  T  F  *  N  R  E  H  A  P  H  S  V  L  L  P  G  E  S  M  M  R  A  W  *  A  A  V  H  G  V  T  Q  S  R  T  *  L  S  D  F  T  F  T  F  Q  F  H  A  L  E  P  E  M  A  T  H  S  S  V  L  A  W  R  I  P  G  M  G  K  P  G  G  L  L  S  M  G  S  Q  R  V  R  H  D  *  S  D  L  A  A  T  F  W  N  K  R  L  R  L  *  T  L  A  *  P  F  T  S  H  G  I  H  *  S  R  P  A  G  M  Q  A <br/>>gi|555|emb|X65215.1|BTMISATN B.taurus microsatellite DNA (624bp)_+3<br/>   L  H  *  L  C  L  *  *  C  S  V  A  P  L  G  S  P  L  M  N  V  P  F  R  K  S  H  T  H  K  Y  T  Y  R  N  K  V  I  F  L  K  *  *  N  K  I  W  K  K  I  F  T  V  L  G  K  V  R  H  W  K  V  A  R  Y  D  I  *  S  C  I  L  K  L  E  L  D  N  *  Y  S  V  I  F  M  L  L  P  S  R  I  E  S  M  H  P  T  L  Y  S  C  L  E  N  P  *  *  E  P  G  R  L  Q  S  M  G  S  H  R  V  G  H  D  *  A  T  S  L  S  L  F  N  F  M  H  W  S  R  K  W  Q  P  T  P  V  F  L  P  G  E  S  Q  G  W  G  S  L  V  G  C  C  L  W  G  R  R  E  S  D  T  T  E  V  T  *  Q  Q  P  S  G  I  N  A  S  G  F  K  L  W  L  D  H  S  L  A  M  G  S  T  R  V  D  L  Q  A  C  K  - <br/>>gi|555|emb|X65215.1|BTMISATN B.taurus microsatellite DNA (624bp)_reverse<br/>GCTTGCATGCCTGCAGGTCGACTCTAGTGGATCCCATGGCTAGTGAATGGTCAAGCCAGAGTTTAAAGCCTGAGGCGTTTATTCCAGAAGGTTGCTGCTAAGTCACTTCAGTCGTGTCTGACTCTCTGCGACCCCATAGACAGCAGCCCACCAGGCTTCCCCATCCCTGGGATTCTCCAGGCAAGAACACTGGAGTGGGTTGCCATTTCCGGCTCCAATGCATGAAATTGAAAAGTGAAAGTGAAGTCGCTCAGTCATGTCCGACTCTGTGTGACCCCATGGACTGCAGCCTACCAGGCTCTCATCATGGATTCTCCAGGCAAGAGTACAGAGTGGGGTGCATGCTCTCGATTCTAGAAGGTAGTAGCATAAATATCACTGAATATCAGTTGTCCAATTCCAGTTTTAAAATGCAACTTCATATGTCATACCTAGCCACCTTCCAATGCCTTACTTTTCCTAAGACAGTAAAAATCTTCTTCCAAATTTTGTTTTATTATTTCAAGAAAATCACCTTGTTTCTATAAGTGTATTTGTGTGTGTGTGATTTGCGGAAAGGCACATTCATTAAAGGGCTTCCCAGTGGTGCTACAGAGCATCACTACAAACAAAGCTAGTGGAGGT<br/>>gi|555|emb|X65215.1|BTMISATN B.taurus microsatellite DNA (624bp)_-1<br/> A  C  M  P  A  G  R  L  *  W  I  P  W  L  V  N  G  Q  A  R  V  *  S  L  R  R  L  F  Q  K  V  A  A  K  S  L  Q  S  C  L  T  L  C  D  P  I  D  S  S  P  P  G  F  P  I  P  G  I  L  Q  A  R  T  L  E  W  V  A  I  S  G  S  N  A  *  N  *  K  V  K  V  K  S  L  S  H  V  R  L  C  V  T  P  W  T  A  A  Y  Q  A  L  I  M  D  S  P  G  K  S  T  E  W  G  A  C  S  R  F  *  K  V  V  A  *  I  S  L  N  I  S  C  P  I  P  V  L  K  C  N  F  I  C  H  T  *  P  P  S  N  A  L  L  F  L  R  Q  *  K  S  S  S  K  F  C  F  I  I  S  R  K  S  P  C  F  Y  K  C  I  C  V  C  V  I  C  G  K  A  H  S  L  K  G  F  P  V  V  L  Q  S  I  T  T  N  K  A  S  G  G <br/>>gi|555|emb|X65215.1|BTMISATN B.taurus microsatellite DNA (624bp)_-2<br/>  L  A  C  L  Q  V  D  S  S  G  S  H  G  *  *  M  V  K  P  E  F  K  A  *  G  V  Y  S  R  R  L  L  L  S  H  F  S  R  V  *  L  S  A  T  P  *  T  A  A  H  Q  A  S  P  S  L  G  F  S  R  Q  E  H  W  S  G  L  P  F  P  A  P  M  H  E  I  E  K  *  K  *  S  R  S  V  M  S  D  S  V  *  P  H  G  L  Q  P  T  R  L  S  S  W  I  L  Q  A  R  V  Q  S  G  V  H  A  L  D  S  R  R  *  *  H  K  Y  H  *  I  S  V  V  Q  F  Q  F  *  N  A  T  S  Y  V  I  P  S  H  L  P  M  P  Y  F  S  *  D  S  K  N  L  L  P  N  F  V  L  L  F  Q  E  N  H  L  V  S  I  S  V  F  V  C  V  *  F  A  E  R  H  I  H  *  R  A  S  Q  W  C  Y  R  A  S  L  Q  T  K  L  V  E  V <br/>>gi|555|emb|X65215.1|BTMISATN B.taurus microsatellite DNA (624bp)_-3<br/>   L  H  A  C  R  S  T  L  V  D  P  M  A  S  E  W  S  S  Q  S  L  K  P  E  A  F  I  P  E  G  C  C  *  V  T  S  V  V  S  D  S  L  R  P  H  R  Q  Q  P  T  R  L  P  H  P  W  D  S  P  G  K  N  T  G  V  G  C  H  F  R  L  Q  C  M  K  L  K  S  E  S  E  V  A  Q  S  C  P  T  L  C  D  P  M  D  C  S  L  P  G  S  H  H  G  F  S  R  Q  E  Y  R  V  G  C  M  L  S  I  L  E  G  S  S  I  N  I  T  E  Y  Q  L  S  N  S  S  F  K  M  Q  L  H  M  S  Y  L  A  T  F  Q  C  L  T  F  P  K  T  V  K  I  F  F  Q  I  L  F  Y  Y  F  K  K  I  T  L  F  L  *  V  Y  L  C  V  C  D  L  R  K  G  T  F  I  K  G  L  P  S  G  A  T  E  H  H  Y  K  Q  S  *  W  R  - </pre>",
    "Codon table": "Here you can select which codon usage table should be used to translate your nucleic acid sequence.",
    //Backtranslator
    "Input   ": "Enter a single protein sequence in FASTA format, for example:<br/><br/><br/><pre>>gi|576099|pdb|1ENA|  Staphylococcal Nuclease (E.C.3.1.31.1) Mutation<br/>LHKEPATLIKAIDGETVKLMYKGQPMTFRLLLVDTPETKHPKKGVEKYGPEASAFTKKMVENAKKIEVEF<br/>DKGQRTDKYGRGLAYIYADGKMVNEALVRQGLAKVAYVYKPNNTHEQHLRKSEAQAKKEKLNIWS</pre><br/><br/>The sequence name (and an optional description) must be contained in a single name line beginning with the > symbol and followed directly by the sequence name. The residue data is contained in one or more lines of arbitrary length following the name line. No empty lines should be used. Upper and lower case letters are allowed. Spaces and gaps (-.) will be removed.",
    "Include amino acid sequence in output": "Activate this option to include the amino acid sequence in the output. If activated amino acids will be printed above their encoding DNA codon. For example, results may look like this:<br/><br/>	<br/><pre>> Possible DNA-sequence of Protein: gi|576099|pdb|1ENA|  Staphylococcal Nuclease (E.C.3.1.31.1) Mutation<br/> L   H   K   E   P   A   T   L   I   K   A   I   D   G   E   T   V   K   L   M  <br/>TTG CAC AAG GAG CCT GCA ACC CTT ATC AAG GCA ATA GAC GGA GAA ACG GTA AAA CTC ATG <br/><br/> Y   K   G   Q   P   M   T   F   R   L   L   L   V   D   T   P   E   T   K   H  <br/>TAC AAA GGG CAG CCT ATG ACG TTT CGG CTA CTT CTG GTC GAC ACG CCG GAA ACG AAA CAT <br/><br/> P   K   K   G   V   E   K   Y   G   P   E   A   S   A   F   T   K   K   M   V  <br/>CCG AAG AAA GGG GTC GAA AAA TAT GGT CCG GAG GCC TCG GCA TTC ACA AAA AAG ATG GTA <br/><br/> E   N   A   K   K   I   E   V   E   F   D   K   G   Q   R   T   D   K   Y   G  <br/>GAG AAT GCT AAA AAA ATT GAG GTG GAG TTT GAT AAA GGT CAA CGA ACT GAC AAG TAT GGT <br/><br/> R   G   L   A   Y   I   Y   A   D   G   K   M   V   N   E   A   L   V   R   Q  <br/>AGA GGT TTA GCT TAC ATC TAC GCT GAT GGC AAG ATG GTC AAT GAA GCC CTC GTT CGG CAA <br/><br/> G   L   A   K   V   A   Y   V   Y   K   P   N   N   T   H   E   Q   H   L   R  <br/>GGC CTT GCA AAA GTG GCT TAT GTG TAT AAA CCG AAC AAT ACT CAC GAA CAA CAC CTT CGG <br/><br/> K   S   E   A   Q   A   K   K   E   K   L   N   I   W   S  <br/>AAA TCA GAG GCT CAG GCA AAA AAG GAA AAA TTA AAT ATC TGG TCG </pre><br/><br/>With the option turned of the above output would look like this:<br/><br/><br/><pre>> Possible DNA-sequence of Protein: gi|576099|pdb|1ENA|  Staphylococcal Nuclease (E.C.3.1.31.1) Mutation <br/>CTTCATAAAGAACCAGCTACCCTTATCAAGGCGATCGACGGTGAAACTGTGAAGCTCATGTACAAGGGGCAAC<br/>CAATGACGTTCCGCCTGCTGCTTGTGGACACACCCGAAACTAAGCACCCCAAAAAAGGTGTAGAGAAATACGG<br/>GCCTGAAGCTTCTGCCTTCACAAAAAAGATGGTGGAAAACGCAAAGAAGATTGAGGTCGAATTCGATAAGGGC<br/>CAAAGAACGGATAAGTACGGTCGAGGGCTGGCCTATATTTACGCTGACGGCAAGATGGTGAATGAAGCGCTCG<br/>TGCGTCAAGGACTGGCCAAAGTAGCATACGTTTACAAACCAAATAATACTCATGAGCAACACCTCCGAAAGAG<br/>TGAGGCGCAAGCTAAAAAGGAAAAGCTCAACATCTGGTCC</pre>",
    "Genetic code": "Select the correct genetic code for your organism. The genetic codes are identical to those available on the <a href = https://www.ncbi.nlm.nih.gov/taxonomy/>NCBI homepage</a>.",
    "Codon usage table": "Select this option if you want to use a codon usage table. If this option is not selected all possible codons will have the same probability.",
    "Codon usage table of organism": "Here you can search for a codon usage table of an organism. The codon usage database (identical to the one of <a href = http://www.kazusa.or.jp/codon/>http://www.kazusa.or.jp/codon/</a>) is based of the <a href = https://www.ncbi.nlm.nih.gov/genbank/>NCBI GenBank</a> (GenBank Release 160.0 [June 15 2007]). Please enter the name of the organism in the box.",
    //HHfilter
    //"Input"
    //"Input format"
    "Max. sequence identity": "Filter by maximum pairwise sequence identity.",
    "Min. sequence identity with query": "Filter by minimum sequence identity with query.",
    "Min. coverage with query": " Filter by minimum coverage with query.",
    "Extract N most dissimilar sequences": " Filter most diverse set of sequences, keeping at least this many sequences in each block of >50 columns.",
    //Seq2ID
    "Input    ": "Enter a file with GI-numbers. This tool finds only numbers with the format: <b>gi|number</b> !<br/>Exception: If a line starts with <b>>NR20|</b>, the <b>gi</b> prefix is not necessary.",
    "Extract only last gi number per input line": "If this option is checked, only at most one gi number per input line (the last one) will be listed. Otherwise, all found gi numbers are listed.",
    //RetSeq
    "Input     ": "Enter a newline separated list of identifiers.",
    "Retrieve identical sequences only once": "With this option the program filters identical identifiers and identical sequences, so that the result sequences are unique."
};

exampleContent = {
    "psiblast": ["Search with an amino acid sequence against protein databases for locally similar sequences.<br/>Similar to Protein BLAST+ but more sensitive. PSI-BLAST+ first performs a BLAST+ search and builds an alignment from the best local hits. This alignment is then used as a query for the next round of search. After each successive round the search alignment is updated.", ["Alignment", "Database", "Matrix", "Number of iterations", "E-Value", "E-value inclusion threshold", "Alignments and descriptions"], "At the top of the results page is a \"View alignment\"-button, that shows the multiple alignment of all hits.<br/>Then there is a list with brief information about all hits that were found.<br/>They are listed together with their scores and e-values.<br/>There you can select the ones, you want to work with. Below this you can find more information.<br/>There are all pairwise alignments and there is also additional information about the method,<br/>the identities and the positives. There you also have the possibility to select the hits, you want to work with.<br/>At the bottom you can find more information about the database and the matrix.<br/>The \"View alignment\"-button shows the multiple alignment of all sequences that was found by PSI-BLAST. The different colours illustrate the identities between the different amino acids. This helps you to decide, whether the found sequences may be really homologue", "Alva V., Nam SZ., Söding J., Lupas AN. (2016) <b>The MPI bioinformatics Toolkit as an integrative platform for advanced protein sequence and structure analysis.</b> Nucleic Acids Res. pii: gkw348. <a href=https://www.ncbi.nlm.nih.gov/pubmed/27131380>PMID: 27131380</a><br/><br/>Altschul S., Madden T., Schaffer A., Zhang J., Miller W., Lipman D. (1997) <b>Gapped BLAST and PSI-BLAST: a new generation of protein database search programs.</b> Nucl Acids Res 25(17):3389-3402. <a href = https://www.ncbi.nlm.nih.gov/pmc/articles/PMC146917/>PMCID: PMC146917</a><br/><br/>Altschul SF., Gish W., Miller W., Myers EW., Lipman DJ. (1990) <b>Basic local alignment search tool.</b> J Mol Biol. 215:403-410. <a href = https://www.ncbi.nlm.nih.gov/pubmed/2231712>PMID: 2231712</a><br/><br/>Camacho C., Coulouris G., Avagyan V., Ma N., Papadopoulos J., Bealer K., Madden TL. (2008) <b>BLAST+: architecture and applications.</b> BMC Bioinformatics 10:421. <a href = https://www.ncbi.nlm.nih.gov/pubmed/20003500>PMID: 20003500</a>"],
    "hhblits": ["For an accurate sequence-based prediction of protein structure and function, a highly sensitive sequence-search method is crucial. Today, profile-profile and HMM-HMM alignment methods are custom when it comes to identification and alignment of templates for 3D-homology modeling. As these methods are generally too slow for iteratively searching through large databases, we offer HHblits as a 'lightning-fast' HMM-HMM-based iterative sequence search tool. The profile-profile alignment prefilter of HHblits reduces the number of full HMM-HMM alignments from many millions to a few thousand, making it faster than PSI-BLAST but still as sensitive as HHsearch. Prefiltering is essential for speed and sensitivity. Compared to PSI-BLAST, HHblits is faster, up to twice as sensitive and produces more accurate alignments.<br/><br/>The HHblits software is part of the open source package HHsuite.", ["Paste sequence or MSA", "Select HMM databases", "Max. number of iterations", "E-value inclusion threshold ", "Minimum probability in hit list", "Max. number of hits in hit list"], "Alva V., Nam SZ., Söding J., Lupas AN. (2016) <b>The MPI bioinformatics Toolkit as an integrative platform for advanced protein sequence and structure analysis.</b> Nucleic Acids Res. pii: gkw348. <a href=https://www.ncbi.nlm.nih.gov/pubmed/27131380>PMID: 27131380</a><br/><br/>Remmert M., Biegert A., Hauser A., Söding J. (2011) <b>HHblits: Lightning-fast iterative protein sequence searching by HMM-HMM alignment.</b> Nat Methods. 9(2):173-5. doi: 10.1038/nmeth.1818. <a href=https://www.ncbi.nlm.nih.gov/pubmed/22198341>PMID: 22198341</a>"],
    "hhpred": ["<h3>About HHpred</h3><br/><h5><b>HHpred performance in the CASP9 structure prediction benchmark</b></h5>In the latest CASP competition that took place in 2010, our HHpred server was assessed as one of the best out of the 81 servers, in particular in template-based structure prediction, the category most relevant for biological applications. (See <a href = http://onlinelibrary.wiley.com/doi/10.1002/prot.23177/abstract>Mariani, 2011</a>, <a href = http://predictioncenter.org/casp9/groups_analysis.cgi?type=server&tbm=on&submit=Filter>CASP9 results table</a>, and figure below). This is remarkable for two reasons: First, with ~4 minutes median response time, HHpred is about 100 times faster than other top servers. Second, most top servers use our freely available software HHsearch to find templates and to align target to template sequences. For CASP8 results, please refer to our publication in Proteins (2009).<br/><div class = \"wraptocenter\"><img src = https://toolkit.tuebingen.mpg.de/images/hhpred/CASP9_results.png></div><br/>Official CASP9 cumulative Z-scores for model quality in the template-based modeling category, plotted over the median response time of the 81 participating servers. (See <a href = http://predictioncenter.org/casp9/groups_analysis.cgi?type=server&tbm=on&submit=Filter>table</a>)<br/><br/><h5><b>HHpred server for protein remote homolog detection and 3D structure prediction</b></h5>The primary aim in developing HHpred was to provide biologists with a method for sequence database searching and structure prediction that is as easy to use as BLAST or PSI-BLAST and that is at the same time much more sensitive in finding remote homologs. In fact, HHpred's sensitivity is competitive with the most powerful servers for structure prediction currently available.<br/>HHpred is the first server that is based on the pairwise comparison of profile hidden Markov models (HMMs). Whereas most conventional sequence search methods search sequence databases such as UniProt or the NR, HHpred searches alignment databases, like Pfam or SMART. This greatly simplifies the list of hits to a number of sequence families instead of a clutter of single sequences. All major publicly available profile and alignment databases are available through HHpred.<br/><br/>HHpred accepts a single query sequence or a multiple alignment as input. Within only a few minutes it returns the search results in an easy-to-read format similar to that of PSI-BLAST. Search options include local or global alignment and scoring secondary structure similarity. HHpred can produce pairwise query-template sequence alignments, merged query-template multiple alignments (e.g. for transitive searches), as well as 3D structural models calculated by the MODELLER software from HHpred alignments.<br/><br/><h5><b>When can HHpred be useful for you?</b></h5>It is well known that sequence search methods such as BLAST, FASTA, or PSI-BLAST are of prime importance for biological research because functional information of a protein or gene can be inferred from homologous proteins or genes identified in a sequence search. But quite often no significant relationship to a protein of known function can be established. This is certainly the case for the most interesting group of proteins, those for which no ortholog has yet been studied.<br/>It is less well known that in cases where conventional sequence search methods fail, the recently developed, highly sensitive methods for homology detection or structure prediction (see descriptions and links at <a href = http://meta.bioinfo.pl/submit_wizard.pl>http://meta.bioinfo.pl/submit_wizard.pl</a>) quite often allow to make inferences from more remotely homologous relationships [<a href = https://www.ncbi.nlm.nih.gov/pubmed/10871397?dopt=Abstract>Venclovas (2000)</a>, <a href = https://www.ncbi.nlm.nih.gov/pubmed/15688435?dopt=Abstract>Zheng (2005)</a>, <a href = https://www.ncbi.nlm.nih.gov/pubmed/14983005?dopt=Abstract>Ginalski (2004)</a>, <a href = https://www.ncbi.nlm.nih.gov/pubmed/15452342?dopt=Abstract>Rand (2004)</a>, <a href = https://www.ncbi.nlm.nih.gov/pubmed/15684412?dopt=Abstract>Pawlak (2005)</a>, <a href = https://www.ncbi.nlm.nih.gov/pubmed/15048836?dopt=Abstract>Kihara (2004)</a>]. If the relationship is so remote that no common function can be assumed, one can generally still derive hypotheses about possible mechanisms, active site positions and residues, or the class of substrate bound [<a href = https://www.ncbi.nlm.nih.gov/pubmed/11286560?dopt=Abstract>Todd (2001)</a>, <a href = https://www.ncbi.nlm.nih.gov/pubmed/10902155?dopt=Abstract>Pawlowski (2000)</a>]. When a homologous protein with known structure can be identified, its stucture can be used as a template to model the 3D structure for the protein of interest [<a href = https://www.ncbi.nlm.nih.gov/pubmed/9710568?dopt=Abstract>Rychlewski (1998)</a>], since even remotely homologous proteins generally have quite similar 3D structure [<a href = https://www.ncbi.nlm.nih.gov/pubmed/12127461?dopt=Abstract>Kinch (2002)</a>]. The 3D model may then help to generate hypotheses to guide experiments.<br/><br/><h5><b>What is HMM-HMM comparison and why is it so powerful?</b></h5>When searching for remote homologs, it is wise to make use of as much information about the query and database proteins as possible in order to better distinguish true from false positives and to produce optimal alignments. This is the reason why sequence-sequence comparison is inferior to profile-sequence comparison. Sequence profiles contain for each column of a multiple alignment the frequencies of the 20 amino acids. They therefore contain detailed information about the conservation of each residue position, i.e. how important each position is for defining other members of the protein family, and about the preferred amino acids. Profile Hidden Markov Models (<b>HMMs</b>) are similar to simple sequence profiles, but in addition to the amino acid frequencies in the columns of a multiple sequence alignment they contain information about the frequency of inserts and deletions at each column. Using profile HMMs in place of simple sequence profiles should therefore further improve sensitivity. HHpred is the first server to employ HMM-HMM comparison, based on a novel statistical method that we have developed recently. Using HMMs both on the query and the database side greatly enhances the sensitivity and selectivity over sequence-profile based methods such as PSI-BLAST.<br/><br/><h5><b>Structure prediction with HHpred</b></h5>The most successful techniques for protein structure prediction rely on identifying homologous sequences with known structure to be used as template. This works so well because structures diverge much more slowly than sequences and homologous proteins may have very similar structures even when their sequences have diverged beyond recognition. But sensitivity in homology detection is crucial for success since many proteins have only remote relatives in the structure database.<br/><br/><h5><b>Most publicly available alignment databases can be searched</b></h5>A large number of publicly available (partly redundant) and manually or automatically annotated databases of protein family alignments can be searched through our webserver, such as <a href = http://pfam.xfam.org/>PFAM</a>, <a href = http://smart.embl-heidelberg.de/>SMART</a>, <a href = http://www.pantherdb.org/>PANTHER</a>, <a href = http://www.jcvi.org/cgi-bin/tigrfams/index.cgi>TIGRFAMs</a>, <a href = http://pir.georgetown.edu/pirwww/dbinfo/iproclass.shtml>PIRSF</a>, or <a href = https://www.ncbi.nlm.nih.gov/COG/>COG/KOG</a>. Two composite databases in HHpred that include most of these original databases are <a href = http://www.ebi.ac.uk/interpro/>InterPro</a> from the EBI and the conserved domain database <a href = https://www.ncbi.nlm.nih.gov/Structure/cdd/cdd.shtml>CDD</a> from the NCBI. A number of alignment databases are built around sequences of known structure, generally by starting with such a sequence as seed and searching for sequence homologs. We built two such databases ourselves, one using full-length sequences from the <a href = http://www.rcsb.org/pdb/home/home.do>Protein Data Bank</a> (PDB) as seeds, and one database that uses PDB sequences cut into structural and evolutionary domains as defined by the <a href = http://scop.mrc-lmb.cam.ac.uk/scop/index.html>SCOP database</a>. These two databases are referred to simply as pdb70 and scop70 in HHpred. An alignment database very much like our scop70 is <a href = http://supfam.mrc-lmb.cam.ac.uk/SUPERFAMILY/>SUPERFAMILY</a>, whereas <a href = http://www.cathdb.info/Gene3D/>CATH/Gene3D</a> is based on <a href = http://scop.mrc-lmb.cam.ac.uk/scop/index.html>CATH</a>, a database for hierarchical classification of structural domains that is similar to SCOP.<br/><br/><h5><b>Benchmark</b></h5>HHsearch was benchmarked together with BLAST, PSI-BLAST, HMMER, the profile-profile tools PROF\_SIM and COMPASS, and another HMM-HMM tool, PRC [M. Madera, <a href = http://supfam.mrc-lmb.cam.ac.uk/PRC>http://supfam.mrc-lmb.cam.ac.uk/PRC</a>}] in an all-against-all comparison of a database of 3691 protein domains from SCOP 1.63 with pairwise sequence identities below 20%. HHsearch is able to detect 2.8--2.9 times more homologs than PSIBLAST or HMMER, twice more than PROF_SIM, 60% more than COMPASS and 7% more than PRC at a fixed error rate of 20% (see figure). By including secondary structure predicted by <a href = https://www.ncbi.nlm.nih.gov/pubmed/10493868?dopt=Abstract>PSIPRED</a> a further improvement of 20% can be achieved. We show further that the improvements in sensitivity are mirrored by an increase in alignment quality.<br/>Thanks to a fast column score and an efficient algorithm, HHsearch runs four times faster than PRC, ten times faster than PROF_SIM (4/2004), and 17 times fasterthan COMPASS (4/2004), scanning a profile HMM with 200 columns against 3691 SCOP domains in 33 s on a 2 GHz AMD64 processor.<br/><br/><div class=\"wraptocenter\"><img src = https://toolkit.tuebingen.mpg.de/images/hhpred/hhpred_bench.png></div><h5><b>Theory of HMM-HMM comparison</b></h5>We have extended the theory of Hidden Markov Models to the case of pairwise comparison of HMMs and developed a method for sequence homology search which is based on this theory. The alignment algorithm maximizes a weighted form of coemission probability, the probability that the two HMMs will emit the same sequence of residues. Amino acids are weighted according to their abundance, rare coemitted amino acids contributing more to the alignment score. This weighting is analogous to the use of a null model with amino acid background probabilities in HMM-sequence comparison.<br/>Secondary structure can be included in the HMM-HMM comparison. We score pairs of aligned secondary structure states in a way analogous to the classical amino acids substitution matrices. We use ten different substitution matrices that we derived from a statistical analysis of the structure database, one for each confidence value given by <a href = https://www.ncbi.nlm.nih.gov/pubmed/10493868?dopt=Abstract>PSIPRED</a>. By making use of confidence values and using a strictly statistical approach we are able to improve the sensitivity gain considerably over previous methods.<br/><br/><h5><b>HHpred calculates E-values as well as probabilities for a match to be a true positive</b></h5>To assert a homologous relationship for a pair of proteins it is necessary to quantify one's uncertainty. Most sequence similarity search programs report the significance of a database match by an E-value (expect-value). The E-value is the average expected number of non-homologous proteins with a score higher than the one obtained for the database match. An E-value much lower than 1 therefore indicates statistical significance.<br/>To derive E-values, a representative database composed of one HMM per SCOP fold is searched with the query HMM before the actual database search. We record the score distibution and fit an Gumbel extreme value distribution (EVD) to it. From this distribution, one could in principle calculate for each score S the P-value as the area under the EVD to the right of S, i.e. P-value = Prob(score>S). Then the E-value could be calculated for each score according to<br/><br/><div align = center><em>E-value N x P-value,</em></div><br/>where N is the number of HMMs in the database to be scanned. But in order to record a score distribution of negatives, we should exclude all proteins that are homologous to the query protein. We do this in two steps. Before fitting an EVD to the score distribution, we exclude the best-scoring three folds. We then calculate preliminary E-values for the representative HMMs and now exclude those HMMs with an E-value better than 0.5. The lambda and mu parameters for the EVD are determined by Maximum Likelihood. In local alignment mode, we use a simple log length correction.<br/>When we add secondary structure scores, we find that the score disctribution of negatives does not strictly follow an EVD anymore. E-values in HHpred therefore do not include secondary structure scores! As an alternative to E-values, HHpred reports probabilities for a homologous relationship. This probability is based on the real-world score distribution for negative and homologous domain pairs in an all-against-all comparison of SCOP. Since the probability also contains the contribution from the secondary structure score, it is more sensitive than the E-value. The probability is calculated according to:<br/><fiv align = center>Probability for homologous relationship = Prob_hom(score=S) / [Prob_neg(score=S) + Prob_hom(score=S)]</div><br/>Homologous pairs are from the same superfamily or have a structural superposition MAXSUB score > 0.1. All other pairs are counted as negatives in this context.<br/>The large number of compared sequences in our benchmark allowed us to quantitatively test the accuracy of reported E-values. Remarkably, E-values below 1E-4 are wrong for all tested tools except BLAST and HMMER. Specifically, when using PSI-BLAST with standard parameters (up to eight iterations at E-value threshold of 1E-4) ~100 times too many sequences were observed with a reported E-value of 1E-4 or better. The probabilities take account of the complications (mainly alignments corrupted by non-homologous sequences) that lead to these deviations. They are therefore a more realistic measure of confidence.", ["Paste single sequence or multiple alignment", "MSA generation method", "Max. MSA generation iterations", "Select HMM databases ", "Score secondary structure", "Alignment mode ", "Realign with MAC algorithm ", "E-value threshold for MSA generation", "Min. coverage of MSA hits", "Min. sequence identity of MSA hits with query", "MAC realignment threshold ", "Minimum probability in hit list", "Max. number of hits in hit list"], "Alva V., Nam SZ., Söding J., Lupas AN. (2016) <b>The MPI Bioinformatics Toolkit as an integrative platform for advanced protein sequence and structure analysis.</b> Nucleic Acids Res. pii: gkw348. <a href=https://www.ncbi.nlm.nih.gov/pubmed/27131380>PMID: 27131380</a><br/><br/>Söding J., Biegert A., Lupas AN. (2005) <b>The HHpred interactive server for protein homology detection and structure prediction.</b> Nucleic Acids Res 33 (Web Server issue), W244-W248. <a href = https://www.ncbi.nlm.nih.gov/pubmed/15980461>PMID: 15980461</a><br/><br/>Söding J. (2005) <b>Protein homology detection by HMM-HMM comparison.</b> Bioinformatics 21: 951-960. <a href = https://www.ncbi.nlm.nih.gov/pubmed/15531603>PMID: 15531603</a><br/><br/>Hildebrand A., Remmert A., Biegert A., Söding J. (2009) <b>Fast and accurate automatic structure prediction with HHpred.</b> Proteins 77(Suppl 9):128-32. doi: 10.1002/prot.22499. <a href = https://www.ncbi.nlm.nih.gov/pubmed/19626712>PMID: 19626712</a><br/><br/>Meier A., Söding J. (2015) <b>Automatic Prediction of Protein 3D Structures by Probabilistic Multi-template Homology Modeling.</b> PLoS Comput Biol. 11(10):e1004343. doi: 10.1371/journal.pcbi.1004343. <a href = https://www.ncbi.nlm.nih.gov/pubmed/26496371>PMID: 26496371</a><br/><br/>Remmert M., Biegert A., Hauser A., Söding J. (2011) <b>HHblits: Lightning-fast iterative protein sequence searching by HMM-HMM alignment.</b> Nat Methods. 9(2):173-5. doi: 10.1038/nmeth.1818. <a href=https://www.ncbi.nlm.nih.gov/pubmed/22198341>PMID: 22198341</a>"],
    "hmmer": ["HMMER3 is a utility for fast and sensitive homology searchers. Input is a single sequence or a multiple alignment in aligned FASTA/CLUSTAL format which is used to build a profile HMM. ", ["Input", "Database ", "Max. MSA generation iterations", "Extract sequences up to E-value", "Max. number of hits in hit list"], "Alva V., Nam SZ., Söding J., Lupas AN. (2016) <b>The MPI bioinformatics Toolkit as an integrative platform for advanced protein sequence and structure analysis.</b> Nucleic Acids Res. pii: gkw348. <a href=https://www.ncbi.nlm.nih.gov/pubmed/27131380>PMID: 27131380</a><br/><br/>Finn RD., Clements J., Eddy SR. (2011) <b>HMMER web server: interactive sequence similarity searching.</b> Nucleic Acids Res. 39(Web Server issue): W29–W37. <a href = https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3125773/>PMCID: PMC3125773</a>"],
    "patsearch": ["Retrieves all sequences which contain a given PROSITE pattern or regular expression.", ["Input format ", "Select database", "Sequence count cutoff"], "Alva V., Nam SZ., Söding J., Lupas AN. (2016) <b>The MPI bioinformatics Toolkit as an integrative platform for advanced protein sequence and structure analysis.</b> Nucleic Acids Res. pii: gkw348. <a href=https://www.ncbi.nlm.nih.gov/pubmed/27131380>PMID: 27131380</a>"],
    "clustalo": ["ClustalO is a general purpose multiple alignment program for DNA or proteins.", ["Input format  "], "Alva V., Nam SZ., Söding J., Lupas AN. (2016) <b>The MPI bioinformatics Toolkit as an integrative platform for advanced protein sequence and structure analysis.</b> Nucleic Acids Res. pii: gkw348. <a href=https://www.ncbi.nlm.nih.gov/pubmed/27131380>PMID: 27131380</a><br/><br/>Sievers F., Wilm A., Dineen D., Gibson TJ., Karplus K., Li W., Lopez R., McWilliam H., Remmert M., Söding J., Thompson JD., Higgins DG. (2011) <b>Fast, scalable generation of high-quality protein multiple sequence alignments using Clustal Omega.</b> Mol Syst Biol 7:539. <a href = https://www.ncbi.nlm.nih.gov/pubmed/21988835>PMID: 21988835</a>"],
    "kalign": ["Kalign - an accurate and fast multiple sequence alignment algorithm.<br/><br/>Timo Lassmann and Erik LL Sonnhammer (2005) Kalign - an accurate and fast multiple sequence alignment algorithm<br/>BMC Bioinformatics 2005, 6:298 doi:10.1186/1471-2105-6-298", ["Input format   ", "Output sequence order", "Gap open penalty", "Gap extension penalty", "Terminal gap penalties", "Bonus score"], "Alva V., Nam SZ., Söding J., Lupas AN. (2016) <b>The MPI bioinformatics Toolkit as an integrative platform for advanced protein sequence and structure analysis.</b> Nucleic Acids Res. pii: gkw348. <a href=https://www.ncbi.nlm.nih.gov/pubmed/27131380>PMID: 27131380</a><br/><br/>Lassmann T., Sonnhammer EL. (2005) <b>Kalign--an accurate and fast multiple sequence alignment algorithm.</b> BMC Bioinformatics. 6:298.<a href = https://www.ncbi.nlm.nih.gov/pubmed/16343337>PMID: 16343337</a>"],
    "mafft": ["Multiple alignment program for amino acid or nucleotide sequences<br/>[MAFFT v7; <a href = http://mafft.cbrc.jp/alignment/software/>http://mafft.cbrc.jp/alignment/software/</a>] ", ["Input format  ", "Alignment order", "Gap open penalty", "Offset"], "Alva V., Nam SZ., Söding J., Lupas AN. (2016) <b>The MPI bioinformatics Toolkit as an integrative platform for advanced protein sequence and structure analysis.</b> Nucleic Acids Res. pii: gkw348. <a href=https://www.ncbi.nlm.nih.gov/pubmed/27131380>PMID: 27131380</a><br/><br/>Katoh K., Misawa K., Kuma K., Miyata T. (2002) <b>MAFFT: a novel method for rapid multiple sequence alignment based on fast Fourier transform.</b> Nucleic Acid Res 30(14):3059-3066. <a href = https://www.ncbi.nlm.nih.gov/pubmed/12136088>PMID: 12136088</a>"],
    "msaprobs": ["MSAProbs is a practical multiple alignment algorithm for protein sequences. The design of MSAProbs is based on a combination of pair hidden Markov models and partition functions to calculate posterior probabilities. Assessed using the popular benchmarks: BAliBASE, PREFAB, SABmark and OXBENCH, MSAProbs achieves statistically significant accuracy improvements over the existing top performing aligners, including ClustalW, MAFFT, MUSCLE, ProbCons and Probalign. Furthermore, MSAProbs is optimized for multi-core CPUs by employing a multi-threaded design, leading to a competitive execution time compared to other aligners.", ["Input format  "], "Alva V., Nam SZ., Söding J., Lupas AN. (2016) <b>The MPI bioinformatics Toolkit as an integrative platform for advanced protein sequence and structure analysis.</b> Nucleic Acids Res. pii: gkw348. <a href=https://www.ncbi.nlm.nih.gov/pubmed/27131380>PMID: 27131380</a><br/><br/>Liu Y., Schmidt B., Maskell DL. (2010) <b>MSAProbs: multiple sequence alignment based on pair hidden Markov models and partition function posterior probabilities.</b> Bioinformatics 26(16): 1958-64. <a href = https://www.ncbi.nlm.nih.gov/pubmed/20576627>PMID: 20576627</a>"],
    "muscle": ["MUSCLE is a multiple alignment program for protein sequences. MUSCLE stands for multiple sequence comparison by log-expectation.<br/>[MUSCLE 3.8.31; <a href = http://www.drive5.com/muscle/>http://www.drive5.com/muscle/</a>]", ["Input format   ", "Maximum number of iterations"], "Alva V., Nam SZ., Söding J., Lupas AN. (2016) <b>The MPI bioinformatics Toolkit as an integrative platform for advanced protein sequence and structure analysis.</b> Nucleic Acids Res. pii: gkw348. <a href=https://www.ncbi.nlm.nih.gov/pubmed/27131380>PMID: 27131380</a><br/><br/>Edgar RC. (2004) <b>MUSCLE: multiple sequence alignment with high accuracy and high throughput.</b> Nucl Acid Res 32(5):1792-1797. <a href = https://www.ncbi.nlm.nih.gov/pubmed/15034147>PMID: 15034147</a> "],
    "tcoffee": ["T-Coffee (Tree-based Consistency Objective Function for alignment Evaluation) is one of the most accurate multiple sequence alignment methods for protein and nucleotide sequences.<br/>[T-Coffee 11.0; <a href = http://www.tcoffee.org/>http://www.tcoffee.org/</a>] ", ["Input format  ", "Output in alignment order"], "Alva V., Nam SZ., Söding J., Lupas AN. (2016) <b>The MPI bioinformatics Toolkit as an integrative platform for advanced protein sequence and structure analysis.</b> Nucleic Acids Res. pii: gkw348. <a href=https://www.ncbi.nlm.nih.gov/pubmed/27131380>PMID: 27131380</a><br/><br/>Magis C., Taly JF., Bussotti G., Chang JM., Di Tommaso P., Erb I., Espinosa-Carrasco J., Notredame C. (2014) <b>T-Coffee: Tree-based consistency objective function for alignment evaluation. </b>Methods Mol Biol 1079:117-129. <a href = https://www.ncbi.nlm.nih.gov/pubmed/24170398>PMID: 24170398</a>"],
    "aln2plot": ["Shows a graphical overview of hydrophobicity and side chain volume along a protein sequence.", ["Input"], "Alva V., Nam SZ., Söding J., Lupas AN. (2016) <b>The MPI bioinformatics Toolkit as an integrative platform for advanced protein sequence and structure analysis.</b> Nucleic Acids Res. pii: gkw348. <a href=https://www.ncbi.nlm.nih.gov/pubmed/27131380>PMID: 27131380</a>"],
    "frpred": ["FRpred is a collection of methods for the prediction of functional residues in protein multiple sequence alignemnets. You can also run it on a single sequence. Then, PSI-BLAST is called to generate an alignment, which is directly used as input for the methods. If you have a PDB file representing the sequences in your alignment, you can specify it in the input parameters. It will be used to visualize the results via RasMol scripts. The methods in FRpred are:<br/><br/><ol><li><b>Shannon entropy:</b> A simple method to score conservation.</li><li><b>FRcons:</b> A novel conservation method that scores fully conserved positions equally, conservative mutaions higher than non-conservative ones, penalizes gap-rich columns and yields a score of 0 for positions distributed like the background frequencies.</li><li><b>Evolutionary Trace (ET):</b> A method yielding conserved and subtype-specific residues. It uses UPGMA to cluster the sequences according to sequence similarity and the sums over teh entropy conservation score of all subgroups defined by dendrogram cuts on all levels of the UPGMA tree.</li><li><b>FRtrace:</b> Following the idea of ET, FRtrace also sums over conservation scores in subgroups defined by a UPGMA tree in order to predict conserved as well as subtype-specific positions. However, FRtrace uses FRcons instead of entropy conservation and weighs the contribution from each subgroup according to its effective number of sequences. Moreover, the method takes only those subgroups into account that are new in the current dendrogram cut when traversing the tree from the root to the leafs. FRtrace offers a family- and a sequence-centered analysis.</li><li><b>FRsubtype:</b> This method predicts only subtype-specific residues. It uses mutual information to score the difference between the amino acid distribution at the same position in different subgroups and, additionally, it scores the conservation within the subgroups using FRcons. It also offers a sequence- and a family-centerd analysis. You can either manually group the alignment into subgroups or - if this information is not available - the method can do its own clustering (UPGMA tree) and sum over the weighted subgroups.</li></ol><br/>If a structure is supplied, DSSP will be used to compute the solvent accessibility. Otherwise, Sable2 is run to predict it. Sable2 runs 3 PSI-BLAST iterations on the query sequence, which constitutes a big part of the total runtime of FRpred. ", ["Input ", "Input format    ", "PDB file", "Chain", "Extend sequence/alignment", "Maximum number of PSI-BLAST iterations", "PSI-BLAST E-value threshold", "Minimum coverage", "Minimum sequence identity", "Catalytic residues", "Sequence-centered analysis"],  "Alva V., Nam SZ., Söding J., Lupas AN. (2016) <b>The MPI bioinformatics Toolkit as an integrative platform for advanced protein sequence and structure analysis.</b> Nucleic Acids Res. pii: gkw348. <a href=https://www.ncbi.nlm.nih.gov/pubmed/27131380>PMID: 27131380</a><br/><br/>Fischer JD., Mayer CE., Söding J. (2008) <b>Prediction of Protein Functional Residues From Sequence by Probability Density Estimation. </b>Bioinformatics 24:613-620. <a href = https://www.ncbi.nlm.nih.gov/pubmed/18174181>PMID: 18174181</a><br/><br/>Mihalek I., Res I., Lichtarge O. (2004) <b>A Family of Evolution-Entropy Hybrid Methods for Ranking Protein Residues by Importance. </b>J Mol Biol 336:1265-1282. <a href = https://www.ncbi.nlm.nih.gov/pubmed/15037084>PMID: 15037084</a><br/><br/>Hannenhalli SS., Russel, RB. (2000) <b>Analysis and Prediction of Functional Sub-types from Protein Sequence Alignments. </b>J Mol Biol 303(1):61-76. <a href = https://www.ncbi.nlm.nih.gov/pubmed/11021970>PMID: 11021970</a><br/><br/>Altschul S., Madden T., Schaffer A., Zhang J., Zhang Z., Miller W., Lipman D. (1997) <b>Gapped BLAST and PSI-BLAST: a New Generation of Protein Database Search Programs. </b>Nucl Acids Res 25:3389-3402. <a href = https://www.ncbi.nlm.nih.gov/pubmed/9254694>PMID: 9254694</a><br/><br/>Kabsch W., Sander C. (1983) <b>Dictionary of Protein Secondary Structure: Pattern Recognition of Hydrogen-Bonded and Geometrical Features.</b> Biopolymers 22:2577-2637. <a href = https://www.ncbi.nlm.nih.gov/pubmed/6667333>PMID: 6667333</a><br/><br/>Adamczak R., Porollo A., Meller J. (2004) <b>Accurate Prediction of Solvent Accessibility Using Neural Networks-Based Regression.</b> Proteins 56:753-767. <a href = https://www.ncbi.nlm.nih.gov/pubmed/15281128>PMID: 15281128</a>"],
    "hhrepid": ["<h3>About HHrepID</h3><br/><h5><b>What is HHrepID good for?</b></h5>HHrepID is a novel automated procedure for the de novo identification of repeats in protein sequences. It is able to detect the sequence signature of structural repeats in many proteins that have not yet been known to possess internal sequence symmetry, such as TIM barrels and outer membrane beta-barrels. HHrepID uses HMM-HMM comparison to exploit evolutionary information in the form of the multiple sequence alignment of homologs, but in contrast to HHrep, the new method has several novel characteristics: (1) automated generation of a multiple alignment of repeats; (2) utilization of the transitive nature of homology through a novel merging procedure based on the fully probabilistic treatment of alignments; (3) superior alignment quality through an algorithm that maximizes the expected accuracy of an alignment; (4) the ability to identify different repeats within complicated architectures or multiple domains through automatic domain boundary detection, (5) new statistical treatment yielding improved sensitivity.<br/><br/><h5><b>Length of submitted sequences</b></h5>HHrepID works best on single-domain seuqences or sequences containing only one kind of tandem repeat. If you don't know where your domains are, perform an HHpred search through SCOP and PfamA. Also, you could perform a preliminary HHrepID analysis to home in on the repeat regions and then cut the sequence further down to include only regions with one kind of repeat structure. Sequences should best have between 100 and 300 residues length. Sequences that are too short will gather too few homologs in the PSI-BLAST searches, thus loosing sensitivity. Also, the more repeat units are contained in a sequence, the more information from transitivity of pairwise alignmemnts (A-B, B-C => A-C) can be used by HHrepID through its \"Merge alignments\" functionality. Sequences longer than 800 residues are truncated by HHrepID for practical reasons.<br/><br/><h5><b>Submitting an alignment</b></h5>If you already have a trustworthy multiple sequence alignment of your query with some homologs it is recommended that you start with this instead of from a single sequence. This will help PSI-BLAST to avoid alignment errors.<br/><br/><h5><b>Why is HHrepID so sensitive?</b></h5>The reason for HHrepID's sensitivity lies in the use of evolutionary information when comparing the query sequence with itself. This evolutionary information comes from the multiple sequence alignment of the sequence with homologous stretches of proteins. The information in the multiple sequence alignment is used in a condensed form, a profile HMM. A profile HMM is almost the same as a sequence profile. This in turn is just a table with 20 rows and L columns (where L is the length of the query protein). Each of the L columns contains the frequencies of the 20 amino acids in the corresponding column of the multiple sequence alignment. This table is then compared with itself in order to find stretches of columns with similar amino acid distributions. By looking at many related sequences with very similar structures that contribute to the sequence profile, profile-profile (or HMM-HMM) comparison effectively averages away those features of the sequence that are there just by chance and not by necessity and concentrates on those that determine the protein family's properties.<br/><br/>A method that also employs HMM-HMM comparison for very sensitive protein-protein comparison is implemented in HHpred, our interactive function and structure prediction server. (For a short description of HHpred, please go to the HHpred help.) ", ["Paste single sequence or multiple alignment ", "Max. number of PSI-BLAST iterations", "Score secondary structure ", "Repeat family P-value threshold", "Self-alignment P-value threshold", "Merge rounds", "MAC threshold", "Alignment stringency", "Domain boundary detection"], "Alva V., Nam SZ., Söding J., Lupas AN. (2016) <b>The MPI bioinformatics Toolkit as an integrative platform for advanced protein sequence and structure analysis.</b> Nucleic Acids Res. pii: gkw348. <a href=https://www.ncbi.nlm.nih.gov/pubmed/27131380>PMID: 27131380</a><br/><br/>Biegert A., Söding J. (2008) <b>HHrepID: de novo protein repeat identification by probabilistic consistency. </b>Bioinformatics 24(6):807-814.<a href = https://academic.oup.com/bioinformatics/article/24/6/807/194276/De-novo-identification-of-highly-diverged-protein>(download pdf)</a> <a href = https://www.ncbi.nlm.nih.gov/pubmed/18245125>PMID: 18245125</a>"],
    "marcoil": ["MARCOIL is a hidden Markov model-based program that predicts existence and location of potential coiled-coil domains in protein sequences. The external MARCOIL Homepage can be found <a href = http://bioinf.wehi.edu.au/folders/mauro/Marcoil/index.html>here</a>. ", ["Input  ", "Matrices"], "Alva V., Nam SZ., Söding J., Lupas AN. (2016) <b>The MPI bioinformatics Toolkit as an integrative platform for advanced protein sequence and structure analysis.</b> Nucleic Acids Res. pii: gkw348. <a href=https://www.ncbi.nlm.nih.gov/pubmed/27131380>PMID: 27131380</a><br/><br/>Delorenzi M., Speed T. (2002) <b>An HMM model for coiled-coil domains and a comparison with PSSM-based predictions. </b>Bioinformatics 18(4):617-625. <a href = https://www.ncbi.nlm.nih.gov/pubmed/12016059>PMID: 12016059</a>"],
    "pcoils": ["<h5><b>Recommendations</b></h5>COILS with the MTIDK matrix is specific for solvent-exposed, left-handed coiled coils. Other types of coiled-coil structure, such as buried coiled coils (e.g the central coiled coil in catabolite repressor protein, or some transmembrane domains) and right-handed coiled coils, are not detected. COILS does not reach yes-or-no decisions based on a threshold value. Rather, it yields a set of probabilities that presumably reflect the coiled-coil forming potential of a sequence. This means that even at high probabilities (e.g. >90%), there will be (and should be) sequences that in fact do not form a coiled coil, though they may have the potential to do so in a different context. COILS is biased towards hydrophilic, highly charged sequences. For this reason, all scans should be performed with a weighted and an unweighted matrix, and the results compared. Differences of more than 20-30 percentage points in the probabilities should be taken to indicate that a coiled-coil structure is unlikely, the elevated scores being mainly due to the high incidence of charged residues (note, though, that this would have marked human mannose-binding protein as a false positive). The MTIDK matrix assigns high probabilities to known coiled coils segments, but identifies different helices at high probability in a database of globular proteins. This is a surprising feature whose reason is as yet unclear, but which can be exploited for predictive purposes. It is therefore useful to compare the results of scans made with the two matrices. Again, differences of more than 20-30 percentage points in the probabilities should be taken to indicate that a coiled-coil structure is unlikely (note, though, that this threshold would make the replication terminator protein a border-line case). The resolution between globular and coiled-coil score distributions decreases strongly with a decreasing size of the scanning window. The prediction of new coiled-coil segments should therefore be made using a 28 residue window, or in special cases a 21 residue window. 14 residue windows should normally be reserved for the analysis of local parameters (such as the frame) in known or predicted coiled coils. The ends of coiled-coil segments appear to be most accurately identified in a 21 residue window. In general, it is assumed that residues with probabilities >50% are part of a coiled-coil segment. Sequences with high coiled-coil probability from globular proteins rarely exceed a length of 30 residues. None is longer than 35 residues. Sequences with probabilities >80-90% that extend for more than 35 residues are therefore more likely to assume a coiled-coil structure than is indicated by the obtained probabilities. Where possible, sequences related to the protein of interest should also be analyzed for predicted coiled-coil segments (e.g. by using a sequence alignment instead of a single sequence as input). It should be kept in mind, though, that the sequences must be related in the region of high scores in order for the comparison to be significant.<br/><br/><h5><b>PSIPRED</b></h5>Comparison of the coiled-coil prediction with predictions of the secondary structure are generally useful, particularly if multiple related sequences are available. PSIPRED predicts the secondary structure as described in <a href =https://www.ncbi.nlm.nih.gov/pubmed/10493868?dopt=Abstract>Jones, J.Mol.Biol. 292:195-202 (1999).</a>", ["Input", "Input format", "Input mode", "Matrix ", "Weighting"], "Alva V., Nam SZ., Söding J., Lupas AN. (2016) <b>The MPI bioinformatics Toolkit as an integrative platform for advanced protein sequence and structure analysis.</b> Nucleic Acids Res. pii: gkw348. <a href=https://www.ncbi.nlm.nih.gov/pubmed/27131380>PMID: 27131380</a><br/><br/>Lupas A., Van Dyke M., Stock J. (1991) <b>Predicting Coiled Coils from Protein Sequences. </b>Science 252:1162-1164. <a href = https://www.ncbi.nlm.nih.gov/pubmed/2031185>PMID: 2031185</a><br/><br/>Lupas A. (1996) <b>Prediction and Analysis of Coiled-Coil Structures. </b>Methods Enzymol 266:513-525. <a href = https://www.ncbi.nlm.nih.gov/pubmed/8743703>PMID: 8743703</a><br/><br/>Parry DA. (1982) <b>Coiled-Coils in Alpha-Helix-Containing Proteins: Analysis of the Residue Types within the Heptad Repeat and the Use of These Data in the Prediction of Coiled-Coils in Other Proteins. </b>Biosci Rep 2(12):1017-24. <a href = https://www.ncbi.nlm.nih.gov/pubmed/7165792>PMID: 7165792</a>"],
    "repper": ["<h3>About REPPER</h3>REPPER is a server that detects and analyses regions with short gapless REPeats in protein sequences. It finds PERiodicities by Fourier Transform (FTwin) and internal homology analysis (REPwin). The exact locations of the periodic patterns are traced by a sliding window. The output is complemented by coiled coil prediction (COILS) and optionally by secondary structure prediction (PSIPRED).<br/><br/><h5><b>FTwin</b></h5>Sequence analysis by a sliding window Fourier transform (FT) method calculating the periodicity spectra for all possible sequence windows. The overview graph shows all significant periodicities and where in the sequence they are located. Periodicities are considered to be significant if their FT intensity is above the given threshold.<br/>Clicking on a colored line on the output graph provides the complete FT spectrum for the sequence fragment defined by the line (the fragment may include closely neighbored lines as well). Alternatively, an arbitrary sequence range may be given (in the boxes under the output graph) in order to get a FT spectrum of this particular sequence fragment. <a href = ftp://ftp.tuebingen.mpg.de/pub/protevo/FTwin/>Download</a><br/><br/><h5><b>REPwin</b></h5>REPWin compares a protein sequence with itself within a sliding window of choosable length W. For each window starting position i and periodicity p it calculates<br/><br/><div align = center><em>Score(i,p) = Sum_k Sum_j S(x[j],x[j+kp])</em></div><br/>S(x[j],x[j+kp]) is the Gonnet substitution matrix element for residues x[j] and x[j+kp]. The sum runs over all k and j such that j and j+kp are inside the window [i,..,i+W-1]. The score S(i,p) is normalized by dividing through the standard deviation of S(i,p) for nonperiodic sequences. The final score value for each residue number i and periodicity p is the maximum over all windows containing residue i.<br/>Clicking on a colored line on the output graph displays the corresponding alignment that shows the internal homology. <a href = ftp://ftp.tuebingen.mpg.de/pub/protevo/REPwin/>Download</a><br/><br/><h5><b>COILS</b></h5>COILS is a program that compares a sequence to a database of known parallel two-stranded coiled-coils and derives a similarity score. By comparing this score to the distribution of scores in globular and coiled-coil proteins, the program then calculates the probability that the sequence will adopt a coiled-coil conformation. COILS is described in <a href = https://www.ncbi.nlm.nih.gov/pubmed/2031185?dopt=Abstract>Lupas, Science 252: 1162-1164 (1991)</a> and Lupas, Meth. Enzymology 266: 513-525 (1996). It is based on a prediction protocol proposed by <a href = https://www.ncbi.nlm.nih.gov/pubmed/7165792?dopt=Abstract>Parry, Biosci. Rep. 2: 1017-1024 (1982)</a>. You can also <a href = ftp://ftp.tuebingen.mpg.de/pub/protevo/COILS/>download</a> the software.<br/><br/><h5><b>PSIPRED</b></h5>PSIPRED predicts the secondary structure as described in <a href = https://www.ncbi.nlm.nih.gov/pubmed/10493868?dopt=Abstract>Jones, J.Mol.Biol. 292:195-202 (1999)</a>.", ["Input", "Input format", "Input mode ", "Window size (FTwin and REPwin", "Periodicity range (FTwin)", "Scale (FTwin)", "FTwin threshold", "REPwin threshold", "COILS parameters"], "Alva V., Nam SZ., Söding J., Lupas AN. (2016) <b>The MPI bioinformatics Toolkit as an integrative platform for advanced protein sequence and structure analysis.</b> Nucleic Acids Res. pii: gkw348.  <a href=https://www.ncbi.nlm.nih.gov/pubmed/27131380>PMID: 27131380</a><br/><br/>Gruber M., Söding J., Lupas AN. (2005) <b>REPPER -- Repeats and their periodicities in fibrous proteins. </b>Nucl Acids Res 33(Web Server issue): W239-W243. <a href = https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1160166/>PMCID: PMC1160166</a>"],
    "tprpred": ["TPRpred uses the profile representation of the known repeats to detect Tetratrico Peptide Repeats (TPRs), Pentatrico Peptide Repeats(PPRs) and SEL1-like repeats from the query sequence and computes the statistical significance for their occurrence.", ["Input format     ", "Maximum whole protein E-value"], "Alva V., Nam SZ., Söding J., Lupas AN. (2016) <b>The MPI bioinformatics Toolkit as an integrative platform for advanced protein sequence and structure analysis.</b> Nucleic Acids Res. pii: gkw348. <a href=https://www.ncbi.nlm.nih.gov/pubmed/27131380>PMID: 27131380</a><br/><br/>Karpenahalli MR., Söding J., Lupas AN. (2007) <b>TPRpred: a tool for prediction of TPR-, PPR- and SEL1-like repeats from protein sequences. </b>BMC Bioinformatics 8:2. <a href = https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1774580/>PMCID: PMC1774580</a>"],
    "ali2d": ["Ali2D which takes a sequence alignment as input, performs PSIPRED and MEMSAT2 on each of the sequences and finally plots information about secondary structure, transmembrane regions and amino acids on the alignemnt.", ["Input", "Input format", "Percentage sequence identity"], "Alva V., Nam SZ., Söding J., Lupas AN. (2016) <b>The MPI bioinformatics Toolkit as an integrative platform for advanced protein sequence and structure analysis.</b> Nucleic Acids Res. pii: gkw348. <a href=https://www.ncbi.nlm.nih.gov/pubmed/27131380>PMID: 27131380</a>"],
    "quick2d": ["Quick2D gives you an overview of secondary structure features like alpha-helices, extended beta-sheets, coiled coils, transmembrane helices and disorder regions. Predictions by PSIPRED, JNET, Prof(Rost), Prof(Ouali), Coils, HMMTOP and DISOPRED2.", ["Input", "Input format", "Do PSI-BLAST search"], "Alva V., Nam SZ., Söding J., Lupas AN. (2016) <b>The MPI bioinformatics Toolkit as an integrative platform for advanced protein sequence and structure analysis.</b> Nucleic Acids Res. pii: gkw348. <a href=https://www.ncbi.nlm.nih.gov/pubmed/27131380>PMID: 27131380</a>"],
    "modeller": ["Modeller is one of the most popular programs for homology modelling. The goal of homology modelling (also called comparative modelling) is to obtain a 3D structure for a protein sequence of interest (the target). One starts from a sequence alignment of the target sequence with one or more template sequences for which the structures are known. Sequences with recognizable sequence similarity (e.g. above 30%) will in general have quite similar 3D structures. Therefore, an alignment with a homologous template of known structure can, in principle, be used to directly predict the main chain coordinates by simply mapping the target sequence into the template main chain structure. Modeller goes a step further in deriving constraints from the multiple alignment of the target with the templates. In addition to a-priori constraints like bond lengths et cetera, these constraints may refer to Calpha-Calpha distances, main-chain dihedral angles, or side-chain dihedral angles. The constraints are formulated probabilistically in terms of probability density functions, that are derived from a statistical analysis of a great number of pairwise structural alignments. Modeller then tries find the parameters (i.e. the model structure) that maximizes the total probability density obtained by combining all constraints. Modeller, as most (if not all) other available programs, has a few important limitations:<br/><br/><ul><li>It can not correct a wrong or suboptimal target-template alignment. Errors in the alignment will directly translate into errors in the model structures. The alignment errors are the leading source of errors for homology models when the sequence similarity is below ~50%.</li><li>The model is, on average, not more similar to the real target structure than the template used. (This sounds frustrating, but it means that normally, the improvements which the homology modelling software achieves do not fully compensate the alignment errors, on average). For this reason, the other limiting factor in model quality, beside the alignment quality, is the selection of the best template. This is a very difficult task for which a large number of methods and servers have been developed. (See our HHpred server and the links in the section on tertiary structure prediction.)</li><li>Side chain placement gets unreliable below sequence identities of 70%, simply because the position of side chains is conserved less and less below this degree of similarity.</li><li>Loops of more than three or four residues for which no corresponding template atoms are available can not be reliably estimated. They are more or less guessed by the programs. There exist specialized loop modelling methods, but they are very time-consuming and can not be offered over the web. Also, loops of more than 8 to 12 residues can not be modelled reliably.</li></ul><br/>In summary, homology modelling software can not correct errors in the alignments or templates. What you get out is almost what you put in.", ["Input format      ", "Modeller key"], "Alva V., Nam SZ., Söding J., Lupas AN. (2016) <b>The MPI bioinformatics Toolkit as an integrative platform for advanced protein sequence and structure analysis.</b> Nucleic Acids Res. pii: gkw348. <a href=https://www.ncbi.nlm.nih.gov/pubmed/27131380>PMID: 27131380</a><br/><br/>Sali A., Potterton L., Yuan F., van Vlijmen H., Karplus M. (1995) <b>Evaluation of comparative protein modeling by MODELLER. </b>Proteins 23(3):318-26. <a href = https://www.ncbi.nlm.nih.gov/pubmed/8710825>PMID: 8710825</a>"],
    "samcc": ["SamCC measures local parameters of (a)symmetric, parallel, and antiparallel four-helical bundles. Briefly, SamCC divides bundle into layers (slices), brings each layer into an idealized state (expected from equations), then performing final calculations on the resulting idealized structure. ", ["Input format       ", "Definition line for helix 1-4", "Periodicity"], "Alva V., Nam SZ., Söding J., Lupas AN. (2016) <b>The MPI bioinformatics Toolkit as an integrative platform for advanced protein sequence and structure analysis.</b> Nucleic Acids Res. pii: gkw348. <a href=https://www.ncbi.nlm.nih.gov/pubmed/27131380>PMID: 27131380</a><br/><br/>Dunin-Horkawicz S., Lupas AN. (2010) <b>Measuring the conformational space of square four-helical bundles with the program samCC. </b>170(2):226-35. doi: 10.1016/j.jsb.2010.01.023. <a href = https://www.ncbi.nlm.nih.gov/pubmed/20139000>PMID: 20139000</a>"],
    "ancescon": ["ANCESCON is a package for distance-based phylogenetic inference and reconstruction of ancestral protein sequences that takes into account the observed variation of evolutionary rates between positions that more precisely describes the evolution of protein families.<br/><br/>[Reconstruction of ancestral protein sequences and its applications, Wei Cai, Jimin Pei, and Nick V Grishin<br/>BMC Evol Biol. 2004; 4: 33. doi: 10.1186/1471-2148-4-33. <a href = https://www.ncbi.nlm.nih.gov/pmc/articles/PMC522809/>Published online 2004 September 17</a>.]", ["Input"], "Alva V., Nam SZ., Söding J., Lupas AN. (2016) <b>The MPI bioinformatics Toolkit as an integrative platform for advanced protein sequence and structure analysis.</b> Nucleic Acids Res. pii: gkw348. <a href=https://www.ncbi.nlm.nih.gov/pubmed/27131380>PMID: 27131380</a><br/><br/>Cai W., Pei J., Grishin NV. (2004) <b>Reconstruction of ancestral protein sequences and its applications. </b>BMC Evol Biol. 4:33. <a href = https://www.ncbi.nlm.nih.gov/pubmed/15377393>PMID: 15377393</a>"],
    "clans": ["A clustering is performed based on all against all BLAST+ similarities and the detected groups are returned.", ["Input format   ", "Matrix"], "Alva V., Nam SZ., Söding J., Lupas AN. (2016) <b>The MPI bioinformatics Toolkit as an integrative platform for advanced protein sequence and structure analysis.</b> Nucleic Acids Res. pii: gkw348. <a href=https://www.ncbi.nlm.nih.gov/pubmed/27131380>PMID: 27131380</a><br/><br/>Frickey T., Lupas AN. (2004) <b>CLANS: a Java application for visualizing protein families based on pairwise similarity. </b>Bioinformatics 20(18):3702-3704. <a href = https://www.ncbi.nlm.nih.gov/pubmed/15284097>PMID: 15284097</a>"],
    "mmseqs2": ["MMseqs2 (Many-against-Many searching) is a software suite to search and cluster huge protein sequence sets. MMseqs2 can run 10000 times faster than BLAST. At 100 times its speed it achieves the same sensitivity. It can also perform profile searches with the same sensitivity as PSI-BLAST but at around 270 times its speed.", [], "Alva V., Nam SZ., Söding J., Lupas AN. (2016) <b>The MPI bioinformatics Toolkit as an integrative platform for advanced protein sequence and structure analysis.</b> Nucleic Acids Res. pii: gkw348. <a href=https://www.ncbi.nlm.nih.gov/pubmed/27131380>PMID: 27131380</a><br/><br/>Steinegger M and Soeding J. Sensitive protein sequence searching for the analysis of massive data sets. bioRxiv, doi: 10.1101/079681 (2017)."],
    "phyml": ["PhyML is a software that estimates maximum likelihood phylogenies from alignments of amino acid sequences.", ["Input", "Model of amino acid replacement", "Number of replicates"], "Alva V., Nam SZ., Söding J., Lupas AN. (2016) <b>The MPI bioinformatics Toolkit as an integrative platform for advanced protein sequence and structure analysis.</b> Nucleic Acids Res. pii: gkw348. <a href=https://www.ncbi.nlm.nih.gov/pubmed/27131380>PMID: 27131380</a><br/><br/>Guindon S., Dufayard J.F., Lefort V., Anisimova M., Hordijk W., Gascuel O. <b>New Algorithms and Methods to Estimate Maximum-Likelihood Phylogenies: Assessing the Performance of PhyML 3.0.</b> Molecular Biology and Evolution, msx149, 2017."],
    "sixframe": ["Performs a six-frame translation on a given nucleotide sequence. A selection of different codon usage tables are provides in the Parameters tab.", ["Input format        ", "Include nucleic acid sequence in output", "Print amino acids in relation to nucleotide sequence", "Codon table"], "Alva V., Nam SZ., Söding J., Lupas AN. (2016) <b>The MPI bioinformatics Toolkit as an integrative platform for advanced protein sequence and structure analysis.</b> Nucleic Acids Res. pii: gkw348. <a href=https://www.ncbi.nlm.nih.gov/pubmed/27131380>PMID: 27131380</a>"],
    "backtrans": ["The Backtranslator translates an amino acid sequence back into a possible DNA sequence.", ["Input   ", "Include amino acid sequence in output", "Genetic code", "Codon usage table of organism"], "Alva V., Nam SZ., Söding J., Lupas AN. (2016) <b>The MPI bioinformatics Toolkit as an integrative platform for advanced protein sequence and structure analysis.</b> Nucleic Acids Res. pii: gkw348. <a href=https://www.ncbi.nlm.nih.gov/pubmed/27131380>PMID: 27131380</a><br/><br/>Nakamura Y., Gojobori T., Ikemura T. (2000) <b>Codon usage tabulated from international DNA sequence databases: status for the year 2000. </b>Nucleic Acids Res 28(1):292. <a href = https://www.ncbi.nlm.nih.gov/pubmed/10592250>PMID: 10592250</a>"],
    "hhfilter": ["Extract a representative set of sequences from an alignment, based of pairwise sequence identity and minimum coverage.", ["Input", "Max. sequence identity", "Min. sequence identity with query", "Min. coverage with query", "Extract N most dissimilar sequences"], "Alva V., Nam SZ., Söding J., Lupas AN. (2016) <b>The MPI bioinformatics Toolkit as an integrative platform for advanced protein sequence and structure analysis.</b> Nucleic Acids Res. pii: gkw348. <a href=https://www.ncbi.nlm.nih.gov/pubmed/27131380>PMID: 27131380</a><br/><br/>Remmert M., Biegert A., Hauser A., Söding J. (2011) <b>HHblits: Lightning-fast iterative protein sequence searching by HMM-HMM alignment. </b>Nat Methods. 9(2):173-5. doi: 10.1038/nmeth.1818. <a href = https://www.ncbi.nlm.nih.gov/pubmed/22198341>PMID: 22198341</a>"],
    "retseq": ["This utility performs a sequence retrieval from the nre and nr database using a list of identifiers.", ["Input     ", "Retrieve identical sequences only once"], "Alva V., Nam SZ., Söding J., Lupas AN. (2016) <b>The MPI bioinformatics Toolkit as an integrative platform for advanced protein sequence and structure analysis.</b> Nucleic Acids Res. pii: gkw348. <a href=https://www.ncbi.nlm.nih.gov/pubmed/27131380>PMID: 27131380</a><br/>"],
    "seq2id": ["Seq2ID is a utility that extracts all GI-numbers from a given file. Each sequence has to be new line separated. eg.:<br/><br/>Sequence 1<br/>Sequence 2<br/>sequence 3<br/>...", ["Input    "], "Alva V., Nam SZ., Söding J., Lupas AN. (2016) <b>The MPI bioinformatics Toolkit as an integrative platform for advanced protein sequence and structure analysis.</b> Nucleic Acids Res. pii: gkw348. <a href=https://www.ncbi.nlm.nih.gov/pubmed/27131380>PMID: 27131380</a><br/>"]
};

helpContent = function(tool : string) {
    if (exampleContent[tool]) {
        return exampleContent[tool];
    } else {
        return ["", [], ""];
    }
};

accordion = function(elem : Element, isInit : boolean) : any {
    if (!isInit) {
        return elem.setAttribute("data-accordion", "data-accordion");
    }
};

accordionItem = function(elem : Element, isInit : boolean) : any {
    if (!isInit) {
        return elem.setAttribute("data-accordion-item", "data-accordion-item");
    }
};

accordionContent = function(elem : Element, isInit : boolean) {
    if (!isInit) {
        return elem.setAttribute("data-tab-content", "data-tab-content");
    }
};

(<any>window).HelpModalComponent = {

    controller: function(){},
    view: function(ctrl : any, args : any) {
        let overview : any, params : any, results : any, references : any;
        overview = helpContent(args.toolname)[0];
        params = helpContent(args.toolname)[1];
        //results = helpContent(args.toolname)[2];
        references = helpContent(args.toolname)[2];
        return m("div", {
            id: "help-" + args.toolname,
            "class": "reveal helpModal",
            config: helpModalReveal
        }, m("div", {
            id: "help-tabs",
            config: helpModalTabs
        }, [
            m("ul", [
                m("li", m("a", {
                    href: "#help-tabs1"
                }, "Overview")), m("li", m("a", {
                    href: "#help-tabs2"
                }, "Parameters")),// m("li", m("a", {
                    //href: "#help-tabs3"
                //},"Results")),
                m("li", m("a", {
                   href: "#help-tabs3"},
                "References")), m("li", {
                    "class": "toolname"
                }, m("a", {
                    href: "#",
                    "class": "not-active"
                }, args.toolnameLong))
            ]), m("div", {
                id: "help-tabs1",
                "class": "greenlink"
            }, m("div", m.trust(overview))), m("div", {
                id: "help-tabs2"
            }, m("ul", {
                "class": "accordion",
                config: accordion
            }, params.map(function(param : any) {
                return m("li", {
                    "class": "accordion-item",
                    config: accordionItem
                }, [
                    m("a", {
                        href: "#",
                        "class": "accordion-title"
                    }, param), m("div", {
                        "class": "accordion-content",
                        config: accordionContent
                    }, [ m.trust(parameterContent[param]) ])
                ]);
            }))),/* m("div", {
                id: "help-tabs3"
            }, m("div", m.trust(results)), m("img", {
                src: '/assets/images/psiblast.png',
                style: {
                    marginTop: '2em'
                }
            })), */m("div", {
                id: "help-tabs3",
                "class": "greenlink"
            }, m("div", m.trust(references)))
        ]));
    }
};