import Component = Mithril.Component;

let accordion : ElementConfig,
    accordionContent : any,
    accordionItem : ElementConfig,
    exampleContent : any,
    helpContent : any,
    helpModalReveal : ElementConfig,
    helpModalTabs : ElementConfig,
    parameterContent : any;

helpModalReveal = function(elem : any, isInit : boolean) : any {
    if (!isInit) {
        elem.setAttribute("data-reveal", "data-reveal");
        return $(elem).foundation();
    }
};

helpModalTabs = function(elem : any, isInit : boolean) : any {
    if (!isInit) {
        return $(elem).tabs();
    }
};

parameterContent = {
    "Alignment": "Enter single protein sequence or protein sequence alignment and choose its format from the dropdown menu.",
    "Database": "The selected databases are used by BLAST to perform the search. <br /> - <b>nr:</b> the non-redundant sequence database at the NCBI, updated weekly. (See NCBI's BLAST tutorial). <br /> - <b>env:</b> NCBI's env database with environnmental sequences, usually from unknown organisms. <br /> - <b>nre:</b> nr+env database. Updated weekly. <br /> - <b>nr90, nr70, env90 etc.:</b> representative sequences, filtered for a maximum pairwise sequence identity of ~90% or 70% with kClust. Updated bimonthly. <br /> - <b>nr_euk, nr_bac, nr_arc, nr_pro, nr_vir:</b> eukaryotic, bacterial, archaeal, prokaryotic, and viral sequences from the non-redundant database nr. Updated weekly. <br /> - <b>uniprot_sprot:</b> http://www.expasy.org/sprot/ Updated weekly. <br /> - <b>uniprot_trembl:</b> http://www.ebi.ac.uk/swissprot/ Updated weekly. <br /> - <b>pdb_nr: Sequences</b> from proteins whose structures have been deposited in the Protein Data Bank PDB. This database uses the more complete SEQRES records of the PDB files (not the ATOM records that contain the atomic coordinates). This is the database used to construct the HHpred PDB alignment database. Updated weekly from the PDB. <br /> ",
    "Matrix": "Specify an alternate scoring matrix for PSIBLAST.",
    "Number of Iterations": "Specify the maximum number of rounds of search. After each successive round the search alignment is updated. ",
    "E-Value": "The statistical significance threshold for reporting matches against database sequences; the default value is 10, meaning that 10 matches are expected to be found merely by chance, according to the stochastic model of Karlin and Altschul (1990). If the statistical significance ascribed to a match is greater than the e-value threshold, the match will not be reported. Lower e-value thresholds are more stringent, leading to fewer chance matches being reported. Increasing the threshold shows less stringent matches. Fractional values are acceptable.",
    "E-value inclusion threshold": "The statistical significance threshold for including a sequence in the model used by PSI-BLAST to create the PSSM on the next iteration.",
    "Filter (low complexity)": "Mask off segments of the query sequence that have low compositional complexity, as determined by the SEG program of Wootton & Federhen (Computers and Chemistry, 1993) or, for BLASTN, by the DUST program of Tatusov and Lipman (in preparation). Filtering can eliminate statistically significant but biologically uninteresting reports from the blast output (e.g., hits against common acidic-, basic- or proline-rich regions), leaving the more biologically interesting regions of the query sequence available for specific matching against database sequences. Filtering is only applied to the query sequence (or its translation products), not to database sequences. Default filtering is DUST for BLASTN, SEG for other programs. It is not unusual for nothing at all to be masked by SEG, when applied to sequences in SWISS-PROT, so filtering should not be expected to always yield an effect. Furthermore, in some cases, sequences are masked in their entirety, indicating that the statistical significance of any matches reported against the unfiltered query sequence should be suspect.",
    "Compute Smith-Waterman alignment": "Compute locally optimal Smith-Waterman alignments.",
    "Use nr70 for all but last iteration": " If this option is selected the program will use the nr70-database for all but last round. In the last round, the database you selected in the database section will be used.",
    "Alignments and descriptions": "Restricts the number of short descriptions of matching sequences reported to the number specified; default limit is 100 descriptions. Restricts also database sequences to the number specified for which high-scoring segment pairs (HSPs) are reported. If more database sequences than this happen to satisfy the statistical significance threshold for reporting, only the matches ascribed the greatest statistical significance are reported."
};

exampleContent = {
    "psiblast": ["Search with an amino acid sequence against protein databases for locally similar sequences.<br/>Similar to Protein BLAST+ but more sensitive. PSI-BLAST+ first performs a BLAST+ search and builds an alignment\nfrom the best local hits. This alignment is then used as a query for the next round of search.\nAfter each successive round the search alignment is updated.", ["Alignment", "Database", "Matrix", "Number of Iterations", "E-Value", "E-value inclusion threshold", "Filter (low complexity)", "Compute Smith-Waterman alignment", "Use nr70 for all but last iteration", "Alignments and descriptions"], "At the top of the results page is a \"View alignment\"-button, that shows the multiple alignment of all hits.<br/>Then there is a list with brief information about all hits that were found.<br/>They are listed together with their scores and e-values.<br/>There you can select the ones, you want to work with. Below this you can find more information.<br/>There are all pairwise alignments and there is also additional information about the method,<br/>the identities and the positives. There you also have the possibility to select the hits, you want to work with.<br/>At the bottom you can find more information about the database and the matrix.<br/>The \"View alignment\"-button shows the multiple alignment of all sequences that was found by PSI-BLAST. The different colours illustrate the identities between the different amino acids. This helps you to decide, whether the found sequences may be really homologue", "Alva V., Nam SZ., Söding J., Lupas AN. (2016) <b>The MPI bioinformatics Toolkit as an integrative platform for advanced protein sequence and structure analysis.</b> Nucleic Acids Res. pii: gkw348. [Epub ahead of print] <a href=https://www.ncbi.nlm.nih.gov/pubmed/27131380>PMID: 27131380</a><br/><br/>Altschul S., Madden T., Schaffer A., Zhang J., Miller W., Lipman D. (1997) <b>Gapped BLAST and PSI-BLAST: a new generation of protein database search programs.</b> Nucl Acids Res 25(17):3389-3402. <a href = https://www.ncbi.nlm.nih.gov/pmc/articles/PMC146917/>PMCID: PMC146917</a><br/><br/>Altschul SF., Gish W., Miller W., Myers EW., Lipman DJ. (1990) <b>Basic local alignment search tool.</b> J Mol Biol. 215:403-410. <a href = https://www.ncbi.nlm.nih.gov/pubmed/2231712>PMID: 2231712</a><br/><br/>Camacho C., Coulouris G., Avagyan V., Ma N., Papadopoulos J., Bealer K., Madden TL. (2008) <b>BLAST+: architecture and applications.</b> BMC Bioinformatics 10:421. <a href = https://www.ncbi.nlm.nih.gov/pubmed/20003500>PMID: 20003500</a>"],
    "hhblits": ["For an accurate sequence-based prediction of protein structure and function, a highly sensitive sequence-search method is crucial. Today, profile-profile and HMM-HMM alignment methods are custom when it comes to identification and alignment of templates for 3D-homology modeling. As these methods are generally too slow for iteratively searching through large databases, we offer HHblits as a 'lightning-fast' HMM-HMM-based iterative sequence search tool. The profile-profile alignment prefilter of HHblits reduces the number of full HMM-HMM alignments from many millions to a few thousand, making it faster than PSI-BLAST but still as sensitive as HHsearch. Prefiltering is essential for speed and sensitivity. Compared to PSI-BLAST, HHblits is faster, up to twice as sensitive and produces more accurate alignments.<br/><br/>The HHblits software is part of the open source package HHsuite.", [], "", "Alva V., Nam SZ., Söding J., Lupas AN. (2016) <b>The MPI bioinformatics Toolkit as an integrative platform for advanced protein sequence and structure analysis.</b> Nucleic Acids Res. pii: gkw348. [Epub ahead of print] <a href=https://www.ncbi.nlm.nih.gov/pubmed/27131380>PMID: 27131380</a><br/><br/>Remmert M., Biegert A., Hauser A., Söding J. (2011) <b>HHblits: Lightning-fast iterative protein sequence searching by HMM-HMM alignment.</b> Nat Methods. 9(2):173-5. doi: 10.1038/nmeth.1818. <a href=https://www.ncbi.nlm.nih.gov/pubmed/22198341>PMID: 22198341</a>"],
    "hhpred": ["<h3>About HHpred</h3><br/><h5><b>HHpred performance in the CASP9 structure prediction benchmark</b></h5>In the latest CASP competition that took place in 2010, our HHpred server was assessed as one of the best out of the 81 servers, in particular in template-based structure prediction, the category most relevant for biological applications. (See <a href = http://onlinelibrary.wiley.com/doi/10.1002/prot.23177/abstract>Mariani, 2011</a>, <a href = http://predictioncenter.org/casp9/groups_analysis.cgi?type=server&tbm=on&submit=Filter>CASP9 results table</a>, and figure below). This is remarkable for two reasons: First, with ~4 minutes median response time, HHpred is about 100 times faster than other top servers. Second, most top servers use our freely available software HHsearch to find templates and to align target to template sequences. For CASP8 results, please refer to our publication in Proteins (2009).<br/><div class = \"wraptocenter\"><img src = https://toolkit.tuebingen.mpg.de/images/hhpred/CASP9_results.png></div><br/>Official CASP9 cumulative Z-scores for model quality in the template-based modeling category, plotted over the median response time of the 81 participating servers. (See <a href = http://predictioncenter.org/casp9/groups_analysis.cgi?type=server&tbm=on&submit=Filter>table</a>)<br/><br/><h5><b>HHpred server for protein remote homolog detection and 3D structure prediction</b></h5>The primary aim in developing HHpred was to provide biologists with a method for sequence database searching and structure prediction that is as easy to use as BLAST or PSI-BLAST and that is at the same time much more sensitive in finding remote homologs. In fact, HHpred's sensitivity is competitive with the most powerful servers for structure prediction currently available.<br/>HHpred is the first server that is based on the pairwise comparison of profile hidden Markov models (HMMs). Whereas most conventional sequence search methods search sequence databases such as UniProt or the NR, HHpred searches alignment databases, like Pfam or SMART. This greatly simplifies the list of hits to a number of sequence families instead of a clutter of single sequences. All major publicly available profile and alignment databases are available through HHpred.<br/><br/>HHpred accepts a single query sequence or a multiple alignment as input. Within only a few minutes it returns the search results in an easy-to-read format similar to that of PSI-BLAST. Search options include local or global alignment and scoring secondary structure similarity. HHpred can produce pairwise query-template sequence alignments, merged query-template multiple alignments (e.g. for transitive searches), as well as 3D structural models calculated by the MODELLER software from HHpred alignments.<br/><br/><h5><b>When can HHpred be useful for you?</b></h5>It is well known that sequence search methods such as BLAST, FASTA, or PSI-BLAST are of prime importance for biological research because functional information of a protein or gene can be inferred from homologous proteins or genes identified in a sequence search. But quite often no significant relationship to a protein of known function can be established. This is certainly the case for the most interesting group of proteins, those for which no ortholog has yet been studied.<br/>It is less well known that in cases where conventional sequence search methods fail, the recently developed, highly sensitive methods for homology detection or structure prediction (see descriptions and links at <a href = http://meta.bioinfo.pl/submit_wizard.pl>http://meta.bioinfo.pl/submit_wizard.pl</a>) quite often allow to make inferences from more remotely homologous relationships [<a href = https://www.ncbi.nlm.nih.gov/pubmed/10871397?dopt=Abstract>Venclovas (2000)</a>, <a href = https://www.ncbi.nlm.nih.gov/pubmed/15688435?dopt=Abstract>Zheng (2005)</a>, <a href = https://www.ncbi.nlm.nih.gov/pubmed/14983005?dopt=Abstract>Ginalski (2004)</a>, <a href = https://www.ncbi.nlm.nih.gov/pubmed/15452342?dopt=Abstract>Rand (2004)</a>, <a href = https://www.ncbi.nlm.nih.gov/pubmed/15684412?dopt=Abstract>Pawlak (2005)</a>, <a href = https://www.ncbi.nlm.nih.gov/pubmed/15048836?dopt=Abstract>Kihara (2004)</a>]. If the relationship is so remote that no common function can be assumed, one can generally still derive hypotheses about possible mechanisms, active site positions and residues, or the class of substrate bound [<a href = https://www.ncbi.nlm.nih.gov/pubmed/11286560?dopt=Abstract>Todd (2001)</a>, <a href = https://www.ncbi.nlm.nih.gov/pubmed/10902155?dopt=Abstract>Pawlowski (2000)</a>]. When a homologous protein with known structure can be identified, its stucture can be used as a template to model the 3D structure for the protein of interest [<a href = https://www.ncbi.nlm.nih.gov/pubmed/9710568?dopt=Abstract>Rychlewski (1998)</a>], since even remotely homologous proteins generally have quite similar 3D structure [<a href = https://www.ncbi.nlm.nih.gov/pubmed/12127461?dopt=Abstract>Kinch (2002)</a>]. The 3D model may then help to generate hypotheses to guide experiments.<br/><br/><h5><b>What is HMM-HMM comparison and why is it so powerful?</b></h5>When searching for remote homologs, it is wise to make use of as much information about the query and database proteins as possible in order to better distinguish true from false positives and to produce optimal alignments. This is the reason why sequence-sequence comparison is inferior to profile-sequence comparison. Sequence profiles contain for each column of a multiple alignment the frequencies of the 20 amino acids. They therefore contain detailed information about the conservation of each residue position, i.e. how important each position is for defining other members of the protein family, and about the preferred amino acids. Profile Hidden Markov Models (<b>HMMs</b>) are similar to simple sequence profiles, but in addition to the amino acid frequencies in the columns of a multiple sequence alignment they contain information about the frequency of inserts and deletions at each column. Using profile HMMs in place of simple sequence profiles should therefore further improve sensitivity. HHpred is the first server to employ HMM-HMM comparison, based on a novel statistical method that we have developed recently. Using HMMs both on the query and the database side greatly enhances the sensitivity and selectivity over sequence-profile based methods such as PSI-BLAST.<br/><br/><h5><b>Structure prediction with HHpred</b></h5>The most successful techniques for protein structure prediction rely on identifying homologous sequences with known structure to be used as template. This works so well because structures diverge much more slowly than sequences and homologous proteins may have very similar structures even when their sequences have diverged beyond recognition. But sensitivity in homology detection is crucial for success since many proteins have only remote relatives in the structure database.<br/><br/><h5><b>Most publicly available alignment databases can be searched</b></h5>A large number of publicly available (partly redundant) and manually or automatically annotated databases of protein family alignments can be searched through our webserver, such as <a href = http://pfam.xfam.org/>PFAM</a>, <a href = http://smart.embl-heidelberg.de/>SMART</a>, <a href = http://www.pantherdb.org/>PANTHER</a>, <a href = http://www.jcvi.org/cgi-bin/tigrfams/index.cgi>TIGRFAMs</a>, <a href = http://pir.georgetown.edu/pirwww/dbinfo/iproclass.shtml>PIRSF</a>, or <a href = https://www.ncbi.nlm.nih.gov/COG/>COG/KOG</a>. Two composite databases in HHpred that include most of these original databases are <a href = http://www.ebi.ac.uk/interpro/>InterPro</a> from the EBI and the conserved domain database <a href = https://www.ncbi.nlm.nih.gov/Structure/cdd/cdd.shtml>CDD</a> from the NCBI. A number of alignment databases are built around sequences of known structure, generally by starting with such a sequence as seed and searching for sequence homologs. We built two such databases ourselves, one using full-length sequences from the <a href = http://www.rcsb.org/pdb/home/home.do>Protein Data Bank</a> (PDB) as seeds, and one database that uses PDB sequences cut into structural and evolutionary domains as defined by the <a href = http://scop.mrc-lmb.cam.ac.uk/scop/index.html>SCOP database</a>. These two databases are referred to simply as pdb70 and scop70 in HHpred. An alignment database very much like our scop70 is <a href = http://supfam.mrc-lmb.cam.ac.uk/SUPERFAMILY/>SUPERFAMILY</a>, whereas <a href = http://www.cathdb.info/Gene3D/>CATH/Gene3D</a> is based on <a href = http://scop.mrc-lmb.cam.ac.uk/scop/index.html>CATH</a>, a database for hierarchical classification of structural domains that is similar to SCOP.<br/><br/><h5><b>Benchmark</b></h5>HHsearch was benchmarked together with BLAST, PSI-BLAST, HMMER, the profile-profile tools PROF\_SIM and COMPASS, and another HMM-HMM tool, PRC [M. Madera, <a href = http://supfam.mrc-lmb.cam.ac.uk/PRC>http://supfam.mrc-lmb.cam.ac.uk/PRC</a>}] in an all-against-all comparison of a database of 3691 protein domains from SCOP 1.63 with pairwise sequence identities below 20%. HHsearch is able to detect 2.8--2.9 times more homologs than PSIBLAST or HMMER, twice more than PROF_SIM, 60% more than COMPASS and 7% more than PRC at a fixed error rate of 20% (see figure). By including secondary structure predicted by <a href = https://www.ncbi.nlm.nih.gov/pubmed/10493868?dopt=Abstract>PSIPRED</a> a further improvement of 20% can be achieved. We show further that the improvements in sensitivity are mirrored by an increase in alignment quality.<br/>Thanks to a fast column score and an efficient algorithm, HHsearch runs four times faster than PRC, ten times faster than PROF_SIM (4/2004), and 17 times fasterthan COMPASS (4/2004), scanning a profile HMM with 200 columns against 3691 SCOP domains in 33 s on a 2 GHz AMD64 processor.<br/><br/><div class=\"wraptocenter\"><img src = https://toolkit.tuebingen.mpg.de/images/hhpred/hhpred_bench.png></div><h5><b>Theory of HMM-HMM comparison</b></h5>We have extended the theory of Hidden Markov Models to the case of pairwise comparison of HMMs and developed a method for sequence homology search which is based on this theory. The alignment algorithm maximizes a weighted form of coemission probability, the probability that the two HMMs will emit the same sequence of residues. Amino acids are weighted according to their abundance, rare coemitted amino acids contributing more to the alignment score. This weighting is analogous to the use of a null model with amino acid background probabilities in HMM-sequence comparison.<br/>Secondary structure can be included in the HMM-HMM comparison. We score pairs of aligned secondary structure states in a way analogous to the classical amino acids substitution matrices. We use ten different substitution matrices that we derived from a statistical analysis of the structure database, one for each confidence value given by <a href = https://www.ncbi.nlm.nih.gov/pubmed/10493868?dopt=Abstract>PSIPRED</a>. By making use of confidence values and using a strictly statistical approach we are able to improve the sensitivity gain considerably over previous methods.<br/><br/><h5><b>HHpred calculates E-values as well as probabilities for a match to be a true positive</b></h5>To assert a homologous relationship for a pair of proteins it is necessary to quantify one's uncertainty. Most sequence similarity search programs report the significance of a database match by an E-value (expect-value). The E-value is the average expected number of non-homologous proteins with a score higher than the one obtained for the database match. An E-value much lower than 1 therefore indicates statistical significance.<br/>To derive E-values, a representative database composed of one HMM per SCOP fold is searched with the query HMM before the actual database search. We record the score distibution and fit an Gumbel extreme value distribution (EVD) to it. From this distribution, one could in principle calculate for each score S the P-value as the area under the EVD to the right of S, i.e. P-value = Prob(score>S). Then the E-value could be calculated for each score according to<br/><br/><div align = center><em>E-value N x P-value,</em></div><br/>where N is the number of HMMs in the database to be scanned. But in order to record a score distribution of negatives, we should exclude all proteins that are homologous to the query protein. We do this in two steps. Before fitting an EVD to the score distribution, we exclude the best-scoring three folds. We then calculate preliminary E-values for the representative HMMs and now exclude those HMMs with an E-value better than 0.5. The lambda and mu parameters for the EVD are determined by Maximum Likelihood. In local alignment mode, we use a simple log length correction.<br/>When we add secondary structure scores, we find that the score disctribution of negatives does not strictly follow an EVD anymore. E-values in HHpred therefore do not include secondary structure scores! As an alternative to E-values, HHpred reports probabilities for a homologous relationship. This probability is based on the real-world score distribution for negative and homologous domain pairs in an all-against-all comparison of SCOP. Since the probability also contains the contribution from the secondary structure score, it is more sensitive than the E-value. The probability is calculated according to:<br/><fiv align = center>Probability for homologous relationship = Prob_hom(score=S) / [Prob_neg(score=S) + Prob_hom(score=S)]</div><br/>Homologous pairs are from the same superfamily or have a structural superposition MAXSUB score > 0.1. All other pairs are counted as negatives in this context.<br/>The large number of compared sequences in our benchmark allowed us to quantitatively test the accuracy of reported E-values. Remarkably, E-values below 1E-4 are wrong for all tested tools except BLAST and HMMER. Specifically, when using PSI-BLAST with standard parameters (up to eight iterations at E-value threshold of 1E-4) ~100 times too many sequences were observed with a reported E-value of 1E-4 or better. The probabilities take account of the complications (mainly alignments corrupted by non-homologous sequences) that lead to these deviations. They are therefore a more realistic measure of confidence.", [], "", "Alva V., Nam SZ., Söding J., Lupas AN. (2016) <b>The MPI Bioinformatics Toolkit as an integrative platform for advanced protein sequence and structure analysis.</b> Nucleic Acids Res. pii: gkw348. [Epub ahead of print] <a href=https://www.ncbi.nlm.nih.gov/pubmed/27131380>PMID: 27131380</a><br/><br/>Söding J., Biegert A., Lupas AN. (2005) <b>The HHpred interactive server for protein homology detection and structure prediction.</b> Nucleic Acids Res 33 (Web Server issue), W244-W248. <a href = https://www.ncbi.nlm.nih.gov/pubmed/15980461>PMID: 15980461</a><br/><br/>Söding J. (2005) <b>Protein homology detection by HMM-HMM comparison.</b> Bioinformatics 21: 951-960. <a href = https://www.ncbi.nlm.nih.gov/pubmed/15531603>PMID: 15531603</a><br/><br/>Hildebrand A., Remmert A., Biegert A., Söding J. (2009) <b>Fast and accurate automatic structure prediction with HHpred.</b> Proteins 77(Suppl 9):128-32. doi: 10.1002/prot.22499. <a href = https://www.ncbi.nlm.nih.gov/pubmed/19626712>PMID: 19626712</a><br/><br/>Meier A., Söding J. (2015) <b>Automatic Prediction of Protein 3D Structures by Probabilistic Multi-template Homology Modeling.</b> PLoS Comput Biol. 11(10):e1004343. doi: 10.1371/journal.pcbi.1004343. <a href = https://www.ncbi.nlm.nih.gov/pubmed/26496371>PMID: 26496371</a><br/><br/>Remmert M., Biegert A., Hauser A., Söding J. (2011) <b>HHblits: Lightning-fast iterative protein sequence searching by HMM-HMM alignment.</b> Nat Methods. 9(2):173-5. doi: 10.1038/nmeth.1818. <a href=https://www.ncbi.nlm.nih.gov/pubmed/22198341>PMID: 22198341</a>"],
    "hmmer": ["HMMER3 is a utility that accelerates searches with HMMER (a HMM software package [1]) by filtering the database with PsiBlast. Input is a multiple alignment in aligned FASTA format which is used to build a profile HMM and to kickstart a PsiBlast-search. All PsiBlast results up to a given E-value are extracted and form the filtered database used for the HMM search. HMMER3 will return results approx. 30 times faster than a standard HMMER search against \"nr\" would.<br/>[1] Sean Eddy: HMMER 2.3.2; <a href = http://hmmer.janelia.org/>http://hmmer.janelia.org/</a>", [], "", "Alva V., Nam SZ., Söding J., Lupas AN. (2016) <b>The MPI bioinformatics Toolkit as an integrative platform for advanced protein sequence and structure analysis.</b> Nucleic Acids Res. pii: gkw348. [Epub ahead of print] <a href=https://www.ncbi.nlm.nih.gov/pubmed/27131380>PMID: 27131380</a><br/><br/>Finn RD., Clements J., Eddy SR. (2011) <b>HMMER web server: interactive sequence similarity searching.</b> Nucleic Acids Res. 39(Web Server issue): W29–W37. <a href = https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3125773/>PMCID: PMC3125773</a>"],
    "patsearch": ["Retrieves all sequences which contain a given PROSITE pattern or regular expression.", [], "", "Alva V., Nam SZ., Söding J., Lupas AN. (2016) <b>The MPI bioinformatics Toolkit as an integrative platform for advanced protein sequence and structure analysis.</b> Nucleic Acids Res. pii: gkw348. [Epub ahead of print] <a href=https://www.ncbi.nlm.nih.gov/pubmed/27131380>PMID: 27131380</a>"],
    "clustalo": ["ClustalO is a general purpose multiple alignment program for DNA or proteins.", [], "", "Alva V., Nam SZ., Söding J., Lupas AN. (2016) <b>The MPI bioinformatics Toolkit as an integrative platform for advanced protein sequence and structure analysis.</b> Nucleic Acids Res. pii: gkw348. [Epub ahead of print] <a href=https://www.ncbi.nlm.nih.gov/pubmed/27131380>PMID: 27131380</a><br/><br/>Sievers F., Wilm A., Dineen D., Gibson TJ., Karplus K., Li W., Lopez R., McWilliam H., Remmert M., Söding J., Thompson JD., Higgins DG. (2011) <b>Fast, scalable generation of high-quality protein multiple sequence alignments using Clustal Omega.</b> Mol Syst Biol 7:539. <a href = https://www.ncbi.nlm.nih.gov/pubmed/21988835>PMID: 21988835</a>"],
    "kalign": ["Kalign - an accurate and fast multiple sequence alignment algorithm.<br/><br/>Timo Lassmann and Erik LL Sonnhammer (2005) Kalign - an accurate and fast multiple sequence alignment algorithm<br/>BMC Bioinformatics 2005, 6:298 doi:10.1186/1471-2105-6-298", [], "", "Alva V., Nam SZ., Söding J., Lupas AN. (2016) <b>The MPI bioinformatics Toolkit as an integrative platform for advanced protein sequence and structure analysis.</b> Nucleic Acids Res. pii: gkw348. [Epub ahead of print] <a href=https://www.ncbi.nlm.nih.gov/pubmed/27131380>PMID: 27131380</a><br/><br/>Lassmann T., Sonnhammer EL. (2005) <b>Kalign--an accurate and fast multiple sequence alignment algorithm.</b> BMC Bioinformatics. 6:298.<a href = https://www.ncbi.nlm.nih.gov/pubmed/16343337>PMID: 16343337</a>"],
    "mafft": ["Multiple alignment program for amino acid or nucleotide sequences<br/>[MAFFT v7; <a href = http://mafft.cbrc.jp/alignment/software/>http://mafft.cbrc.jp/alignment/software/</a>] ", [], "", "Alva V., Nam SZ., Söding J., Lupas AN. (2016) <b>The MPI bioinformatics Toolkit as an integrative platform for advanced protein sequence and structure analysis.</b> Nucleic Acids Res. pii: gkw348. [Epub ahead of print] <a href=https://www.ncbi.nlm.nih.gov/pubmed/27131380>PMID: 27131380</a><br/><br/>Katoh K., Misawa K., Kuma K., Miyata T. (2002) <b>MAFFT: a novel method for rapid multiple sequence alignment based on fast Fourier transform.</b> Nucleic Acid Res 30(14):3059-3066. <a href = https://www.ncbi.nlm.nih.gov/pubmed/12136088>PMID: 12136088</a>"],
    "msaprobs": ["MSAProbs is a practical multiple alignment algorithm for protein sequences. The design of MSAProbs is based on a combination of pair hidden Markov models and partition functions to calculate posterior probabilities. Assessed using the popular benchmarks: BAliBASE, PREFAB, SABmark and OXBENCH, MSAProbs achieves statistically significant accuracy improvements over the existing top performing aligners, including ClustalW, MAFFT, MUSCLE, ProbCons and Probalign. Furthermore, MSAProbs is optimized for multi-core CPUs by employing a multi-threaded design, leading to a competitive execution time compared to other aligners.", [], "", "Alva V., Nam SZ., Söding J., Lupas AN. (2016) <b>The MPI bioinformatics Toolkit as an integrative platform for advanced protein sequence and structure analysis.</b> Nucleic Acids Res. pii: gkw348. [Epub ahead of print] <a href=https://www.ncbi.nlm.nih.gov/pubmed/27131380>PMID: 27131380</a><br/><br/>Liu Y., Schmidt B., Maskell DL. (2010) <b>MSAProbs: multiple sequence alignment based on pair hidden Markov models and partition function posterior probabilities.</b> Bioinformatics 26(16): 1958-64. <a href = https://www.ncbi.nlm.nih.gov/pubmed/20576627>PMID: 20576627</a>"],
    "muscle": ["MUSCLE is a multiple alignment program for protein sequences. MUSCLE stands for multiple sequence comparison by log-expectation.<br/>[MUSCLE 3.8.31; <a href = http://www.drive5.com/muscle/>http://www.drive5.com/muscle/</a>]", [], "", "Alva V., Nam SZ., Söding J., Lupas AN. (2016) <b>The MPI bioinformatics Toolkit as an integrative platform for advanced protein sequence and structure analysis.</b> Nucleic Acids Res. pii: gkw348. [Epub ahead of print] <a href=https://www.ncbi.nlm.nih.gov/pubmed/27131380>PMID: 27131380</a><br/><br/>Edgar RC. (2004) <b>MUSCLE: multiple sequence alignment with high accuracy and high throughput.</b> Nucl Acid Res 32(5):1792-1797. <a href = https://www.ncbi.nlm.nih.gov/pubmed/15034147>PMID: 15034147</a> "],
    "tcoffee": ["T-Coffee (Tree-based Consistency Objective Function for alignment Evaluation) is one of the most accurate multiple sequence alignment methods for protein and nucleotide sequences.<br/>[T-Coffee 11.0; <a href = http://www.tcoffee.org/>http://www.tcoffee.org/</a>] ", [], "", "Alva V., Nam SZ., Söding J., Lupas AN. (2016) <b>The MPI bioinformatics Toolkit as an integrative platform for advanced protein sequence and structure analysis.</b> Nucleic Acids Res. pii: gkw348. [Epub ahead of print] <a href=https://www.ncbi.nlm.nih.gov/pubmed/27131380>PMID: 27131380</a><br/><br/>Magis C., Taly JF., Bussotti G., Chang JM., Di Tommaso P., Erb I., Espinosa-Carrasco J., Notredame C. (2014) <b>T-Coffee: Tree-based consistency objective function for alignment evaluation. </b>Methods Mol Biol 1079:117-129. <a href = https://www.ncbi.nlm.nih.gov/pubmed/24170398>PMID: 24170398</a>"],
    "aln2plot": ["Shows a graphical overview of hydrophobicity and side chain volume along a protein sequence.", [], "", "Alva V., Nam SZ., Söding J., Lupas AN. (2016) <b>The MPI bioinformatics Toolkit as an integrative platform for advanced protein sequence and structure analysis.</b> Nucleic Acids Res. pii: gkw348. [Epub ahead of print] <a href=https://www.ncbi.nlm.nih.gov/pubmed/27131380>PMID: 27131380</a>"],
    "frpred": ["FRpred is a collection of methods for the prediction of functional residues in protein multiple sequence alignemnets. You can also run it on a single sequence. Then, PSI-BLAST is called to generate an alignment, which is directly used as input for the methods. If you have a PDB file representing the sequences in your alignment, you can specify it in the input parameters. It will be used to visualize the results via RasMol scripts. The methods in FRpred are:<br/><br/><ol><li><b>Shannon entropy:</b> A simple method to score conservation.</li><li><b>FRcons:</b> A novel conservation method that scores fully conserved positions equally, conservative mutaions higher than non-conservative ones, penalizes gap-rich columns and yields a score of 0 for positions distributed like the background frequencies.</li><li><b>Evolutionary Trace (ET):</b> A method yielding conserved and subtype-specific residues. It uses UPGMA to cluster the sequences according to sequence similarity and the sums over teh entropy conservation score of all subgroups defined by dendrogram cuts on all levels of the UPGMA tree.</li><li><b>FRtrace:</b> Following the idea of ET, FRtrace also sums over conservation scores in subgroups defined by a UPGMA tree in order to predict conserved as well as subtype-specific positions. However, FRtrace uses FRcons instead of entropy conservation and weighs the contribution from each subgroup according to its effective number of sequences. Moreover, the method takes only those subgroups into account that are new in the current dendrogram cut when traversing the tree from the root to the leafs. FRtrace offers a family- and a sequence-centered analysis.</li><li><b>FRsubtype:</b> This method predicts only subtype-specific residues. It uses mutual information to score the difference between the amino acid distribution at the same position in different subgroups and, additionally, it scores the conservation within the subgroups using FRcons. It also offers a sequence- and a family-centerd analysis. You can either manually group the alignment into subgroups or - if this information is not available - the method can do its own clustering (UPGMA tree) and sum over the weighted subgroups.</li></ol><br/>If a structure is supplied, DSSP will be used to compute the solvent accessibility. Otherwise, Sable2 is run to predict it. Sable2 runs 3 PSI-BLAST iterations on the query sequence, which constitutes a big part of the total runtime of FRpred. ", [], "", "Alva V., Nam SZ., Söding J., Lupas AN. (2016) <b>The MPI bioinformatics Toolkit as an integrative platform for advanced protein sequence and structure analysis.</b> Nucleic Acids Res. pii: gkw348. [Epub ahead of print] <a href=https://www.ncbi.nlm.nih.gov/pubmed/27131380>PMID: 27131380</a><br/><br/>Fischer JD., Mayer CE., Söding J. (2008) <b>Prediction of Protein Functional Residues From Sequence by Probability Density Estimation. </b>Bioinformatics 24:613-620. <a href = https://www.ncbi.nlm.nih.gov/pubmed/18174181>PMID: 18174181</a><br/><br/>Mihalek I., Res I., Lichtarge O. (2004) <b>A Family of Evolution-Entropy Hybrid Methods for Ranking Protein Residues by Importance. </b>J Mol Biol 336:1265-1282. <a href = https://www.ncbi.nlm.nih.gov/pubmed/15037084>PMID: 15037084</a><br/><br/>Hannenhalli SS., Russel, RB. (2000) <b>Analysis and Prediction of Functional Sub-types from Protein Sequence Alignments. </b>J Mol Biol 303(1):61-76. <a href = https://www.ncbi.nlm.nih.gov/pubmed/11021970>PMID: 11021970</a><br/><br/>Altschul S., Madden T., Schaffer A., Zhang J., Zhang Z., Miller W., Lipman D. (1997) <b>Gapped BLAST and PSI-BLAST: a New Generation of Protein Database Search Programs. </b>Nucl Acids Res 25:3389-3402. <a href = https://www.ncbi.nlm.nih.gov/pubmed/9254694>PMID: 9254694</a><br/><br/>Kabsch W., Sander C. (1983) <b>Dictionary of Protein Secondary Structure: Pattern Recognition of Hydrogen-Bonded and Geometrical Features.</b> Biopolymers 22:2577-2637. <a href = https://www.ncbi.nlm.nih.gov/pubmed/6667333>PMID: 6667333</a><br/><br/>Adamczak R., Porollo A., Meller J. (2004) <b>Accurate Predicition of Solvent Accessibility Using Neural Networks-Based Regression.</b> Proteins 56:753-767. <a href = https://www.ncbi.nlm.nih.gov/pubmed/15281128>PMID: 15281128</a>"],
    "hhrepid": ["<h3>About HHrepID</h3><br/><h5><b>What is HHrepID good for?</b></h5>HHrepID is a novel automated procedure for the de novo identification of repeats in protein sequences. It is able to detect the sequence signature of structural repeats in many proteins that have not yet been known to possess internal sequence symmetry, such as TIM barrels and outer membrane beta-barrels. HHrepID uses HMM-HMM comparison to exploit evolutionary information in the form of the multiple sequence alignment of homologs, but in contrast to HHrep, the new method has several novel characteristics: (1) automated generation of a multiple alignment of repeats; (2) utilization of the transitive nature of homology through a novel merging procedure based on the fully probabilistic treatment of alignments; (3) superior alignment quality through an algorithm that maximizes the expected accuracy of an alignment; (4) the ability to identify different repeats within complicated architectures or multiple domains through automatic domain boundary detection, (5) new statistical treatment yielding improved sensitivity.<br/><br/><h5><b>Length of submitted sequences</b></h5>HHrepID works best on single-domain seuqences or sequences containing only one kind of tandem repeat. If you don't know where your domains are, perform an HHpred search through SCOP and PfamA. Also, you could perform a preliminary HHrepID analysis to home in on the repeat regions and then cut the sequence further down to include only regions with one kind of repeat structure. Sequences should best have between 100 and 300 residues length. Sequences that are too short will gather too few homologs in the PSI-BLAST searches, thus loosing sensitivity. Also, the more repeat units are contained in a sequence, the more information from transitivity of pairwise alignmemnts (A-B, B-C => A-C) can be used by HHrepID through its \"Merge alignments\" functionality. Sequences longer than 800 residues are truncated by HHrepID for practical reasons.<br/><br/><h5><b>Submitting an alignment</b></h5>If you already have a trustworthy mulitple sequence alignment of your query with some homologs it is recommended that you start with this instead of from a single sequence. This will help PSI-BLAST to avoid alignment errors.<br/><br/><h5><b>Why is HHrepID so sensitive?</b></h5>The reason for HHrepID's sensitivity lies in the use of evolutionary information when comparing the query sequence with itself. This evolutionary information comes from the multiple sequence alignment of the sequence with homologous stretches of proteins. The information in the multiple sequence alignment is used in a condensed form, a profile HMM. A profile HMM is almost the same as a sequence profile. This in turn is just a table with 20 rows and L columns (where L is the length of the query protein). Each of the L columns contains the frequencies of the 20 amino acids in the corresponding column of the multiple sequence alignment. This table is then compared with itself in order to find stretches of columns with similar amino acid distributions. By looking at many related sequences with very similar structures that contribute to the sequence profile, profile-profile (or HMM-HMM) comparison effectively averages away those features of the sequence that are there just by chance and not by necessity and concentrates on those that determine the protein family's properties.<br/><br/>A method that also employs HMM-HMM comparison for very sensitive protein-protein comparison is implemented in HHpred, our interactive function and structure prediction server. (For a short description of HHpred, please go to the HHpred help.) ", [], "", "Alva V., Nam SZ., Söding J., Lupas AN. (2016) <b>The MPI bioinformatics Toolkit as an integrative platform for advanced protein sequence and structure analysis.</b> Nucleic Acids Res. pii: gkw348. [Epub ahead of print] <a href=https://www.ncbi.nlm.nih.gov/pubmed/27131380>PMID: 27131380</a><br/><br/>Biegert A., Söding J. (2008) <b>HHrepID: de novo protein repeat identification by probabilistic consistency. </b>Bioinformatics 24(6):807-814.<a href = https://academic.oup.com/bioinformatics/article/24/6/807/194276/De-novo-identification-of-highly-diverged-protein>(download pdf)</a> <a href = https://www.ncbi.nlm.nih.gov/pubmed/18245125>PMID: 18245125</a>"],
    "marcoil": ["MARCOIL is a hidden Markov model-based program that predicts existence and location of potential coiled-coil domains in protein sequences. The external MARCOIL Homepage can be found <a href = http://bioinf.wehi.edu.au/folders/mauro/Marcoil/index.html>here</a>. ", [], "", "Alva V., Nam SZ., Söding J., Lupas AN. (2016) <b>The MPI bioinformatics Toolkit as an integrative platform for advanced protein sequence and structure analysis.</b> Nucleic Acids Res. pii: gkw348. [Epub ahead of print] <a href=https://www.ncbi.nlm.nih.gov/pubmed/27131380>PMID: 27131380</a><br/><br/>Delorenzi M., Speed T. (2002) <b>An HMM model for coiled-coil domains and a comparison with PSSM-based predictions. </b>Bioinformatics 18(4):617-625. <a href = https://www.ncbi.nlm.nih.gov/pubmed/12016059>PMID: 12016059</a>"],
    "pcoils": ["<h5><b>Recommendations</b></h5>COILS with the MTIDK matrix is specific for solvent-exposed, left-handed coiled coils. Other types of coiled-coil structure, such as buried coiled coils (e.g the central coiled coil in catabolite repressor protein, or some transmembrane domains) and right-handed coiled coils, are not detected. COILS does not reach yes-or-no decisions based on a threshold value. Rather, it yields a set of probabilities that presumably reflect the coiled-coil forming potential of a sequence. This means that even at high probabilities (e.g. >90%), there will be (and should be) sequences that in fact do not form a coiled coil, though they may have the potential to do so in a different context. COILS is biased towards hydrophilic, highly charged sequences. For this reason, all scans should be performed with a weighted and an unweighted matrix, and the results compared. Differences of more than 20-30 percentage points in the probabilities should be taken to indicate that a coiled-coil structure is unlikely, the elevated scores being mainly due to the high incidence of charged residues (note, though, that this would have marked human mannose-binding protein as a false positive). The MTIDK matrix assigns high probabilities to known coiled coils segments, but identifies different helices at high probability in a database of globular proteins. This is a surprising feature whose reason is as yet unclear, but which can be exploited for predictive purposes. It is therefore useful to compare the results of scans made with the two matrices. Again, differences of more than 20-30 percentage points in the probabilities should be taken to indicate that a coiled-coil structure is unlikely (note, though, that this threshold would make the replication terminator protein a border-line case). The resolution between globular and coiled-coil score distributions decreases strongly with a decreasing size of the scanning window. The prediction of new coiled-coil segments should therefore be made using a 28 residue window, or in special cases a 21 residue window. 14 residue windows should normally be reserved for the analysis of local parameters (such as the frame) in known or predicted coiled coils. The ends of coiled-coil segments appear to be most accurately identified in a 21 residue window. In general, it is assumed that residues with probabilities >50% are part of a coiled-coil segment. Sequences with high coiled-coil probabilitiy from globular proteins rarely exceed a length of 30 residues. None is longer than 35 residues. Sequences with probabilities >80-90% that extend for more than 35 residues are therefore more likely to assume a coiled-coil structure than is indicated by the obtained probabilities. Where possible, sequences related to the protein of interest should also be analyzed for predicted coiled-coil segments (e.g. by using a sequence alignment instead of a single sequence as input). It should be kept in mind, though, that the sequences must be related in the region of high scores in order for the comparison to be significant.<br/><br/><h5><b>PSIPRED</b></h5>Comparison of the coiled-coil prediction with predictions of the secondary structure are generally useful, particularly if multiple related sequences are available. PSIPRED predicts the secondary structure as described in <a href =https://www.ncbi.nlm.nih.gov/pubmed/10493868?dopt=Abstract>Jones, J.Mol.Biol. 292:195-202 (1999).</a>", [], "", "Alva V., Nam SZ., Söding J., Lupas AN. (2016) <b>The MPI bioinformatics Toolkit as an integrative platform for advanced protein sequence and structure analysis.</b> Nucleic Acids Res. pii: gkw348. [Epub ahead of print] <a href=https://www.ncbi.nlm.nih.gov/pubmed/27131380>PMID: 27131380</a><br/><br/>Lupas A., Van Dyke M., Stock J. (1991) <b>Predicting Coiled Coils from Protein Sequences. </b>Science 252:1162-1164. <a href = https://www.ncbi.nlm.nih.gov/pubmed/2031185>PMID: 2031185</a><br/><br/>Lupas A. (1996) <b>Prediction and Analysis of Coiled-Coil Structures. </b>Methods Enzymol 266:513-525. <a href = https://www.ncbi.nlm.nih.gov/pubmed/8743703>PMID: 8743703</a><br/><br/>Parry DA. (1982) <b>Coiled-Coils in Alpha-Helix-Containing Proteins: Analysis of the Residue Types within the Heptad Repeat and the Use of These Data in the Prediction of Coiled-Coils in Other Proteins. </b>Biosci Rep 2(12):1017-24. <a href = https://www.ncbi.nlm.nih.gov/pubmed/7165792>PMID: 7165792</a>"],
    "repper": ["<h3>About REPPER</h3>REPPER is a server that detects and analyses regions with short gapless REPeats in protein sequences. It finds PERiodicities by Fourier Transform (FTwin) and internal homology analysis (REPwin). The exact locations of the periodic patterns are traced by a sliding window. The output is complemented by coiled coil prediction (COILS) and optionally by secondary structure prediction (PSIPRED).<br/><br/><h5><b>FTwin</b></h5>Sequence analysis by a sliding window Fourier transform (FT) method calculating the periodicity spectra for all possible sequence windows. The overview graph shows all significant periodicities and where in the sequence they are located. Periodicities are considered to be significant if their FT intensity is above the given threshold.<br/>Clicking on a colored line on the output graph provides the complete FT spectrum for the sequence fragment defined by the line (the fragment may include closely neighbored lines as well). Alternatively, an arbitrary sequence range may be given (in the boxes under the output graph) in order to get a FT spectrum of this particular sequence fragment. <a href = ftp://ftp.tuebingen.mpg.de/pub/protevo/FTwin/>Download</a><br/><br/><h5><b>REPwin</b></h5>REPWin compares a protein sequence with itself within a sliding window of choosable length W. For each window starting position i and periodicity p it calculates<br/><br/><div align = center><em>Score(i,p) = Sum_k Sum_j S(x[j],x[j+kp])</em></div><br/>S(x[j],x[j+kp]) is the Gonnet substitution matrix element for residues x[j] and x[j+kp]. The sum runs over all k and j such that j and j+kp are inside the window [i,..,i+W-1]. The score S(i,p) is normalized by dividing through the standard deviation of S(i,p) for nonperiodic sequences. The final score value for each residue number i and periodicity p is the maximum over all windows containing residue i.<br/>Clicking on a colored line on the output graph displays the corresponding alignment that shows the internal homology. <a href = ftp://ftp.tuebingen.mpg.de/pub/protevo/REPwin/>Download</a><br/><br/><h5><b>COILS</b></h5>COILS is a program that compares a sequence to a database of known parallel two-stranded coiled-coils and derives a similarity score. By comparing this score to the distribution of scores in globular and coiled-coil proteins, the program then calculates the probability that the sequence will adopt a coiled-coil conformation. COILS is described in <a href = https://www.ncbi.nlm.nih.gov/pubmed/2031185?dopt=Abstract>Lupas, Science 252: 1162-1164 (1991)</a> and Lupas, Meth. Enzymology 266: 513-525 (1996). It is based on a prediction protocol proposed by <a href = https://www.ncbi.nlm.nih.gov/pubmed/7165792?dopt=Abstract>Parry, Biosci. Rep. 2: 1017-1024 (1982)</a>. You can also <a href = ftp://ftp.tuebingen.mpg.de/pub/protevo/COILS/>download</a> the software.<br/><br/><h5><b>PSIPRED</b></h5>PSIPRED predicts the secondary structure as described in <a href = https://www.ncbi.nlm.nih.gov/pubmed/10493868?dopt=Abstract>Jones, J.Mol.Biol. 292:195-202 (1999)</a>.", [], "", "Alva V., Nam SZ., Söding J., Lupas AN. (2016) <b>The MPI bioinformatics Toolkit as an integrative platform for advanced protein sequence and structure analysis.</b> Nucleic Acids Res. pii: gkw348. [Epub ahead of print] <a href=https://www.ncbi.nlm.nih.gov/pubmed/27131380>PMID: 27131380</a><br/><br/>Gruber M., Söding J., Lupas AN. (2005) <b>REPPER -- Repeats and their periodicities in fibrous proteins. </b>Nucl Acids Res 33(Web Server issue): W239-W243. <a href = https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1160166/>PMCID: PMC1160166</a>"],
    "tprpred": ["TPRpred uses the profile representation of the known repeats to detect Tetratrico Peptide Repeats (TPRs), Pentatrico Peptide Repeats(PPRs) and SEL1-like repeats from the query sequence and computes the statistical significance for their occurrence.", [], "", "Alva V., Nam SZ., Söding J., Lupas AN. (2016) <b>The MPI bioinformatics Toolkit as an integrative platform for advanced protein sequence and structure analysis.</b> Nucleic Acids Res. pii: gkw348. [Epub ahead of print] <a href=https://www.ncbi.nlm.nih.gov/pubmed/27131380>PMID: 27131380</a><br/><br/>Karpenahalli MR., Söding J., Lupas AN. (2007) <b>TPRpred: a tool for prediction of TPR-, PPR- and SEL1-like repeats from protein sequences. </b>BMC Bioinformatics 8:2. <a href = https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1774580/>PMCID: PMC1774580</a>"],
    "ali2d": ["Ali2D which takes a sequence alignment as input, performs PSIPRED and MEMSAT2 on each of the sequences and finally plots information about secondary structure, transmembrane regions and amino acids on the alignemnt.", [], "", "Alva V., Nam SZ., Söding J., Lupas AN. (2016) <b>The MPI bioinformatics Toolkit as an integrative platform for advanced protein sequence and structure analysis.</b> Nucleic Acids Res. pii: gkw348. [Epub ahead of print] <a href=https://www.ncbi.nlm.nih.gov/pubmed/27131380>PMID: 27131380</a>"],
    "quick2d": ["Quick2D gives you an overview of secondary structure features like alpha-helices, extended beta-sheets, coiled coils, transmembrane helices and disorder regions. Predictions by PSIPRED, JNET, Prof(Rost), Prof(Ouali), Coils, HMMTOP and DISOPRED2.", [], "", "Alva V., Nam SZ., Söding J., Lupas AN. (2016) <b>The MPI bioinformatics Toolkit as an integrative platform for advanced protein sequence and structure analysis.</b> Nucleic Acids Res. pii: gkw348. [Epub ahead of print] <a href=https://www.ncbi.nlm.nih.gov/pubmed/27131380>PMID: 27131380</a>"],
    "modeller": ["Modeller is one of the most popular programs for homology modelling. The goal of homology modelling (also called comparative modelling) is to obtain a 3D structure for a protein sequence of interest (the target). One starts from a sequence alignment of the target sequence with one or more template sequences for which the structures are known. Sequences with recognizable sequence similarity (e.g. above 30%) will in general have quite similar 3D structures. Therefore, an alignment with a homologous template of known structure can, in principle, be used to directly predict the main chain coordinates by simply mapping the target sequence into the template main chain structure. Modeller goes a step further in deriving constraints from the multiple alignment of the target with the templates. In addition to a-priori constraints like bond lengths et cetera, these constraints may refer to Calpha-Calpha distances, main-chain dihedral angles, or side-chain dihedral angles. The constraints are formulated probabilistically in terms of probability densitiy functions, that are derived from a statistical analysis of a great number of pairwise structural alignments. Modeller then tries find the parameters (i.e. the model structure) that maximizes the total probability density obtained by combining all constraints. Modeller, as most (if not all) other available programs, has a few important limitations:<br/><br/><ul><li>It can not correct a wrong or suboptimal target-template alignment. Errors in the alignment will directly translate into errors in the model structures. The alignment errors are the leading source of errors for homology models when the sequence similarity is below ~50%.</li><li>The model is, on average, not more similar to the real target structure than the template used. (This sounds frustrating, but it means that normally, the improvements which the homology modelling software achieves do not fully compensate the alignment errors, on average). For this reason, the other limiting factor in model quality, beside the alignment quality, is the selection of the best template. This is a very difficult task for which a large number of methods and servers have been developed. (See our HHpred server and the links in the section on tertiary structure prediction.)</li><li>Side chain placement gets unreliable below sequence identities of 70%, simply because the position of side chains is conserved less and less below this degree of similarity.</li><li>Loops of more than three or four residues for which no corresponding template atoms are available can not be reliably estimated. They are more or less guessed by the programs. There exist specialized loop modelling methods, but they are very time-consuming and can not be offered over the web. Also, loops of more than 8 to 12 residues can not be modelled reliably.</li></ul><br/>In summary, homology modelling software can not correct errors in the alignments or templates. What you get out is almost what you put in.", [], "", "Alva V., Nam SZ., Söding J., Lupas AN. (2016) <b>The MPI bioinformatics Toolkit as an integrative platform for advanced protein sequence and structure analysis.</b> Nucleic Acids Res. pii: gkw348. [Epub ahead of print] <a href=https://www.ncbi.nlm.nih.gov/pubmed/27131380>PMID: 27131380</a><br/><br/>Sali A., Potterton L., Yuan F., van Vlijmen H., Karplus M. (1995) <b>Evaluation of comparative protein modeling by MODELLER. </b>Proteins 23(3):318-26. <a href = https://www.ncbi.nlm.nih.gov/pubmed/8710825>PMID: 8710825</a>"],
    "samcc": ["SamCC measures local parameters of (a)symmetric, parallel, and antiparallel four-helical bundles. Briefly, SamCC divides bundle into layers (slices), brings each layer into an idealized state (expected from equations), then performing final calculations on the resulting idealized structure. ", [], "", "Alva V., Nam SZ., Söding J., Lupas AN. (2016) <b>The MPI bioinformatics Toolkit as an integrative platform for advanced protein sequence and structure analysis.</b> Nucleic Acids Res. pii: gkw348. [Epub ahead of print] <a href=https://www.ncbi.nlm.nih.gov/pubmed/27131380>PMID: 27131380</a><br/><br/>Dunin-Horkawicz S., Lupas AN. (2010) <b>Measuring the conformational space of square four-helical bundles with the program samCC. </b>170(2):226-35. doi: 10.1016/j.jsb.2010.01.023. <a href = https://www.ncbi.nlm.nih.gov/pubmed/20139000>PMID: 20139000</a>"],
    "ancescon": ["ANCESCON is a package for distance-based phylogenetic inference and reconstruction of ancestral protein sequences that takes into account the observed variation of evolutionary rates between positions that more precisely describes the evolution of protein families.<br/><br/>[Reconstruction of ancestral protein sequences and its applications, Wei Cai, Jimin Pei, and Nick V Grishin<br/>BMC Evol Biol. 2004; 4: 33. doi: 10.1186/1471-2148-4-33. <a href = https://www.ncbi.nlm.nih.gov/pmc/articles/PMC522809/>Published online 2004 September 17</a>.]", [], "", "Alva V., Nam SZ., Söding J., Lupas AN. (2016) <b>The MPI bioinformatics Toolkit as an integrative platform for advanced protein sequence and structure analysis.</b> Nucleic Acids Res. pii: gkw348. [Epub ahead of print] <a href=https://www.ncbi.nlm.nih.gov/pubmed/27131380>PMID: 27131380</a><br/><br/>Cai W., Pei J., Grishin NV. (2004) <b>Reconstruction of ancestral protein sequences and its applications. </b>BMC Evol Biol. 4:33. <a href = https://www.ncbi.nlm.nih.gov/pubmed/15377393>PMID: 15377393</a>"],
    "clans": ["A clustering is performed based on all against all BLAST similarities and the detected groups are returned. Only BLAST similarities better than a specified cutoff are used to perform the cluster analysis.", [], "", "Alva V., Nam SZ., Söding J., Lupas AN. (2016) <b>The MPI bioinformatics Toolkit as an integrative platform for advanced protein sequence and structure analysis.</b> Nucleic Acids Res. pii: gkw348. [Epub ahead of print] <a href=https://www.ncbi.nlm.nih.gov/pubmed/27131380>PMID: 27131380</a><br/><br/>Frickey T., Lupas AN. (2004) <b>CLANS: a Java application for visualizing protein families based on pairwise similarity. </b>Bioinformatics 20(18):3702-3704. <a href = https://www.ncbi.nlm.nih.gov/pubmed/15284097>PMID: 15284097</a>"],
    "mmseqs2": ["", [], "", ""],
    "phylip": ["This tool provides quick and easy access to standard distance based phylogenetic methods. NEIGHBOR is a module of the PHYLIP package and can be used for the computation of unrooted Neighbor-Joining and rooted UPGMA (Average Linkage clustering) trees assuming a molecular clock in the latter case. DNA or protein alignments are preprocessed with <a href = http://evolution.genetics.washington.edu/phylip/doc/dnadist.html>DNADIST</a> or <a href = http://evolution.genetics.washington.edu/phylip/doc/protdist.html>PROTDIST</a>. For bootstrapping the program <a href = http://evolution.genetics.washington.edu/phylip/doc/seqboot.html>SEQBOOT</a> is used for resampling and <a href = http://evolution.genetics.washington.edu/phylip/doc/consense.html>CONSENSE is employed to evaluate a consensus tree.</a>", [], "", "Alva V., Nam SZ., Söding J., Lupas AN. (2016) <b>The MPI bioinformatics Toolkit as an integrative platform for advanced protein sequence and structure analysis.</b> Nucleic Acids Res. pii: gkw348. [Epub ahead of print] <a href=https://www.ncbi.nlm.nih.gov/pubmed/27131380>PMID: 27131380</a><br/><br/>Felsenstein, J. (1991-2012) <b>Neighbor -- Neighbor-Joining and UPGMA methods.</b> Retrieved May 23, 2016, from <a href = http://evolution.genetics.washington.edu/phylip/doc/neighbor.html>http://evolution.genetics.washington.edu/phylip/doc/neighbor.html</a>"],
    "6frametranslation": ["Performs a six-frame translation on a nucleotide sequence.", [], "", "Alva V., Nam SZ., Söding J., Lupas AN. (2016) <b>The MPI bioinformatics Toolkit as an integrative platform for advanced protein sequence and structure analysis.</b> Nucleic Acids Res. pii: gkw348. [Epub ahead of print] <a href=https://www.ncbi.nlm.nih.gov/pubmed/27131380>PMID: 27131380</a>"],
    "backtrans": ["The Backtranslator translates an amino acid sequence back into a possible DNA sequence.", [], "", "Alva V., Nam SZ., Söding J., Lupas AN. (2016) <b>The MPI bioinformatics Toolkit as an integrative platform for advanced protein sequence and structure analysis.</b> Nucleic Acids Res. pii: gkw348. [Epub ahead of print] <a href=https://www.ncbi.nlm.nih.gov/pubmed/27131380>PMID: 27131380</a><br/><br/>Nakamura Y., Gojobori T., Ikemura T. (2000) <b>Codon usage tabulated from international DNA sequence databases: status for the year 2000. </b>Nucleic Acids Res 28(1):292. <a href = https://www.ncbi.nlm.nih.gov/pubmed/10592250>PMID: 10592250</a>"],
    "hhfilter": ["Extract a representative set of sequences from an alignment.", [], "", "Alva V., Nam SZ., Söding J., Lupas AN. (2016) <b>The MPI bioinformatics Toolkit as an integrative platform for advanced protein sequence and structure analysis.</b> Nucleic Acids Res. pii: gkw348. [Epub ahead of print] <a href=https://www.ncbi.nlm.nih.gov/pubmed/27131380>PMID: 27131380</a><br/><br/>Remmert M., Biegert A., Hauser A., Söding J. (2011) <b>HHblits: Lightning-fast iterative protein sequence searching by HMM-HMM alignment. </b>Nat Methods. 9(2):173-5. doi: 10.1038/nmeth.1818. <a href = https://www.ncbi.nlm.nih.gov/pubmed/22198341>PMID: 22198341</a>"],
    "retseq": ["This utility performs a sequence retrieval from the nre and nr database using a list of identifiers.", [], "", ""],
    "seq2id": ["Extract GIs is a utility that extracts all GI-numbers from a given file. Each sequence has to be new line separated. eg.:<br/><br/>Sequence 1<br/>Sequence 2<br/>sequence 3<br/>...", [], "", ""]
};

helpContent = function(tool : string) {
    if (exampleContent[tool]) {
        return exampleContent[tool];
    } else {
        return ["", [], "", ""];
    }
};

accordion = function(elem : Element, isInit : boolean) : any {
    if (!isInit) {
        return elem.setAttribute("data-accordion", "data-accordion");
    }
};

accordionItem = function(elem : Element, isInit : boolean) : any {
    if (!isInit) {
        return elem.setAttribute("data-accordion-item", "data-accordion-item");
    }
};

accordionContent = function(elem : Element, isInit : boolean) {
    if (!isInit) {
        return elem.setAttribute("data-tab-content", "data-tab-content");
    }
};

(<any>window).HelpModalComponent = {
    controller: function(){},
    view: function(ctrl : any, args : any) {
        let overview : any, params : any, results : any, references : any;
        overview = helpContent(args.toolname)[0];
        params = helpContent(args.toolname)[1];
        results = helpContent(args.toolname)[2];
        references = helpContent(args.toolname)[3];
        return m("div", {
            id: "help-" + args.toolname,
            "class": "reveal",
            config: helpModalReveal
        }, m("div", {
            id: "help-tabs",
            config: helpModalTabs
        }, [
            m("ul", [
                m("li", m("a", {
                    href: "#help-tabs1"
                }, "Overview")), m("li", m("a", {
                    href: "#help-tabs2"
                }, "Parameters")), m("li", m("a", {
                    href: "#help-tabs3"
                }, "Results")), m("li", m("a", {
                    href: "#help-tabs4"
                }, "References")), m("li", {
                    "class": "toolname"
                }, m("a", {
                    href: "#",
                    "class": "not-active"
                }, args.toolnameLong))
            ]), m("div", {
                id: "help-tabs1"
            }, m("div", m.trust(overview))), m("div", {
                id: "help-tabs2"
            }, m("ul", {
                "class": "accordion",
                config: accordion
            }, params.map(function(param : any) {
                return m("li", {
                    "class": "accordion-item",
                    config: accordionItem
                }, [
                    m("a", {
                        href: "#",
                        "class": "accordion-title"
                    }, param), m("div", {
                        "class": "accordion-content",
                        config: accordionContent
                    }, [ m.trust(parameterContent[param]) ])
                ]);
            }))), m("div", {
                id: "help-tabs3"
            }, m("div", m.trust(results)), m("img", {
                src: '/assets/images/psiblast.png',
                style: {
                    marginTop: '2em'
                }
            })), m("div", {
                id: "help-tabs4"
            }, m("div", m.trust(references)))
        ]));
    }
};