<div id="recentUpdatesContent">
    <h5>Recent Updates</h5>
    <hr>
    <div class="row section">
        <h6><a id="#arc_2017_12">December 5, 2017</a></h6>
        <p>
            HHpred: profile HMM DBs of six archaeal proteomes (<em>Archaeoglobus fulgidus</em>, <em>Halobacterium jilantaiense</em>,
            <em>Methanocaldococcus jannaschii</em>, <em>Methanothermus fervidus</em>, <em>Pyrococcus horikoshii</em>,
            and <em>Sulfolobus solfataricus</em>) are online.
        </p>
    </div>
    <hr>
    <div class="row section">
        <h6><a id="#nr30_2017_11">November 13, 2017</a></h6>
        <p>In addition to nr50, we now also offer nr30 for PSI-BLAST, HMMER, and PatternSearch.</p>
    </div>
    <hr>
    <div class="row section">
        <h6><a id="#uniclust_2017_10">October 31, 2017</a></h6>
        <p>
        HHpred: the Uniclust30 DB is now available for query A3M generation.</p>
    </div>
    <hr>
    <div class="row section">
        <h6><a id="uniclust_2017_7">September 29, 2017</a></h6>
        <p>
        HHblits: a new version of the Uniclust30 DB is online.</p>
    </div>
    <hr>
    <div class="row section">
        <h6><a id="fisc">September 24, 2017</a></h6>
        <p>
        HHpred: profile HMM DBs of <em>Fischerella muscicola</em>, <em>Frankia alni</em>, <em>Streptomyces scabiei</em>, and <em>Thermus aquaticus</em> are online.</p>
    </div>
    <hr>
    <div class="row section">
        <h6><a id="aqui">September 19, 2017</a></h6>
        <p>
        HHpred: profile HMM DBs of <em>Aquifex aeolicus</em>, <em>Deinococcus radiodurans</em>, <em>Dictyostelium discoideum</em>, <em>Schizosaccharomyces pombe</em>, and <em>Thermus thermophilus</em> are online.</p>
    </div>
    <hr>
    <div class="row section">
        <h6><a id="ecod">September 9, 2017</a></h6>
        <p>
        HHpred: profile HMMs of the ECOD database (Grishin Lab) are online.        </p>
    </div>
    <hr>
    <div class="row section">
        <h6><a>August 31, 2017</a></h6>
        <p>
        HHpred: HMM database of <em>Giardia lamblia</em> is online.        </p>
    </div>
    <hr>
    <div class="row section">
        <h6><a id="patens">August 28, 2017</a></h6>
        <p>
        HHpred: HMM database of <em>Physcomitrella patens</em> is online.        </p>
    </div>
    <hr>
    <div class="row section">
        <h6><a id="tetra">August 23, 2017</a></h6>
        <p>
        HHpred: HMM databases of <em>Brachypodium distachyon</em> and <em>Tetrahymena thermophila</em> are online.        </p>
    </div>
    <hr>
    <div class="row section">
        <h6><a>July 21, 2017</a></h6>
        <p>
            Our Quick2D tool for the annotation of protein secondary structural features is back.
        </p>
    </div>
    <hr>
    <div class="row section">
        <h6><a>July 18, 2017</a></h6>
        <p>
        HMM databases of <em>Caenorhabitis elegans</em> and <em>Drosophila melanogaster</em> are online.        </p>
    </div>
    <hr>
    <div class="row section">
        <h6><a>July 10, 2017</a></h6>
        <p>
            HMM databases of the COG, KOG, SMART, and NCBI CD databases are online now.
        </p>
    </div>
    <hr>

    <div class="row section">
        <h6><a>July 4, 2017</a></h6>
        <p>
            Profile HMM databases of the <b>human proteome</b> and <em>Chaetomium thermophilum</em> are online.
        </p>
    </div>
    <hr>

    <div class="row section">
        <h6><a>June 20, 2017</a></h6>
        <p>
             Profile HMM database of <em>Arabidopsis thaliana</em> is online.
        </p>
    </div>
    <hr>

    <div class="row section">
        <h6><a>June 10, 2017</a></h6>
        <p>
            Profile HMM databases of many bacterial proteomes are online.
        </p>
    </div>
    <hr>

    <div class="row section">
        <h6><a>May 27, 2017</a></h6>
        <p>
            Profile HMM databases of <em>Mycobacterium tuberculosis</em> is now available for HHpred.
        </p>
    </div>
    <hr>

    <div class="row section">
        <h6><a id="releaseNotes">May 19, 2017</a></h6>
        <p>
            Release Notes: Welcome to the new version of the MPI Bioinformatics Toolkit
        </p>
        <p>
            Dear Toolkit Users,<br>
            <br>
            welcome to the new version of the MPI Bioinformatics Toolkit hosted at the Department of Protein Evolution,
            Max Planck Institute for Developmental Biology, Germany. The first version of the Toolkit was released in 2005 and
            has serviced over 2.5 million external queries to date. We had to rework the Toolkit entirely because of mounting
            software incompatibility with our computing infrastructure and also took this opportunity to improve some features
            and prune a few of the outdated and less-widely used tools. To allow an easy transition to the new Toolkit, we have
            retained the basic structure and UI features of the earlier version, and have added additional features, some of
            which are listed below, to enhance user experience.
        </p>
        <p>
            <em>New tools</em>
        </p>
        <ul>
            <li>PhyML is offered instead of PHYLIP-NEIGHBOR for phylogenetic inference.</li>
            <li>MMseq2s is offered instead of BLASTClust for clustering of up to 20,000 sequences.</li>
        </ul>
        <p>
            <em>New features/changes</em>
        </p>
        <ul>
            <li>HHpred allows comparison of two sequences or MSAs, replacing HHalign.</li>
            <li>CLANS allows clustering of up to 10,000 sequences.</li>
            <li>BLAST+ and PSI-BLAST+ are offered as a single tool.</li>
            <li>Caching of results for quick retrieval of pre-computed jobs.</li>
            <li>Search box on the index page for quick access to tools and jobs.</li>
            <li>Job manager in addition to joblist for easy management of jobs.</li>
            <li>E-mail notifications are only available to logged-in users.</li>
            <li>Comprehensive client-side input validation.</li>
            <li>Significantly enhanced performance, owing to upgraded computational infrastructure.</li>
        </ul>
        <p>
            <em>Coming soon</em>
            <br>
            We are going to bring back FRpred, HHomp, Quick2D, daTAA, and REPPER within the next few weeks. Additionally, we
            are going to offer many more proteome databases for HHpred over the next months and are also working on making the
            help pages more comprehensive.
        </p>
        <p>
            The new Toolkit is still under active development and we invite you to send us your feature requests and bug
            reports (<a href="mailto:mpi-toolkit&commat;tuebingen.mpg.de">mpi-toolkit&commat;tuebingen.mpg.de</a>
            ).
        </p>
        <p>
            If you find our Toolkit useful for your research, please cite us
            <a href="https://academic.oup.com/nar/article-lookup/doi/10.1093/nar/gkw348" target="_blank" rel="noopener"> (Alva et al., 2016) </a>
            ; it helps us keep this service running for you.
        </p>
        <p>
            Best wishes,<br>
            Your Toolkit Team
        </p>
    </div>
    <hr>
</div>