@*
 * Copyright 2018 Dept. Protein Evolution, Max Planck Institute for Developmental Biology
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *@

@(version: String)
<div id="tabs">
    <ul>
        <li><a href="#tabs-1">Overview</a></li>
        <li><a href="#tabs-2">Input & Parameters</a></li>
        <li><a href="#tabs-3">References</a></li>
        <li><a href="#tabs-4">Version</a></li>
    </ul>
    <div id="tabs-1" class="greenlink">
        <p>DeepCoil is a method for the accurate prediction of coiled coil domains in protein sequences. A local
        copy and further details on DeepCoil can here obtained here: <a href = https://github.com/labstructbioinf/DeepCoil
        target="_blank" rel="noopener">https://github.com/labstructbioinf/DeepCoil</a>.</p>

        <p>DeepCoil predictions can be made based on a single sequence or based on a multiple sequence alignment (MSA). Users
            are provided with the option to enter their own alignment as well as with an option to have the alignment generated
            for their input sequence ('Run_PSI-BLAST' mode). DeepCoil generates a graphical summary of the predictions and
            a text output with probability value for each residue in the sequence to be in a coiled coil domain.
        </p>

        <p>Please check the 'Input & Parameters' tab for details on allowed input formats.</p>

    </div>
    <div id="tabs-2" class="greenlink">
        <ul class="accordion" data-accordion data-allow-all-closed='true'>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Input</a>
                <div class="accordion-content" data-tab-content>
                    You have the option to enter or upload either a single protein sequence or a multiple sequence
                    alignment (MSA) in FASTA or CLUSTAL format.<br/><br/>
                    @views.html.help.common.cc_singleseq()
                    @views.html.help.common.cc_msa()
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Input mode</a>
                <div class="accordion-content" data-tab-content>
                    <p>Use_input_sequence: prediction is performed on the basis of the given sequence.</p>
                    <p>Use_input_msa: prediction is performed on the basis of the given multiple sequence alignment (MSA).</p>
                    <p>Run_PSI-BLAST: In this mode, a PSI-BLAST search, with 'evalue' set to 0.001 and 'num_iterations' to 3,
                        against the nr90 database is performed for the generation of a PSSM for the given query. </p>
                </div>
            </li>
        </ul>
    </div>
    <div id="tabs-3" class="greenlink">
        @views.html.help.common.toolkit_citation()
    </div>
    <div id="tabs-4" class="greenlink">
        <strong>Version:</strong> @version
    </div>
</div>
