@*
 * Copyright 2018 Dept. Protein Evolution, Max Planck Institute for Developmental Biology
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *@

@(version: String)
<div id="tabs">
    <ul>
        <li><a href="#tabs-1">Overview</a></li>
        <li><a href="#tabs-2">Input & Parameters</a></li>
        <li><a href="#tabs-3">References</a></li>
        <li><a href="#tabs-4">Version</a></li>
    </ul>
    <div id="tabs-1" class="greenlink">
        <a href = https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3125773 target="_blank" rel="noopener">HMMER</a> is a
        tool for fast and sensitive homology searching based on profile Hidden Markov models. It accepts a single
        sequence or a multiple alignment in aligned FASTA/CLUSTAL format as input. In case the input is a single sequence,
        HHblits is run over Uniprot20 to build an alignment required for the calculation of query HMM.
    </div>
    <div id="tabs-2" class="greenlink">
        <ul class="accordion" data-accordion data-allow-all-closed='true'>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Input</a>
                <div class="accordion-content" data-tab-content>
                    You have the option to enter or upload either a single protein sequence or a multiple sequence
                    alignment (MSA) in FASTA or CLUSTAL format.<br/><br/>
                    @views.html.help.common.singleseq()
                    @views.html.help.common.msa()
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Select database</a>
                <div class="accordion-content" data-tab-content>
                    The selected database is used by HMMER to perform the search.
                        <ul>
                            <li><b>nr50, nr70, nr90:</b> NCBI's nonredundant protein sequence database filtered for a
                                maximum pairwise sequence identity of 50%, 70% and 90%, respectively.</li>
                            <li><b>env70, env90:</b> NCBI's environmental sequence database, usually from unknown organisms,
                                filtered for a maximum pairwise sequence identity of 70% and 90%, respectively.</li>
                            <li><b>nr_euk*, nr_bac*, nr_arc*, nr_pro*, nr_vir*:</b> eukaryotic, bacterial, archaeal,
                                prokaryotic, and viral sequences from the nonredundant database nr.</li>
                            <li><b>pdb_nr: </b>Sequences from proteins whose structures have been deposited in the Protein
                                Data Bank PDB.</li>
                            <li><b>Uniprot_sprot:</b> manually annotated and reviewed sequences from the
                                <a href="http://www.expasy.org/sprot/" target="_blank" rel="noopener">UniProtKB.</a></li>
                            <li><b>Uniprot_trembl:</b> automatically annotated and not reviewed sequences from the
                                <a href="http://www.ebi.ac.uk/swissprot/" target="_blank" rel="noopener">UniProtKB.</a></li>
                        </ul>
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">MSA enrichment iterations using HHblits</a>
                <div class="accordion-content" data-tab-content>
                    Specifies the number of HHblits iterations over Uniprot20 to build an MSA when the input is a single sequence.
                </div>
            </li>

            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">E-value</a>
                <div class="accordion-content" data-tab-content>
                    Report target sequences with an E-value of <= the specified value.
                </div>
        <li class="accordion-item" data-accordion-item>
            <a class="accordion-title">No. of target sequences</a>
            <div class="accordion-content" data-tab-content>
                This parameter controls how many matches will be displayed in the results.
            </div>
        </li>
        </ul>
    </div>
    <div id="tabs-3" class="greenlink">
        @views.html.help.common.toolkit_citation()
        <p>Finn RD., Clements J., Eddy SR. (2011) HMMER web server: interactive sequence similarity searching. Nucleic Acids Res. 39(Web Server issue): W29â€“W37. <a href = https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3125773 target="_blank" rel="noopener">PMCID: PMC3125773</a></p>
    </div>
    <div id="tabs-4" class="greenlink">
        <strong>Version:</strong> @version
    </div>
</div>
