@*
 * Copyright 2018 Dept. Protein Evolution, Max Planck Institute for Developmental Biology
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *@

@(version: String)
<div id="tabs">
    <ul>
        <li><a href="#tabs-1">Overview</a></li>
        <li><a href="#tabs-2">Input & Parameters</a></li>
        <li><a href="#tabs-3">References</a></li>
        <li><a href="#tabs-4">Version</a></li>
    </ul>
    <div id="tabs-1" class="greenlink">
        PhyML is a software that estimates maximum likelihood phylogenies from alignments of amino acid sequences.
    </div>
    <div id="tabs-2" class="greenlink">
        <ul class="accordion" data-accordion data-allow-all-closed='true'>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Input</a>
                <div class="accordion-content" data-tab-content>
                    You have the option to enter a multiple sequence alignment (MSA) in FASTA or CLUSTAL format.<br/><br/>
                    @views.html.help.common.msa()
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Number of replicates</a>
                <div class="accordion-content" data-tab-content>
                    Number of resampled datasets asked by <a href = http://evolution.genetics.washington.edu/phylip/doc/seqboot.html target="_blank" rel="noopener">SEQBOOT</a>.
                </div>
            </li>

            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Model of AminoAcid replacement</a>
                <div class="accordion-content" data-tab-content>
                    Amino acid substitution model that is used for the computation of the distance matrix by <a href = http://evolution.genetics.washington.edu/phylip/doc/protdist.html target="_blank" rel="noopener">PROTDIST</a> (Please have a look at the link for a more detailed description).
                </div>
            </li>

        </ul>
    </div>
    <div id="tabs-3" class="greenlink">
        @views.html.help.common.toolkit_citation()
        <p>Guindon S., Dufayard J.F., Lefort V., Anisimova M., Hordijk W., Gascuel O. <b>New Algorithms and Methods to Estimate Maximum-Likelihood Phylogenies: Assessing the Performance of PhyML 3.0.</b> Molecular Biology and Evolution, msx149, 2017.</p>
    </div>
    <div id="tabs-4" class="greenlink">
        <strong>Version:</strong> @version
    </div>
</div>
