@*
 * Copyright 2018 Dept. Protein Evolution, Max Planck Institute for Developmental Biology
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *@

@(version: String)
<div id="tabs">
    <ul>
        <li><a href="#tabs-1">Overview</a></li>
        <li><a href="#tabs-2">Input & Parameters</a></li>
        <li><a href="#tabs-3">References</a></li>
        <li><a href="#tabs-4">Version</a></li>
    </ul>
    <div id="tabs-1" class="greenlink">
        <p>This tools offers the BLAST+ and PSI-BLAST+ methods of NCBI for searching locally similar sequences in a chosen
            protein sequence database. The input for this tools is either a single protein sequence or a multiple
            sequence alignment. By default, one search iteration is performed, offering the BLAST+ functionality. To obtain
            the functionality of PSI-BLAST+, the hits found in the first search iteration could be forwarded to this tool
            to start the next search iteration. Alternatively, the number of iteration can also be set to a value > 1.</p>
    </div>
    <div id="tabs-2" class="greenlink">
        <ul class="accordion" data-accordion data-allow-all-closed='true'>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Input</a>
                <div class="accordion-content" data-tab-content>
                    You have the option to enter or upload either a single protein sequence or a multiple sequence
                    alignment (MSA) in FASTA or CLUSTAL format.<br/><br/>
                    @views.html.help.common.singleseq()
                    @views.html.help.common.msa()
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Select standard database</a>
                <div class="accordion-content" data-tab-content>
                    <p>The selected databases are used to perform the search.</p>
                    <ul>
                        <li><b>nr:</b> the non-redundant sequence database at the NCBI, updated weekly. (See NCBI's BLAST tutorial).</li>
                        <li><b>env:</b> NCBI's env database with environmental sequences, usually from unknown organisms.</li>
                        <li><b>nre:</b> nr+env database. Updated weekly.</li>
                        <li><b>nr90, nr70, env90 etc.:</b> representative sequences, filtered for a maximum pairwise sequence identity of ~90% or 70% with kClust. Updated bimonthly.</li>
                        <li><b>nr_euk, nr_bac, nr_arc, nr_pro, nr_vir:</b> eukaryotic, bacterial, archaeal, prokaryotic, and viral sequences from the non-redundant database nr. Updated weekly.</li>
                        <li><b>pdb_nr: </b>Sequences from proteins whose structures have been deposited in the Protein
                            Data Bank PDB.</li>
                        <li><b>Uniprot_sprot:</b> manually annotated and reviewed sequences from the
                            <a href="http://www.expasy.org/sprot/" target="_blank" rel="noopener">UniProtKB.</a></li>
                        <li><b>Uniprot_trembl:</b> automatically annotated and not reviewed sequences from the
                            <a href="http://www.ebi.ac.uk/swissprot/" target="_blank" rel="noopener">UniProtKB.</a></li>
                    </ul>
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Scoring Matrix</a>
                <div class="accordion-content" data-tab-content>
                    Specify an alternate scoring matrix.
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">Max. number of iterations</a>
                <div class="accordion-content" data-tab-content>
                    Specify the maximum number of rounds of search.
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">E-value</a>
                <div class="accordion-content" data-tab-content>
                    The statistical significance threshold for reporting matches against database sequences. If the
                    statistical significance ascribed to a match is greater than the e-value threshold, the match will
                    not be reported. Lower e-value thresholds are more stringent, leading to fewer chance matches being
                    reported. Increasing the threshold shows less stringent matches. Fractional values are acceptable.
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">E-value inclusion threshold</a>
                <div class="accordion-content" data-tab-content>
                    The statistical significance threshold for including a sequence in the model used to
                    create the PSSM for the next iteration.
                </div>
            </li>
            <li class="accordion-item" data-accordion-item>
                <a class="accordion-title">No. of target sequences</a>
                <div class="accordion-content" data-tab-content>
                    This parameter controls how many matches will be displayed in the results.
                </div>
            </li>
        </ul>
    </div>

    <div id="tabs-3" class="greenlink">
        @views.html.help.common.toolkit_citation()
        <p>Altschul S., Madden T., Schaffer A., Zhang J., Miller W., Lipman D. (1997) Gapped BLAST and PSI-BLAST: a new
            generation of protein database search programs. Nucl Acids Res 25(17):3389-3402.
            <a href = https://www.ncbi.nlm.nih.gov/pmc/articles/PMC146917 target="_blank" rel="noopener">PMCID: PMC146917</a></p>
        <p>Altschul SF., Gish W., Miller W., Myers EW., Lipman DJ. (1990) Basic local alignment search tool.
            J Mol Biol. 215:403-410. <a href = https://www.ncbi.nlm.nih.gov/pubmed/2231712 target="_blank" rel="noopener">PMID: 2231712</a></p>
        <p>Camacho C., Coulouris G., Avagyan V., Ma N., Papadopoulos J., Bealer K., Madden TL. (2008) BLAST+:
            architecture and applications. BMC Bioinformatics 10:421. <a href = https://www.ncbi.nlm.nih.gov/pubmed/20003500
        target="_blank" rel="noopener">PMID: 20003500</a></p>
    </div>
    <div id="tabs-4" class="greenlink">
        <strong>Version:</strong> @version
    </div>
</div>