<style>
    h3.search {
    background-color: #F4EEDF;
    }

    h3 {
    font-family: 'Noto Sans','Lucida Grande','Lucida Sans Unicode',Geneva,Verdana,Sans-Serif;
    clear: both;
    display: block;
    font-weight: bold;
    background-color: #E6F2F2;
    font-size: 1.2em;
    color: #000;
    margin-top: 10px;
    padding: 5px 0px 4px 8px;
    }
    h4 {
    font-family: 'Noto Sans','Lucida Grande','Lucida Sans Unicode',Geneva,Verdana,Sans-Serif;
    font-size: 1.0em;
    font-weight: bold;
    padding-left: 2%;
    float: left;
    width: 150px;
    text-align: left;
    }
    .tooldescr {
    font-family: 'Noto Sans','Lucida Grande','Lucida Sans Unicode',Geneva,Verdana,Sans-Serif;
    padding-left: 20%;
    font-size: 0.9em;
    text-align: justify;
    display: block;
    padding-right: 20%;
    }

    div.hr {

    clear: both;
    margin: 0px 0px 10px 0px;
    border-bottom: 1px solid #E6F2F2;
    }

    .close-button {
    position: absolute;
    color: #8a8a8a;
    right: 0.5rem;
    top: 0.2rem;
    font-size: 2em;
    line-height: 1;
    cursor: pointer;
    }


    .top-bar, .top-bar ul {
    background-color: #fff;
    }

    #navisection ul {

    font-size: 10px;
    display: block;
    margin: auto;
    border-width: 0px;
    padding-right: 0%;


    }

    div.ocker a {
    border-bottom-color: #D0BA89;
    text-decoration: none;
    line-height: inherit;
    cursor: pointer;
    color: #000;

    }


    hr {
    display: block;
    margin-top: 0em;
    margin-bottom: 2em;
    margin-left: auto;
    margin-right: auto;
    border-style: inset;
    border-width: 1px;
    }

    .dotted {border: 1px dotted #D0BA89; border-style: none none dotted; color: #fff; background-color: #fff; }

</style>
<h3 class="search">Search Tools</h3>

<div class="top-bar">
    <div id="navisection" class="top-bar-left ocker">
        <ul class="dropdown menu" data-dropdown-menu>

            <li  ><a href="/#/tools/csblast" data-close>CS-BLAST</a></li>

            <li  ><a href="/hhblits">HHblits</a></li>

            <li  ><a href="/#/tools/hhpred" data-close>HHpred</a></li>

            <li  ><a href="/hhsenser">HHsenser</a></li>

            <li  ><a href="/#/tools/hmmer3" data-close>HMMER3</a></li>

            <li  ><a href="/patsearch">PatternSearch</a></li>

            <li  ><a href="/prot_blastp">ProtBLAST+</a></li>

            <li  ><a href="/#/tools/psiblast" data-close>PSI-BLAST+</a></li>

            <li  ><a href="/simshift">SimShiftDB</a></li>

        </ul>
    </div>
</div>

<hr class="dotted">

<h4><a href="/#/tools/csblast" data-close>CS-BLAST</a></h4>
<p class="tooldescr">
	CS-BLAST is an extension to standard NCBI BLAST that allows to increase its sensitivity by a factor of more than two on remote homologs at the same speed.
    CS-BLAST first adds context-specific pseudocounts to the input sequence and then jumpstarts PSI-BLAST with the resulting profile.
    The output is identical to BLAST and contains a list of closest homologs with alignments.
    <br>The software can be downloaded <a href="http://wwwuser.gwdg.de/~compbiol/data/csblast/releases" target="_blank">here</a>.

</p>
<div class="hr">&nbsp;</div>



<h4><a href="/hhblits">HHblits</a></h4>
<p class="tooldescr">
	Remote homology detection method based on iterative HMM-HMM comparison. HHblits can build high-quality MSAs starting from single sequences or from MSAs. It transforms these into a query HMM and iteratively searches through uniprot20 or nr20 databases by adding significantly similar sequences from the previous search to the updated query HMM for the next search iteration. Compared to PSI-BLAST, HHblits is faster, up to twice as sensitive and produces more accurate alignments.

    <br>The HHblits software is part of the open source package <a href="http://wwwuser.gwdg.de/~compbiol/data/" target="_self">HHsuite</a>.
</p>
<div class="hr">&nbsp;</div>


<h4><a href="/#/tools/hhpred" data-close>HHpred</a></h4>
<p class="tooldescr">
	Sensitive protein homology detection and structure prediction by HMM-HMM-comparison. Starting from a query sequence, HHpred builds a multiple sequence alignment
    using HHblits and turns it into a profile HMM. This is then compared it with a database of HMMs representing proteins with known structure (e.g. PDB, SCOP) or annotated protein
    families (e.g. PFAM, SMART, CDD, COGs, KOGs). The output is a list of closest homologs with alignments.
    HHpred can also build 3D homology models using the identified templates in the PDB database. It can optimize template picking and query-template alignments for
    homology modeling.
    <br>The HHblits software is part of the open source package <a href="http://wwwuser.gwdg.de/~compbiol/data/" target="_self">HHsuite</a>.
</p>
<div class="hr">&nbsp;</div>


<h4><a href="/hhsenser">HHsenser</a></h4>
<p class="tooldescr">
	Sensitive intermediate profile search using HMM-HMM comparison. HHsenser takes a sequence or an alignment as input and builds an alignment with as many near or remote homologs as possible. It uses representative sequences in the vicinity of the input alignment as seeds to build new alignments with PSI-BLAST. The process continues until all seeds have been processed. The runtime may be 30 min up to several hours depending on input parameters.

</p>
<div class="hr">&nbsp;</div>





<h4><a href="/#/tools/hmmer3" data-close>HMMER3</a></h4>
<p class="tooldescr">
	Fast, PSI-BLAST accelerated <a href="http://hmmer.janelia.org/" target="blank">HMMER</a> search.
    About 30 times faster for the nr database.

	</p>
<div class="hr">&nbsp;</div>



<h4><a href="/patsearch">PatternSearch</a></h4>
<p class="tooldescr">
	Retrieves all sequences which contain a given PROSITE pattern or regular expression.
</p>
<div class="hr">&nbsp;</div>


<h4><a href="/prot_blast">ProtBLAST</a></h4>
<p class="tooldescr">
	Search with an amino acid sequence against protein databases for locally similar sequences. Uses legacy BLAST internally.
</p>
<div class="hr">&nbsp;</div>





<h4><a href="/prot_blastp">ProtBLAST+</a></h4>
<p class="tooldescr">
	Search with an amino acid sequence against protein databases for locally similar sequences.
</p>
<div class="hr">&nbsp;</div>





<h4><a href="/#/tools/psi_blast">PSI-BLAST</a></h4>
<p class="tooldescr">
	Search with an amino acid sequence against protein databases for locally similar sequences. Similar to ProteinBLAST but more sensitive. PSI-BLAST first performs a BLAST search and builds an alignment from the best local hits. This alignment is then used as a query for the next round of search. After each successive round the search alignment is updated.
</p>
<div class="hr">&nbsp;</div>





<h4><a href="/psi_blastp" data-close>PSI-BLAST+</a></h4>
<p class="tooldescr">
	Search with an amino acid sequence against protein databases for locally similar sequences. Similar to ProteinBLAST+ but more sensitive. PSI-BLAST+ first performs a BLAST+ search and builds an alignment from the best local hits. This alignment is then used as a query for the next round of search. After each successive round the search alignment is updated.
</p>
<div class="hr">&nbsp;</div>


<h4><a href="/simshift">SimShiftDB</a></h4>
<p class="tooldescr">
	SimShiftDB builds alignments between a query protein with unknown
    structure (for which chemical shifts for HA,H,N,CA,CB, or CO are available)
    and a database of template proteins whose 3D structure is known.
    The alignments are scored by an E-Value indication (similar to the E-Value
    calculated by BLAST).

</p>
<div class="hr">&nbsp;</div>



<h4><a target="_blank" href="/pdbalert">PDBalert</a></h4>
<p class="tooldescr">
PDBalert is a web-based automatic system that alerts users as soon as a pdb structure with homology to a protein of interest becomes available. Users can upload their personal protein sequences of interest. Once a week, when new proteins are released to the PDB database, PDBalert compares these with the users' sequences. When a significant match is found, the user is alerted by an email containing a link to the search results.
</p>
<div class="hr">&nbsp;</div>

<h3 class="search">External servers for nucleotide search</h3>

<h4><a target="_blank" href="http://blast.ncbi.nlm.nih.gov/Blast.cgi?PROGRAM=blastn&BLAST_PROGRAMS=megaBlast&PAGE_TYPE=BlastSearch&SHOW_DEFAULTS=on&LINK_LOC=blasthome">Nucleotide Blast</a></h4>
<p class="tooldescr">
        Search with a nucleotide sequence against nucleotide databases for locally
    similar sequences with tblastn.
</p>
<div class="hr">&nbsp;</div>


<h3 class="search">External servers for protein family search</h3>

<h4><a target="_blank" href="http://cbdm.mdc-berlin.de/~ard2">ARD2</a></h4>
<p class="tooldescr">
        Identify alpha-rod repeats, also defined as alpha-solenoids (HEAT, Armadillo among others) in protein sequences using a neural network. Alpha-solenoids are stacks of pairs of anti-parallel helices which form together a flexible rod or wire called alpha-solenoid.
</p>
<div class="hr">&nbsp;</div>

<h4><a target="_blank" href="http://www.ebi.ac.uk/InterProScan/">InterPro</a></h4>
<p class="tooldescr">
	Search your protein with the InterPro database of protein families, domains and
    functional sites in which identifiable features found in known proteins can
    be applied to unknown protein sequences. InterPro contains many databases of
    HMMs or patterns, such as Pfam, PANTHER, SMART, TIGRFAMs, PIRSF, PRODOM, CATH, SUPERFAMILY,
    PRINTS, and PROSITE.
</p>
<div class="hr">&nbsp;</div>

<h4><a target="_blank" href="http://pfam.xfam.org">Pfam</a></h4>
<p class="tooldescr">
	Search your protein sequence with the Pfam database of HMMs for known protein
    domains (including modular domains and repeat families). The Pfam database
    contains a large collection of curated and relatively well-annotated alignments/HMMs.
</p>
<div class="hr">&nbsp;</div>

<h4><a target="_blank" href="http://smart.embl-heidelberg.de">SMART</a></h4>
<p class="tooldescr">
	Search your protein sequence with the SMART database of HMMs for known modular
    domains and repeat families. The database alignments/HMMs are hand-curated and very
    well annotated.
</p>
<div class="hr">&nbsp;</div>

<h3 class="search">External servers for linear motif search</h3>

<h4><a target="_blank" href="http://elm.eu.org/">ELM</a></h4>
<p class="tooldescr">
	Prediction of Eukaryotic Linear Motifs, a special class of functional sites in
    eukaryotic proteins. Putative functional sites are identified by patterns. To
    improve the predictive power, context-based rules and logical filters are applied.
</p>
<div class="hr">&nbsp;</div>

<h4><a target="_blank" href="http://scansite.mit.edu/abtscansite.phtml">ScanSite</a></h4>
<p class="tooldescr">
	Search your protein for motifs likely to be phosphorylated by a specific protein kinases
    or to bind to domains such as SH2 domains, 14-3-3 domains or PDZ domains. Uses
    sequence profiles instead of patterns.
</p>
<div class="hr">&nbsp;</div>