@*
 * Copyright 2018 Dept. Protein Evolution, Max Planck Institute for Developmental Biology
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *@

@import better.files._
@import de.proteinevolution.results.results.HHPredResult
@import de.proteinevolution.tools.Tool
@import play.twirl.api.Html

@(jobID: String, result: HHPredResult, tool: Tool, htmlPath: String)()

@if(result.num_hits < 1) {
    <div class="note">
        No hits found! Please re-run HHpred with an increased number of MSA generation steps or with a custom-built
        multiple sequence alignment as input (set 'Maximal no. of MSA generation steps' to 0). You could also
        consider picking a different target database and/or using different parameters.
    </div>
} else {
    <div class="scrollContainer scrollContainerWhite">
        <div class="scrollContainerDiv scrollContainerDivWhite">
            <div id="scrollLinks">
                <a id="visualizationScroll" name="#visualization">Vis</a>
                <a id="hitlistScroll" name="#hitlist">Hits</a>
                <a id="alignmentsScroll" name="#alignments">Aln</a>
            </div>
            <span class="headerDivider"></span>
            <div id="controlls">
                <a class="selectAllSeqBar extraWide">Select all</a>
                <a class="forwardModalOpenBtn" data-modal-type="simple">Forward</a>
                <a class="forwardModalOpenBtn" data-modal-type="hhsuite">Forward Query A3M</a>
                @*only show create model when chosen db=mmcif70/pdb70*@
                @if((result.db.matches("mmcif70/pdb70") || result.db.matches("mmcif30/pdb30"))
                        && result.proteomes.matches("false")) {
                    <a onclick="@{tool.toolNameShort}_manual()">Model using selection</a>
                }
                <a onclick="downloadHHR()" class="downloadHHR">HHR file</a>
                <a class="colorAA">Color Seqs</a>
                <a id="wrap">Wrap Seqs</a>
            </div>
        </div>
    </div>
    <br>
    <span class="notePlain">Number of hits: <b>@result.num_hits</b></span>
    <br>
    @if(result.TMPRED == "1" || result.TMPRED.toInt > 1 || result.COILPRED == "0") {
        <br>
        <span class="note"><b> We have detected</b></span>

    }
    @if(result.TMPRED == "1") {

        <span class="note"><b> @result.TMPRED transmembrane helix</b></span>
    }
    @if(result.TMPRED.toInt > 1) {

        <span class="note"><b> @result.TMPRED transmembrane helices</b></span>
    }
    @if((result.TMPRED == "1" || result.TMPRED.toInt > 1) && result.COILPRED == "0") {
        <span class="note"><b> and </b></span>

    }
    @if(result.COILPRED == "0") {
        <span class="note"><b>coiled-coil segments </b></span>
    }

    @if(result.TMPRED == "1" || result.TMPRED.toInt > 1 || result.COILPRED == "0") {
        <span class="note"><b> in your query protein!</b></span>
        <br>

    }

    @if(result.QA3M_COUNT < 20) {
        <br>
        <span class="note">Note: your query alignment consists of only @result.QA3M_COUNT sequence(s).</span>
        @if(result.MSA_GEN == "uniprot20") {

            <span class="note">You could improve the sensitivity of your search vastly by building a larger query
                alignment either with HHblits over Uniclust30 or with PSI-BLAST over nre70. You could also consider
                increasing the number of MSA generation iterations and/or relaxing the inclusion threshold.</span>
        }
        @if(result.MSA_GEN == "uniclust30") {
            <span class="note">You could improve the sensitivity of your search vastly by building a larger
                query alignment with PSI-BLAST over nre70. You could also consider increasing the number of MSA
                generation iterations and/or relaxing the inclusion threshold.</span>
        }
        @if(result.MSA_GEN == "psiblast") {
            <span class="note">You could improve the sensitivity of your search vastly by building a larger
                query alignment; for instance, by increasing the number of MSA generation iterations and/or relaxing the
                inclusion threshold. Alternatively, you could input your own alignment.</span>
        }
        @if(result.MSA_GEN == "custom") {
            <span class="note">
                You could improve the sensitivity of your search vastly by providing a larger query alignment.</span>
        }

    }
    @* Load html Map used for the image representation *@
    <div id="visualization">
        <h5 id="visualizationTitle">Visualization</h5>
        <div>
            <a class="button tiny resubmitSection" type="button" id="resubmitSection">
                Resubmit section
            </a>
        </div>
        <div id='blastviz'>
            <div class='flat-slider' id='flat-slider'></div>
            @Html(htmlPath.toFile.contentAsString)
            <p><img src="results/files/@{
                jobID + "/" + jobID + ".png"
            }" border="0" alt="blasthits" usemap="#blastmap"></p>
        </div>
    </div>
    <div id="hitlist">
        <h5 id="hitlistTitle">Hitlist</h5>
        <table id="htb" class="display" cellspacing="0" width="100%">
            <thead>
                <tr>
                    <th>Nr</th>
                    <th>Hit</th>
                    <th>Name</th>
                    <th>Probability</th>
                    <th>E-value</th>
                    <th>SS</th>
                    <th>Cols</th>
                    <th>Target Length</th>
                </tr>
            </thead>
        </table>
    </div>
    <div id="alignmentResult">
        <div id="alignments">
            <h5 id="alignmentsTitleHHpred">Alignments</h5>
            <form id="hhpred_templates">
                <table class="unstriped">
                    <tbody id="alignmentTable">
                    </tbody>
                    <tr>
                        <td colspan="4" ><a id="loadHits">Load Hits</a></td>
                    </tr>
                    <tr>
                        <td colspan="4"><div id="loadingHits">Loading Hits...</div></td>
                    </tr>
                </table>
            </form>
        </div>
    </div>
    <br>
    <div id="toolReferences">
        <h5 id="refTitle">References</h5>
        If you use our HHpred server for your research, please cite:
        <div class="citation">
        @views.html.modals.toolkit_citation_results()
        </div>
        And one or more of the following:
        <div class="citation">
            <div>Protein homology detection by HMM-HMM comparison.<br>
                Söding J.<a href="https://academic.oup.com/bioinformatics/article-lookup/doi/10.1093/bioinformatics/bti125" target="_blank" rel="noopener">
                    Bioinformatics. 2005 Apr 1;21(7):951-60</a>,</div><br>
            <div>Fast and accurate automatic structure prediction with HHpred.<br>
                Hildebrand A, Remmert M, Biegert A, Söding J.<a href="http://onlinelibrary.wiley.com/doi/10.1002/prot.22499/abstract;jsessionid=6FF63B8F61E35391B79F4681BB12DB2C.f02t04" target="_blank" rel="noopener">
                    Proteins. 2009;77 Suppl 9:128-32</a>,</div><br>
            <div>Automatic Prediction of Protein 3D Structures by Probabilistic Multi-template Homology Modeling.<br>
                Meier A, Söding J.<a href="http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1004343" target="_blank" rel="noopener">
                    PLoS Comput Biol. 2015 Oct 23;11(10):e1004343</a>.</div><br>
        </div>
    </div>
    <hr class="horizontal-line">
    @*MODALS*@
    <div class="reveal" id="structureModal" data-reveal>
        <p class="custom-modal-header">Template 3D structure</p>
        <div id="accessionStructure"> </div>

        <div id="viewport"></div>
        <button class="close-button" data-close aria-label="Close modal" type="button">
            <span aria-hidden="true">&times;</span>
        </button>
    </div>

}
<script>
        var shownHits = 100;
        var jobID = "@{jobID}";

        @if(result.num_hits > 0){
            $(function () {
                var context = {
                    numHits: @result.num_hits,
                    toolName: "@tool.toolNameShort",
                    firstQueryStart: @result.HSPS.head.query.start,
                    firstQueryEnd: @result.HSPS.head.query.end,
                    query: {
                        seq: "@result.query.seq",
                        accession: "@result.query.accession"
                    }
                };
                Toolkit.resultView = new NormalResultView($("#jobview"), jobID, shownHits, true, true, context);
            });
            new StructureModal($("#jobview"), $("#structureModal"));
        }

        window["downloadHHR"] = function () {
            var downloadFilename = "hhpred_" + jobID + ".hhr";
            var fileURL = "/results/files/" + jobID + "/" + jobID + ".hhr";
            $.ajax({
                method: "GET",
                url: fileURL
            }).done(function (data) {
                DownloadHelper.download(downloadFilename, data);
            })
        };

        function @{tool.toolNameShort}_manual() {
            // Get selected templates and append parentID
            var formData = new FormData(document.getElementById("hhpred_templates"));
            if ($('input:checkbox[name=alignment_elem]:checked').length === 0) {
                $("input:checkbox[name=alignment_elem][value=1]").prop("checked", true);
                AlertService.alert("No templates selected! Using the first template for modelling.", "alert-warning");
            }
            formData.append("templates", Toolkit.resultView.getSelectedValues.join(" ")); // TODO move checkbox handling
            formData.append("parent_id", jobID);

            m.request({
                method: "POST",
                url: "/api/jobs/?toolName=hhpred_manual",
                data: formData,
                serialize: function (submissionReturnData) {
                    return submissionReturnData;
                }
            }).then(function (submissionReturnData) {
                if (submissionReturnData.successful) {
                    jobID = submissionReturnData.jobID;
                    var jobListComp = JobListComponent.Job(
                            {jobID: jobID, state: 0, dateCreated: Date.now().valueOf(), tool: "hhpred_manual"}
                    );
                    if (!JobListComponent.contains(jobID)) {
                        JobListComponent.pushJob(jobListComp, true);
                    }
                }
                $(".submitJob").prop("disabled", false);
                JobSubmissionComponent.submitting = false;
            }).catch(function (error) {
                console.log("Error while submitting:", error);
                $(".submitJob").prop("disabled", false);
                JobSubmissionComponent.submitting = false;
            });
        }

</script>
