@import de.proteinevolution.results.results.Common
@import de.proteinevolution.results.results.Quick2D.Quick2DResult
@import play.twirl.api.Html
@(result: Quick2DResult)()

<h5>Protein ID: @{result.query.accession.slice(0, 50)}</h5>

@if(result.signal.seq == "yes"){
    <br>
    <span class="note"><b> We have detected a potential signal peptide in your query protein!</b></span>
}
<br />
<br />
<br />

<table class="unstriped" id="resultTableQ2D">
    <tbody class="alignmentTBody">
    @Html(Common.quick2dWrapped(result, 0, 90))
    </tbody>
</table>
<hr class="horizontal-line">
<br>
<div align="center">
SS = <span class="ss_h_b">&nbsp;&alpha;-helix&nbsp;</span><span class="ss_e_b">&nbsp;&beta;-strand&nbsp;</span><span class="ss_pihelix">&nbsp;&pi;-helix&nbsp;</span>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;CC = <span class="CC_b">Coiled Coils</span>
    &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;TM = <span class="CC_m">Transmembrane</span>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;DO = <span class="CC_do">Disorder</span>
    </div>
<br>
<br>
<hr class="horizontal-line">
<div id="toolReferences">
    <h5 id="refTitle">References</h5>
    If you use our Quick2D server for your research, please cite:
    <div class="citation">
        @views.html.modals.toolkit_citation_results()
    </div>
    And one or more of the following as appropriate:
    <div class="citation">
        <div>PSIPRED: Protein secondary structure prediction based on position-specific scoring matrices.<br>
            Jones DT.<a href="http://www.sciencedirect.com/science/article/pii/S0022283699930917?via%3Dihub" target="_blank" rel="noopener"> J Mol Biol. 1999 Sep 17;292(2):195-202</a>,</div><br>
        <div>SPIDER2: A Package to Predict Secondary Structure, Accessible Surface Area, and Main-Chain Torsional Angles by Deep Neural Networks.<br>
            Yang et al.<a href="https://link.springer.com/protocol/10.1007%2F978-1-4939-6406-2_6" target="_blank" rel="noopener"> Methods Mol Biol. 2017;1484:55-63</a>,</div><br>
        <div>PSSPred: Protein Secondary Structure PREDiction.<br>
            Zhang Y.<a href="http://zhanglab.ccmb.med.umich.edu/PSSpred" target="_blank" rel="noopener"> PSSPred Webserver</a>,</div><br>
        <div>DeepCNF-SS: Protein Secondary Structure Prediction Using Deep Convolutional Neural Fields.<br>
            Wang S, Peng J, Ma J, Xu J.<a href="https://www.nature.com/articles/srep18962" target="_blank" rel="noopener"> Sci Rep. 2016 Jan 11;6:18962</a>,</div><br>
        <div>COILS: Predicting coiled coils from protein sequences.<br>
            Lupas A, Van Dyke M, Stock J.<a href="http://science.sciencemag.org/content/252/5009/1162" target="_blank" rel="noopener"> Science. 1991 May 24;252(5009):1162-4</a>,</div><br>
        <div>PCOILS: Comparative analysis of coiled-coil prediction methods.<br>
            Gruber M, Söding J, Lupas AN.<a href="http://www.sciencedirect.com/science/article/pii/S1047847706000815?via%3Dihub" target="_blank" rel="noopener"> J Struct Biol. 2006 Aug;155(2):140-5</a>,</div><br>
        <div>MARCOIL: An HMM model for coiled-coil domains and a comparison with PSSM-based predictions.<br>
            Delorenzi M, Speed T.<a href="https://www.ncbi.nlm.nih.gov/pubmed/12016059?dopt=Abstract" target="_blank" rel="noopener"> Bioinformatics. 2002 Apr;18(4):617-25</a>,</div><br>
        <div>TMHMM: Predicting transmembrane protein topology with a hidden Markov model: application to complete genomes.<br>
            Krogh A, Larsson B, von Heijne G, Sonnhammer EL.<a href="http://www.sciencedirect.com/science/article/pii/S0022283600943158?via%3Dihub" target="_blank" rel="noopener"> J Mol Biol. 2001 Jan 19;305(3):567-80</a>,</div><br>
        <div>Phobius: A combined transmembrane topology and signal peptide prediction method.<br>
            Käll L, Krogh A, Sonnhammer EL.<a href="http://www.sciencedirect.com/science/article/pii/S0022283604002943?via%3Dihub" target="_blank" rel="noopener"> J Mol Biol. 2004 May 14;338(5):1027-36</a>,</div><br>
        <div>PolyPhobius: An HMM posterior decoder for sequence feature prediction that includes homology information.<br>
            Käll L, Krogh A, Sonnhammer EL.<a href="https://academic.oup.com/bioinformatics/article-lookup/doi/10.1093/bioinformatics/bti1014" target="_blank" rel="noopener"> Bioinformatics. 2005 Jun;21</a>,</div><br>
        <div>DISOPRED3: precise disordered region predictions with annotated protein-binding activity.<br>
            Jones DT, Cozzetto D.<a href="https://academic.oup.com/bioinformatics/article/31/6/857/215129/DISOPRED3-precise-disordered-region-predictions" target="_blank" rel="noopener"> Bioinformatics. 2015 Mar 15;31(6):857-63</a>,</div><br>
        <div>SPOT-Disorder: Improving protein disorder prediction by deep bidirectional long short-term memory recurrent neural networks.<br>
            Hanson J, Yang Y, Paliwal K, Zhou Y.<a href="https://academic.oup.com/bioinformatics/article-lookup/doi/10.1093/bioinformatics/btw678" target="_blank" rel="noopener"> Bioinformatics. 2017 Mar 1;33(5):685-692</a>,</div><br>
        <div>IUPred: The pairwise energy content estimated from amino acid composition discriminates between folded and intrinsically unstructured proteins.<br>
            Dosztányi Z, Csizmók V, Tompa P, Simon I.<a href="http://www.sciencedirect.com/science/article/pii/S0022283605001294?via%3Dihub" target="_blank" rel="noopener"> J Mol Biol. 2005 Apr 8;347(4):827-39</a>.</div><br>
    </div>
</div>
<hr class="horizontal-line">
