export default [
    {
        date: '2020-04-09',
        html: `<p>
        HHpred: profile HMM database for a new version of the
        <a href="https://www.ncbi.nlm.nih.gov/Structure/cdd/cdd.shtml" target="_blank">
        NCBI Conserved Domains Database</a>, version 3.18, is now online. It is also available for download at our
         <a href="ftp://ftp.tuebingen.mpg.de/pub/protevo/toolkit/databases/hhsuite_dbs/" target="_blank">
         FTP site</a>.
        </p>`,
    },
    {
        date: '2020-03-31',
        html: `<p>
        HHpred: profile HMM database for the proteome of <em>SARS coronavirus 2 (SARS-CoV-2)</em>
        is now online.
        </p>`,
    },
    {
        date: '2020-03-23',
        html: `<p>
        HHpred: profile HMM database for the proteome of <em>Capsaspora owczarzaki </em>
        (strain ATCC 30864), a close unicellular relative of animals, is now online.
        </p>`,
    },
    {
        date: '2020-03-10',
        html: `<p>
        HHpred: profile HMM database for a new version of the
        <a href="https://www.ncbi.nlm.nih.gov/Structure/cdd/cdd.shtml" target="_blank">
        NCBI Conserved Domains database</a>, version 3.17, is now online. It is also available for download at our
         <a href="ftp://ftp.tuebingen.mpg.de/pub/protevo/toolkit/databases/hhsuite_dbs/" target="_blank">
         FTP site</a>.
        </p>`,
    },
    {
        date: '2020-03-09',
        html: `<p>
        HHpred: profile HMM database for a new version of the
        <a href="http://prodata.swmed.edu/ecod/" target="_blank"> ECOD_F70 database</a>, version 20200221 (develop265),
        is now online. It is also available for download at our
         <a href="ftp://ftp.tuebingen.mpg.de/pub/protevo/toolkit/databases/hhsuite_dbs/" target="_blank">
         FTP site</a>.
        </p>`,
    },
    {
        date: '2020-02-12',
        html: `<p>
        HHpred, HMMER, HHrepID, and HHomp now use the
        <a href="http://gwdu111.gwdg.de/~compbiol/uniclust/current_release/" target="_blank"> UniRef30
        </a> database (instead of UniClust30) for query MSA generation. Additionally, UniRef30 is also
        offered as the standard database for HHblits.
        </p>`,
    },
    {
        date: '2020-02-06',
        html: ` <p>
        Pre-formatted nr, nr90, nr70, nr50, and nr30 BLAST databases are now available for download at our
         <a href="ftp://ftp.tuebingen.mpg.de/pub/protevo/toolkit/databases/blast/" target="_blank">
         FTP site</a>.
        </p>`,
    },
    {
        date: '2019-08-13',
        html: ` <p>
        The UI of the Toolkit was completely reimplemented using <a href="https://vuejs.org/"
        target="_blank">Vue.js</a>.
        </p>`,
    },
    {
        date: '2019-07-28',
        html: ` <p>
        HHpred: Profile HMM database of the <em>Solanum lycopersicum</em> (tomato) proteome is now online.
        </p>`,
    },
    {
        date: '2019-05-06',
        html: ` <p>
        HHpred: profile HMM DBs for the proteomes of <em>Bombyx mori</em> (domestic silkmoth) and the human gut
        bacterium <em>Christensenella minuta</em> are online.
        </p>`,
    },
    {
        date: '2019-05-03',
        html: ` <p>
        We have included a new tool, PiPred, for the prediction of π-helices in protein sequences.
        </p>`,
    },
    {
        date: '2019-01-04',
        html: ` <p>
        We have included a further tool, DeepCoil, for the prediction of coiled coils in protein sequences.
        </p>`,
    },
    {
        date: '2018-12-09',
        html: ` <p>
        HHpred: profile HMM database of the unicellular ciliate <em>Paramecium tetraurelia</em> is online.
        </p>`,
    },
    {
        date: '2018-11-22',
        html: ` <p>
        HHpred: profile HMM databases of <em>Salmonella enterica</em> serovar Typhi (CT18) and <em>Trichomonas
        vaginalis</em> are online.
        </p>`,
    },
    {
        date: '2018-11-10',
        html: ` <p>
        HHpred: profile HMM database of the latest version of Pfam (v32.0) is online.
        </p>`,
    },
    {
        date: '2018-10-31',
        html: ` <p>
        HHpred: profile HMM database of the composite metagenome-assembled genome <em>Lokiarchaeum sp. GC14_75</em>,
        a reference for exploring eukaryogenesis, is online.
        </p>`,
    },
    {
        date: '2018-06-01',
        html: ` <p>
        HHpred: profile HMM databases of <em>Bacteriovorax sp., Bdellovibrio bacteriovorus, Enterococcus faecalis,
        Helicobacter pylori, Leptospira interrogans, Neisseria gonorrhoeae, Phycisphaerae bacterium,
        Plesiocystis pacifica, and Waddlia chondrophila</em> are online.
        </p>`,
    },
    {
        date: '2018-05-10',
        html: `<p>
        HHpred: profile HMM database of <em>Toxoplasma gondii</em> is online.
        </p>`,
    },
    {
        date: '2018-03-29',
        html: `<p>
        HHpred: profile HMM database of <em>Acinetobacter baumannii</em> is online.
        </p>`,
    },
    {
        date: '2018-03-07',
        html: `<p>
        HHpred: a new version of the SCOPe database, <em><a href="http://scop.berkeley.edu/statistics/ver=2.07"
        target="_blank" rel="noopener">version 2.07</a></em>, is now online.
        </p>`,
    },
    {
        date: '2018-03-04',
        html: `<p>
        HHpred: a new version of the ECOD database, version 20180219 (develop204), is now online.
        </p>`,
    },
    {
        date: '2017-12-24',
        html: `<p>
        Our paper on the new Toolkit is out:
        <em>
        A Completely Reimplemented MPI Bioinformatics Toolkit with a New HHpred Server at its Core.
        Zimmermann L, Stephens A, Nam SZ, Rau D, Kübler J, Lozajic M, Gabler F, Söding J, Lupas AN, Alva V.
        <a href="https://www.sciencedirect.com/science/article/pii/S0022283617305879" target="_blank" rel="noopener">
            J Mol Biol. 2017 Dec 16.
        </a>
        </em>
        </p>`,
    },
    {
        date: '2017-12-06',
        html: `<p>
        HHpred: the PDB_mmCIF70 DB is now built using Uniclust30 instead of Uniprot20. Additionally, we now also
        offer PDB_mmCIF30, containing fewer representatives.
        </p>`,
    },
    {
        date: '2017-12-05',
        html: `<p>
        HHpred: profile HMM DBs of six archaeal proteomes (<em>Archaeoglobus fulgidus</em>,
        <em>Halobacterium jilantaiense</em>, <em>Methanocaldococcus jannaschii</em>, <em>Methanothermus fervidus</em>,
        <em>Pyrococcus horikoshii</em>, and <em>Sulfolobus solfataricus</em>) are online.
        </p>`,
    },
    {
        date: '2017-11-13',
        html: `<p>
        In addition to nr50, we now also offer nr30 for PSI-BLAST, HMMER, and PatternSearch.
        </p>`,
    },
    {
        date: '2017-10-31',
        html: `<p>
        HHpred: the Uniclust30 DB is now available for query A3M generation.
        </p>`,
    },
    {
        date: '2017-09-29',
        html: `<p>
        HHblits: a new version of the Uniclust30 DB is online.
        </p>`,
    },
    {
        date: '2017-09-24',
        html: `<p>
        HHpred: profile HMM DBs of <em>Fischerella muscicola</em>, <em>Frankia alni</em>,
        <em>Streptomyces scabiei</em>, and <em>Thermus aquaticus</em> are online.
        </p>`,
    },
    {
        date: '2017-09-19',
        html: `<p>
        HHpred: profile HMM DBs of <em>Aquifex aeolicus</em>, <em>Deinococcus radiodurans</em>,
        <em>Dictyostelium discoideum</em>, <em>Schizosaccharomyces pombe</em>, and
        <em>Thermus thermophilus</em> are online.
        </p>`,
    },
    {
        date: '2017-09-09',
        html: `<p>
        HHpred: profile HMMs of the ECOD database (Grishin Lab) are online.
        </p>`,
    },
    {
        date: '2017-08-31',
        html: `<p>
        HHpred: HMM database of <em>Giardia lamblia</em> is online.
        </p>`,
    },
    {
        date: '2017-08-28',
        html: `<p>
        HHpred: HMM database of <em>Physcomitrella patens</em> is online.
        </p>`,
    },
    {
        date: '2017-08-23',
        html: `<p>
        HHpred: HMM databases of <em>Brachypodium distachyon</em> and <em>Tetrahymena thermophila</em> are online.
        </p>`,
    },
    {
        date: '2017-07-21',
        html: `<p>
        Our Quick2D tool for the annotation of protein secondary structural features is back.
        </p>`,
    },
    {
        date: '2018-07-18',
        html: `<p>
        HMM databases of <em>Caenorhabitis elegans</em> and <em>Drosophila melanogaster</em> are online.
        </p>`,
    },
    {
        date: '2017-07-10',
        html: `<p>
        HMM databases of the COG, KOG, SMART, and NCBI CD databases are online now.
        </p>`,
    },
    {
        date: '2017-07-04',
        html: `<p>
        Profile HMM databases of the <b>human proteome</b> and <em>Chaetomium thermophilum</em> are online.
        </p>`,
    },
    {
        date: '2017-06-20',
        html: `<p>
        Profile HMM database of <em>Arabidopsis thaliana</em> is online.
        </p>`,
    },
    {
        date: '2017-06-10',
        html: `<p>
        Profile HMM databases of many bacterial proteomes are online.
        </p>`,
    },
    {
        date: '2017-05-27',
        html: `<p>
        Profile HMM databases of <em>Mycobacterium tuberculosis</em> is now available for HHpred.
        </p>`,
    },
    {
        date: '2017-05-19',
        html: `<p>
            <b>Release Notes</b>: Welcome to the new version of the
            MPI Bioinformatics Toolkit
        </p>
        <p>
            Dear Toolkit Users,<br>
            <br>
            welcome to the new version of the MPI Bioinformatics Toolkit hosted at the Department of Protein Evolution,
            Max Planck Institute for Developmental Biology, Germany.
            The first version of the Toolkit was released in 2005
            and has serviced over 2.5 million external queries to date.
            We had to rework the Toolkit entirely because of mounting software incompatibility with our
            computing infrastructure and also took this opportunity to improve some features
            and prune a few of the outdated and less-widely used tools.
            To allow an easy transition to the new Toolkit, we have retained the basic structure and
            UI features of the earlier version, and have added additional features, some of
            which are listed below, to enhance user experience.
        </p>
        <p>
            <em>New tools</em>
        </p>
        <ul>
            <li>PhyML is offered instead of PHYLIP-NEIGHBOR for phylogenetic inference.</li>
            <li>MMseq2s is offered instead of BLASTClust for clustering of up to 20,000 sequences.</li>
        </ul>
        <p>
            <em>New features/changes</em>
        </p>
        <ul>
            <li>HHpred allows comparison of two sequences or MSAs, replacing HHalign.</li>
            <li>CLANS allows clustering of up to 10,000 sequences.</li>
            <li>BLAST+ and PSI-BLAST+ are offered as a single tool.</li>
            <li>Caching of results for quick retrieval of pre-computed jobs.</li>
            <li>Search box on the index page for quick access to tools and jobs.</li>
            <li>Job manager in addition to joblist for easy management of jobs.</li>
            <li>E-mail notifications are only available to logged-in users.</li>
            <li>Comprehensive client-side input validation.</li>
            <li>Significantly enhanced performance, owing to upgraded computational infrastructure.</li>
        </ul>
        <p>
            <em>Coming soon</em>
            <br>
            We are going to bring back FRpred, HHomp, Quick2D, daTAA, and REPPER within the next few weeks.
            Additionally, we are going to offer many more proteome databases for
            HHpred over the next months and are also working on making the
            help pages more comprehensive.
        </p>
        <p>
            The new Toolkit is still under active development and
            we invite you to send us your feature requests and bug reports
            (<a href="mailto:mpi-toolkit&commat;tuebingen.mpg.de">mpi-toolkit&commat;tuebingen.mpg.de</a>).
        </p>
        <p>
            If you find our Toolkit useful for your research, please cite us
            <a href="https://academic.oup.com/nar/article-lookup/doi/10.1093/nar/gkw348" target="_blank" rel="noopener">
            (Alva et al., 2016)
            </a>; it helps us keep this service running for you.
        </p>
        <p>
            Best wishes,<br>
            Your Toolkit Team
        </p>`,
    },
];
