<search>
<h3 class="search">Search</h3>

<%= render(:partial => "shared/tool_descriptions") %>

<h4><a target="_blank" href="/pdbalert">PDBalert</a></h4>
<p class="tooldescr">
PDBalert is a web-based automatic system that alerts users as soon as a pdb structure with homology to a protein of interest becomes available. Users can upload their personal protein sequences of interest. Once a week, when new proteins are released to the PDB database, PDBalert compares these with the users' sequences. When a significant match is found, the user is alerted by an email containing a link to the search results.
</p>
<div class="hr">&nbsp;</div>

<h3 class="search">External servers for nucleotide search</h3>

<h4><a target="_blank" href="http://blast.ncbi.nlm.nih.gov/Blast.cgi?PROGRAM=blastn&BLAST_PROGRAMS=megaBlast&PAGE_TYPE=BlastSearch&SHOW_DEFAULTS=on&LINK_LOC=blasthome">Nucleotide Blast</a></h4>
<p class="tooldescr">
        Search with a nucleotide sequence against nucleotide databases for locally
    similar sequences with tblastn.
</p>
<div class="hr">&nbsp;</div>


<h3 class="search">External servers for protein family search</h3>

<h4><a target="_blank" href="http://www.ebi.ac.uk/InterProScan/">InterPro</a></h4>
<p class="tooldescr">
	Search your protein with the InterPro database of protein families, domains and
    functional sites in which identifiable features found in known proteins can
    be applied to unknown protein sequences. InterPro contains many databases of
    HMMs or patterns, such as Pfam, PANTHER, SMART, TIGRFAMs, PIRSF, PRODOM, CATH, SUPERFAMILY,
    PRINTS, and PROSITE.
</p>
<div class="hr">&nbsp;</div>

<h4><a target="_blank" href="http://pfam.xfam.org">Pfam</a></h4>
<p class="tooldescr">
	Search your protein sequence with the Pfam database of HMMs for known protein
    domains (including modular domains and repeat families). The Pfam database
    contains a large collection of curated and relatively well-annotated alignments/HMMs.
</p>
<div class="hr">&nbsp;</div>

<h4><a target="_blank" href="http://smart.embl-heidelberg.de">SMART</a></h4>
<p class="tooldescr">
	Search your protein sequence with the SMART database of HMMs for known modular
    domains and repeat families. The database alignments/HMMs are hand-curated and very
    well annotated.
</p>
<div class="hr">&nbsp;</div>

<h4><a target="_blank" href="http://www.embl-heidelberg.de/~andrade/papers/rep/search.html">REP</a></h4>
<p class="tooldescr">
	Search your protein sequence with a database of HMMs for known repeat families, such as
    Ankyrin, Armadillo, HAT, HEAT, Kelch, LRR, PFTA, PFTB, RCC1, TPR, WD40.
</p>
<div class="hr">&nbsp;</div>

<h3 class="search">External servers for linear motif search</h3>

<h4><a target="_blank" href="http://elm.eu.org/">ELM</a></h4>
<p class="tooldescr">
	Prediction of Eukaryotic Linear Motifs, a special class of functional sites in
    eukaryotic proteins. Putative functional sites are identified by patterns. To
    improve the predictive power, context-based rules and logical filters are applied.
</p>
<div class="hr">&nbsp;</div>

<h4><a target="_blank" href="http://scansite.mit.edu/abtscansite.phtml">ScanSite</a></h4>
<p class="tooldescr">
	Search your protein for motifs likely to be phosphorylated by a specific protein kinases
    or to bind to domains such as SH2 domains, 14-3-3 domains or PDZ domains. Uses
    sequence profiles instead of patterns.
</p>
<div class="hr">&nbsp;</div>
</search>
